diff --cc CMakeLists.txt
index 536ca45,c7e9b1c..0000000
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@@ -33,16 -33,17 +33,17 @@@ cet_set_compiler_flags(DIAGS VIGILAN
  
  # these are minimum required versions, not the actual product versions
  find_ups_boost(v1_53_0)
 -#find_ups_product (cetbuildtools) ### MIGRATE-ACTION-REQUIRED: remove ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
 -find_ups_product( canvas v1_00_00 ) ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
 -find_ups_product( cetlib v1_08_00 ) ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
 -find_ups_product( cetpkgsupport v1_08_03 ) ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
 -find_ups_product( clhep v2_2_0 ) ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
 -find_ups_product( fhiclcpp v3_04_00 ) ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
 -find_ups_product( messagefacility v1_14_01 ) ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
 -find_ups_product( sqlite v3_08_05_00 ) ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
 -find_ups_product( tbb ) ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
 +find_ups_product( canvas v1_00_00 )
 +find_ups_product( cetlib v1_08_00 )
 +find_ups_product( cetpkgsupport v1_08_03 )
 +find_ups_product( clhep v2_2_0 )
 +find_ups_product( fhiclcpp v3_04_00 )
 +find_ups_product( messagefacility v1_14_01 )
 +find_ups_product( sqlite v3_08_05_00 )
 +find_ups_product( tbb )
++
  find_ups_root( v5_34_20 )
 -find_ups_product( TRACE v3_03_00 ) ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
 +find_ups_product( TRACE v3_03_00 )
  
  
  string(REGEX MATCHALL "[0-9][0-9]?[0-9]?" MESSAGEFACILITY_VERSION_LIST "$ENV{MESSAGEFACILITY_VERSION}")
@@@ -75,19 -76,19 +76,19 @@@ if(BTYPE_UC STREQUAL DEBUG
  		if($ENV{USE_ASAN} GREATER 0)
  			message("!!! Address Sanitizer Activated!!!")
  			add_compile_options(-fsanitize=address -Wno-unused-command-line-argument)
- 			add_link_options(-lasan)
 -			add_link_options(-lasan) ### MIGRATE-ACTION-RECOMMENDED: use target_link_options()
++			add_link_options(-lasan) 
  		endif()
  	endif()
  	if(DEFINED ENV{USE_TSAN})
  		if($ENV{USE_TSAN} GREATER 0)
  			message("!!! Thread Sanitizer Activated!!!")
  			add_compile_options(-fsanitize=thread -Wno-unused-command-line-argument)
- 			add_link_options(-ltsan)
 -			add_link_options(-ltsan) ### MIGRATE-ACTION-RECOMMENDED: use target_link_options()
++			add_link_options(-ltsan) 
  		endif()
  	endif()
  	if(DEFINED ENV{USE_GCOV})
  		add_compile_options(-fprofile-arcs -ftest-coverage --coverage)
- 		add_link_options(--coverage)
 -		add_link_options(--coverage) ### MIGRATE-ACTION-RECOMMENDED: use target_link_options()
++		add_link_options(--coverage) 
  	endif()
  endif()
  
@@@ -98,7 -99,7 +99,7 @@@ if(DEFINED ENV{USE_EXCEPTIONSTACKTRACE}
  		add_definitions(-D EXCEPTIONSTACKTRACE)
  		set( EXCEPTIONSTACKTRACE_LIB  artdaq-core_Utilities_ExceptionStackTrace )
  		add_compile_options( -rdynamic -Wno-unused-command-line-argument )
- 		add_link_options( -rdynamic )
 -		add_link_options( -rdynamic ) ### MIGRATE-ACTION-RECOMMENDED: use target_link_options()
++		add_link_options( -rdynamic ) 
  	endif()
  endif()
  
diff --cc ups/setup_for_development
index d705c1b,f1ddd2e..0000000
--- a/ups/setup_for_development
+++ b/ups/setup_for_development
diff --git a/artdaq-core/Core/CMakeLists.txt b/artdaq-core/Core/CMakeLists.txt
index 504f4ee..e2d553d 100644
--- a/artdaq-core/Core/CMakeLists.txt
+++ b/artdaq-core/Core/CMakeLists.txt
@@ -7,10 +7,10 @@
 # ----------------------------------------------------------------------
 # Build this project's library:
 
-cet_make( ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
+cet_make( 
   LIBRARIES
-  artdaq-core_Data
-  artdaq-core_Utilities
+  artdaq_core::artdaq-core_Data
+  artdaq_core::artdaq-core_Utilities
   pthread
   Boost::thread
   )
diff --git a/artdaq-core/Data/CMakeLists.txt b/artdaq-core/Data/CMakeLists.txt
index 33f83cd..824aa13 100644
--- a/artdaq-core/Data/CMakeLists.txt
+++ b/artdaq-core/Data/CMakeLists.txt
@@ -1,7 +1,7 @@
-cet_make( ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
+cet_make( 
 NO_DICTIONARY
   LIBRARIES
-  artdaq-core_Utilities
+  artdaq_core::artdaq-core_Utilities
   EXCLUDE Artdaq_fragmentNameHelper.cc
   )
 
@@ -9,7 +9,7 @@ basic_plugin(Artdaq "fragmentNameHelper" artdaq-core_Data)
 
 cet_remove_compiler_flags(-fsanitize=address -lasan)
 cet_remove_compiler_flags(-fsanitize=thread -ltsan)
-  art_dictionary(DICTIONARY_LIBRARIES artdaq-core_Data)
+  art_dictionary(DICTIONARY_LIBRARIES artdaq_core::artdaq-core_Data)
 
 install_headers(SUBDIRS detail)
 install_source(SUBDIRS detail)
diff --git a/artdaq-core/Generators/CMakeLists.txt b/artdaq-core/Generators/CMakeLists.txt
index 4222def..d53b677 100644
--- a/artdaq-core/Generators/CMakeLists.txt
+++ b/artdaq-core/Generators/CMakeLists.txt
@@ -1,5 +1,5 @@
 
-cet_make( ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
+cet_make( 
   LIBRARIES
   artdaq_core::artdaq-core_Data
   )
diff --git a/artdaq-core/Utilities/CMakeLists.txt b/artdaq-core/Utilities/CMakeLists.txt
index b625426..cd022aa 100644
--- a/artdaq-core/Utilities/CMakeLists.txt
+++ b/artdaq-core/Utilities/CMakeLists.txt
@@ -1,5 +1,5 @@
 
-cet_make( ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
+cet_make( 
 	EXCLUDE GenFile_mfPlugin.cc ExceptionStackTrace.cc
   LIBRARIES
   messagefacility::MF_MessageLogger
@@ -20,7 +20,7 @@ cet_make_library(
 
 basic_plugin( "GenFile" "mfPlugin"
   messagefacility::MF_MessageLogger
-  cetlib:: cetlib
+  cetlib::cetlib
   cetlib_except::cetlib_except
   fhiclcpp::fhiclcpp
 )
diff --git a/test/Core/CMakeLists.txt b/test/Core/CMakeLists.txt
index 563271e..c89c1b8 100644
--- a/test/Core/CMakeLists.txt
+++ b/test/Core/CMakeLists.txt
@@ -2,11 +2,11 @@
 if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
 
 cet_test(SharedMemoryManager_t USE_BOOST_UNIT
-  LIBRARIES artdaq-core_Core
+  LIBRARIES artdaq_core::artdaq-core_Core
   )
 cet_test(SharedMemoryFragmentManager_t USE_BOOST_UNIT
-  LIBRARIES artdaq-core_Core 
-  artdaq-core_Data
+  LIBRARIES artdaq_core::artdaq-core_Core 
+  artdaq_core::artdaq-core_Data
   )
 
-  endif()
\ No newline at end of file
+  endif()
diff --git a/test/Data/CMakeLists.txt b/test/Data/CMakeLists.txt
index bffd58e..a8ff6c1 100644
--- a/test/Data/CMakeLists.txt
+++ b/test/Data/CMakeLists.txt
@@ -1,20 +1,20 @@
 
 cet_test(Fragment_t USE_BOOST_UNIT
-  LIBRARIES artdaq-core_Data
+  LIBRARIES artdaq_core::artdaq-core_Data
   )
 
 cet_test(ContainerFragment_t USE_BOOST_UNIT
-  LIBRARIES artdaq-core_Data
+  LIBRARIES artdaq_core::artdaq-core_Data
   )
 
 cet_test(RawEvent_t USE_BOOST_UNIT
-  LIBRARIES artdaq-core_Data
+  LIBRARIES artdaq_core::artdaq-core_Data
   )
 
 cet_test(BuildInfo_t USE_BOOST_UNIT
-  LIBRARIES artdaq-core_Data
+  LIBRARIES artdaq_core::artdaq-core_Data
   )
 
 cet_test(FragmentNameHelper_t USE_BOOST_UNIT
-  LIBRARIES artdaq-core_Data
-  )
\ No newline at end of file
+  LIBRARIES artdaq_core::artdaq-core_Data
+  )
diff --git a/test/Generators/CMakeLists.txt b/test/Generators/CMakeLists.txt
index 059e1af..ee64902 100644
--- a/test/Generators/CMakeLists.txt
+++ b/test/Generators/CMakeLists.txt
@@ -1,4 +1,4 @@
 
 cet_test(FragmentGenerator_t USE_BOOST_UNIT
-  LIBRARIES artdaq-core_Generators artdaq-core_Data
+  LIBRARIES artdaq_core::artdaq-core_Generators artdaq_core::artdaq-core_Data
   )
diff --git a/test/Utilities/CMakeLists.txt b/test/Utilities/CMakeLists.txt
index 895801c..60fe360 100644
--- a/test/Utilities/CMakeLists.txt
+++ b/test/Utilities/CMakeLists.txt
@@ -3,28 +3,28 @@ cet_test(TimeUtils_t USE_BOOST_UNIT
 )
 
 cet_test(ExceptionHandler_t USE_BOOST_UNIT
-	LIBRARIES artdaq_core::aartdaq-core_Utilities
+	LIBRARIES artdaq_core::artdaq-core_Utilities
 	cetlib_except::cetlib_except
 	canvas::canvas
 )
 
 cet_test(SimpleLookupPolicy_t USE_BOOST_UNIT
 	DATAFILES fcl/LookupTarget.fcl
-	LIBRARIES artdaq_core::aartdaq-core_Utilities
+	LIBRARIES artdaq_core::artdaq-core_Utilities
   messagefacility::MF_MessageLogger
 )
 
 cet_test(ExceptionStackTrace_t USE_BOOST_UNIT
-	LIBRARIES artdaq_core::aartdaq-core_Utilities_ExceptionStackTrace
+	LIBRARIES artdaq_core::artdaq-core_Utilities_ExceptionStackTrace
   messagefacility::MF_MessageLogger
 )
 
 cet_test(configureMessageFacility_t USE_BOOST_UNIT
-	LIBRARIES artdaq_core::aartdaq-core_Utilities
+	LIBRARIES artdaq_core::artdaq-core_Utilities
   messagefacility::MF_MessageLogger
 )
 
 cet_test(GenFile_t USE_BOOST_UNIT
-	LIBRARIES artdaq_core::aartdaq-core_Utilities
+	LIBRARIES artdaq_core::artdaq-core_Utilities
  messagefacility::MF_MessageLogger
 )
