diff --git a/CMakeLists.txt b/CMakeLists.txt
index ab80715..95cc85f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,28 +13,21 @@
 # use cmake 3.14 or later
 cmake_minimum_required (VERSION 3.12...3.14 FATAL_ERROR)
 
-project(larsoftobj LANGUAGES NONE)
+project(larsoftobj VERSION 08.26.07 LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
-IF (NOT CETBUILDTOOLS_VERSION)
-    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
-ENDIF()
-
-set(CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules
-		      $ENV{CETBUILDTOOLS_DIR}/Modules
-		      ${CMAKE_MODULE_PATH})
+find_package(cetmodules)
 
 include(CetCMakeEnv)
 cet_cmake_env()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( cetbuildtools )
+find_package(nusimdata REQUIRED)
+find_package(lardataalg REQUIRED)
+find_package(lardataobj REQUIRED)
+find_package(larcorealg REQUIRED)
+find_package(larcoreobj REQUIRED)
 
 # larsoftobj bundle scripts
 add_subdirectory(bundle)
-# ups - table and config files
-add_subdirectory(ups)
 
-# packaging utility
-include(UseCPack)
+add_subdirectory(cmakeConfig)
diff --git a/bundle/CMakeLists.txt b/bundle/CMakeLists.txt
index ab25dc9..3df8b31 100644
--- a/bundle/CMakeLists.txt
+++ b/bundle/CMakeLists.txt
@@ -1,4 +1,5 @@
-
+# if the CETBUILDTOOLS_VERSION environmental variable is defined, use cetbuildtools
+if(DEFINED ENV{CETBUILDTOOLS_VERSION})
 # variables to be used in cmake templates
 macro( create_product_variables productname productversion )
    string(TOUPPER ${productname} PRODUCTNAME_UC )
@@ -36,3 +37,4 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/larsoftobj-cfg.in
   ${CMAKE_CURRENT_BINARY_DIR}/larsoftobj-cfg-${LARSOFTOBJ_DOT_VERSION} @ONLY )
 CONFIGURE_FILE ( ${CMAKE_CURRENT_SOURCE_DIR}/larsoftobj.html.in
   ${CMAKE_CURRENT_BINARY_DIR}/larsoftobj-${LARSOFTOBJ_VERSION}.html @ONLY )
+endif()
diff --git a/cmakeConfig/CMakeLists.txt b/cmakeConfig/CMakeLists.txt
new file mode 100644
index 0000000..11194ed
--- /dev/null
+++ b/cmakeConfig/CMakeLists.txt
@@ -0,0 +1,2 @@
+# Create package configuration and version files.
+cet_cmake_config(CONFIG_FRAGMENTS larsoftobj-fragment.cmake.in)
diff --git a/cmakeConfig/larsoftobj-fragment.cmake.in b/cmakeConfig/larsoftobj-fragment.cmake.in
new file mode 100644
index 0000000..727ca19
--- /dev/null
+++ b/cmakeConfig/larsoftobj-fragment.cmake.in
@@ -0,0 +1,5 @@
+find_package(nusimdata REQUIRED)
+find_package(lardataalg REQUIRED)
+find_package(lardataobj REQUIRED)
+find_package(larcorealg REQUIRED)
+find_package(larcoreobj REQUIRED)
