diff --git a/CMakeLists.txt b/CMakeLists.txt
index 755e1a3..9c9d4f6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,15 +8,13 @@
 # LArSoftObj is an empty product
 # It sets up the larsoftobj ups products (lardataobj, larcoreobj)
 # ======================================================================
-
 cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
 
 project(larsoftobj VERSION 09.12.00.01 LANGUAGES NONE)
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
+find_package(cetmodules REQUIRED)
 
-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules)
+project(larsoftobj VERSION 09.11.00 LANGUAGES NONE)
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -24,7 +22,8 @@ cet_cmake_env()
 # larsoftobj bundle scripts
 add_subdirectory(bundle)
 # ups - table and config files
-add_subdirectory(ups)
+if(WANT_UPS)
+endif()
 
 # packaging utility
-include(UseCPack)
+cet_cmake_config()
diff --git a/bundle/CMakeLists.txt b/bundle/CMakeLists.txt
index d97a151..88fae06 100644
--- a/bundle/CMakeLists.txt
+++ b/bundle/CMakeLists.txt
@@ -1,4 +1,5 @@
-
+# if the CETBUILDTOOLS_VERSION environmental variable is defined, use cetbuildtools ### MIGRATE-ACTION-REQUIRED: remove
+if(DEFINED ENV{CETBUILDTOOLS_VERSION}) ### MIGRATE-ACTION-REQUIRED: remove
 # variables to be used in cmake templates
 macro( create_product_variables productname productversion )
    string(TOUPPER ${productname} PRODUCTNAME_UC )
@@ -65,7 +66,7 @@ create_product_variables( nusimdata v1_24_06 )
 create_product_variables( mrb v5_19_05 )
 create_product_variables( hub         v2_14_2 )
 
-# CMake, cetbuildtools, and cetmodules are special: 
+# CMake, cetbuildtools, and cetmodules are special:  ### MIGRATE-ACTION-REQUIRED: remove
 #  define as many versions as necessary.
 #create_version_variables(cmake LIST v3_17_3 v3_18_2 v3_19_6 v3_20_0)
 create_version_variables(cetbuildtools LIST v8_18_00 v8_18_01 v8_18_04 v8_18_05)
@@ -76,3 +77,4 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/larsoftobj-cfg.in
   ${CMAKE_CURRENT_BINARY_DIR}/larsoftobj-cfg-${LARSOFTOBJ_DOT_VERSION} @ONLY )
 CONFIGURE_FILE ( ${CMAKE_CURRENT_SOURCE_DIR}/larsoftobj.html.in
   ${CMAKE_CURRENT_BINARY_DIR}/larsoftobj-${LARSOFTOBJ_VERSION}.html @ONLY )
+endif()
