diff --git a/CMakeLists.txt b/CMakeLists.txt
index c38c2b6..4eaeeac 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,22 +11,20 @@
 #  make install
 #  make package (builds distribution tarfile)
 # ======================================================================
+# use cmake 3.12-3.14
+cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
+cmake_policy(VERSION 3.14)
 
 
-# use cmake 3.12-3.14
-cmake_minimum_required (VERSION 3.12...3.14 FATAL_ERROR)
 
-project(icarusalg LANGUAGES CXX)
+find_package(cetmodules 2.25.05 REQUIRED)
+project(icarusalg VERSION 09.28.01 LANGUAGES CXX)
 
 message(STATUS
   "\n-- ============================================================================="
   "\n-- === *** Start of configuration of ${PROJECT_NAME} ***"
   "\n-- ===")
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
-
-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules)
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -37,18 +35,23 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
   EXTRA_FLAGS -pedantic -I $ENV{CLANG_FQ_DIR}/include/c++/v1
 )
 
-cet_report_compiler_flags()
+cet_report_compiler_flags(REPORT_THRESHOLD VERBOSE)
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( larcorealg )
-find_ups_product( larcoreobj )
-find_ups_product( lardataalg )
-find_ups_product( lardataobj )
-find_ups_product( nusimdata  )
-find_ups_product( canvas     )
-find_ups_product( guideline_sl )
-find_ups_root()
-
+find_package( larcorealg )
+find_package( larcoreobj )
+find_package( lardataalg )
+find_package( lardataobj )
+find_package( nusimdata  )
+find_package( celib )
+find_package( cetlib_except )
+find_package( canvas     )
+find_package( canvas_root_io )
+find_package( messagefacility )
+find_package( guideline_sl )
+find_package( ROOT )
+find_package( Boost COMPONENTS unit_test_framework)
+find_package( CLHEP )
 # macros for dictionary and simple_plugin
 include(ArtDictionary)
 include(CetMake)
@@ -65,10 +68,8 @@ add_subdirectory(test)
 add_subdirectory(fcl)
 
 # ups - table and config files
-add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
 
 message(STATUS
   "\n-- ==="
@@ -77,3 +78,4 @@ message(STATUS
   "\n-- "
   )
 
+cet_cmake_config()
diff --git a/icarusalg/Geometry/CMakeLists.txt b/icarusalg/Geometry/CMakeLists.txt
index 0413d95..2273741 100644
--- a/icarusalg/Geometry/CMakeLists.txt
+++ b/icarusalg/Geometry/CMakeLists.txt
@@ -2,16 +2,18 @@
 cet_make(
   SUBDIRS "details"
   LIBRARIES
-          larcorealg_Geometry
-          larcorealg_CoreUtils
-          ${MF_MESSAGELOGGER}
-          ${FHICLCPP}
-          cetlib
-          cetlib_except
+          larcorealg::Geometry
+          larcorealg::CoreUtils
+          larcoreobj::SimpleTypesAndConstants
+          messagefacility::MF_MessageLogger
+          fhiclcpp::fhiclcpp
+          cetlib::cetlib
+          cetlib_except::cetlib_except
           ROOT::Core
           ROOT::Physics
           ROOT::Geom
           ROOT::GenVector
+          CLHEP::CLHEP
         )
 
 
diff --git a/icarusalg/Geometry/gdml/GDMLSchema/CMakeLists.txt b/icarusalg/Geometry/gdml/GDMLSchema/CMakeLists.txt
index bbf0035..6851239 100644
--- a/icarusalg/Geometry/gdml/GDMLSchema/CMakeLists.txt
+++ b/icarusalg/Geometry/gdml/GDMLSchema/CMakeLists.txt
@@ -1,3 +1,3 @@
 # install schema files
 file(GLOB gdml_schema *.xsd)
-install( FILES ${gdml_schema} DESTINATION ${${product}_gdml_dir}/GDMLSchema )
+install( FILES ${gdml_schema} DESTINATION ${${CETMODULES_CURRENT_PROJECT_NAME}_GDML_DIR}/GDMLSchema )
diff --git a/icarusalg/Utilities/CMakeLists.txt b/icarusalg/Utilities/CMakeLists.txt
index cb2502d..00d0e93 100644
--- a/icarusalg/Utilities/CMakeLists.txt
+++ b/icarusalg/Utilities/CMakeLists.txt
@@ -1,8 +1,9 @@
-
-cet_make(
+include_directories(${CPPGSL_INC})
+cet_make( ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
   LIBRARIES
-    ${CETLIB_EXCEPT}
-    ${CLHEP}
+    cetlib_except::cetlib_except
+    CLHEP::CLHEP
+    larcorealg::CoreUtils
   )
 
 install_headers()
diff --git a/icarusalg/gallery/CMakeLists.txt b/icarusalg/gallery/CMakeLists.txt
index df3e923..e59f4b9 100644
--- a/icarusalg/gallery/CMakeLists.txt
+++ b/icarusalg/gallery/CMakeLists.txt
@@ -9,10 +9,11 @@ foreach(ExampleDir IN ITEMS galleryAnalysis DetectorActivityRatePlots )
   
   # cet_install_files() will reproduce the intermediate subdirectories from the
   # source list into the destination directory
-  cet_install_files(
-    LIST ${GalleryExamples}
-    DIRNAME examples/gallery/${ExampleDir}
-    )
+## FIXME:  replace with install() as cet_install_files is no longer available
+##  cet_install_files(
+##    LIST ${GalleryExamples}
+##    DIRNAME examples/gallery/${ExampleDir}
+##    )
   
 endforeach()
 
diff --git a/icarusalg/gallery/DetectorActivityRatePlots/C++/CMakeLists.txt b/icarusalg/gallery/DetectorActivityRatePlots/C++/CMakeLists.txt
index 1130558..8c7434b 100644
--- a/icarusalg/gallery/DetectorActivityRatePlots/C++/CMakeLists.txt
+++ b/icarusalg/gallery/DetectorActivityRatePlots/C++/CMakeLists.txt
@@ -51,21 +51,21 @@ project(DetectorActivityRatePlots CXX)
 
 
 # install in the same directory as the source
-set(CMAKE_INSTALL_PREFIX ${CMAKE_CURRENT_SOURCE_DIR})
+set(CMAKE_INSTALL_PREFIX ${CMAKE_CURRENT_SOURCE_DIR}) 
 
 #
 # include directories
 #
-include_directories($ENV{BOOST_INC})
-include_directories($ENV{ROOT_INC} $ENV{CLHEP_INC})
-include_directories($ENV{CETLIB_INC} $ENV{CETLIB_EXCEPT_INC} $ENV{CANVAS_INC} $ENV{CANVAS_ROOT_IO_INC})
-include_directories($ENV{FHICLCPP_INC} $ENV{MESSAGEFACILITY_INC})
-include_directories($ENV{HEP_CONCURRENCY_INC})
-include_directories($ENV{GALLERY_INC})
-include_directories($ENV{NUSIMDATA_INC} $ENV{LARCOREOBJ_INC} $ENV{LARDATAOBJ_INC})
-include_directories($ENV{LARCOREALG_INC} $ENV{LARDATAALG_INC})
+include_directories($ENV{BOOST_INC}) 
+include_directories($ENV{ROOT_INC} $ENV{CLHEP_INC}) 
+include_directories($ENV{CETLIB_INC} $ENV{CETLIB_EXCEPT_INC} $ENV{CANVAS_INC} $ENV{CANVAS_ROOT_IO_INC}) 
+include_directories($ENV{FHICLCPP_INC} $ENV{MESSAGEFACILITY_INC}) 
+include_directories($ENV{HEP_CONCURRENCY_INC}) 
+include_directories($ENV{GALLERY_INC}) 
+include_directories($ENV{NUSIMDATA_INC} $ENV{LARCOREOBJ_INC} $ENV{LARDATAOBJ_INC}) 
+include_directories($ENV{LARCOREALG_INC} $ENV{LARDATAALG_INC}) 
 
-include_directories($ENV{ICARUSALG_INC} $ENV{ICARUSCODE_INC})
+include_directories($ENV{ICARUSALG_INC} $ENV{ICARUSCODE_INC}) 
 
 #
 # link directories and libraries
@@ -77,8 +77,8 @@ set(LARSOFT_LIBDIRS $ENV{CLHEP_LIB_DIR} $ENV{LARCOREALG_LIB} $ENV{LARDATAALG_LIB
 set(SBNDCODE_LIBDIRS $ENV{SBNDCODE_LIB})
 set(ICARUS_LIBDIRS $ENV{ICARUSALG_LIB})
 
-link_directories(${GALLERY_LIBDIRS} ${LARSOFTOBJ_LIBDIRS} ${LARSOFT_LIBDIRS})
-link_directories(${ICARUS_LIBDIRS})
+link_directories(${GALLERY_LIBDIRS} ${LARSOFTOBJ_LIBDIRS} ${LARSOFT_LIBDIRS}) 
+link_directories(${ICARUS_LIBDIRS}) 
 
 set(GALLERY_LIBS MF_MessageLogger fhiclcpp canvas cetlib_except cetlib gallery)
 set(ROOT_CORELIBS Core RIO Net Hist Graf Graf3d Gpad Tree Rint Postscript Matrix Physics MathCore Thread MultiProc pthread)
diff --git a/icarusalg/gallery/MCTruthBase/CMakeLists.txt b/icarusalg/gallery/MCTruthBase/CMakeLists.txt
index c109864..f77dfc6 100644
--- a/icarusalg/gallery/MCTruthBase/CMakeLists.txt
+++ b/icarusalg/gallery/MCTruthBase/CMakeLists.txt
@@ -1,19 +1,24 @@
+cet_find_library( DK2NU_TREE NAMES dk2nuTree PATHS ENV DK2NUDATA_LIB NO_DEFAULT_PATH ) ### MIGRATE-ACTION-RECOMMENDED: use of ENV ... may be UPS-dependent ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
+include_directories( $ENV{DK2NUDATA_INC} ) ### MIGRATE-ACTION-RECOMMENDED: use target_link_directories() with target semantics
+
 cet_enable_asserts()
-cet_make(
+cet_make( ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
   LIBRARIES
-          lardataobj_RecoBase
-          larcorealg_Geometry
-          lardataalg_DetectorInfo
-          nusimdata_SimulationBase
-          canvas
-          ${MF_MESSAGELOGGER}
-          ${FHICLCPP}
-          cetlib
-          cetlib_except
+          lardataobj::RecoBase
+          larcorealg::Geometry
+          larcoreobj::SimpleTypesAndConstants
+          lardataalg::DetectorInfo
+          nusimdata::SimulationBase
+          canvas::canvas
+          messagefacility::MF_MessageLogger
+          fhiclcpp::fhiclcpp
+          cetlib::cetlib
+          cetlib_except::cetlib_except
           ROOT::Core
           ROOT::Physics
           ROOT::Geom
           ROOT::GenVector
+          CLHEP::CLHEP
         )
 
 install_headers()
diff --git a/icarusalg/gallery/galleryAnalysis/C++/CMakeLists.txt b/icarusalg/gallery/galleryAnalysis/C++/CMakeLists.txt
index bcad54a..a7cdfe8 100644
--- a/icarusalg/gallery/galleryAnalysis/C++/CMakeLists.txt
+++ b/icarusalg/gallery/galleryAnalysis/C++/CMakeLists.txt
@@ -50,21 +50,21 @@ project(galleryAnalysis CXX)
 
 
 # install in the same directory as the source
-set(CMAKE_INSTALL_PREFIX ${CMAKE_CURRENT_SOURCE_DIR})
+set(CMAKE_INSTALL_PREFIX ${CMAKE_CURRENT_SOURCE_DIR}) 
 
 #
 # include directories
 #
-include_directories($ENV{BOOST_INC})
-include_directories($ENV{ROOT_INC} $ENV{CLHEP_INC})
-include_directories($ENV{CETLIB_INC} $ENV{CETLIB_EXCEPT_INC} $ENV{CANVAS_INC} $ENV{CANVAS_ROOT_IO_INC})
-include_directories($ENV{FHICLCPP_INC} $ENV{MESSAGEFACILITY_INC})
-include_directories($ENV{HEP_CONCURRENCY_INC})
-include_directories($ENV{GALLERY_INC})
-include_directories($ENV{NUSIMDATA_INC} $ENV{LARCOREOBJ_INC} $ENV{LARDATAOBJ_INC})
-include_directories($ENV{LARCOREALG_INC} $ENV{LARDATAALG_INC})
+include_directories($ENV{BOOST_INC}) 
+include_directories($ENV{ROOT_INC} $ENV{CLHEP_INC}) 
+include_directories($ENV{CETLIB_INC} $ENV{CETLIB_EXCEPT_INC} $ENV{CANVAS_INC} $ENV{CANVAS_ROOT_IO_INC}) 
+include_directories($ENV{FHICLCPP_INC} $ENV{MESSAGEFACILITY_INC}) 
+include_directories($ENV{HEP_CONCURRENCY_INC}) 
+include_directories($ENV{GALLERY_INC}) 
+include_directories($ENV{NUSIMDATA_INC} $ENV{LARCOREOBJ_INC} $ENV{LARDATAOBJ_INC}) 
+include_directories($ENV{LARCOREALG_INC} $ENV{LARDATAALG_INC}) 
 
-include_directories($ENV{ICARUSCODE_INC})
+include_directories($ENV{ICARUSCODE_INC}) 
 
 #
 # link directories and libraries
@@ -75,7 +75,7 @@ set(LARSOFTOBJ_LIBDIRS $ENV{NUSIMDATA_LIB} $ENV{LARCOREOBJ_LIB} $ENV{LARDATAOBJ_
 set(LARSOFT_LIBDIRS $ENV{CLHEP_LIB_DIR} $ENV{LARCOREALG_LIB} $ENV{LARDATAALG_LIB})
 set(ICARUSCODE_LIBDIRS $ENV{ICARUSCODE_LIB})
 
-link_directories(${GALLERY_LIBDIRS} ${LARSOFTOBJ_LIBDIRS} ${LARSOFT_LIBDIRS} ${ICARUSCODE_LIBDIRS})
+link_directories(${GALLERY_LIBDIRS} ${LARSOFTOBJ_LIBDIRS} ${LARSOFT_LIBDIRS} ${ICARUSCODE_LIBDIRS}) 
 
 set(GALLERY_LIBS MF_MessageLogger fhiclcpp canvas cetlib_except cetlib gallery)
 set(ROOT_CORELIBS Core RIO Net Hist Graf Graf3d Gpad Tree Rint Postscript Matrix Physics MathCore Thread MultiProc pthread)
diff --git a/icarusalg/gallery/helpers/python/CMakeLists.txt b/icarusalg/gallery/helpers/python/CMakeLists.txt
index b666725..4400c3e 100644
--- a/icarusalg/gallery/helpers/python/CMakeLists.txt
+++ b/icarusalg/gallery/helpers/python/CMakeLists.txt
@@ -20,5 +20,5 @@ if(nonexes AND exes)
 endif(nonexes AND exes)
 if(nonexes)
   message(STATUS "Non-executable python modules: ${nonexes}")
-  install(FILES ${nonexes} DESTINATION "${product}/${version}/python")
+  install(FILES ${nonexes} DESTINATION "python")
 endif(nonexes)
diff --git a/test/Geometry/CMakeLists.txt b/test/Geometry/CMakeLists.txt
index 215767f..574505b 100644
--- a/test/Geometry/CMakeLists.txt
+++ b/test/Geometry/CMakeLists.txt
@@ -8,14 +8,13 @@
 cet_test(geometry_icarus_test
   SOURCES geometry_icarus_test.cxx
   TEST_ARGS test_geometry_icarus.fcl
-  LIBRARIES icarusalg_Geometry
-            larcorealg_Geometry
-            GeometryTestLib
-            ${MF_MESSAGELOGGER}
-            ${MF_UTILITIES}
-            ${FHICLCPP}
-            cetlib_except
-	    ${ROOT_CORE}
+  LIBRARIES icarusalg::Geometry
+            larcorealg::Geometry
+            larcorealg::GeometryTestLib
+            messagefacility::MF_MessageLogger
+            fhiclcpp::fhiclcpp
+            cetlib_except::cetlib_except
+	    ROOT::Core
 #  OPTIONAL_GROUPS Broken
 )
 
@@ -25,14 +24,13 @@ cet_test(geometry_icarus_test
 cet_test(geometry_iterator_icarus_test
   SOURCES geometry_iterator_icarus_test.cxx
   TEST_ARGS test_geometry_iterators_icarus.fcl
-  LIBRARIES icarusalg_Geometry
-            larcorealg_Geometry
-            GeometryTestLib
-            ${MF_MESSAGELOGGER}
-            ${MF_UTILITIES}
-            ${FHICLCPP}
-            cetlib_except
-	    ${ROOT_CORE}
+  LIBRARIES icarusalg::Geometry
+            larcorealg::GeometryTestLib
+            larcorealg::Geometry
+            messagefacility::MF_MessageLogger
+            fhiclcpp::fhiclcpp
+            cetlib_except::cetlib_except
+	    ROOT::Core
   USE_BOOST_UNIT
 )
 
@@ -41,14 +39,13 @@ cet_test(geometry_iterator_icarus_test
 cet_test(geometry_iterator_loop_icarus_test
   SOURCES geometry_iterator_loop_icarus_test.cxx
   TEST_ARGS test_geometry_iterators_icarus.fcl
-  LIBRARIES icarusalg_Geometry
-            larcorealg_Geometry
-            GeometryTestLib
-            ${MF_MESSAGELOGGER}
-            ${MF_UTILITIES}
-            ${FHICLCPP}
-            cetlib_except
-	    ${ROOT_CORE}
+  LIBRARIES icarusalg::Geometry
+            larcorealg::Geometry
+            larcorealg::GeometryTestLib
+            messagefacility::MF_MessageLogger
+            fhiclcpp::fhiclcpp
+            cetlib_except::cetlib_except
+	    ROOT::Core
 )
 
 
diff --git a/test/PMT/Utilities/WaveformOperations_test.cc b/test/PMT/Utilities/WaveformOperations_test.cc
index 26f5ac1..ff19147 100644
--- a/test/PMT/Utilities/WaveformOperations_test.cc
+++ b/test/PMT/Utilities/WaveformOperations_test.cc
@@ -8,8 +8,7 @@
 
 // Boost libraries
 #define BOOST_TEST_MODULE WaveformOperationsTest
-#include <cetlib/quiet_unit_test.hpp> // BOOST_AUTO_TEST_CASE()
-#include <boost/test/test_tools.hpp> // BOOST_CHECK(), BOOST_CHECK_EQUAL()
+#include <boost/test/unit_test.hpp>
 
 // ICARUS libraries
 #include "icarusalg/Utilities/WaveformOperations.h"
diff --git a/test/Utilities/CMakeLists.txt b/test/Utilities/CMakeLists.txt
index 5a056da..8a71f4f 100644
--- a/test/Utilities/CMakeLists.txt
+++ b/test/Utilities/CMakeLists.txt
@@ -1,23 +1,29 @@
-cet_test(rounding_test USE_BOOST_UNIT)
-cet_test(ChangeMonitor_test USE_BOOST_UNIT)
+include_directories(${CPPGSL_INC})
+add_compile_options(-Wno-narrowing)
+cet_test(rounding_test LIBRARIES cetlib::cetlib USE_BOOST_UNIT)
+cet_test(ChangeMonitor_test LIBRARIES cetlib::cetlib USE_BOOST_UNIT)
 
 cet_test(FastAndPoorGauss_test
   LIBRARIES
-    ${ROOT_MATHCORE}
-    ${ROOT_HIST}
-    ${ROOT_RIO}
-    ${ROOT_CORE}
+    larcorealg::CoreUtils
+    ROOT::Core
+    ROOT::Hist
+    ROOT::RIO
+    ROOT::MathCore
+    cetlib::cetlib
   USE_BOOST_UNIT
   )
 
-cet_test(SampledFunction_test USE_BOOST_UNIT)
+cet_test(SampledFunction_test LIBRARIES cetlib::cetlib larcorealg::CoreUtils  USE_BOOST_UNIT)
 
-cet_test(FixedBins_test USE_BOOST_UNIT)
-cet_test(IntegerRanges_test USE_BOOST_UNIT)
+cet_test(FixedBins_test LIBRARIES cetlib::cetlib USE_BOOST_UNIT)
+cet_test(IntegerRanges_test LIBRARIES cetlib::cetlib larcorealg::CoreUtils USE_BOOST_UNIT)
 
 cet_test(NonRandomCounter_test
   LIBRARIES
-    icarusalg_Utilities
-    ${CLHEP}
+    icarusalg::Utilities
+    CLHEP::CLHEP
+    canvas::canvas
+    cetlib::cetlib
   USE_BOOST_UNIT
   )
diff --git a/test/Utilities/ChangeMonitor_test.cc b/test/Utilities/ChangeMonitor_test.cc
index 22a59e3..3bedae1 100644
--- a/test/Utilities/ChangeMonitor_test.cc
+++ b/test/Utilities/ChangeMonitor_test.cc
@@ -8,8 +8,7 @@
 
 // Boost libraries
 #define BOOST_TEST_MODULE ChangeMonitorTest
-#include <cetlib/quiet_unit_test.hpp> // BOOST_AUTO_TEST_CASE()
-#include <boost/test/test_tools.hpp> // BOOST_CHECK(), BOOST_CHECK_EQUAL()
+#include <boost/test/unit_test.hpp>
 
 // ICARUS libraries
 #include "icarusalg/Utilities/ChangeMonitor.h"
@@ -46,38 +45,38 @@ void documentationTest() {
    */
 
   icarus::ns::util::ChangeMonitor<int> monitor;
-  BOOST_CHECK(!monitor.hasReference());
+  BOOST_CHECK((!monitor.hasReference()));
   
   // first check just establishes the reference
   int var = 0;
   auto&& res1 = monitor(var); // this is also a `update()`, which returns no value
-  BOOST_CHECK(!res1);
-  BOOST_CHECK(monitor.hasReference());
-  BOOST_CHECK_EQUAL(monitor.reference(), var);
+  BOOST_CHECK((!res1));
+  BOOST_CHECK((monitor.hasReference()));
+  BOOST_TEST((monitor.reference() ==  var));
   
   // reference is 0, new value is 1: a change is detected
   auto&& res2 = monitor(1);
-  BOOST_CHECK(res2);
-  BOOST_CHECK_EQUAL(res2.value(), 0);
-  BOOST_CHECK(monitor.hasReference());
-  BOOST_CHECK_EQUAL(monitor.reference(), 1);
+  BOOST_CHECK((!!res2));
+  BOOST_TEST((res2.value() ==  0));
+  BOOST_CHECK((monitor.hasReference()));
+  BOOST_TEST((monitor.reference() ==  1));
   
   var = 5; // this does not change the monitoring
   // reference is now 1, new value is 1: no change is detected
   auto&& res3 = monitor(1);
-  BOOST_CHECK(!res3);
-  BOOST_CHECK(monitor.hasReference());
-  BOOST_CHECK_EQUAL(monitor.reference(), 1);
+  BOOST_CHECK((!res3));
+  BOOST_CHECK((monitor.hasReference()));
+  BOOST_TEST((monitor.reference() ==  1));
   
   bool detected = false;
   if (auto prevVal = monitor(2); prevVal) {
     detected = true;
-    BOOST_CHECK(prevVal);
-    BOOST_CHECK_EQUAL(prevVal.value(), 1);
-    BOOST_CHECK(monitor.hasReference());
-    BOOST_CHECK_EQUAL(monitor.reference(), 2);
+    BOOST_CHECK((!!prevVal));
+    BOOST_TEST((prevVal.value() ==  1));
+    BOOST_CHECK((monitor.hasReference()));
+    BOOST_TEST((monitor.reference() ==  2));
   }
-  BOOST_CHECK(detected);
+  BOOST_CHECK((detected));
  
 } // documentationTest()
 
@@ -88,38 +87,38 @@ void ThreadSafeChangeMonitor_documentationTest() {
   // same test as ChangeMonitor_documentationTest()
   
   icarus::ns::util::ThreadSafeChangeMonitor<int> monitor;
-  BOOST_CHECK(!monitor.hasReference());
+  BOOST_CHECK((!monitor.hasReference()));
   
   // first check just establishes the reference
   int var = 0;
   auto&& res1 = monitor(var); // this is also a `update()`, which returns no value
-  BOOST_CHECK(!res1);
-  BOOST_CHECK(monitor.hasReference());
-  BOOST_CHECK_EQUAL(monitor.reference(), var);
+  BOOST_CHECK((!res1));
+  BOOST_CHECK((monitor.hasReference()));
+  BOOST_TEST((monitor.reference() ==  var));
   
   // reference is 0, new value is 1: a change is detected
   auto&& res2 = monitor(1);
-  BOOST_CHECK(res2);
-  BOOST_CHECK_EQUAL(res2.value(), 0);
-  BOOST_CHECK(monitor.hasReference());
-  BOOST_CHECK_EQUAL(monitor.reference(), 1);
+  BOOST_CHECK((!!res2));
+  BOOST_TEST((res2.value() ==  0));
+  BOOST_CHECK((monitor.hasReference()));
+  BOOST_TEST((monitor.reference() ==  1));
   
   var = 5; // this does not change the monitoring
   // reference is now 1, new value is 1: no change is detected
   auto&& res3 = monitor(1);
-  BOOST_CHECK(!res3);
-  BOOST_CHECK(monitor.hasReference());
-  BOOST_CHECK_EQUAL(monitor.reference(), 1);
+  BOOST_CHECK((!res3));
+  BOOST_CHECK((monitor.hasReference()));
+  BOOST_TEST((monitor.reference() ==  1));
   
   bool detected = false;
   if (auto prevVal = monitor(2); prevVal) {
     detected = true;
-    BOOST_CHECK(prevVal);
-    BOOST_CHECK_EQUAL(prevVal.value(), 1);
-    BOOST_CHECK(monitor.hasReference());
-    BOOST_CHECK_EQUAL(monitor.reference(), 2);
+    BOOST_CHECK((!!prevVal));
+    BOOST_TEST((prevVal.value() ==  1));
+    BOOST_CHECK((monitor.hasReference()));
+    BOOST_TEST((monitor.reference() ==  2));
   }
-  BOOST_CHECK(detected);
+  BOOST_CHECK((detected));
  
 } // ThreadSafeChangeMonitor_documentationTest()
 
diff --git a/test/Utilities/FastAndPoorGauss_test.cc b/test/Utilities/FastAndPoorGauss_test.cc
index f2a2026..c0f576c 100644
--- a/test/Utilities/FastAndPoorGauss_test.cc
+++ b/test/Utilities/FastAndPoorGauss_test.cc
@@ -8,8 +8,7 @@
 
 // Boost libraries
 #define BOOST_TEST_MODULE FastAndPoorGauss
-#include <cetlib/quiet_unit_test.hpp> // BOOST_AUTO_TEST_CASE()
-#include <boost/test/test_tools.hpp> // BOOST_CHECK(), BOOST_CHECK_EQUAL()
+#include <boost/test/unit_test.hpp>
 #include <boost/test/tools/floating_point_comparison.hpp> // BOOST_CHECK_CLOSE()
 
 // ICARUS libraries
diff --git a/test/Utilities/FixedBins_test.cc b/test/Utilities/FixedBins_test.cc
index b2885d2..7f60a42 100644
--- a/test/Utilities/FixedBins_test.cc
+++ b/test/Utilities/FixedBins_test.cc
@@ -9,8 +9,7 @@
 
 // Boost libraries
 #define BOOST_TEST_MODULE FixedBins
-#include <cetlib/quiet_unit_test.hpp> // BOOST_AUTO_TEST_CASE()
-#include <boost/test/test_tools.hpp> // BOOST_CHECK(), BOOST_CHECK_EQUAL(), ...
+#include <boost/test/unit_test.hpp>
 
 // ICARUS libraries
 #include "icarusalg/Utilities/FixedBins.h"
@@ -22,48 +21,48 @@ void BasicTest() {
   icarus::ns::util::FixedBins bins { 2.0, -1.0 }; // initializes doubles
   std::vector<unsigned int> expectedContent;
   
-  BOOST_CHECK_EQUAL(bins.binWidth(), 2.0);
-  BOOST_CHECK_EQUAL(bins.offset(), -1.0);
+  BOOST_TEST((bins.binWidth() ==  2.0));
+  BOOST_TEST((bins.offset() ==  -1.0));
   
-  BOOST_CHECK(bins.empty());
-  BOOST_CHECK(empty(bins));
-  BOOST_CHECK_EQUAL(bins.size(), 0U);
-  BOOST_CHECK_EQUAL(size(bins), 0U);
-  BOOST_CHECK(bins.cbegin() == bins.cend());
-  BOOST_CHECK(bins.begin() == bins.end());
-  BOOST_CHECK_EQUAL(bins.nBins(), 0U);
-  BOOST_CHECK_EQUAL(bins.range(), 0.0);
-  BOOST_CHECK_EQUAL(bins.count(-2), 0U);
-  BOOST_CHECK_EQUAL(bins.count(-1), 0U);
-  BOOST_CHECK_EQUAL(bins.count( 0), 0U);
-  BOOST_CHECK_EQUAL(bins.count( 1), 0U);
-  BOOST_CHECK_EQUAL(bins.count( 2), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-5.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-4.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-3.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-2.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-1.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-0.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+0.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+1.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+2.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+3.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+4.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+5.0), 0U);
-  BOOST_CHECK_EQUAL(bins[-2], 0U);
-  BOOST_CHECK_EQUAL(bins[-1], 0U);
-  BOOST_CHECK_EQUAL(bins[ 0], 0U);
-  BOOST_CHECK_EQUAL(bins[ 1], 0U);
-  BOOST_CHECK_EQUAL(bins[ 2], 0U);
+  BOOST_CHECK((bins.empty()));
+  BOOST_CHECK((empty(bins)));
+  BOOST_TEST((bins.size() ==  0U));
+  BOOST_TEST((size(bins) ==  0U));
+  BOOST_TEST((bins.cbegin() == bins.cend()));
+  BOOST_TEST((bins.begin() == bins.end()));
+  BOOST_TEST((bins.nBins() ==  0U));
+  BOOST_TEST((bins.range() ==  0.0));
+  BOOST_TEST((bins.count(-2) ==  0U));
+  BOOST_TEST((bins.count(-1) ==  0U));
+  BOOST_TEST((bins.count( 0) ==  0U));
+  BOOST_TEST((bins.count( 1) ==  0U));
+  BOOST_TEST((bins.count( 2) ==  0U));
+  BOOST_TEST((bins.countFor(-5.0) ==  0U));
+  BOOST_TEST((bins.countFor(-4.0) ==  0U));
+  BOOST_TEST((bins.countFor(-3.0) ==  0U));
+  BOOST_TEST((bins.countFor(-2.0) ==  0U));
+  BOOST_TEST((bins.countFor(-1.0) ==  0U));
+  BOOST_TEST((bins.countFor(-0.0) ==  0U));
+  BOOST_TEST((bins.countFor(+0.0) ==  0U));
+  BOOST_TEST((bins.countFor(+1.0) ==  0U));
+  BOOST_TEST((bins.countFor(+2.0) ==  0U));
+  BOOST_TEST((bins.countFor(+3.0) ==  0U));
+  BOOST_TEST((bins.countFor(+4.0) ==  0U));
+  BOOST_TEST((bins.countFor(+5.0) ==  0U));
+  BOOST_TEST((bins[-2] ==  0U));
+  BOOST_TEST((bins[-1] ==  0U));
+  BOOST_TEST((bins[ 0] ==  0U));
+  BOOST_TEST((bins[ 1] ==  0U));
+  BOOST_TEST((bins[ 2] ==  0U));
   
   
   // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   bins.add(3.5);
   expectedContent = { 1 };
-  BOOST_CHECK(!bins.empty());
-  BOOST_CHECK(!empty(bins));
-  BOOST_CHECK_EQUAL(bins.size(), 1U);
-  BOOST_CHECK_EQUAL(size(bins), 1U);
+  BOOST_CHECK((!bins.empty()));
+  BOOST_CHECK((!empty(bins)));
+  BOOST_TEST((bins.size() ==  1U));
+  BOOST_TEST((size(bins) ==  1U));
   BOOST_CHECK_EQUAL_COLLECTIONS
     (bins.begin(), bins.end(), expectedContent.begin(), expectedContent.end());
   BOOST_CHECK_EQUAL_COLLECTIONS
@@ -72,43 +71,43 @@ void BasicTest() {
     (begin(bins), end(bins), expectedContent.begin(), expectedContent.end());
   BOOST_CHECK_EQUAL_COLLECTIONS
     (cbegin(bins), cend(bins), expectedContent.cbegin(), expectedContent.cend());
-  BOOST_CHECK_EQUAL(bins.nBins(), 1U);
-  BOOST_CHECK_EQUAL(bins.range(), 2.0);
-  BOOST_CHECK_EQUAL(bins.minBin(), 0);
-  BOOST_CHECK_EQUAL(bins.maxBin(), 0);
-  BOOST_CHECK_EQUAL(bins.min(), +3.0);
-  BOOST_CHECK_EQUAL(bins.max(), +5.0);
-  BOOST_CHECK_EQUAL(bins.count(-2), 0U);
-  BOOST_CHECK_EQUAL(bins.count(-1), 0U);
-  BOOST_CHECK_EQUAL(bins.count( 0), 1U);
-  BOOST_CHECK_EQUAL(bins.count( 1), 0U);
-  BOOST_CHECK_EQUAL(bins.count( 2), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-5.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-4.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-3.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-2.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-1.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-0.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+0.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+1.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+2.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+3.0), 1U);
-  BOOST_CHECK_EQUAL(bins.countFor(+4.0), 1U);
-  BOOST_CHECK_EQUAL(bins.countFor(+5.0), 0U);
-  BOOST_CHECK_EQUAL(bins[-2], 0U);
-  BOOST_CHECK_EQUAL(bins[-1], 0U);
-  BOOST_CHECK_EQUAL(bins[ 0], 1U);
-  BOOST_CHECK_EQUAL(bins[ 1], 0U);
-  BOOST_CHECK_EQUAL(bins[ 2], 0U);
+  BOOST_TEST((bins.nBins() ==  1U));
+  BOOST_TEST((bins.range() ==  2.0));
+  BOOST_TEST((bins.minBin() ==  0));
+  BOOST_TEST((bins.maxBin() ==  0));
+  BOOST_TEST((bins.min() ==  +3.0));
+  BOOST_TEST((bins.max() ==  +5.0));
+  BOOST_TEST((bins.count(-2) ==  0U));
+  BOOST_TEST((bins.count(-1) ==  0U));
+  BOOST_TEST((bins.count( 0) ==  1U));
+  BOOST_TEST((bins.count( 1) ==  0U));
+  BOOST_TEST((bins.count( 2) ==  0U));
+  BOOST_TEST((bins.countFor(-5.0) ==  0U));
+  BOOST_TEST((bins.countFor(-4.0) ==  0U));
+  BOOST_TEST((bins.countFor(-3.0) ==  0U));
+  BOOST_TEST((bins.countFor(-2.0) ==  0U));
+  BOOST_TEST((bins.countFor(-1.0) ==  0U));
+  BOOST_TEST((bins.countFor(-0.0) ==  0U));
+  BOOST_TEST((bins.countFor(+0.0) ==  0U));
+  BOOST_TEST((bins.countFor(+1.0) ==  0U));
+  BOOST_TEST((bins.countFor(+2.0) ==  0U));
+  BOOST_TEST((bins.countFor(+3.0) ==  1U));
+  BOOST_TEST((bins.countFor(+4.0) ==  1U));
+  BOOST_TEST((bins.countFor(+5.0) ==  0U));
+  BOOST_TEST((bins[-2] ==  0U));
+  BOOST_TEST((bins[-1] ==  0U));
+  BOOST_TEST((bins[ 0] ==  1U));
+  BOOST_TEST((bins[ 1] ==  0U));
+  BOOST_TEST((bins[ 2] ==  0U));
   
   
   // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   bins.add(-4.0);
   expectedContent = { 1U, 0U, 0U, 0U, 1U };
-  BOOST_CHECK(!bins.empty());
-  BOOST_CHECK(!empty(bins));
-  BOOST_CHECK_EQUAL(bins.size(), 5U);
-  BOOST_CHECK_EQUAL(size(bins), 5U);
+  BOOST_CHECK((!bins.empty()));
+  BOOST_CHECK((!empty(bins)));
+  BOOST_TEST((bins.size() ==  5U));
+  BOOST_TEST((size(bins) ==  5U));
   BOOST_CHECK_EQUAL_COLLECTIONS
     (bins.begin(), bins.end(), expectedContent.begin(), expectedContent.end());
   BOOST_CHECK_EQUAL_COLLECTIONS
@@ -117,49 +116,49 @@ void BasicTest() {
     (begin(bins), end(bins), expectedContent.begin(), expectedContent.end());
   BOOST_CHECK_EQUAL_COLLECTIONS
     (cbegin(bins), cend(bins), expectedContent.cbegin(), expectedContent.cend());
-  BOOST_CHECK_EQUAL(bins.nBins(), 5U);
-  BOOST_CHECK_EQUAL(bins.range(), 10.0);
-  BOOST_CHECK_EQUAL(bins.minBin(), -4);
-  BOOST_CHECK_EQUAL(bins.maxBin(),  0);
-  BOOST_CHECK_EQUAL(bins.min(), -5.0);
-  BOOST_CHECK_EQUAL(bins.max(), +5.0);
-  BOOST_CHECK_EQUAL(bins.count(-5), 0U);
-  BOOST_CHECK_EQUAL(bins.count(-4), 1U);
-  BOOST_CHECK_EQUAL(bins.count(-3), 0U);
-  BOOST_CHECK_EQUAL(bins.count(-2), 0U);
-  BOOST_CHECK_EQUAL(bins.count(-1), 0U);
-  BOOST_CHECK_EQUAL(bins.count( 0), 1U);
-  BOOST_CHECK_EQUAL(bins.count( 1), 0U);
-  BOOST_CHECK_EQUAL(bins.count( 2), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-5.0), 1U);
-  BOOST_CHECK_EQUAL(bins.countFor(-4.0), 1U);
-  BOOST_CHECK_EQUAL(bins.countFor(-3.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-2.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-1.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-0.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+0.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+1.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+2.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+3.0), 1U);
-  BOOST_CHECK_EQUAL(bins.countFor(+4.0), 1U);
-  BOOST_CHECK_EQUAL(bins.countFor(+5.0), 0U);
-  BOOST_CHECK_EQUAL(bins[-5], 0U);
-  BOOST_CHECK_EQUAL(bins[-4], 1U);
-  BOOST_CHECK_EQUAL(bins[-3], 0U);
-  BOOST_CHECK_EQUAL(bins[-2], 0U);
-  BOOST_CHECK_EQUAL(bins[-1], 0U);
-  BOOST_CHECK_EQUAL(bins[ 0], 1U);
-  BOOST_CHECK_EQUAL(bins[ 1], 0U);
-  BOOST_CHECK_EQUAL(bins[ 2], 0U);
+  BOOST_TEST((bins.nBins() ==  5U));
+  BOOST_TEST((bins.range() ==  10.0));
+  BOOST_TEST((bins.minBin() ==  -4));
+  BOOST_TEST((bins.maxBin() ==   0));
+  BOOST_TEST((bins.min() ==  -5.0));
+  BOOST_TEST((bins.max() ==  +5.0));
+  BOOST_TEST((bins.count(-5) ==  0U));
+  BOOST_TEST((bins.count(-4) ==  1U));
+  BOOST_TEST((bins.count(-3) ==  0U));
+  BOOST_TEST((bins.count(-2) ==  0U));
+  BOOST_TEST((bins.count(-1) ==  0U));
+  BOOST_TEST((bins.count( 0) ==  1U));
+  BOOST_TEST((bins.count( 1) ==  0U));
+  BOOST_TEST((bins.count( 2) ==  0U));
+  BOOST_TEST((bins.countFor(-5.0) ==  1U));
+  BOOST_TEST((bins.countFor(-4.0) ==  1U));
+  BOOST_TEST((bins.countFor(-3.0) ==  0U));
+  BOOST_TEST((bins.countFor(-2.0) ==  0U));
+  BOOST_TEST((bins.countFor(-1.0) ==  0U));
+  BOOST_TEST((bins.countFor(-0.0) ==  0U));
+  BOOST_TEST((bins.countFor(+0.0) ==  0U));
+  BOOST_TEST((bins.countFor(+1.0) ==  0U));
+  BOOST_TEST((bins.countFor(+2.0) ==  0U));
+  BOOST_TEST((bins.countFor(+3.0) ==  1U));
+  BOOST_TEST((bins.countFor(+4.0) ==  1U));
+  BOOST_TEST((bins.countFor(+5.0) ==  0U));
+  BOOST_TEST((bins[-5] ==  0U));
+  BOOST_TEST((bins[-4] ==  1U));
+  BOOST_TEST((bins[-3] ==  0U));
+  BOOST_TEST((bins[-2] ==  0U));
+  BOOST_TEST((bins[-1] ==  0U));
+  BOOST_TEST((bins[ 0] ==  1U));
+  BOOST_TEST((bins[ 1] ==  0U));
+  BOOST_TEST((bins[ 2] ==  0U));
   
   
   // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   bins.add(-4.5);
   expectedContent = { 2U, 0U, 0U, 0U, 1U };
-  BOOST_CHECK(!bins.empty());
-  BOOST_CHECK(!empty(bins));
-  BOOST_CHECK_EQUAL(bins.size(), 5U);
-  BOOST_CHECK_EQUAL(size(bins), 5U);
+  BOOST_CHECK((!bins.empty()));
+  BOOST_CHECK((!empty(bins)));
+  BOOST_TEST((bins.size() ==  5U));
+  BOOST_TEST((size(bins) ==  5U));
   BOOST_CHECK_EQUAL_COLLECTIONS
     (bins.begin(), bins.end(), expectedContent.begin(), expectedContent.end());
   BOOST_CHECK_EQUAL_COLLECTIONS
@@ -168,77 +167,77 @@ void BasicTest() {
     (begin(bins), end(bins), expectedContent.begin(), expectedContent.end());
   BOOST_CHECK_EQUAL_COLLECTIONS
     (cbegin(bins), cend(bins), expectedContent.cbegin(), expectedContent.cend());
-  BOOST_CHECK_EQUAL(bins.nBins(), 5U);
-  BOOST_CHECK_EQUAL(bins.range(), 10.0);
-  BOOST_CHECK_EQUAL(bins.minBin(), -4);
-  BOOST_CHECK_EQUAL(bins.maxBin(),  0);
-  BOOST_CHECK_EQUAL(bins.min(), -5.0);
-  BOOST_CHECK_EQUAL(bins.max(), +5.0);
-  BOOST_CHECK_EQUAL(bins.count(-5), 0U);
-  BOOST_CHECK_EQUAL(bins.count(-4), 2U);
-  BOOST_CHECK_EQUAL(bins.count(-3), 0U);
-  BOOST_CHECK_EQUAL(bins.count(-2), 0U);
-  BOOST_CHECK_EQUAL(bins.count(-1), 0U);
-  BOOST_CHECK_EQUAL(bins.count( 0), 1U);
-  BOOST_CHECK_EQUAL(bins.count( 1), 0U);
-  BOOST_CHECK_EQUAL(bins.count( 2), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-5.0), 2U);
-  BOOST_CHECK_EQUAL(bins.countFor(-4.0), 2U);
-  BOOST_CHECK_EQUAL(bins.countFor(-3.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-2.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-1.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-0.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+0.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+1.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+2.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+3.0), 1U);
-  BOOST_CHECK_EQUAL(bins.countFor(+4.0), 1U);
-  BOOST_CHECK_EQUAL(bins.countFor(+5.0), 0U);
-  BOOST_CHECK_EQUAL(bins[-5], 0U);
-  BOOST_CHECK_EQUAL(bins[-4], 2U);
-  BOOST_CHECK_EQUAL(bins[-3], 0U);
-  BOOST_CHECK_EQUAL(bins[-2], 0U);
-  BOOST_CHECK_EQUAL(bins[-1], 0U);
-  BOOST_CHECK_EQUAL(bins[ 0], 1U);
-  BOOST_CHECK_EQUAL(bins[ 1], 0U);
-  BOOST_CHECK_EQUAL(bins[ 2], 0U);
+  BOOST_TEST((bins.nBins() ==  5U));
+  BOOST_TEST((bins.range() ==  10.0));
+  BOOST_TEST((bins.minBin() ==  -4));
+  BOOST_TEST((bins.maxBin() ==   0));
+  BOOST_TEST((bins.min() ==  -5.0));
+  BOOST_TEST((bins.max() ==  +5.0));
+  BOOST_TEST((bins.count(-5) ==  0U));
+  BOOST_TEST((bins.count(-4) ==  2U));
+  BOOST_TEST((bins.count(-3) ==  0U));
+  BOOST_TEST((bins.count(-2) ==  0U));
+  BOOST_TEST((bins.count(-1) ==  0U));
+  BOOST_TEST((bins.count( 0) ==  1U));
+  BOOST_TEST((bins.count( 1) ==  0U));
+  BOOST_TEST((bins.count( 2) ==  0U));
+  BOOST_TEST((bins.countFor(-5.0) ==  2U));
+  BOOST_TEST((bins.countFor(-4.0) ==  2U));
+  BOOST_TEST((bins.countFor(-3.0) ==  0U));
+  BOOST_TEST((bins.countFor(-2.0) ==  0U));
+  BOOST_TEST((bins.countFor(-1.0) ==  0U));
+  BOOST_TEST((bins.countFor(-0.0) ==  0U));
+  BOOST_TEST((bins.countFor(+0.0) ==  0U));
+  BOOST_TEST((bins.countFor(+1.0) ==  0U));
+  BOOST_TEST((bins.countFor(+2.0) ==  0U));
+  BOOST_TEST((bins.countFor(+3.0) ==  1U));
+  BOOST_TEST((bins.countFor(+4.0) ==  1U));
+  BOOST_TEST((bins.countFor(+5.0) ==  0U));
+  BOOST_TEST((bins[-5] ==  0U));
+  BOOST_TEST((bins[-4] ==  2U));
+  BOOST_TEST((bins[-3] ==  0U));
+  BOOST_TEST((bins[-2] ==  0U));
+  BOOST_TEST((bins[-1] ==  0U));
+  BOOST_TEST((bins[ 0] ==  1U));
+  BOOST_TEST((bins[ 1] ==  0U));
+  BOOST_TEST((bins[ 2] ==  0U));
   
   
   // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
   bins.clear();
-  BOOST_CHECK_EQUAL(bins.binWidth(), 2.0);
-  BOOST_CHECK_EQUAL(bins.offset(), -1.0);
+  BOOST_TEST((bins.binWidth() ==  2.0));
+  BOOST_TEST((bins.offset() ==  -1.0));
   
-  BOOST_CHECK(bins.empty());
-  BOOST_CHECK(empty(bins));
-  BOOST_CHECK_EQUAL(bins.size(), 0U);
-  BOOST_CHECK_EQUAL(size(bins), 0U);
-  BOOST_CHECK(bins.cbegin() == bins.cend());
-  BOOST_CHECK(bins.begin() == bins.end());
-  BOOST_CHECK_EQUAL(bins.nBins(), 0U);
-  BOOST_CHECK_EQUAL(bins.range(), 0.0);
-  BOOST_CHECK_EQUAL(bins.count(-2), 0U);
-  BOOST_CHECK_EQUAL(bins.count(-1), 0U);
-  BOOST_CHECK_EQUAL(bins.count( 0), 0U);
-  BOOST_CHECK_EQUAL(bins.count( 1), 0U);
-  BOOST_CHECK_EQUAL(bins.count( 2), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-5.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-4.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-3.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-2.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-1.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(-0.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+0.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+1.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+2.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+3.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+4.0), 0U);
-  BOOST_CHECK_EQUAL(bins.countFor(+5.0), 0U);
-  BOOST_CHECK_EQUAL(bins[-2], 0U);
-  BOOST_CHECK_EQUAL(bins[-1], 0U);
-  BOOST_CHECK_EQUAL(bins[ 0], 0U);
-  BOOST_CHECK_EQUAL(bins[ 1], 0U);
-  BOOST_CHECK_EQUAL(bins[ 2], 0U);
+  BOOST_CHECK((bins.empty()));
+  BOOST_CHECK((empty(bins)));
+  BOOST_TEST((bins.size() ==  0U));
+  BOOST_TEST((size(bins) ==  0U));
+  BOOST_TEST((bins.cbegin() == bins.cend()));
+  BOOST_TEST((bins.begin() == bins.end()));
+  BOOST_TEST((bins.nBins() ==  0U));
+  BOOST_TEST((bins.range() ==  0.0));
+  BOOST_TEST((bins.count(-2) ==  0U));
+  BOOST_TEST((bins.count(-1) ==  0U));
+  BOOST_TEST((bins.count( 0) ==  0U));
+  BOOST_TEST((bins.count( 1) ==  0U));
+  BOOST_TEST((bins.count( 2) ==  0U));
+  BOOST_TEST((bins.countFor(-5.0) ==  0U));
+  BOOST_TEST((bins.countFor(-4.0) ==  0U));
+  BOOST_TEST((bins.countFor(-3.0) ==  0U));
+  BOOST_TEST((bins.countFor(-2.0) ==  0U));
+  BOOST_TEST((bins.countFor(-1.0) ==  0U));
+  BOOST_TEST((bins.countFor(-0.0) ==  0U));
+  BOOST_TEST((bins.countFor(+0.0) ==  0U));
+  BOOST_TEST((bins.countFor(+1.0) ==  0U));
+  BOOST_TEST((bins.countFor(+2.0) ==  0U));
+  BOOST_TEST((bins.countFor(+3.0) ==  0U));
+  BOOST_TEST((bins.countFor(+4.0) ==  0U));
+  BOOST_TEST((bins.countFor(+5.0) ==  0U));
+  BOOST_TEST((bins[-2] ==  0U));
+  BOOST_TEST((bins[-1] ==  0U));
+  BOOST_TEST((bins[ 0] ==  0U));
+  BOOST_TEST((bins[ 1] ==  0U));
+  BOOST_TEST((bins[ 2] ==  0U));
   
   
   // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
diff --git a/test/Utilities/IntegerRanges_test.cc b/test/Utilities/IntegerRanges_test.cc
index f78067a..b9429c5 100644
--- a/test/Utilities/IntegerRanges_test.cc
+++ b/test/Utilities/IntegerRanges_test.cc
@@ -9,8 +9,7 @@
 
 // Boost libraries
 #define BOOST_TEST_MODULE IntegerRanges
-#include <cetlib/quiet_unit_test.hpp> // BOOST_AUTO_TEST_CASE()
-#include <boost/test/test_tools.hpp> // BOOST_CHECK(), BOOST_CHECK_EQUAL(), ...
+#include <boost/test/unit_test.hpp>
 
 // ICARUS libraries
 #include "icarusalg/Utilities/IntegerRanges.h"
diff --git a/test/Utilities/NonRandomCounter_test.cc b/test/Utilities/NonRandomCounter_test.cc
index c6ee6b2..214bcc4 100644
--- a/test/Utilities/NonRandomCounter_test.cc
+++ b/test/Utilities/NonRandomCounter_test.cc
@@ -8,8 +8,7 @@
 
 // Boost libraries
 #define BOOST_TEST_MODULE NonRandomCounterTest
-#include <cetlib/quiet_unit_test.hpp> // BOOST_AUTO_TEST_CASE()
-#include <boost/test/test_tools.hpp> // BOOST_CHECK(), BOOST_CHECK_EQUAL()
+#include <boost/test/unit_test.hpp>
 
 // ICARUS libraries
 #include "icarusalg/Utilities/NonRandomCounter.h"
diff --git a/test/Utilities/SampledFunction_test.cc b/test/Utilities/SampledFunction_test.cc
index dd76292..67a2764 100644
--- a/test/Utilities/SampledFunction_test.cc
+++ b/test/Utilities/SampledFunction_test.cc
@@ -8,9 +8,8 @@
 
 // Boost libraries
 #define BOOST_TEST_MODULE SampledFunction
-#include <cetlib/quiet_unit_test.hpp> // BOOST_AUTO_TEST_CASE()
-#include <boost/test/test_tools.hpp> // BOOST_CHECK(), BOOST_CHECK_EQUAL()
-#include <boost/test/tools/floating_point_comparison.hpp> // BOOST_CHECK_CLOSE()
+#include <boost/test/unit_test.hpp>
+namespace tt = boost::test_tools;
 
 // ICARUS libraries
 #include "icarusalg/Utilities/SampledFunction.h"
@@ -76,13 +75,14 @@ void IdentityTest() {
   //
   // Query
   //
-  BOOST_CHECK_EQUAL(sampled.size(), nSamples);
-  BOOST_CHECK_EQUAL(sampled.nSubsamples(), nSubsamples);
-  BOOST_CHECK_EQUAL(sampled.lower(), min);
-  BOOST_CHECK_EQUAL(sampled.upper(), max);
-  BOOST_CHECK_CLOSE(sampled.rangeSize(), max - min, 1e-6);
-  BOOST_CHECK_CLOSE(sampled.stepSize(), step, 1e-6);
-  BOOST_CHECK_CLOSE(sampled.substepSize(), substep, 1e-6);
+  auto const close = tt::tolerance(1.e-6);
+  BOOST_TEST(sampled.size() == nSamples);
+  BOOST_TEST(sampled.nSubsamples() == nSubsamples);
+  BOOST_TEST(sampled.lower() == min);
+  BOOST_TEST(sampled.upper() == max);
+  BOOST_TEST(sampled.rangeSize() == max - min, close);
+  BOOST_TEST(sampled.stepSize() == step, close);
+  BOOST_TEST(sampled.substepSize() == substep, close);
 
   for (auto const iSub: util::counter(nSubsamples)) BOOST_TEST_CONTEXT("Subsample: " << iSub)
   {
@@ -103,8 +103,8 @@ void IdentityTest() {
       double const expected_value = identity(expected_x); // I wonder how much
 
       if (bInRange) {
-        BOOST_CHECK_EQUAL(sampled.value(iSample, iSub), expected_value);
-        BOOST_CHECK_EQUAL(*itSample, expected_value);
+        BOOST_TEST(sampled.value(iSample, iSub) == expected_value);
+        BOOST_TEST(*itSample == expected_value);
         BOOST_TEST_MESSAGE("[" << iSample << "] " << *itSample);
         ++itSample;
       }
@@ -116,19 +116,19 @@ void IdentityTest() {
         gsl::index const stepIndex
           = sampled.stepIndex(expected_x_in_the_middle, iSub);
 
-        BOOST_CHECK_EQUAL(sampled.isValidStepIndex(stepIndex), bInRange);
-        BOOST_CHECK_EQUAL(stepIndex, iSample);
+        BOOST_TEST(sampled.isValidStepIndex(stepIndex) == bInRange);
+        BOOST_TEST(stepIndex == iSample);
 
-        BOOST_CHECK_EQUAL
-          (sampled.closestSubsampleIndex(expected_x_in_the_middle), iSub);
+        BOOST_TEST
+          (sampled.closestSubsampleIndex(expected_x_in_the_middle) == iSub);
 
       } // for shift
 
     } // for all samples in the subsample
 
-    BOOST_CHECK(itSample == subSample.end());
+    BOOST_TEST((itSample == subSample.end()), "itSample != subSample.end()");
 
-    BOOST_CHECK(!sampled.isValidStepIndex
+    BOOST_TEST(!sampled.isValidStepIndex
       (sampled.stepIndex(subsampleStart + max - min, iSub))
       );
 
@@ -171,14 +171,14 @@ void ExtendedRangeTest() {
   //
   // Query
   //
-  BOOST_CHECK_EQUAL(sampled.nSubsamples(), nSubsamples);
-  BOOST_CHECK_EQUAL(sampled.lower(), min);
-  BOOST_CHECK_CLOSE(sampled.stepSize(), step, 1e-6);
-  BOOST_CHECK_CLOSE(sampled.substepSize(), substep, 1e-6);
+  BOOST_TEST(sampled.nSubsamples() == nSubsamples);
+  BOOST_TEST(sampled.lower() == min);
+  BOOST_TEST(sampled.stepSize() == step, close);
+  BOOST_TEST(sampled.substepSize() == substep, close);
 
-  BOOST_CHECK_CLOSE(sampled.upper(), expected_max, 1e-6);
-  BOOST_REQUIRE_EQUAL(sampled.size(), expected_nSamples);
-  BOOST_CHECK_CLOSE(sampled.rangeSize(), expected_range, 1e-6);
+  BOOST_TEST(sampled.upper() == expected_max, close);
+  BOOST_TEST_REQUIRE(sampled.size() == expected_nSamples);
+  BOOST_TEST(sampled.rangeSize() == expected_range, close);
 
 
   auto const nSamples = sampled.size();
@@ -202,17 +202,17 @@ void ExtendedRangeTest() {
       double const expected_value = identity(expected_x); // I wonder how much
 
       if (bInRange) {
-        BOOST_CHECK_EQUAL(sampled.value(iSample, iSub), expected_value);
-        BOOST_CHECK_EQUAL(*itSample, expected_value);
+        BOOST_TEST(sampled.value(iSample, iSub) == expected_value);
+        BOOST_TEST(*itSample == expected_value);
         BOOST_TEST_MESSAGE("[" << iSample << "] " << *itSample);
         ++itSample;
       }
 
     } // for all samples in the subsample
 
-    BOOST_CHECK(itSample == subSample.end());
+    BOOST_TEST((itSample == subSample.end()), "itSample != subSample.end()");
 
-    BOOST_CHECK(!sampled.isValidStepIndex
+    BOOST_TEST(!sampled.isValidStepIndex
       (sampled.stepIndex(subsampleStart + expected_max - min, iSub))
       );
 
diff --git a/test/Utilities/rounding_test.cc b/test/Utilities/rounding_test.cc
index ec22293..8b8cf7a 100644
--- a/test/Utilities/rounding_test.cc
+++ b/test/Utilities/rounding_test.cc
@@ -8,8 +8,7 @@
 
 // Boost libraries
 #define BOOST_TEST_MODULE RoundingTest
-#include <cetlib/quiet_unit_test.hpp> // BOOST_AUTO_TEST_CASE()
-#include <boost/test/test_tools.hpp> // BOOST_CHECK(), BOOST_CHECK_EQUAL()
+#include <boost/test/unit_test.hpp>
 
 // ICARUS libraries
 #include "icarusalg/Utilities/rounding.h"
diff --git a/ups/CMakeLists.txt b/ups/CMakeLists.txt
deleted file mode 100644
index fe33cc7..0000000
--- a/ups/CMakeLists.txt
+++ /dev/null
@@ -1,6 +0,0 @@
-
-# create package configuration and version files
-
-process_ups_files()
-
-cet_cmake_config()
diff --git a/ups/product-config.cmake.in b/ups/product-config.cmake.in
deleted file mode 100644
index 4e22a3b..0000000
--- a/ups/product-config.cmake.in
+++ /dev/null
@@ -1,12 +0,0 @@
-
-set( @product@_VERSION @cet_dot_version@ )
-set( @product@_UPS_VERSION @version@ )
-
-@PACKAGE_INIT@
-
-@CONFIG_FIND_UPS_COMMANDS@
-
-@CONFIG_FIND_LIBRARY_COMMANDS@
-
-check_required_components(@product@)
-
diff --git a/ups/product_deps b/ups/product_deps
index 8e5e525..b50f790 100644
--- a/ups/product_deps
+++ b/ups/product_deps
@@ -33,10 +33,14 @@ table_fragment_end
 # Add the dependent product and version
 
 product             version
+<<<<<<< HEAD
 larsoftobj          v09_07_01
+=======
+larsoftobj          v09_07_01_01
+>>>>>>> origin/feature/team_for_art_v3_09
 guideline_sl        v3_1_0
 
-cetbuildtools	    v7_17_01	-	only_for_build
+cetbuildtools	    v8_15_00	-	only_for_build
 end_product_list
 
 # Restore this temporarily...
diff --git a/ups/setup_deps b/ups/setup_deps
deleted file mode 100644
index 72c2e32..0000000
--- a/ups/setup_deps
+++ /dev/null
@@ -1,89 +0,0 @@
-# NO USER-SERVICEABLE PARTS BELOW.
-#
-# There should be as little as possible here,
-# with most of the heavy lifting done by other small scripts
-#
-# When sourcing this file from a script, you may have to tell this
-# source file where it is via the fw_db shell (or env) variable.
-# I.e.:
-#    set fw_db=/some/path; source /some/path/this_file
-# or  fw_db=/some/path; . /some/path/this_file
-
-test $?shell = 1 && set ss=csh || ss=sh
-#echo Shell type is $ss.
-
-# make some things similar. need to use set_ because sh builtin set would hide function set
-# Note: perhaps the trickiest thing in this file is the sh set_ function tries to return the same status
-#       as at the start of the function (which most likely is the result of a back-tick expression
-test "$ss" = csh && alias set_ set && alias vecho_ 'if ($?vv == 1) echo \!*' || eval 'vecho_() { test -n "${vv-}" && echo "$@"; return 0; }'
-test "$ss" =  sh && eval 'set_() { sts=$?;for xx in "$@";do var=`expr "$xx" : "\([^=]*\)"`;val=`expr "$xx" : "[^=]*=\(.*\)"`;eval "$var=\"$val\"";done;return $sts; }'
-test "$ss" =  sh && eval 'setenv() { export $1;eval "$1=\"\${2-}\""; }; source() { file=$1; shift; . $file "$@"; }; unsetenv_() { unset "$@"; }'
-test "$ss" =  sh && eval 'tnotnull() { eval "test -n \"\${$1-}\""; }'                             && eval 'nullout() { "$@" >/dev/null 2>&1; }'
-test "$ss" = csh && alias tnotnull "eval '"'test $?'"\!* -eq 1' && eval '"'test -n "$'"\!*"'"'"'" && alias nullout "\!* >& /dev/null" && alias unsetenv_ unsetenv
-test "$ss" = csh && alias return exit
-
-set_ msg1='ERROR: You MUST setup ups'
-test -z $UPS_DIR && ( echo ""; echo "$msg1"; echo "" ) && return
-
-set_ msg3='ERROR: You MUST specify either -o, -p, or -d'
-set_ msg4='Usage: setup_deps <-d|-o|-p> <qualifier list>'
-test -z "$1" && ( echo ""; echo "$msg4"; echo "$msg3"; echo "" ) && return
-
-# make sure we know where this script lives
-# now if we were passed a path or if not, we can figure it out (maybe if we are interactive)
-#   bash, zsh, ksh and tcsh pass params to a source file, csh does not. but someone will be writing csh scripts
-set_ msg2='Please set shell or env. variable fw_db (to be the path to the framework source code). Fix this and other errors may go away.'
-tnotnull fw_db && set_ db=`sh -c "cd $fw_db >/dev/null 2>&1 && pwd"` && vecho_ 'setting db from fw_db variable' || set_ db=
-
-test -z "$db" && tnotnull BASH_SOURCE && set_ me=`dirname $BASH_SOURCE` && set_ db=`sh -c "cd $me >/dev/null 2>&1 && pwd"` && vecho_ 'setting db via BASH_SOURCE'
-
-# history is applicable only for interactive t/csh
-test -z "$db" -a "$ss" = csh && test $?history = 0 && set history=5  # make sure history 1 works
-test -z "$db" -a "$ss" = csh && set me=`history 1|sed 's/^[ 	0-9:]*//'` && test -n "$me" && set me=`dirname $me[2]` \
-    && set db=`sh -c "cd $me >/dev/null 2>&1 && pwd"` && vecho_ 'setting db via interactive history'
-#echo This script lives in $db
-
-test -z "$db" && echo "$msg2" || vecho_ "db=$db"
-test -n "$db" && setenv CETPKG_SOURCE `dirname $db`
-
-# make sure we know the current directory
-set_ myname=`whoami`
-setenv CETPKG_BUILD /tmp/$myname/setup_deps
-rm -rf $CETPKG_BUILD
-mkdir -p $CETPKG_BUILD
-#echo Build directory is $CETPKG_BUILD
-
-echo The source code directory is $CETPKG_SOURCE
-
-set_ msg5='ERROR: setup of required products has failed'
-
-echo ----------- check this block for errors -----------------------
-set_ setup_fail="false"
-set_ exit_now="false"
-set_ cetb=` grep -e '^[ \t]*cetbuildtools' $CETPKG_SOURCE/ups/product_deps | grep  only_for_build| awk '{ print $1 }' `
-set_ cetv=` grep -e '^[ \t]*cetbuildtools' $CETPKG_SOURCE/ups/product_deps | grep  only_for_build| awk '{ print $2 }' `
-#echo Found $cetb $cetv
-setup -B $cetb $cetv
-test "$?" = 0 || set_ setup_fail="true"
-# now get the rest of the products
-set_ cmd="$CETBUILDTOOLS_DIR/bin/set_dev_products $CETPKG_SOURCE $CETPKG_BUILD $*"
-#echo Ready to run $cmd
-source `$cmd`
-test "$?" = 0 || set_ setup_fail="true"
-#echo "$cmd returned $setup_fail"
-test "$setup_fail" = "true" && echo "$msg5"
-test "$setup_fail" = "true" && set_ exit_now="true"
-test -e "$CETPKG_BUILD/diag_report" && cat $CETPKG_BUILD/diag_report
-echo ----------------------------------------------------------------
-
-test "${exit_now}" = "true" && test "$ss" = csh && unalias tnotnull nullout set_ vecho_ return
-test "${exit_now}" = "true" && unset ss db me thisdir msg1 msg2 msg3 msg4 msg5 setup_fail set_ setenv unsetenv_ tnotnull nullout vecho_
-test "${exit_now}" = "true" && return 1
-
-# final sanity check and report
-source $CETBUILDTOOLS_DIR/bin/set_dep_check_report
-
-# cleanup before exiting
-test "$ss" = csh && unalias tnotnull nullout set_ vecho_ return
-unset ss db me thisdir msg1 msg2 msg3 msg4 msg5 setup_fail
-unset set_ setenv unsetenv_ tnotnull nullout vecho_
diff --git a/ups/setup_for_development b/ups/setup_for_development
index 92c3135..afba3a4 100644
--- a/ups/setup_for_development
+++ b/ups/setup_for_development
@@ -1,94 +1,239 @@
-# NO USER-SERVICEABLE PARTS BELOW.
+########################################################################
+# This script is shell-agnostic, which involves much acrobatics to avoid
+# causing errors in one interpreter or another (*stares in csh*).
 #
-# There should be as little as possible here,
-# with most of the heavy lifting done by other small scripts
+# There should be as little as possible here, with most of the heavy
+# lifting done by other small scripts.
 #
 # When sourcing this file from a script, you may have to tell this
-# source file where it is via the fw_db shell (or env) variable.
-# I.e.:
+# source file where it is via the fw_db shell (or env) variable, i.e.
 #    set fw_db=/some/path; source /some/path/this_file
 # or  fw_db=/some/path; . /some/path/this_file
+#
+# Generated by cetmodules 2.25.05 at Wed Sep 01 20:47:45 CDT
+########################################################################
+
+########################################################################
+# NO USER-SERVICEABLE PARTS BELOW.
+########################################################################
 
-test $?shell = 1 && set ss=csh || ss=sh
-#echo Shell type is $ss.
-
-# make some things similar. need to use set_ because sh builtin set would hide function set
-# Note: perhaps the trickiest thing in this file is the sh set_ function tries to return the same status
-#       as at the start of the function (which most likely is the result of a back-tick expression
-test "$ss" = csh && alias set_ set && alias vecho_ 'if ($?vv == 1) echo \!*' || eval 'vecho_() { test -n "${vv-}" && echo "$@"; return 0; }'
-test "$ss" =  sh && eval 'set_() { sts=$?;for xx in "$@";do var=`expr "$xx" : "\([^=]*\)"`;val=`expr "$xx" : "[^=]*=\(.*\)"`;eval "$var=\"$val\"";done;return $sts; }'
-test "$ss" =  sh && eval 'setenv() { export $1;eval "$1=\"\${2-}\""; }; source() { file=$1; shift; . $file "$@"; }; unsetenv_() { unset "$@"; }'
-test "$ss" =  sh && eval 'tnotnull() { eval "test -n \"\${$1-}\""; }'                             && eval 'nullout() { "$@" >/dev/null 2>&1; }'
-test "$ss" = csh && alias tnotnull "eval '"'test $?'"\!* -eq 1' && eval '"'test -n "$'"\!*"'"'"'" && alias nullout "\!* >& /dev/null" && alias unsetenv_ unsetenv
-test "$ss" = csh && alias return exit
-
-set_ msg1='ERROR: You MUST setup ups'
-test -z $UPS_DIR && ( echo ""; echo "$msg1"; echo "" ) && return
-
-set_ msg3='ERROR: You MUST specify either -o, -p, or -d'
-set_ msg4='Usage: setup_for_development <-d|-o|-p> <qualifier list>'
-test -z "$1" && ( echo ""; echo "$msg4"; echo "$msg3"; echo "" ) && return
-
-# make sure we know where this script lives
-# now if we were passed a path or if not, we can figure it out (maybe if we are interactive)
-#   bash, zsh, ksh and tcsh pass params to a source file, csh does not. but someone will be writing csh scripts
-set_ msg2='Please set shell or env. variable fw_db (to be the path to the framework source code). Fix this and other errors may go away.'
-tnotnull fw_db && set_ db=`sh -c "cd $fw_db >/dev/null 2>&1 && pwd"` && vecho_ 'setting db from fw_db variable' || set_ db=
-
-test -z "$db" && tnotnull BASH_SOURCE && set_ me=`dirname $BASH_SOURCE` && set_ db=`sh -c "cd $me >/dev/null 2>&1 && pwd"` && vecho_ 'setting db via BASH_SOURCE'
-
-# history is applicable only for interactive t/csh
-test -z "$db" -a "$ss" = csh && test $?history = 0 && set history=5  # make sure history 1 works
-test -z "$db" -a "$ss" = csh && set me=`history 1|sed 's/^[ 	0-9:]*//'` && test -n "$me" && set me=`dirname $me[2]` \
-    && set db=`sh -c "cd $me >/dev/null 2>&1 && pwd"` && vecho_ 'setting db via interactive history'
-#echo This script lives in $db
-
-test -z "$db" && echo "$msg2" || vecho_ "db=$db"
-test -n "$db" && setenv CETPKG_SOURCE `dirname $db`
-
-# make sure we know the current directory
-setenv CETPKG_BUILD `pwd`
-#echo Build directory is $CETPKG_BUILD
-
-echo The working build directory is $CETPKG_BUILD
-echo The source code directory is $CETPKG_SOURCE
-
-set_ msg5='ERROR: setup of required products has failed'
-
-echo ----------- check this block for errors -----------------------
-set_ setup_fail="false"
-set_ exit_now="false"
-set_ cetb=` grep -e '^[ \t]*cetbuildtools' $CETPKG_SOURCE/ups/product_deps | grep  only_for_build| awk '{ print $1 }' `
-set_ cetv=` grep -e '^[ \t]*cetbuildtools' $CETPKG_SOURCE/ups/product_deps | grep  only_for_build| awk '{ print $2 }' `
-#echo Found $cetb $cetv
-setup -B $cetb $cetv
-test "$?" = 0 || set_ setup_fail="true"
-# now get the rest of the products
-set_ cmd="$CETBUILDTOOLS_DIR/bin/set_dev_products $CETPKG_SOURCE $CETPKG_BUILD $*"
-#echo Ready to run $cmd
-source `$cmd`
-test "$?" = 0 || set_ setup_fail="true"
-#echo "$cmd returned $setup_fail"
-test "$setup_fail" = "true" && echo "$msg5"
-test "$setup_fail" = "true" && set_ exit_now="true"
-test -e "$CETPKG_BUILD/diag_report" && cat $CETPKG_BUILD/diag_report
-echo ----------------------------------------------------------------
-
-test "${exit_now}" = "true" && test "$ss" = csh && unalias tnotnull nullout set_ vecho_ return
-test "${exit_now}" = "true" && unset ss db me thisdir msg1 msg2 msg3 msg4 msg5 setup_fail set_ setenv unsetenv_ tnotnull nullout vecho_
-test "${exit_now}" = "true" && return 1
-
-# add lib to LD_LIBRARY_PATH
-source $CETBUILDTOOLS_DIR/bin/set_dev_lib
-# add bin to path
-source $CETBUILDTOOLS_DIR/bin/set_dev_bin
-# set FHICL_FILE_PATH
-source $CETBUILDTOOLS_DIR/bin/set_dev_fhicl
-
-# final sanity check and report
-source $CETBUILDTOOLS_DIR/bin/set_dev_check_report
-
-# cleanup before exiting
-test "$ss" = csh && unalias tnotnull nullout set_ vecho_ return
-unset ss db me thisdir msg1 msg2 msg3 msg4 msg5 setup_fail
-unset set_ setenv unsetenv_ tnotnull nullout vecho_
+# Determine shell flavor.
+true && test $?shell = 1 && set _cetpkg_shell_flavor=csh || _cetpkg_shell_flavor=sh
+
+####################################
+# A common set of commands to use to hide as many shell flavor
+# differences as possible.
+
+##################
+# IMPORTANT NOTE:
+#
+# Following extensive tests in multiple shells, I have reached the
+# conclusion that it is not actually possible for a function to mirror
+# the semantics of an assignment with respect to the status code of ``
+# and compound statements joined with "||" or "&&". To handle this, we
+# write a large chunk of shell code to a temporary file in the current
+# directory a.k.a. CETPKG_BUILD with shell variable assignments correct
+# according to the shell flavor, and then source it. This also allows us
+# to deal with the fact that some shells don't pass arguments to sourced
+# scripts *stares harder in csh*.
+#
+# 2020-06-24 CHG.
+##################
+
+# If we're C-flavored, set up the commands using alias.
+test "$_cetpkg_shell_flavor" = csh && \
+  set _cetpkg_nobuild="" && \
+  set _cetpkg_shell_var_set=set\\1 && \
+  alias vecho_ 'if ($?vv == 1) echo \>\> \!*' && \
+  alias unsetenv_ unsetenv && \
+  alias tnotnull "eval '"'test $?'"\!* -eq 1' && eval '"'test -n "$'"\!*"'"'"'" && \
+  alias nullout "\!* >& /dev/null" && \
+  alias __TOP_EOF__ '' && \
+  alias return exit
+
+# For Bourne-flavored shells, use functions.
+test "$_cetpkg_shell_flavor" = sh && \
+  _cetpkg_nobuild="" && \
+  _cetpkg_shell_var_set= && \
+  eval 'vecho_() { test -n "${vv-}" && echo ">> $*"; return 0; }' && \
+  eval 'setenv() { eval "$1=\"\${2-}\"; export $1"; }' && \
+  eval 'unsetenv_() { unset "$@"; }' && \
+  eval 'source() { . "$@"; }' && \
+  eval 'tnotnull() { eval "test -n \"\${$1-}\""; }' && \
+  eval 'nullout() { "$@" >/dev/null 2>&1; }'
+####################################
+
+##################
+# Check for UPS (uses alias for return in C-flavored shells).
+test -z "$UPS_DIR" && \
+  ( echo ""; echo "ERROR: you MUST set up UPS"; echo "" ) && \
+  return 1 || true
+##################
+
+####################################
+# Need to find ourselves to set CETPKG_SOURCE.
+
+# Define empty variables for C-flavor shells to make things easier.
+test "$_cetpkg_shell_flavor" = csh && \
+  set _cetpkg_fw_db= _cetpkg_myself= _cetpkg_mydir=
+
+# If user has set fw_db, identify our top-level source directory now.
+test "$_cetpkg_shell_flavor" = csh && tnotnull fw_db && \
+  set _cetpkg_fw_db=`sh -c "cd $fw_db:q >/dev/null 2>&1 && /bin/pwd -P"` && \
+  vecho_ "found ourselves through fw_db" || true
+test "$_cetpkg_shell_flavor" = sh && tnotnull fw_db && \
+  _cetpkg_fw_db=`sh -c "cd \"$fw_db\" >/dev/null 2>&1 && /bin/pwd -P"` && \
+  vecho_ "found ourselves through fw_db" || true
+
+##################
+# For Bash and zsh, identify ourselves now so we don't have to go
+# through acrobatics later in the sourced script.
+test "$_cetpkg_shell_flavor" = sh && tnotnull BASH_SOURCE && \
+  _cetpkg_myself="$BASH_SOURCE" && test -n "$BASH_SOURCE" && \
+  vecho_ "found ourselves through BASH_SOURCE" || true
+test "$_cetpkg_shell_flavor" = sh -a -z "$_cetpkg_myself" && \
+  tnotnull ZSH_VERSION && eval '_cetpkg_myself="${(%):-%x}"' && \
+  test -n "$_cetpkg_myself" && \
+  vecho_ "found ourselves through ZSH prompt logic" || true
+test "$_cetpkg_shell_flavor" = sh && tnotnull _cetpkg_myself && \
+  _cetpkg_mydir=`dirname "$_cetpkg_myself"` && \
+  _cetpkg_fw_db=`sh -c "cd $_cetpkg_mydir >/dev/null 2>&1 && /bin/pwd -P"` || true
+##################
+
+##################
+# History is applicable only for interactive t/csh.
+test -z "$_cetpkg_fw_db" -a "$_cetpkg_shell_flavor" = csh -a $?history = 0 && \
+  set history=5  || true
+test -z "$_cetpkg_fw_db" -a "$_cetpkg_shell_flavor" = csh && \
+  set _cetpkg_mydir=`history 1|sed -Ee 's/^[[:space:][:digit:]:]*.*\bsource[[:space:]]+//'` && \
+  test -n "$_cetpkg_mydir" && set _cetpkg_mydir=`dirname $_cetpkg_mydir[1]:q` && \
+  set _cetpkg_fw_db=`sh -c "cd $_cetpkg_mydir:q >/dev/null 2>&1 && /bin/pwd -P"` && \
+  vecho_ 'setting db via interactive history' || true
+##################
+
+# Set CETPKG_SOURCE.
+test -n "$_cetpkg_fw_db" && \
+  vecho_ "_cetpkg_fw_db=$_cetpkg_fw_db" && \
+  setenv CETPKG_SOURCE `dirname $_cetpkg_fw_db` || \
+  printf "WARNING: please set shell or env. variable fw_db (to be the path to the\n\
+         framework source code). Fix this and other errors may go away.\n"
+####################################
+
+##################
+# Set CETPKG_BUILD.
+setenv CETPKG_BUILD `/bin/pwd -P`
+##################
+
+########################################################################
+# Write the rest of ourselves to a file and source it to handle status
+# code semantics for variable assignments.
+#
+# Note that even here we have to play a trick or two due to the fact
+# that we have to deal with the difference in the way sh- and csh-flavor
+# shells handle a quoted delimiter word for a HERE document: the latter
+# expect the delimiter to include the quoting; the former do not. This
+# is the reason for the C-flavored __TOP_EOF__ alias, and the explicit
+# return at the end of the sourced script.
+#
+########################################################################
+
+sed -Ee 's&\b__SET__([[:space:]]+)&'$_cetpkg_shell_var_set'&g' \
+  -e 's&__SETUP_ARGS__&'"$*"'&g' \
+  > ./setup_common.$_cetpkg_shell_flavor <<\__TOP_EOF__
+########################################################################
+
+####################################
+# Personality check: are we setup_deps or setup_for_development?
+#
+# If we're sourcing setup.in directly (e.g. for cetmodules itself) then
+# we select "setup_for_development" behavior, otherwise _cetpkg_nobuild
+# will be set appropriately by CMake's CONFIGURE() command for
+# setup_for_development or setup_deps.
+
+test -z "$_cetpkg_nobuild" -o "$_cetpkg_nobuild" = "--nobuild" || \
+  __SET__ _cetpkg_nobuild="" || true
+####################################
+
+####################################
+# Initial info.
+
+printf "\n";
+test -z "$_cetpkg_nobuild" && \
+  echo "The working build directory is $CETPKG_BUILD" || true
+cat <<EOF
+The source code directory is $CETPKG_SOURCE
+
+---------------------- check this block for errors ---------------------
+EOF
+
+__SET__ _cetpkg_status=""
+####################################
+
+####################################
+# Attempt to find a suitable cetmodules to set up.
+
+__SET__ _cetpkg_cetmsetup=`sed -nE 's&^[[:space:]]*(cetmodules)[[:space:]]+([^[:space:]]+).*\bonly_for_build\b.*$&\1 \2&p' "$CETPKG_SOURCE/ups/product_deps"`
+test -z "_cetpkg_cetmsetup" -a ! -x "$CETPKG_SOURCE/libexec/set_dev_products" && \
+  __SET__ _cetpkg_status=1 && \
+  echo "ERROR: unable to find suitable cetmodules >= 2.00.00" || true
+test -n "$_cetpkg_status" -o -z "$_cetpkg_cetmsetup" || setup -B $_cetpkg_cetmsetup
+test $? != 0 && __SET__ _cetpkg_status=1 && \
+  echo "ERROR: setup of $_cetpkg_cetmsetup has failed" || true
+test -z "$_cetpkg_status$_cetpkg_cetmsetup" && setenv PATH `dropit -sfe "$CETPKG_SOURCE/bin"` && __SET__ _cetpkg_cettop="$CETPKG_SOURCE" || true
+test -z "$_cetpkg_status$_cetpkg_cettop" -a -n "$CETMODULES_DIR" && __SET__ _cetpkg_cettop="$CETMODULES_DIR"
+####################################
+
+####################################
+# Generate the setup script to set up dependencies.
+
+test -n "$_cetpkg_status" || \
+  __SET__ _cetpkg_setup="`${_cetpkg_cettop}/libexec/set_dev_products $_cetpkg_nobuild __SETUP_ARGS__`"
+test $? != 0 -o -z "$_cetpkg_setup" && __SET__ _cetpkg_status=1 && \
+  echo "ERROR: generation of setup script has failed" || true
+####################################
+
+####################################
+# Source the setup script to set up dependencies.
+
+test -n "$_cetpkg_status" || source "$_cetpkg_setup"
+test $? != 0 && __SET__ _cetpkg_status=1 && \
+  echo "ERROR: setup of required products has failed" || true
+__SET__ _cetpkg_setup_errlog="$CETPKG_BUILD/cetpkg_setup.err"
+test -s "$_cetpkg_setup_errlog" && cat "$_cetpkg_setup_errlog" || \
+  rm -f "$_cetpkg_setup_errlog"
+####################################
+
+cat <<EOF
+------------------------------------------------------------------------
+
+EOF
+
+# Final sanity check and report.
+test -n "$_cetpkg_status" -o -n "$_cetpkg_nobuild" || \
+  "$_cetpkg_cettop/libexec/verify_build_environment" "$CETPKG_BUILD"
+test $? != 0 && __SET__ _cetpkg_status=1 && \
+  echo "ERROR: unable to verify build environment" || true
+
+####################################
+# Clean up before exiting.
+
+# Clean up C-flavor aliases and prepare to return^H^H^H^H^H^Hexit.
+test "$_cetpkg_shell_flavor" = csh && \
+  unalias vecho_ unsetenv_ tnotnull nullout __TOP_EOF__ return && \
+  set return=exit || return=return
+
+# Finish cleanup and exit. Note the use of eval so we can unset
+# everything including $return and $_cetpkg_status.
+eval "unset vecho_ setenv unsetenv_ source tnotnull nullout "\
+"_cetpkg_nobuild _cetpkg_shell_var_set "\
+"_cetpkg_fw_db _cetpkg_myself _cetpkg_mydir _cetpkg_cetmsetup "\
+"_cetpkg_cettop _cetpkg_setup _cetpkg_setup_errlog "\
+"_cetpkg_shell_flavor return _cetpkg_status; \
+$return $_cetpkg_status"
+####################################
+
+########################################################################
+\__TOP_EOF__
+__TOP_EOF__
+
+source ./setup_common.$_cetpkg_shell_flavor
