--- icarusalg-09_28_01/CMakeLists.txt	2021-08-18 09:19:54.000000000 -0500
+++ spack-src/CMakeLists.txt	2021-09-10 10:29:27.673444910 -0500
@@ -16,7 +16,7 @@
 # use cmake 3.12-3.14
 cmake_minimum_required (VERSION 3.12...3.14 FATAL_ERROR)
 
-project(icarusalg LANGUAGES CXX)
+project(icarusalg VERSION 09.28.01 LANGUAGES CXX )
 
 message(STATUS
   "\n-- ============================================================================="
@@ -49,6 +49,9 @@
 find_ups_product( guideline_sl )
 find_ups_root()
 
+find_package(CLHEP REQUIRED)
+include_directories(${CLHEP_INCLUDE_DIRS})
+
 # macros for dictionary and simple_plugin
 include(ArtDictionary)
 include(CetMake)
@@ -65,7 +68,9 @@
 add_subdirectory(fcl)
 
 # ups - table and config files
+if(WANT_UPS)
 add_subdirectory(ups)
+endif()
 
 # packaging utility
 include(UseCPack)
--- spack-src/icarusalg/gallery/CMakeLists.txt.orig	2021-09-15 10:59:43.056913908 -0500
+++ spack-src/icarusalg/gallery/CMakeLists.txt	2021-09-15 11:05:53.774202574 -0500
@@ -2,19 +2,11 @@
 add_subdirectory(helpers)
 
 # examples
-foreach(ExampleDir IN ITEMS galleryAnalysis DetectorActivityRatePlots )
-  
-  # take *all* the files from the listed directory
-  file(GLOB_RECURSE GalleryExamples RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} ${ExampleDir}/* )
-  
-  # cet_install_files() will reproduce the intermediate subdirectories from the
-  # source list into the destination directory
-  cet_install_files(
-    LIST ${GalleryExamples}
-    DIRNAME examples/gallery/${ExampleDir}
+install(
+    DIRECTORY galleryAnalysis DetectorAnalysis
+    DESTINATION examples/gallery
+    USE_SOURCE_PERMISSIONS
     )
-  
-endforeach()
 
 # other stuff
 add_subdirectory(MCTruthBase)
