diff --git a/cmakeConfig/CMakeLists.txt b/cmakeConfig/CMakeLists.txt
new file mode 100644
index 0000000..f26fd62
--- /dev/null
+++ b/cmakeConfig/CMakeLists.txt
@@ -0,0 +1,3 @@
+
+# create package configuration and version files
+cet_cmake_config(CONFIG_FRAGMENTS larexamples-fragment.cmake.in)
diff --git a/cmakeConfig/larexamples-fragment.cmake.in b/cmakeConfig/larexamples-fragment.cmake.in
new file mode 100644
index 0000000..23d09c7
--- /dev/null
+++ b/cmakeConfig/larexamples-fragment.cmake.in
@@ -0,0 +1,4 @@
+find_package( ROOT REQUIRED COMPONENTS Core Physics Hist MathMore RIO GenVector Geom Minuit Minuit2 XMLIO Gdml TMVA EG)
+set(HAVE_ROOT6 BOOL:True)
+set(ROOTCLING ROOT::rootcling)
+find_package( larsim REQUIRED )
diff --git a/CMakeLists.txt b/CMakeLists.txt
index e27b298..1ee8e1d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,17 +10,10 @@
 # use cmake 3.14 or later
 cmake_minimum_required (VERSION 3.12...3.14 FATAL_ERROR)
 
-project(larexamples LANGUAGES CXX)
+project(larexamples VERSION 08.06.13 LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
-IF (NOT CETBUILDTOOLS_VERSION)
-    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
-ENDIF()
-
-set(CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules
-		      $ENV{CETBUILDTOOLS_DIR}/Modules
-		      ${CMAKE_MODULE_PATH})
+find_package( cetmodules )
+set(fcl_dir job)
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -34,18 +27,10 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( larcoreobj )
-find_ups_product( larcorealg )
-find_ups_product( larcore )
-find_ups_product( lardataobj )
-find_ups_product( lardataalg )
-find_ups_product( lardata )
-find_ups_product( larevt )
-find_ups_product( larsim )
-find_ups_product( nutools )
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product( cetbuildtools )
+find_package( ROOT REQUIRED COMPONENTS Core Physics Hist MathMore RIO GenVector Geom Minuit Minuit2 XMLIO Gdml TMVA EG Tree)
+set(HAVE_ROOT6 BOOL:True)
+set(ROOTCLING ROOT::rootcling)
+find_package( larsim REQUIRED )
 
 # macros for dictionary and simple_plugin
 include(ArtDictionary)
@@ -55,11 +40,11 @@ include(BuildPlugins)
 # source
 add_subdirectory(larexamples)
 add_subdirectory(test)
-
+add_subdirectory(cmakeConfig)
 # ups - table and config files
-add_subdirectory(ups)
+#add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
+#include(UseCPack)
 
 
diff --git a/larexamples/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt b/larexamples/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt
index 09e75c3..33914c0 100644
--- a/larexamples/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt
+++ b/larexamples/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt
@@ -2,7 +2,7 @@ art_make(
   LIB_LIBRARIES
     larcorealg_Geometry
     cetlib_except
-    ${ROOT_CORE}
+    ROOT::Core
   MODULE_LIBRARIES
     larexamples_Algorithms_RemoveIsolatedSpacePoints
     ${ART_FRAMEWORK_SERVICES_REGISTRY}
diff --git a/larexamples/Algorithms/TotallyCheatTracks/CMakeLists.txt b/larexamples/Algorithms/TotallyCheatTracks/CMakeLists.txt
index 9b4de64..27d33ee 100644
--- a/larexamples/Algorithms/TotallyCheatTracks/CMakeLists.txt
+++ b/larexamples/Algorithms/TotallyCheatTracks/CMakeLists.txt
@@ -2,12 +2,12 @@ add_subdirectory(CheatTrackData)
 
 art_make(
   LIB_LIBRARIES
-    ${ROOT_CORE}
-    ${ROOT_PHYSICS}
+    ROOT::Core
+    ROOT::Physics
     lardataobj_RecoBase
   MODULE_LIBRARIES
     ${MF_MESSAGELOGGER}
-    ${ROOT_CORE}
+    ROOT::Core
     larexamples_Algorithms_TotallyCheatTracks
     nusimdata_SimulationBase
   )
diff --git a/larexamples/AnalysisExample/CMakeLists.txt b/larexamples/AnalysisExample/CMakeLists.txt
index b1b3ad1..26fed17 100644
--- a/larexamples/AnalysisExample/CMakeLists.txt
+++ b/larexamples/AnalysisExample/CMakeLists.txt
@@ -57,10 +57,10 @@ simple_plugin(AnalysisExample "module"
               ${ART_ROOT_IO_TFILESERVICE_SERVICE}
               ${ART_ROOT_IO_TFILE_SUPPORT}
               ${MF_MESSAGELOGGER}
-              ${ROOT_CORE}
-              ${ROOT_HIST}
-              ${ROOT_PHYSICS}
-              ${ROOT_TREE}
+              ROOT::Core
+              ROOT::Hist
+              ROOT::Physics
+              ROOT::Tree
               larcorealg_Geometry
               nusimdata_SimulationBase
 )
diff --git a/larexamples/RecoProxyUsageExample/CMakeLists.txt b/larexamples/RecoProxyUsageExample/CMakeLists.txt
index 9fd5b2a..9b3cbe1 100644
--- a/larexamples/RecoProxyUsageExample/CMakeLists.txt
+++ b/larexamples/RecoProxyUsageExample/CMakeLists.txt
@@ -1,7 +1,7 @@
 simple_plugin(RecoProxyUsageExample "module"
               ${MF_MESSAGELOGGER}
-              ${ROOT_CORE}
-              ${ROOT_GENVECTOR}
+              ROOT::Core
+              ROOT::GenVector
               lardataobj_RecoBase
 )
 
diff --git a/larexamples/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt b/larexamples/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt
index be55a86..480f054 100644
--- a/larexamples/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt
+++ b/larexamples/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt
@@ -1,9 +1,9 @@
 art_make(
   LIB_LIBRARIES
-    ${ROOT_CORE}
-    ${ROOT_HIST}
-    ${ROOT_MATHMORE}
-    ${ROOT_RIO}
+    ROOT::Core
+    ROOT::Hist
+    ROOT::MathMore
+    ROOT::RIO
     cetlib
     cetlib_except
     lardataobj_RecoBase
diff --git a/test/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt b/test/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt
index cbc2864..a4fecf5 100644
--- a/test/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt
+++ b/test/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt
@@ -5,7 +5,7 @@ art_make(
     PointIsolationAlgStress_test.cc
   LIB_LIBRARIES
     lardataobj_RecoBase
-    ${ROOT_CORE}
+    ROOT::Core
   MODULE_LIBRARIES
     lardataobj_RecoBase
     larcorealg_Geometry
diff --git a/test/Algorithms/TotallyCheatTracks/CMakeLists.txt b/test/Algorithms/TotallyCheatTracks/CMakeLists.txt
index 0126542..4c438de 100644
--- a/test/Algorithms/TotallyCheatTracks/CMakeLists.txt
+++ b/test/Algorithms/TotallyCheatTracks/CMakeLists.txt
@@ -6,8 +6,8 @@ simple_plugin(ParticleMaker "module"
     canvas
     ${MF_MESSAGELOGGER}
     ${FHICLCPP}
-    ${ROOT_CORE}
-    ${ROOT_PHYSICS}
+    ROOT::Core
+    ROOT::Physics
   NO_INSTALL
   )
 
@@ -20,8 +20,8 @@ simple_plugin(DumpCheatTracks "module"
     canvas
     ${MF_MESSAGELOGGER}
     ${FHICLCPP}
-    ${ROOT_CORE}
-    ${ROOT_EG}
+    ROOT::Core
+    ROOT::EG
   USE_BOOST_UNIT
   NO_INSTALL
   )
diff --git a/test/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt b/test/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt
index 4fea9d4..edd3a89 100644
--- a/test/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt
+++ b/test/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt
@@ -2,9 +2,8 @@
 cet_test(CreateTestShowerCalibrationFromPID_test
   SOURCES CreateTestShowerCalibrationFromPID.cxx CreateTestShowerCalibrationFromPID_test.cc
   LIBRARIES
-    ${ROOT_BASIC_LIB_LIST}
-    ${ROOT_CORE}
-    ${ROOT_HIST}
+    ROOT::Hist
+    ${ROOT_LIBRARIES}
   )
 
 cet_test(ShowerCalibrationGaloreFromPID_test
@@ -16,9 +15,8 @@ cet_test(ShowerCalibrationGaloreFromPID_test
     
     fhiclcpp
     cetlib cetlib_except
-    ${ROOT_BASIC_LIB_LIST}
-    ${ROOT_CORE}
-    ${ROOT_HIST}
+    ROOT::Hist
+    ${ROOT_LIBRARIES}
   )
 
 cet_test(ShowerCalibrationGaloreScale_test
@@ -29,9 +27,9 @@ cet_test(ShowerCalibrationGaloreScale_test
     
     fhiclcpp
     cetlib cetlib_except
-    ${ROOT_BASIC_LIB_LIST}
-    ${ROOT_CORE}
-    ${ROOT_HIST}
+    ${ROOT_LIBRARIES}
+    ROOT::Core
+    ROOT::Hist
   )
 
 
diff --git a/test/Services/ShowerCalibrationGalore/Services/CMakeLists.txt b/test/Services/ShowerCalibrationGalore/Services/CMakeLists.txt
index c94ecac..8b31f80 100644
--- a/test/Services/ShowerCalibrationGalore/Services/CMakeLists.txt
+++ b/test/Services/ShowerCalibrationGalore/Services/CMakeLists.txt
@@ -6,7 +6,7 @@ simple_plugin(ShowerCalibrationGaloreTest "module"
   mf_MessageLogger
   fhiclcpp
   cetlib cetlib_except 
-  ${ROOT_BASIC_LIB_LIST}
+  ${ROOT_LIBRARIES}
 )
 
 
