commit 8afd4145f3424a8f72157144a01f8dffb032c0d2
Author: Patrick Gartung <gartung@fnal.gov>
Date:   Sat Apr 4 21:01:11 2020 -0500

    Cetmodules

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f8723d7..cdecae4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,21 +13,13 @@
 # ======================================================================
 
 
-# use cmake 2.8 or later
-cmake_minimum_required (VERSION 2.8)
+# use cmake 3.11 or later
+cmake_minimum_required (VERSION 3.11)
 
-project(larexamples)
-
-# cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
-IF (NOT CETBUILDTOOLS_VERSION)
-    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
-ENDIF()
-
-set(CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules
-		      $ENV{CETBUILDTOOLS_DIR}/Modules
-		      ${CMAKE_MODULE_PATH})
+project(larexamples VERSION 08.00.00 LANGUAGES CXX)
 
+find_package( cetmodules )
+set(fcl_dir job)
 include(CetCMakeEnv)
 cet_cmake_env()
 
@@ -40,18 +32,10 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( larcoreobj )
-find_ups_product( larcorealg )
-find_ups_product( larcore )
-find_ups_product( lardataobj )
-find_ups_product( lardataalg )
-find_ups_product( lardata )
-find_ups_product( larevt )
-find_ups_product( larsim )
-find_ups_product( nutools )
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product( cetbuildtools )
+find_package( ROOT REQUIRED COMPONENTS Core Physics Hist MathMore RIO GenVector Geom Minuit Minuit2 XMLIO Gdml TMVA EG)
+set(HAVE_ROOT6 BOOL:True)
+set(ROOTCLING ROOT::rootcling)
+find_package( larsim REQUIRED )
 
 # macros for dictionary and simple_plugin
 include(ArtDictionary)
@@ -61,11 +45,11 @@ include(BuildPlugins)
 # source
 add_subdirectory(larexamples)
 add_subdirectory(test)
-
+add_subdirectory(cmakeConfig)
 # ups - table and config files
-add_subdirectory(ups)
+#add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
+#include(UseCPack)
 
 
diff --git a/cmakeConfig/CMakeLists.txt b/cmakeConfig/CMakeLists.txt
new file mode 100644
index 0000000..f26fd62
--- /dev/null
+++ b/cmakeConfig/CMakeLists.txt
@@ -0,0 +1,3 @@
+
+# create package configuration and version files
+cet_cmake_config(CONFIG_FRAGMENTS larexamples-fragment.cmake.in)
diff --git a/cmakeConfig/larexamples-fragment.cmake.in b/cmakeConfig/larexamples-fragment.cmake.in
new file mode 100644
index 0000000..23d09c7
--- /dev/null
+++ b/cmakeConfig/larexamples-fragment.cmake.in
@@ -0,0 +1,4 @@
+find_package( ROOT REQUIRED COMPONENTS Core Physics Hist MathMore RIO GenVector Geom Minuit Minuit2 XMLIO Gdml TMVA EG)
+set(HAVE_ROOT6 BOOL:True)
+set(ROOTCLING ROOT::rootcling)
+find_package( larsim REQUIRED )
diff --git a/larexamples/Algorithms/CMakeLists.txt b/larexamples/Algorithms/CMakeLists.txt
index 80e40d1..82b7bf4 100644
--- a/larexamples/Algorithms/CMakeLists.txt
+++ b/larexamples/Algorithms/CMakeLists.txt
@@ -1,5 +1,4 @@
 add_subdirectory(RemoveIsolatedSpacePoints)
 add_subdirectory(TotallyCheatTracks)
 
-install_source()
 
diff --git a/larexamples/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt b/larexamples/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt
index 09e75c3..651a352 100644
--- a/larexamples/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt
+++ b/larexamples/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt
@@ -2,7 +2,7 @@ art_make(
   LIB_LIBRARIES
     larcorealg_Geometry
     cetlib_except
-    ${ROOT_CORE}
+    ROOT::Core
   MODULE_LIBRARIES
     larexamples_Algorithms_RemoveIsolatedSpacePoints
     ${ART_FRAMEWORK_SERVICES_REGISTRY}
@@ -20,4 +20,3 @@ file(GLOB TESTFHICLFILES
 install_source(EXTRAS ${TESTFHICLFILES})
 unset(TESTFHICLFILES)
 
-install_source()
diff --git a/larexamples/Algorithms/TotallyCheatTracks/CMakeLists.txt b/larexamples/Algorithms/TotallyCheatTracks/CMakeLists.txt
index 9b4de64..d4bb352 100644
--- a/larexamples/Algorithms/TotallyCheatTracks/CMakeLists.txt
+++ b/larexamples/Algorithms/TotallyCheatTracks/CMakeLists.txt
@@ -2,12 +2,12 @@ add_subdirectory(CheatTrackData)
 
 art_make(
   LIB_LIBRARIES
-    ${ROOT_CORE}
-    ${ROOT_PHYSICS}
+    ROOT::Core
+    ROOT::Physics
     lardataobj_RecoBase
   MODULE_LIBRARIES
     ${MF_MESSAGELOGGER}
-    ${ROOT_CORE}
+    ROOT::Core
     larexamples_Algorithms_TotallyCheatTracks
     nusimdata_SimulationBase
   )
@@ -23,4 +23,3 @@ file(GLOB TESTFHICLFILES
 install_source(EXTRAS ${TESTFHICLFILES})
 unset(TESTFHICLFILES)
 
-install_source()
diff --git a/larexamples/Algorithms/TotallyCheatTracks/CheatTrackData/CMakeLists.txt b/larexamples/Algorithms/TotallyCheatTracks/CheatTrackData/CMakeLists.txt
index a9e9363..47e7d92 100644
--- a/larexamples/Algorithms/TotallyCheatTracks/CheatTrackData/CMakeLists.txt
+++ b/larexamples/Algorithms/TotallyCheatTracks/CheatTrackData/CMakeLists.txt
@@ -1,5 +1,3 @@
 # this will create both the library and the dictionary
 art_make()
-
-install_source()
 install_headers()
diff --git a/larexamples/AnalysisExample/CMakeLists.txt b/larexamples/AnalysisExample/CMakeLists.txt
index b1b3ad1..8a23e62 100644
--- a/larexamples/AnalysisExample/CMakeLists.txt
+++ b/larexamples/AnalysisExample/CMakeLists.txt
@@ -57,10 +57,10 @@ simple_plugin(AnalysisExample "module"
               ${ART_ROOT_IO_TFILESERVICE_SERVICE}
               ${ART_ROOT_IO_TFILE_SUPPORT}
               ${MF_MESSAGELOGGER}
-              ${ROOT_CORE}
-              ${ROOT_HIST}
-              ${ROOT_PHYSICS}
-              ${ROOT_TREE}
+              ROOT::Core
+              ROOT::Hist
+              ROOT::Physics
+              ROOT::Tree
               larcorealg_Geometry
               nusimdata_SimulationBase
 )
@@ -87,4 +87,3 @@ simple_plugin(AnalysisExample "module"
 install_headers()
 install_fhicl()
 install_scripts()
-install_source()
diff --git a/larexamples/CMakeLists.txt b/larexamples/CMakeLists.txt
index cfb5923..059c97a 100644
--- a/larexamples/CMakeLists.txt
+++ b/larexamples/CMakeLists.txt
@@ -4,4 +4,3 @@ add_subdirectory(Algorithms)
 add_subdirectory(RecoProxyUsageExample)
 add_subdirectory(DebuggingExamples)
 
-install_source()
diff --git a/larexamples/DebuggingExamples/CMakeLists.txt b/larexamples/DebuggingExamples/CMakeLists.txt
index b28342f..c785725 100644
--- a/larexamples/DebuggingExamples/CMakeLists.txt
+++ b/larexamples/DebuggingExamples/CMakeLists.txt
@@ -1,3 +1,2 @@
 add_subdirectory(CatchException)
 
-install_source()
diff --git a/larexamples/DebuggingExamples/CatchException/CMakeLists.txt b/larexamples/DebuggingExamples/CatchException/CMakeLists.txt
index 7eb9135..6788eec 100644
--- a/larexamples/DebuggingExamples/CatchException/CMakeLists.txt
+++ b/larexamples/DebuggingExamples/CatchException/CMakeLists.txt
@@ -1,4 +1,3 @@
 art_make( MODULE_LIBRARIES ${MF_MESSAGELOGGER})
 
-install_source()
 install_fhicl()
diff --git a/larexamples/RecoProxyUsageExample/CMakeLists.txt b/larexamples/RecoProxyUsageExample/CMakeLists.txt
index 9fd5b2a..5776505 100644
--- a/larexamples/RecoProxyUsageExample/CMakeLists.txt
+++ b/larexamples/RecoProxyUsageExample/CMakeLists.txt
@@ -1,11 +1,10 @@
 simple_plugin(RecoProxyUsageExample "module"
               ${MF_MESSAGELOGGER}
-              ${ROOT_CORE}
-              ${ROOT_GENVECTOR}
+              ROOT::Core
+              ROOT::GenVector
               lardataobj_RecoBase
 )
 
 install_headers()
 install_fhicl()
 install_scripts()
-install_source()
diff --git a/larexamples/Services/AtomicNumber/CMakeLists.txt b/larexamples/Services/AtomicNumber/CMakeLists.txt
index b977f01..29df46f 100644
--- a/larexamples/Services/AtomicNumber/CMakeLists.txt
+++ b/larexamples/Services/AtomicNumber/CMakeLists.txt
@@ -3,5 +3,4 @@ add_subdirectory(Services)
 
 install_fhicl()
 install_headers()
-install_source()
 
diff --git a/larexamples/Services/CMakeLists.txt b/larexamples/Services/CMakeLists.txt
index 84c8426..a7113f1 100644
--- a/larexamples/Services/CMakeLists.txt
+++ b/larexamples/Services/CMakeLists.txt
@@ -1,4 +1,3 @@
 add_subdirectory(AtomicNumber)
 add_subdirectory(ShowerCalibrationGalore)
 
-install_source()
diff --git a/larexamples/Services/ShowerCalibrationGalore/CMakeLists.txt b/larexamples/Services/ShowerCalibrationGalore/CMakeLists.txt
index 5e2adab..da2fb5d 100644
--- a/larexamples/Services/ShowerCalibrationGalore/CMakeLists.txt
+++ b/larexamples/Services/ShowerCalibrationGalore/CMakeLists.txt
@@ -1,4 +1,3 @@
 add_subdirectory(Providers)
 add_subdirectory(Services)
 
-install_source()
diff --git a/larexamples/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt b/larexamples/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt
index be55a86..480f054 100644
--- a/larexamples/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt
+++ b/larexamples/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt
@@ -1,9 +1,9 @@
 art_make(
   LIB_LIBRARIES
-    ${ROOT_CORE}
-    ${ROOT_HIST}
-    ${ROOT_MATHMORE}
-    ${ROOT_RIO}
+    ROOT::Core
+    ROOT::Hist
+    ROOT::MathMore
+    ROOT::RIO
     cetlib
     cetlib_except
     lardataobj_RecoBase
diff --git a/test/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt b/test/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt
index cbc2864..4b7d4e8 100644
--- a/test/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt
+++ b/test/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt
@@ -5,7 +5,7 @@ art_make(
     PointIsolationAlgStress_test.cc
   LIB_LIBRARIES
     lardataobj_RecoBase
-    ${ROOT_CORE}
+    ROOT::Core
   MODULE_LIBRARIES
     lardataobj_RecoBase
     larcorealg_Geometry
@@ -18,7 +18,7 @@ art_make(
     art_Persistency_Provenance
     art_Utilities canvas
     ${MF_MESSAGELOGGER}
-    
+    ${ROOT_LIBRARIES} 
     ${FHICLCPP}
   )
 
diff --git a/test/Algorithms/TotallyCheatTracks/CMakeLists.txt b/test/Algorithms/TotallyCheatTracks/CMakeLists.txt
index 0126542..4c438de 100644
--- a/test/Algorithms/TotallyCheatTracks/CMakeLists.txt
+++ b/test/Algorithms/TotallyCheatTracks/CMakeLists.txt
@@ -6,8 +6,8 @@ simple_plugin(ParticleMaker "module"
     canvas
     ${MF_MESSAGELOGGER}
     ${FHICLCPP}
-    ${ROOT_CORE}
-    ${ROOT_PHYSICS}
+    ROOT::Core
+    ROOT::Physics
   NO_INSTALL
   )
 
@@ -20,8 +20,8 @@ simple_plugin(DumpCheatTracks "module"
     canvas
     ${MF_MESSAGELOGGER}
     ${FHICLCPP}
-    ${ROOT_CORE}
-    ${ROOT_EG}
+    ROOT::Core
+    ROOT::EG
   USE_BOOST_UNIT
   NO_INSTALL
   )
diff --git a/test/Services/AtomicNumber/CMakeLists.txt b/test/Services/AtomicNumber/CMakeLists.txt
index 70295c8..da2fb5d 100644
--- a/test/Services/AtomicNumber/CMakeLists.txt
+++ b/test/Services/AtomicNumber/CMakeLists.txt
@@ -1,4 +1,3 @@
 add_subdirectory(Providers)
 add_subdirectory(Services)
 
-install_source(EXTRAS CMakeLists.txt)
diff --git a/test/Services/ShowerCalibrationGalore/CMakeLists.txt b/test/Services/ShowerCalibrationGalore/CMakeLists.txt
index 70295c8..da2fb5d 100644
--- a/test/Services/ShowerCalibrationGalore/CMakeLists.txt
+++ b/test/Services/ShowerCalibrationGalore/CMakeLists.txt
@@ -1,4 +1,3 @@
 add_subdirectory(Providers)
 add_subdirectory(Services)
 
-install_source(EXTRAS CMakeLists.txt)
diff --git a/test/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt b/test/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt
index 4fea9d4..1d21a8f 100644
--- a/test/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt
+++ b/test/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt
@@ -2,9 +2,9 @@
 cet_test(CreateTestShowerCalibrationFromPID_test
   SOURCES CreateTestShowerCalibrationFromPID.cxx CreateTestShowerCalibrationFromPID_test.cc
   LIBRARIES
-    ${ROOT_BASIC_LIB_LIST}
-    ${ROOT_CORE}
-    ${ROOT_HIST}
+    ${ROOT_LIBRARIES}
+    ROOT::Core
+    ROOT::Hist
   )
 
 cet_test(ShowerCalibrationGaloreFromPID_test
@@ -16,9 +16,9 @@ cet_test(ShowerCalibrationGaloreFromPID_test
     
     fhiclcpp
     cetlib cetlib_except
-    ${ROOT_BASIC_LIB_LIST}
-    ${ROOT_CORE}
-    ${ROOT_HIST}
+    ${ROOT_LIBRARIES}
+    ROOT::Core
+    ROOT::Hist
   )
 
 cet_test(ShowerCalibrationGaloreScale_test
@@ -29,9 +29,9 @@ cet_test(ShowerCalibrationGaloreScale_test
     
     fhiclcpp
     cetlib cetlib_except
-    ${ROOT_BASIC_LIB_LIST}
-    ${ROOT_CORE}
-    ${ROOT_HIST}
+    ${ROOT_LIBRARIES}
+    ROOT::Core
+    ROOT::Hist
   )
 
 
diff --git a/test/Services/ShowerCalibrationGalore/Services/CMakeLists.txt b/test/Services/ShowerCalibrationGalore/Services/CMakeLists.txt
index c94ecac..8b31f80 100644
--- a/test/Services/ShowerCalibrationGalore/Services/CMakeLists.txt
+++ b/test/Services/ShowerCalibrationGalore/Services/CMakeLists.txt
@@ -6,7 +6,7 @@ simple_plugin(ShowerCalibrationGaloreTest "module"
   mf_MessageLogger
   fhiclcpp
   cetlib cetlib_except 
-  ${ROOT_BASIC_LIB_LIST}
+  ${ROOT_LIBRARIES}
 )
 
 
