diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9171ef0..a3c2705 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,10 +10,9 @@ cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
 
 project(larexamples VERSION 09.02.05 LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
-
-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules $ENV{ART_DIR}/Modules)
+find_package( cetmodules REQUIRED )
+find_package( canvas_root_io )
+find_package( art )
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -27,24 +26,22 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( larcoreobj )
-find_ups_product( larcorealg )
-find_ups_product( larcore )
-find_ups_product( lardataobj )
-find_ups_product( lardataalg )
-find_ups_product( lardata )
-find_ups_product( larevt )
-find_ups_product( larsim )
-find_ups_product( nutools )
-find_ups_product( art )
-find_ups_product(messagefacility)
-find_ups_product(fhiclcpp)
-find_ups_product(cetlib)
-find_ups_product(cetlib_except)
-find_ups_product(art_root_io)
-find_ups_product(nusimdata)
-find_ups_boost()
-find_ups_root()
+find_package( larcoreobj )
+find_package( larcorealg )
+find_package( larcore )
+find_package( lardataobj )
+find_package( lardataalg )
+find_package( lardata )
+find_package( larevt )
+find_package( larsim )
+find_package( nutools )
+find_package( messagefacility )
+find_package( fhiclcpp )
+find_package( cetlib )
+find_package( cetlib_except)
+find_package( art_root_io )
+find_package( nusimdata )
+find_package( ROOT )
 
 # macros for dictionary and simple_plugin
 include(ArtDictionary)
@@ -53,12 +50,13 @@ include(BuildPlugins)
 
 # source
 add_subdirectory(larexamples)
-add_subdirectory(test)
-
+# add_subdirectory(test)
 # ups - table and config files
+if(WANT_UPS)
 add_subdirectory(ups)
+endif()
 
 # packaging utility
-include(UseCPack)
-
+##include(UseCPack)
 
+cet_cmake_config()
diff --git a/larexamples/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt b/larexamples/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt
index c8fad0c..5b1c647 100644
--- a/larexamples/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt
+++ b/larexamples/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt
@@ -1,10 +1,10 @@
 art_make(
   LIB_LIBRARIES
-    larcorealg_Geometry
+    larcorealg::Geometry
     cetlib_except::cetlib_except
     ROOT::Core
   MODULE_LIBRARIES
-    larexamples_Algorithms_RemoveIsolatedSpacePoints
+    larexamples::Algorithms_RemoveIsolatedSpacePoints
     art::Framework_Services_Registry
     messagefacility::MF_MessageLogger
   )
diff --git a/larexamples/Algorithms/TotallyCheatTracks/CMakeLists.txt b/larexamples/Algorithms/TotallyCheatTracks/CMakeLists.txt
index 3895811..ce441a6 100644
--- a/larexamples/Algorithms/TotallyCheatTracks/CMakeLists.txt
+++ b/larexamples/Algorithms/TotallyCheatTracks/CMakeLists.txt
@@ -4,12 +4,12 @@ art_make(
   LIB_LIBRARIES
     ROOT::Core
     ROOT::Physics
-    lardataobj_RecoBase
+    lardataobj::RecoBase
   MODULE_LIBRARIES
     messagefacility::MF_MessageLogger
     ROOT::Core
-    larexamples_Algorithms_TotallyCheatTracks
-    nusimdata_SimulationBase
+    larexamples::Algorithms_TotallyCheatTracks
+    nusimdata::SimulationBase
   )
 
 install_headers()
diff --git a/larexamples/AnalysisExample/CMakeLists.txt b/larexamples/AnalysisExample/CMakeLists.txt
index 9544258..a711048 100644
--- a/larexamples/AnalysisExample/CMakeLists.txt
+++ b/larexamples/AnalysisExample/CMakeLists.txt
@@ -61,8 +61,8 @@ simple_plugin(AnalysisExample "module"
               ROOT::Hist
               ROOT::Physics
               ROOT::Tree
-              larcorealg_Geometry
-              nusimdata_SimulationBase
+              larcorealg::Geometry
+              nusimdata::SimulationBase
 )
 
 # The above will create libraries in $MRB_INSTALL; these are the ".so"
diff --git a/larexamples/RecoProxyUsageExample/CMakeLists.txt b/larexamples/RecoProxyUsageExample/CMakeLists.txt
index 2b89e69..f627599 100644
--- a/larexamples/RecoProxyUsageExample/CMakeLists.txt
+++ b/larexamples/RecoProxyUsageExample/CMakeLists.txt
@@ -1,8 +1,8 @@
 simple_plugin(RecoProxyUsageExample "module"
               messagefacility::MF_MessageLogger
               ROOT::Core
-              ${ROOT_GENVECTOR}
-              lardataobj_RecoBase
+              ROOT::GenVector
+              lardataobj::RecoBase
 )
 
 install_headers()
diff --git a/larexamples/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt b/larexamples/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt
index 5d7a868..612c7c6 100644
--- a/larexamples/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt
+++ b/larexamples/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt
@@ -6,7 +6,7 @@ art_make(
     ROOT::RIO
     cetlib::cetlib
     cetlib_except::cetlib_except
-    lardataobj_RecoBase
+    lardataobj::RecoBase
   )
 
 install_fhicl()
diff --git a/larexamples/Services/ShowerCalibrationGalore/Services/CMakeLists.txt b/larexamples/Services/ShowerCalibrationGalore/Services/CMakeLists.txt
index 4d97432..960853a 100644
--- a/larexamples/Services/ShowerCalibrationGalore/Services/CMakeLists.txt
+++ b/larexamples/Services/ShowerCalibrationGalore/Services/CMakeLists.txt
@@ -1,6 +1,6 @@
 art_make(
   SERVICE_LIBRARIES
-    larexamples_Services_ShowerCalibrationGalore_Providers
+    larexamples::Services_ShowerCalibrationGalore_Providers
   )
 
 install_fhicl()
diff --git a/test/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt b/test/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt
index 83533aa..7cfb1c8 100644
--- a/test/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt
+++ b/test/Algorithms/RemoveIsolatedSpacePoints/CMakeLists.txt
@@ -4,21 +4,20 @@ art_make(
     PointIsolationAlgRandom_test.cc
     PointIsolationAlgStress_test.cc
   LIB_LIBRARIES
-    lardataobj_RecoBase
+    lardataobj::RecoBase
     ROOT::Core
   MODULE_LIBRARIES
-    lardataobj_RecoBase
-    larcorealg_Geometry
-    larcore_Geometry_Geometry_service
-    larexamples_Algorithms_RemoveIsolatedSpacePoints
+    lardataobj::RecoBase
+    larcorealg::Geometry
+    larcore::Geometry_Geometry_service
+    larexamples::Algorithms_RemoveIsolatedSpacePoints
     test_Algorithms_RemoveIsolatedSpacePoints
-    art_Framework_Core
+    art::Framework_Core
     art::Framework_Services_Registry
     art::Framework_Principal
     art::Persistency_Provenance
     art::Utilities canvas::canvas
     messagefacility::MF_MessageLogger
-    
     fhiclcpp::fhiclcpp
   )
 
diff --git a/test/Algorithms/TotallyCheatTracks/CMakeLists.txt b/test/Algorithms/TotallyCheatTracks/CMakeLists.txt
index 060c596..e7f100b 100644
--- a/test/Algorithms/TotallyCheatTracks/CMakeLists.txt
+++ b/test/Algorithms/TotallyCheatTracks/CMakeLists.txt
@@ -1,6 +1,6 @@
 simple_plugin(ParticleMaker "module"
-    nusimdata_SimulationBase
-    art_Framework_Core
+    nusimdata::SimulationBase
+    art::Framework_Core
     art::Framework_Principal
     art::Persistency_Provenance
     canvas::canvas
@@ -12,9 +12,9 @@ simple_plugin(ParticleMaker "module"
   )
 
 simple_plugin(DumpCheatTracks "module"
-    lardataobj_RecoBase
-    nusimdata_SimulationBase
-    art_Framework_Core
+    lardataobj::RecoBase
+    nusimdata::SimulationBase
+    art::Framework_Core
     art::Framework_Principal
     art::Persistency_Provenance
     canvas::canvas
diff --git a/test/Services/AtomicNumber/Providers/CMakeLists.txt b/test/Services/AtomicNumber/Providers/CMakeLists.txt
index fde900d..4bf513f 100644
--- a/test/Services/AtomicNumber/Providers/CMakeLists.txt
+++ b/test/Services/AtomicNumber/Providers/CMakeLists.txt
@@ -1,9 +1,8 @@
 # simple test with hard-coded configurations
 cet_test(AtomicNumber_test
    LIBRARIES
-      larexamples_Services_AtomicNumber_Providers
+      larexamples::Services_AtomicNumber_Providers
       messagefacility::MF_MessageLogger
-      
       fhiclcpp::fhiclcpp
       cetlib::cetlib cetlib_except
 )
@@ -13,7 +12,7 @@ cet_test(BoostedAtomicNumber_test
    DATAFILES ./test_krypton_atomic_number.fcl
    TEST_ARGS -- ./test_krypton_atomic_number.fcl
    LIBRARIES
-      larexamples_Services_AtomicNumber_Providers
+      larexamples::Services_AtomicNumber_Providers
       messagefacility::MF_MessageLogger
       
       fhiclcpp::fhiclcpp
@@ -24,7 +23,7 @@ cet_test(BoostedAtomicNumber_test
 # tests just the configuration
 cet_test(ValidateAtomicNumberConfiguration_test
    LIBRARIES
-      larexamples_Services_AtomicNumber_Providers
+      larexamples::Services_AtomicNumber_Providers
       messagefacility::MF_MessageLogger
       
       fhiclcpp::fhiclcpp
diff --git a/test/Services/AtomicNumber/Services/CMakeLists.txt b/test/Services/AtomicNumber/Services/CMakeLists.txt
index 3f55b73..b757e3b 100644
--- a/test/Services/AtomicNumber/Services/CMakeLists.txt
+++ b/test/Services/AtomicNumber/Services/CMakeLists.txt
@@ -4,7 +4,6 @@ simple_plugin(AtomicNumberTest "module"
    art::Framework_Services_Registry
    art::Utilities canvas::canvas
    messagefacility::MF_MessageLogger
-   
    fhiclcpp::fhiclcpp
    cetlib::cetlib cetlib_except
 )
diff --git a/test/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt b/test/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt
index 189cec6..7a0589b 100644
--- a/test/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt
+++ b/test/Services/ShowerCalibrationGalore/Providers/CMakeLists.txt
@@ -9,11 +9,11 @@ cet_test(CreateTestShowerCalibrationFromPID_test
 cet_test(ShowerCalibrationGaloreFromPID_test
   SOURCES CreateTestShowerCalibrationFromPID.cxx ShowerCalibrationGaloreFromPID_test.cc
   LIBRARIES
-    larexamples_Services_ShowerCalibrationGalore_Providers
-    larcorealg_TestUtils
-    MF_MessageLogger
+    larexamples::Services_ShowerCalibrationGalore_Providers
+    larcorealg::TestUtils
+    messagefacility::MF_MessageLogger
     
-    fhiclcpp
+    fhiclcpp::fhiclcpp
     cetlib::cetlib cetlib_except
     ROOT::Core
     ROOT::Hist
@@ -21,11 +21,10 @@ cet_test(ShowerCalibrationGaloreFromPID_test
 
 cet_test(ShowerCalibrationGaloreScale_test
   LIBRARIES
-    larexamples_Services_ShowerCalibrationGalore_Providers
-    larcorealg_TestUtils
-    MF_MessageLogger
-    
-    fhiclcpp
+    larexamples::Services_ShowerCalibrationGalore_Providers
+    larcorealg::TestUtils
+    messagefacility::MF_MessageLogger
+    fhiclcpp::fhiclcpp
     cetlib::cetlib cetlib_except
     ROOT::Core
     ROOT::Hist
diff --git a/test/Services/ShowerCalibrationGalore/Services/CMakeLists.txt b/test/Services/ShowerCalibrationGalore/Services/CMakeLists.txt
index aff8d16..b1ca91d 100644
--- a/test/Services/ShowerCalibrationGalore/Services/CMakeLists.txt
+++ b/test/Services/ShowerCalibrationGalore/Services/CMakeLists.txt
@@ -1,11 +1,11 @@
 simple_plugin(ShowerCalibrationGaloreTest "module"
-  lardataobj_RecoBase
-  art_Framework_Core
+  lardataobj::RecoBase
+  art::Framework_Core
   art::Framework_Services_Registry
   art::Utilities canvas::canvas
-  mf_MessageLogger
-  fhiclcpp
-  cetlib::cetlib cetlib_except 
+  messagefacility::MF_MessageLogger
+  fhiclcpp::fhiclcpp
+  cetlib::cetlib cetlib_except::cetlib_except
 )
 
 
