diff --git a/CMakeLists.txt b/CMakeLists.txt
index 50326e6..0e17382 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,10 +9,11 @@
 
 
 # use cmake 3.7 or later
-cmake_minimum_required (VERSION 3.7)
+cmake_minimum_required (VERSION 3.11)
 
-project(sbndaq-artdaq-core)
+project(sbndaq-artdaq-core VERSION 0.05.00 LANGUAGES CXX )
 
+find_package(cetmodules REQUIRED)
 #optional build setup
 
 string(FIND $ENV{MRB_QUALS} "icarus" ICARUS_FOUND)
@@ -37,12 +38,6 @@
    SET(ICARUS_BUILD 1)
 ENDIF()
 
-# cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
-IF (NOT CETBUILDTOOLS_VERSION)
-    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
-ENDIF()
-
 set(CMAKE_MODULE_PATH 
         $ENV{ART_DIR}/Modules
         $ENV{CETBUILDTOOLS_DIR}/Modules
@@ -68,17 +63,17 @@
 
 # these are minimum required versions, not the actual product versions
 #find_ups_product( icarus_artdaq_base v0_08_04 )
-find_ups_product( artdaq_core v3_05_07)
-find_ups_product( TRACE v3_15_03 )
-find_ups_product( cetlib v3_07_02 )
-find_ups_product( cetlib_except v1_03_03 )
-find_ups_boost( v1_69_0 filesystem
+find_package(artdaq_core)
+cet_find_simple_package( trace INCPATH_VAR TRACE_INC )
+find_package( cetlib )
+find_package( cetlib_except)
+find_package( Boost REQUIRED 
+               COMPONENTS filesystem
                         program_options
                         unit_test_framework
  )
 
 
-find_ups_product( cetbuildtools v7_13_02 )
 
 #experiment specific products needed
 IF(ICARUS_BUILD)
@@ -102,8 +97,6 @@
 # testing
 # add_subdirectory(test)
 
-# ups - table and config files
-add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
+cet_cmake_config()
