diff --git a/CMakeLists.txt b/CMakeLists.txt
index 50326e6..0e17382 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,9 +9,12 @@
 
 
 # use cmake 3.7 or later
-cmake_minimum_required (VERSION 3.7)
+cmake_minimum_required(VERSION 3.7)
 
-project(sbndaq-artdaq-core)
+project(sbndaq-artdaq-core VERSION 0.06.00 LANGUAGES CXX C)
+#some need this for install_fhicl(), install_gdml()
+set(fcl_dir job)
+set(gdml_dir gdml)
 
 #optional build setup
 
@@ -38,17 +41,18 @@
 ENDIF()
 
 # cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
+SET ( CETBUILDTOOLS_VERSION $CMAKE_CACHE_MAJOR_VERSION )
 IF (NOT CETBUILDTOOLS_VERSION)
     MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
 ENDIF()
 
-set(CMAKE_MODULE_PATH 
+set(CMAKE_MODULE_PATH
         $ENV{ART_DIR}/Modules
         $ENV{CETBUILDTOOLS_DIR}/Modules
         ${CMAKE_MODULE_PATH})
 
 
+find_package(cetmodules)
 include(CetCMakeEnv)
 cet_cmake_env()
 
@@ -68,17 +72,18 @@
 
 # these are minimum required versions, not the actual product versions
 #find_ups_product( icarus_artdaq_base v0_08_04 )
-find_ups_product( artdaq_core v3_05_07)
-find_ups_product( TRACE v3_15_03 )
-find_ups_product( cetlib v3_07_02 )
-find_ups_product( cetlib_except v1_03_03 )
+find_package( artdaq_core )
+find_package( messagefacility )
+cet_find_simple_package( trace INCPATH_VAR TRACE_INC )
+cet_find_simple_package( cetlib INCPATH_VAR CETLIB_INC )
+find_package( cetlib_except )
+find_package( Boost REQUIRED COMPONENTS filesystem
-find_ups_boost( v1_69_0 filesystem
                         program_options
                         unit_test_framework
  )
 
 
-find_ups_product( cetbuildtools v7_13_02 )
+find_package( cetmodules )
 
 #experiment specific products needed
 IF(ICARUS_BUILD)
@@ -103,7 +107,6 @@
 # add_subdirectory(test)
 
 # ups - table and config files
-add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
+cet_cmake_config()
