diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7b52fbd..d3f266a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,18 +11,12 @@
 #  make install
 #  make package (builds distribution tarfile)
 # ======================================================================
-
-
 cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
 
-project(icarus_signal_processing VERSION 09.37.01 LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
+find_package(cetmodules 2.25.05 REQUIRED)
 
-list(APPEND CMAKE_MODULE_PATH
-#     ${CMAKE_CURRENT_SOURCE_DIR}/Modules
-)
+project(icarus_signal_processing VERSION 09.37.01 LANGUAGES CXX)
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -34,23 +28,18 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
   EXTRA_CXX_FLAGS -Wno-unused-local-typedefs
 )
 
-cet_report_compiler_flags()
+cet_report_compiler_flags(REPORT_THRESHOLD VERBOSE)
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( fftw )
-
+find_package( PkgConfig REQUIRED)
+find_package( ROOT )
 # macros for dictionary and simple_plugin
 include(CetMake)
 
 # add cet_find_library commands here when needed
 
-##cet_find_library(FFTW_LIBRARY NAMES fftw3f fftw3 fftw3q fftw3l fftw3_threads fftw3f_threads fftw3q_threads PATHS $ENV{FFTW_LIBRARY} )
-cet_find_library(ICARUS_FFTW3_LIBRARY  NAMES fftw3  PATHS ENV FFTW_LIBRARY NO_DEFAULT_PATH )
-cet_find_library(ICARUS_FFTW3F_LIBRARY NAMES fftw3f PATHS ENV FFTW_LIBRARY NO_DEFAULT_PATH )
-cet_find_library(ICARUS_FFTW3L_LIBRARY NAMES fftw3l PATHS ENV FFTW_LIBRARY NO_DEFAULT_PATH )
-cet_find_library(ICARUS_FFTW3Q_LIBRARY NAMES fftw3q PATHS ENV FFTW_LIBRARY NO_DEFAULT_PATH )
-set(ICARUS_FFTW_LIBRARIES ICARUS_FFTW3_LIBRARY ICARUS_FFTW3F_LIBRARY ICARUS_FFTW3L_LIBRARY ICARUS_FFTW3Q_LIBRARY)
-include_directories( $ENV{EIGEN_INC} )
+pkg_search_module(FFTW3 REQUIRED IMPORTED_TARGET GLOBAL fftw3f fftw3 fftw3q fftw3l fftw3_threads fftw3f_threeads fftw3q_threads)
+set(ICARUS_FFTW_LIBRARIES PkgConfig::FFTW3)
 
 # ADD SOURCE CODE SUBDIRECTORIES HERE
 add_subdirectory(icarus_signal_processing)
@@ -60,7 +49,6 @@ add_subdirectory(test)
 
 # ups - table and config files
 # must be AFTER all other subdirectories
-add_subdirectory(ups)
 
 # packaging utility - must be last
-include(UseCPack)
+cet_cmake_config()
diff --git a/icarus_signal_processing/CMakeLists.txt b/icarus_signal_processing/CMakeLists.txt
old mode 100755
new mode 100644
index 819c2a7..3fc1cad
--- a/icarus_signal_processing/CMakeLists.txt
+++ b/icarus_signal_processing/CMakeLists.txt
@@ -2,17 +2,23 @@ add_subdirectory(Filters)
 add_subdirectory(Detection)
 
 add_definitions(-DEIGEN_FFTW_DEFAULT)
+include_directories(${FFTW3_INCLUDE_DIRS})
+link_directories(${FFTW3_LIBRARY_DIRS})
 
-cet_make( 
+cet_make_library( SOURCE 
+                          Denoising.cxx
+                          ROIFinder2D.cxx
+                          WaveformTools.cxx
+          HEADERS_TARGET
           LIBRARIES 
                           icarus_signal_processing_Detection
                           icarus_signal_processing_Filters
                           ${ICARUS_FFTW_LIBRARIES}
-                          ${ROOT_GEOM}
-                          ${ROOT_XMLIO}
-                          ${ROOT_GDML}
-                          ${ROOT_FFTW}
-                          ${ROOT_BASIC_LIB_LIST}
+                          ROOT::Geom
+                          ROOT::XMLIO
+                          ROOT::Gdml
+                          ROOT::FFTW
+                          ROOT::Core
        )
 
 install_headers()
diff --git a/icarus_signal_processing/Detection/CMakeLists.txt b/icarus_signal_processing/Detection/CMakeLists.txt
old mode 100755
new mode 100644
index 8a62ecf..2b5fa3c
--- a/icarus_signal_processing/Detection/CMakeLists.txt
+++ b/icarus_signal_processing/Detection/CMakeLists.txt
@@ -1,12 +1,12 @@
 
-cet_make( 
+cet_make(  ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
           LIBRARIES 
                           icarus_signal_processing_Filters
-                          ${ROOT_GEOM}
-                          ${ROOT_XMLIO}
-                          ${ROOT_GDML}
-                          ${ROOT_FFTW}
-                          ${ROOT_BASIC_LIB_LIST}
+                          ROOT::Geom
+                          ROOT::XMLIO
+                          ROOT::Gdml
+                          ROOT::FFTW
+                          ROOT::Core
        )
 
 install_headers()
diff --git a/icarus_signal_processing/Filters/CMakeLists.txt b/icarus_signal_processing/Filters/CMakeLists.txt
old mode 100755
new mode 100644
index 0d1537c..5c96851
--- a/icarus_signal_processing/Filters/CMakeLists.txt
+++ b/icarus_signal_processing/Filters/CMakeLists.txt
@@ -1,13 +1,14 @@
 add_definitions(-DEIGEN_FFTW_DEFAULT)
-
-cet_make( 
+include_directories(${FFTW3_INCLUDE_DIRS})
+link_directories(${FFTW3_LIBRARY_DIRS})
+cet_make(  ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
           LIBRARIES 
                           ${ICARUS_FFTW_LIBRARIES}
-                          ${ROOT_GEOM}
-                          ${ROOT_XMLIO}
-                          ${ROOT_GDML}
-                          ${ROOT_FFTW}
-                          ${ROOT_BASIC_LIB_LIST}
+                          ROOT::Geom
+                          ROOT::XMLIO
+                          ROOT::Gdml
+                          ROOT::FFTW
+                          ROOT::Core
        )
 
 install_headers()
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 7966219..67b6d87 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -3,4 +3,4 @@
 # Enable asserts
 cet_enable_asserts()
 
-# Add test items here
\ No newline at end of file
+# Add test items here
