diff --git a/CMakeLists.txt b/CMakeLists.txt
index 65291fc..abb326f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,10 +10,8 @@ cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
 
 project(lardataobj VERSION 09.03.03 LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
-
-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules)
+find_package( cetmodules REQUIRED )
+find_package( canvas_root_io )
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -27,15 +25,15 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( canvas )
-find_ups_product(messagefacility)
-find_ups_product( larcoreobj )
-find_ups_product( larcorealg )
-find_ups_product( nusimdata )
-find_ups_product( cetlib )
-find_ups_product( cetlib_except )
-find_ups_boost( )
-find_ups_root()
+find_package( canvas )
+find_package( messagefacility )
+find_package( larcoreobj )
+find_package( larcorealg )
+find_package( nusimdata )
+find_package( cetlib )
+find_package( cetlib_except )
+find_package( Boost )
+find_package( ROOT )
 
 # macros for artdaq_dictionary and simple_plugin
 include(ArtDictionary)
@@ -47,9 +45,4 @@ add_subdirectory(lardataobj)
 
 # tests
 add_subdirectory(test)
-
-# ups - table and config files
-add_subdirectory(ups)
-
-# packaging utility
-include(UseCPack)
+cet_cmake_config()
diff --git a/lardataobj/AnalysisBase/CMakeLists.txt b/lardataobj/AnalysisBase/CMakeLists.txt
index b6b3b20..c58cb8b 100644
--- a/lardataobj/AnalysisBase/CMakeLists.txt
+++ b/lardataobj/AnalysisBase/CMakeLists.txt
@@ -1,6 +1,7 @@
-cet_make(LIBRARIES canvas::canvas cetlib_except ROOT::Core NO_DICTIONARY)
+cet_make_library(SOURCE  Calorimetry.cxx  CosmicTag.cxx  FlashMatch.cxx  MVAOutput.cxx  ParticleID.cxx  T0.cxx 
+                 LIBRARIES canvas::canvas cetlib_except::cetlib_except ROOT::Core)
 
-art_dictionary(DICTIONARY_LIBRARIES lardataobj_AnalysisBase nusimdata_SimulationBase)
+build_dictionary(DICTIONARY_LIBRARIES canvas::canvas lardataobj::AnalysisBase nusimdata::SimulationBase)
 
 install_headers()
 install_source()
diff --git a/lardataobj/MCBase/CMakeLists.txt b/lardataobj/MCBase/CMakeLists.txt
index 85bf0cb..30c3f57 100644
--- a/lardataobj/MCBase/CMakeLists.txt
+++ b/lardataobj/MCBase/CMakeLists.txt
@@ -1,11 +1,11 @@
-cet_make(NO_DICTIONARY
+cet_make_library(SOURCE  MCBaseException.cxx  MCHitCollection.cxx  MCHit.cxx  MCShower.cxx  MCTrack.cxx  MCWireCollection.cxx  MCWire.cxx
          LIBRARIES
-           nusimdata_SimulationBase
+           nusimdata::SimulationBase
            ROOT::Core
            ROOT::Physics
          )
 
-art_dictionary(DICTIONARY_LIBRARIES lardataobj_MCBase)
+build_dictionary(DICTIONARY_LIBRARIES canvas::canvas lardataobj::MCBase)
 
 install_headers()
 install_source()
diff --git a/lardataobj/OpticalDetectorData/CMakeLists.txt b/lardataobj/OpticalDetectorData/CMakeLists.txt
index f2f4070..33d0f00 100644
--- a/lardataobj/OpticalDetectorData/CMakeLists.txt
+++ b/lardataobj/OpticalDetectorData/CMakeLists.txt
@@ -1,6 +1,5 @@
-cet_make(NO_DICTIONARY)
+cet_make_library(SOURCE OpticalDetectorData.cxx HEADERS_TARGET)
 
-art_dictionary(DICTIONARY_LIBRARIES lardataobj_OpticalDetectorData)
+build_dictionary(DICTIONARY_LIBRARIES canvas::canvas lardataobj::OpticalDetectorData)
 
 install_headers()
-install_source()
diff --git a/lardataobj/RawData/CMakeLists.txt b/lardataobj/RawData/CMakeLists.txt
index b3f9816..c05339e 100644
--- a/lardataobj/RawData/CMakeLists.txt
+++ b/lardataobj/RawData/CMakeLists.txt
@@ -1,12 +1,12 @@
-cet_make(NO_DICTIONARY
+cet_make_library(SOURCE AuxDetDigit.cxx  BeamInfo.cxx  DAQHeader.cxx  ExternalTrigger.cxx  OpDetPulse.cxx  raw.cxx  RawDigit.cxx  TriggerData.cxx 
          LIBRARIES
-           larcoreobj_SummaryData
+           larcoreobj::SummaryData
            canvas::canvas
            messagefacility::MF_MessageLogger
            cetlib_except::cetlib_except
            ROOT::Core)
 
-art_dictionary(DICTIONARY_LIBRARIES lardataobj_RawData)
+build_dictionary(DICTIONARY_LIBRARIES canvas::canvas lardataobj::RawData)
 
 install_headers()
 install_source()
diff --git a/lardataobj/RecoBase/AssnsDicts/CMakeLists.txt b/lardataobj/RecoBase/AssnsDicts/CMakeLists.txt
index e979cec..ba1fc83 100644
--- a/lardataobj/RecoBase/AssnsDicts/CMakeLists.txt
+++ b/lardataobj/RecoBase/AssnsDicts/CMakeLists.txt
@@ -1,4 +1,4 @@
-art_dictionary(DICTIONARY_LIBRARIES lardataobj_RecoBase)
+art_dictionary(DICTIONARY_LIBRARIES lardataobj::RecoBase)
 
 install_headers()
 install_source()
diff --git a/lardataobj/RecoBase/CMakeLists.txt b/lardataobj/RecoBase/CMakeLists.txt
index 55b4588..d15cdbe 100644
--- a/lardataobj/RecoBase/CMakeLists.txt
+++ b/lardataobj/RecoBase/CMakeLists.txt
@@ -1,13 +1,34 @@
-cet_make(LIBRARIES lardataobj_RawData
+cet_make_library(SOURCE Cluster.cxx
+                        EndPoint2D.cxx
+                        Hit.cxx
+                        OpHit.cxx
+                        PFParticle.cxx
+                        Seed.cxx
+                        Slice.cxx
+                        Track.cxx
+                        TrackTrajectory.cxx
+                        TrajectoryPointFlags.cxx
+                        Wire.cxx
+                        Edge.cxx
+                        Event.cxx
+                        OpFlash.cxx
+                        PCAxis.cxx
+                        PFParticleMetadata.cxx
+                        Shower.cxx
+                        SpacePoint.cxx
+                        TrackingPlane.cxx
+                        Trajectory.cxx
+                        Vertex.cxx
+         LIBRARIES lardataobj::RawData
                    canvas::canvas
                    messagefacility::MF_MessageLogger
                    cetlib_except::cetlib_except
                    ROOT::Physics
                    ROOT::Matrix
                    ROOT::GenVector
-                   NO_DICTIONARY)
+                   )
 
-art_dictionary(DICTIONARY_LIBRARIES lardataobj_RecoBase)
+build_dictionary(DICTIONARY_LIBRARIES canvas::canvas lardataobj::RecoBase)
 
 add_subdirectory(AssnsDicts)
 add_subdirectory(TrackingDicts)
diff --git a/lardataobj/RecoBase/TrackingDicts/CMakeLists.txt b/lardataobj/RecoBase/TrackingDicts/CMakeLists.txt
index e979cec..ba1fc83 100644
--- a/lardataobj/RecoBase/TrackingDicts/CMakeLists.txt
+++ b/lardataobj/RecoBase/TrackingDicts/CMakeLists.txt
@@ -1,4 +1,4 @@
-art_dictionary(DICTIONARY_LIBRARIES lardataobj_RecoBase)
+art_dictionary(DICTIONARY_LIBRARIES lardataobj::RecoBase)
 
 install_headers()
 install_source()
diff --git a/lardataobj/Simulation/CMakeLists.txt b/lardataobj/Simulation/CMakeLists.txt
index cd005cd..f3ce760 100644
--- a/lardataobj/Simulation/CMakeLists.txt
+++ b/lardataobj/Simulation/CMakeLists.txt
@@ -1,6 +1,6 @@
-cet_make(NO_DICTIONARY
+cet_make_library(SOURCE AuxDetSimChannel.cxx  OpDetBacktrackerRecord.cxx  SimChannel.cxx  SimPhotons.cxx  SupernovaTruth.cxx
          LIBRARIES
-           nusimdata_SimulationBase
+           nusimdata::SimulationBase
            canvas::canvas
            messagefacility::MF_MessageLogger
            ROOT::GenVector
@@ -8,7 +8,7 @@ cet_make(NO_DICTIONARY
            ROOT::Physics
          )
 
-art_dictionary(DICTIONARY_LIBRARIES lardataobj_Simulation)
+build_dictionary(DICTIONARY_LIBRARIES canvas::canvas lardataobj::Simulation)
 
 install_headers()
 install_source()
diff --git a/test/RawData/CMakeLists.txt b/test/RawData/CMakeLists.txt
index 4f36b9e..e2d6136 100644
--- a/test/RawData/CMakeLists.txt
+++ b/test/RawData/CMakeLists.txt
@@ -1,11 +1,11 @@
 # test raw data compression
 cet_test(raw_test USE_BOOST_UNIT
-  LIBRARIES lardataobj_RawData
+  LIBRARIES lardataobj::RawData
   )
 
 # test data products
 cet_test(RawDigit_test USE_BOOST_UNIT
-  LIBRARIES lardataobj_RawData
+  LIBRARIES lardataobj::RawData
   )
 
 install_headers()
diff --git a/test/RecoBase/CMakeLists.txt b/test/RecoBase/CMakeLists.txt
index af8f11c..480249d 100644
--- a/test/RecoBase/CMakeLists.txt
+++ b/test/RecoBase/CMakeLists.txt
@@ -1,30 +1,30 @@
 cet_test(Wire_test USE_BOOST_UNIT
-  LIBRARIES lardataobj_RecoBase
+  LIBRARIES lardataobj::RecoBase
   )
 
 cet_test(Hit_test USE_BOOST_UNIT
-  LIBRARIES lardataobj_RecoBase
+  LIBRARIES lardataobj::RecoBase
   )
 
 cet_test(Cluster_test USE_BOOST_UNIT
-  LIBRARIES lardataobj_RecoBase
+  LIBRARIES lardataobj::RecoBase
   )
 
 cet_test(PointCharge_test USE_BOOST_UNIT
-  LIBRARIES lardataobj_RecoBase
+  LIBRARIES lardataobj::RecoBase
   )
 
 cet_test(Edge_test USE_BOOST_UNIT
-  LIBRARIES lardataobj_RecoBase
+  LIBRARIES lardataobj::RecoBase
   )
 
 cet_test(TrajectoryPointFlags_test USE_BOOST_UNIT
-  LIBRARIES lardataobj_RecoBase
+  LIBRARIES lardataobj::RecoBase
   )
 
 cet_test(Trajectory_test USE_BOOST_UNIT
   LIBRARIES
-    lardataobj_RecoBase
+    lardataobj::RecoBase
     ROOT::GenVector
     ROOT::Physics
     ROOT::Matrix
@@ -32,7 +32,7 @@ cet_test(Trajectory_test USE_BOOST_UNIT
 
 cet_test(TrackTrajectory_test USE_BOOST_UNIT
   LIBRARIES
-    lardataobj_RecoBase
+    lardataobj::RecoBase
     ROOT::GenVector
     ROOT::Physics
     ROOT::Matrix
