diff --git a/CMakeLists.txt b/CMakeLists.txt
index a53c475..bc703d3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,21 +13,13 @@
 # ======================================================================
 
 
-# use cmake 3.3 or later
-cmake_minimum_required (VERSION 3.3)
-
-project(larcorealg)
-
-# cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
-IF (NOT CETBUILDTOOLS_VERSION)
-    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
-ENDIF()
-
-set(CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules
-		      $ENV{CETBUILDTOOLS_DIR}/Modules
-		      ${CMAKE_MODULE_PATH})
+# use cmake 3.12 or later
+cmake_minimum_required (VERSION 3.12)
 
+project(larcorealg VERSION 08.20.02 LANGUAGES CXX)
+set(fcl_dir job)
+set(gdml_dir gdml)
+find_package(cetmodules REQUIRED)
 include(CetCMakeEnv)
 cet_cmake_env()
 
@@ -40,10 +32,9 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( canvas )
-find_ups_product( cetbuildtools )
-find_ups_product( larcoreobj )
-find_ups_root( )
+find_package(Boost REQUIRED COMPONENTS date_time filesystem graph program_options
+  regex system unit_test_framework)
+find_package( larcoreobj REQUIRED)
 
 # macros for artdaq_dictionary and simple_plugin
 include(ArtDictionary)
@@ -55,9 +46,4 @@ add_subdirectory(larcorealg)
 
 # tests 
 add_subdirectory(test)
-
-# ups - table and config files
-add_subdirectory(ups)
-
-# packaging utility
-include(UseCPack)
+add_subdirectory(cmakeConfig)
diff --git a/cmakeConfig/CMakeLists.txt b/cmakeConfig/CMakeLists.txt
new file mode 100644
index 0000000..84089f3
--- /dev/null
+++ b/cmakeConfig/CMakeLists.txt
@@ -0,0 +1,2 @@
+# Create package configuration and version files.
+cet_cmake_config(CONFIG_FRAGMENTS larcorealg-fragment.cmake.in)
diff --git a/cmakeConfig/larcorealg-fragment.cmake.in b/cmakeConfig/larcorealg-fragment.cmake.in
new file mode 100644
index 0000000..0cb3abc
--- /dev/null
+++ b/cmakeConfig/larcorealg-fragment.cmake.in
@@ -0,0 +1 @@
+find_package(larcoreobj REQUIRED)
diff --git a/larcorealg/CoreUtils/CMakeLists.txt b/larcorealg/CoreUtils/CMakeLists.txt
index aff287f..4fa7937 100644
--- a/larcorealg/CoreUtils/CMakeLists.txt
+++ b/larcorealg/CoreUtils/CMakeLists.txt
@@ -1,4 +1,3 @@
 cet_make(LIBRARIES cetlib_except)
 
 install_headers()
-install_source()
diff --git a/larcorealg/GeoAlgo/CMakeLists.txt b/larcorealg/GeoAlgo/CMakeLists.txt
index 999612b..0702cdc 100644
--- a/larcorealg/GeoAlgo/CMakeLists.txt
+++ b/larcorealg/GeoAlgo/CMakeLists.txt
@@ -1,11 +1,10 @@
 cet_make(NO_DICTIONARY
          LIBRARIES
-           ${ROOT_CORE}
-           ${ROOT_PHYSICS}
+           ROOT::Core
+           ROOT::Physics
          )
 
 art_dictionary(DICTIONARY_LIBRARIES larcorealg_GeoAlgo)
 
 
 install_headers()
-install_source()
diff --git a/larcorealg/Geometry/CMakeLists.txt b/larcorealg/Geometry/CMakeLists.txt
index d3e2ced..0f5ced0 100644
--- a/larcorealg/Geometry/CMakeLists.txt
+++ b/larcorealg/Geometry/CMakeLists.txt
@@ -6,14 +6,13 @@ cet_make(
           ${FHICLCPP}
           cetlib
           cetlib_except
-          ${ROOT_CORE}
-          ${ROOT_PHYSICS}
-          ${ROOT_GEOM}
-          ${ROOT_GENVECTOR}
-        )
+          ROOT::Core
+          ROOT::Physics
+          ROOT::Geom
+          ROOT::GenVector
+          )
 
 install_headers(SUBDIRS "details")
 install_fhicl(SUBDIRS "details")
-install_source(SUBDIRS "details")
 
 add_subdirectory(gdml)
diff --git a/larcorealg/TestUtils/CMakeLists.txt b/larcorealg/TestUtils/CMakeLists.txt
index efd3d41..aa027c0 100644
--- a/larcorealg/TestUtils/CMakeLists.txt
+++ b/larcorealg/TestUtils/CMakeLists.txt
@@ -1,4 +1,3 @@
 cet_make(LIBRARIES canvas cetlib_except)
 
 install_headers()
-install_source()
diff --git a/test/CoreUtils/CMakeLists.txt b/test/CoreUtils/CMakeLists.txt
index 44c3038..bde3d67 100644
--- a/test/CoreUtils/CMakeLists.txt
+++ b/test/CoreUtils/CMakeLists.txt
@@ -3,7 +3,7 @@
 # Testing
 #
 # ======================================================================
-
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-sign-compare")
 cet_test( printBacktrace_test
   LIBRARIES
     larcorealg_CoreUtils
@@ -20,8 +20,8 @@ cet_test( NumericUtils_test USE_BOOST_UNIT )
 cet_test( DereferenceIterator_test USE_BOOST_UNIT )
 cet_test( DumpUtils_test
   LIBRARIES
-    ${ROOT_CORE}
-    ${ROOT_PHYSICS}
+    ROOT::Core
+    ROOT::Physics
   USE_BOOST_UNIT
   )
 cet_test( SortByPointers_test USE_BOOST_UNIT )
diff --git a/test/Geometry/CMakeLists.txt b/test/Geometry/CMakeLists.txt
index 5a3a447..2e11b4d 100644
--- a/test/Geometry/CMakeLists.txt
+++ b/test/Geometry/CMakeLists.txt
@@ -2,9 +2,9 @@ cet_enable_asserts()
 
 cet_test( geo_vectors_utils_test
   LIBRARIES
-    ${ROOT_CORE}
-    ${ROOT_PHYSICS}
-    ${ROOT_GENVECTOR}
+    ROOT::Core
+    ROOT::Physics
+    ROOT::GenVector
   USE_BOOST_UNIT
   )
 
@@ -20,10 +20,9 @@ cet_make_library( LIBRARY_NAME GeometryTestLib
                         larcorealg_TestUtils
                         canvas
                         ${MF_MESSAGELOGGER}
-                        
                         ${FHICLCPP}
                         cetlib cetlib_except
-                        ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY}
+                        Boost::unit_test_framework
           )
 
 
@@ -36,7 +35,6 @@ cet_test(geometry_iterator_test
   LIBRARIES larcorealg_Geometry
             GeometryTestLib
             ${MF_MESSAGELOGGER}
-            
             ${FHICLCPP}
             cetlib cetlib_except
   USE_BOOST_UNIT
@@ -51,7 +49,6 @@ cet_test(geometry_geoid_test
   LIBRARIES larcorealg_Geometry
             GeometryTestLib
             ${MF_MESSAGELOGGER}
-            
             ${FHICLCPP}
             cetlib cetlib_except
   USE_BOOST_UNIT
@@ -67,7 +64,6 @@ cet_test(geometry_test
   LIBRARIES larcorealg_Geometry
             GeometryTestLib
             ${MF_MESSAGELOGGER}
-            
             ${FHICLCPP}
             cetlib cetlib_except
 )
@@ -80,7 +76,6 @@ cet_test(geometry_loader_test
   LIBRARIES larcorealg_Geometry
             GeometryTestLib
             ${MF_MESSAGELOGGER}
-            
             ${FHICLCPP}
             cetlib cetlib_except
 )
@@ -93,7 +88,6 @@ cet_test(geometry_iterator_loop_test
   LIBRARIES larcorealg_Geometry
             GeometryTestLib
             ${MF_MESSAGELOGGER}
-            
             ${FHICLCPP}
             cetlib cetlib_except
 )
@@ -105,7 +99,6 @@ cet_test(geometry_standardchannelmapping_test
   LIBRARIES larcorealg_Geometry
             GeometryTestLib
             ${MF_MESSAGELOGGER}
-            
             ${FHICLCPP}
             cetlib cetlib_except
   USE_BOOST_UNIT
@@ -120,7 +113,6 @@ cet_test(geometry_thirdplaneslope_test
   SOURCES geometry_thirdplaneslope_test.cxx
   LIBRARIES larcorealg_Geometry
             ${MF_MESSAGELOGGER}
-            
             ${FHICLCPP}
             cetlib cetlib_except
   USE_BOOST_UNIT
@@ -133,9 +125,9 @@ cet_test(geometry_thirdplaneslope_test
 cet_test(Decomposer_test
   SOURCES Decomposer_test.cxx
   LIBRARIES
-    ${ROOT_PHYSICS}
-    ${ROOT_GENVECTOR}
-    ${ROOT_CORE} # TVersionCheck
+    ROOT::Physics
+    ROOT::GenVector
+    ROOT::Core # TVersionCheck
   USE_BOOST_UNIT
   )
 
@@ -151,7 +143,6 @@ cet_test(driftvolumes_test
   TEST_ARGS ./test_geometry.fcl
   LIBRARIES larcorealg_Geometry
             ${MF_MESSAGELOGGER}
-            
             ${FHICLCPP}
             cetlib cetlib_except
   USE_BOOST_UNIT
@@ -194,4 +185,3 @@ endif()
 
 install_headers()
 install_fhicl()
-install_source()
