diff --git a/CMakeLists.txt b/CMakeLists.txt
index a53c475..861c1b1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,21 +13,13 @@
 # ======================================================================
 
 
-# use cmake 3.3 or later
-cmake_minimum_required (VERSION 3.3)
-
-project(larcorealg)
-
-# cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
-IF (NOT CETBUILDTOOLS_VERSION)
-    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
-ENDIF()
-
-set(CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules
-		      $ENV{CETBUILDTOOLS_DIR}/Modules
-		      ${CMAKE_MODULE_PATH})
+# use cmake 3.10 or later
+cmake_minimum_required (VERSION 3.10)
 
+project(larcorealg VERSION 08.05.00 LANGUAGES CXX)
+set(fcl_dir job)
+set(gdml_dir gdml)
+find_package(cetmodules REQUIRED)
 include(CetCMakeEnv)
 cet_cmake_env()
 
@@ -40,10 +32,9 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( canvas )
-find_ups_product( cetbuildtools )
-find_ups_product( larcoreobj )
-find_ups_root( )
+find_package(Boost REQUIRED COMPONENTS date_time filesystem graph program_options
+  regex system unit_test_framework)
+find_package( larcoreobj REQUIRED)
 
 # macros for artdaq_dictionary and simple_plugin
 include(ArtDictionary)
@@ -55,9 +46,4 @@ add_subdirectory(larcorealg)
 
 # tests 
 add_subdirectory(test)
-
-# ups - table and config files
-add_subdirectory(ups)
-
-# packaging utility
-include(UseCPack)
+cet_cmake_config()
