diff --git a/CMakeLists.txt b/CMakeLists.txt
index 58bdb36..d3c0d43 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,10 +10,8 @@ cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
 
 project(larcorealg VERSION 09.03.01 LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
-
-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules)
+find_package(cetmodules REQUIRED)
+find_package(canvas_root_io)
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -27,15 +25,15 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_boost()
-find_ups_root()
+find_package( Boost )
+find_package( ROOT )
 
-find_ups_product( canvas )
-find_ups_product( cetlib )
-find_ups_product( cetlib_except )
-find_ups_product(messagefacility)
-find_ups_product( larcoreobj )
-find_ups_product(clhep)
+find_package( canvas )
+find_package( cetlib )
+find_package( cetlib_except )
+find_package( messagefacility )
+find_package( larcoreobj )
+find_package( CLHEP )
 
 # macros for artdaq_dictionary and simple_plugin
 include(ArtDictionary)
@@ -49,7 +47,8 @@ add_subdirectory(larcorealg)
 add_subdirectory(test)
 
 # ups - table and config files
-add_subdirectory(ups)
+#add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
+#include(UseCPack)
+cet_cmake_config()
diff --git a/larcorealg/CoreUtils/CMakeLists.txt b/larcorealg/CoreUtils/CMakeLists.txt
index 008b110..d8a9bb6 100644
--- a/larcorealg/CoreUtils/CMakeLists.txt
+++ b/larcorealg/CoreUtils/CMakeLists.txt
@@ -1,4 +1,3 @@
-cet_make(LIBRARIES cetlib_except::cetlib_except)
+cet_make_library(HEADERS_TARGET LIBRARIES cetlib_except::cetlib_except SOURCE DebugUtils.cxx)
 
 install_headers()
-install_source()
diff --git a/larcorealg/GeoAlgo/CMakeLists.txt b/larcorealg/GeoAlgo/CMakeLists.txt
index 4b0131d..c0208ac 100644
--- a/larcorealg/GeoAlgo/CMakeLists.txt
+++ b/larcorealg/GeoAlgo/CMakeLists.txt
@@ -1,10 +1,22 @@
-cet_make(NO_DICTIONARY
+cet_make_library( HEADERS_TARGET
          LIBRARIES
            ROOT::Core
            ROOT::Physics
+          SOURCE GeoAABox.cxx
+                 GeoAlgo.cxx
+                 GeoCone.cxx
+                 GeoCylinder.cxx
+                 GeoDirectedLine.cxx
+                 GeoHalfLine.cxx
+                 GeoLine.cxx
+                 GeoLineSegment.cxx
+                 GeoObjCollection.cxx
+                 GeoSphere.cxx  
+                 GeoTrajectory.cxx
+                 GeoVector.cxx
          )
 
-art_dictionary(DICTIONARY_LIBRARIES larcorealg_GeoAlgo)
+build_dictionary(DICTIONARY_LIBRARIES canvas::canvas larcorealg::GeoAlgo)
 
 
 install_headers()
diff --git a/larcorealg/Geometry/CMakeLists.txt b/larcorealg/Geometry/CMakeLists.txt
index 1d536ad..51e3e9d 100644
--- a/larcorealg/Geometry/CMakeLists.txt
+++ b/larcorealg/Geometry/CMakeLists.txt
@@ -1,8 +1,8 @@
-cet_make(
+cet_make_library( HEADERS_TARGET
   SUBDIRS
     "details"
   LIBRARIES
-          larcoreobj_SimpleTypesAndConstants
+          larcoreobj::SimpleTypesAndConstants
           messagefacility::MF_MessageLogger
           fhiclcpp::fhiclcpp
           cetlib::cetlib
@@ -11,6 +11,29 @@ cet_make(
           ROOT::Physics
           ROOT::Geom
           ROOT::GenVector
+   SOURCE
+          AuxDetChannelMapAlg.cxx
+          AuxDetGeo.cxx
+          AuxDetGeometryCore.cxx
+          AuxDetSensitiveGeo.cxx
+          BoxBoundedGeo.cxx
+          ChannelMapAlg.cxx
+          ChannelMapStandardAlg.cxx
+          CryostatGeo.cxx
+          DriftPartitions.cxx
+          GeoNodePath.cxx
+          GeoObjectSorter.cxx
+          GeoObjectSorterStandard.cxx
+          GeoVectorLocalTransformation.cxx
+          GeometryBuilderStandard.cxx
+          GeometryBuilderWireless.cxx
+          GeometryCore.cxx
+          LocalTransformation.cxx
+          OpDetGeo.cxx
+          PlaneGeo.cxx
+          StandaloneGeometrySetup.cxx
+          TPCGeo.cxx
+          WireGeo.cxx
         )
 
 install_headers(SUBDIRS "details")
diff --git a/larcorealg/TestUtils/CMakeLists.txt b/larcorealg/TestUtils/CMakeLists.txt
index 139a0e4..200f7a7 100644
--- a/larcorealg/TestUtils/CMakeLists.txt
+++ b/larcorealg/TestUtils/CMakeLists.txt
@@ -1,4 +1,3 @@
-cet_make(LIBRARIES canvas::canvas cetlib_except)
+cet_make_library(LIBRARIES canvas::canvas cetlib_except::cetlib_except SOURCE NameSelector.cxx)
 
 install_headers()
-install_source()
diff --git a/test/CoreUtils/CMakeLists.txt b/test/CoreUtils/CMakeLists.txt
index 1c14e51..e20fe03 100644
--- a/test/CoreUtils/CMakeLists.txt
+++ b/test/CoreUtils/CMakeLists.txt
@@ -3,10 +3,10 @@
 # Testing
 #
 # ======================================================================
-
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-sign-compare")
 cet_test( printBacktrace_test
   LIBRARIES
-    larcorealg_CoreUtils
+    larcorealg::CoreUtils
   )
 
 cet_test(UncopiableAndUnmovableClass_test
diff --git a/test/Geometry/CMakeLists.txt b/test/Geometry/CMakeLists.txt
index 055c505..9dd8de0 100644
--- a/test/Geometry/CMakeLists.txt
+++ b/test/Geometry/CMakeLists.txt
@@ -22,13 +22,13 @@ cet_make_library( LIBRARY_NAME GeometryTestLib
                     GeometryIteratorTestAlg.cxx
                     GeometryIteratorLoopTestAlg.cxx
                     ChannelMapStandardTestAlg.cxx
-                  LIBRARIES larcorealg_Geometry
-                        larcorealg_TestUtils
+                  LIBRARIES larcorealg::Geometry
+                        larcorealg::TestUtils
                         canvas::canvas
                         messagefacility::MF_MessageLogger
                         fhiclcpp::fhiclcpp
-                        cetlib::cetlib cetlib_except
-                        ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY}
+                        cetlib::cetlib cetlib_except::cetlib_except
+                        Boost::unit_test_framework
           )
 
 
@@ -38,12 +38,11 @@ cet_make_library( LIBRARY_NAME GeometryTestLib
 # this uses BOOST for the test
 cet_test(geometry_iterator_test
   SOURCES geometry_iterator_test.cxx
-  LIBRARIES larcorealg_Geometry
-            GeometryTestLib
+  LIBRARIES larcorealg::Geometry
+            larcorealg::GeometryTestLib
             messagefacility::MF_MessageLogger
             fhiclcpp::fhiclcpp
-            cetlib::cetlib cetlib_except
-  USE_BOOST_UNIT
+            cetlib::cetlib cetlib_except::cetlib_except
   DATAFILES test_geometry.fcl
   TEST_ARGS -- ./test_geometry.fcl
 )
@@ -52,12 +51,11 @@ cet_test(geometry_iterator_test
 # this uses BOOST for the test
 cet_test(geometry_geoid_test
   SOURCES geometry_geoid_test.cxx
-  LIBRARIES larcorealg_Geometry
-            GeometryTestLib
+  LIBRARIES larcorealg::Geometry
+            larcorealg::GeometryTestLib
             messagefacility::MF_MessageLogger
             fhiclcpp::fhiclcpp
-            cetlib::cetlib cetlib_except
-  USE_BOOST_UNIT
+            cetlib::cetlib cetlib_except::cetlib_except
   DATAFILES test_geometry.fcl
   TEST_ARGS -- ./test_geometry.fcl
 )
@@ -67,11 +65,11 @@ cet_test(geometry_test
   SOURCES geometry_test.cxx
   DATAFILES test_geometry.fcl
   TEST_ARGS ./test_geometry.fcl
-  LIBRARIES larcorealg_Geometry
-            GeometryTestLib
+  LIBRARIES larcorealg::Geometry
+            larcorealg::GeometryTestLib
             messagefacility::MF_MessageLogger
             fhiclcpp::fhiclcpp
-            cetlib::cetlib cetlib_except
+            cetlib::cetlib cetlib_except::cetlib_except
 )
 
 # test of standalone geometry loading (use the hard-coded channel mapping for "standard" LArTPCdetector)
@@ -79,11 +77,11 @@ cet_test(geometry_loader_test
   SOURCES geometry_loader_test.cxx
   DATAFILES test_geometry.fcl
   TEST_ARGS ./test_geometry.fcl
-  LIBRARIES larcorealg_Geometry
-            GeometryTestLib
+  LIBRARIES larcorealg::Geometry
+            larcorealg::GeometryTestLib
             messagefacility::MF_MessageLogger
             fhiclcpp::fhiclcpp
-            cetlib::cetlib cetlib_except
+            cetlib::cetlib cetlib_except::cetlib_except
 )
 
 # test to verify loops on geometry elements by geometry iterators (hard-coded channel mapping for "standard" LArTPCdetector)
@@ -91,23 +89,22 @@ cet_test(geometry_iterator_loop_test
   SOURCES geometry_iterator_loop_test.cxx
   DATAFILES test_geometry_iterator_loop.fcl
   TEST_ARGS ./test_geometry_iterator_loop.fcl
-  LIBRARIES larcorealg_Geometry
-            GeometryTestLib
+  LIBRARIES larcorealg::Geometry
+            larcorealg::GeometryTestLib
             messagefacility::MF_MessageLogger
             fhiclcpp::fhiclcpp
-            cetlib::cetlib cetlib_except
+            cetlib::cetlib cetlib_except::cetlib_except
 )
 
 
 # test for standard channel mapping (BOOST unit test)
 cet_test(geometry_standardchannelmapping_test
   SOURCES geometry_standardchannelmapping_test.cxx
-  LIBRARIES larcorealg_Geometry
-            GeometryTestLib
+  LIBRARIES larcorealg::Geometry
+            larcorealg::GeometryTestLib
             messagefacility::MF_MessageLogger
             fhiclcpp::fhiclcpp
-            cetlib::cetlib cetlib_except
-  USE_BOOST_UNIT
+            cetlib::cetlib cetlib_except::cetlib_except
   DATAFILES test_geometry.fcl
   TEST_ARGS ./test_geometry.fcl
 )
@@ -117,10 +114,10 @@ cet_test(geometry_standardchannelmapping_test
 # this uses BOOST for the test
 cet_test(geometry_thirdplaneslope_test
   SOURCES geometry_thirdplaneslope_test.cxx
-  LIBRARIES larcorealg_Geometry
+  LIBRARIES larcorealg::Geometry
             messagefacility::MF_MessageLogger
             fhiclcpp::fhiclcpp
-            cetlib::cetlib cetlib_except
+            cetlib::cetlib cetlib_except::cetlib_except
   USE_BOOST_UNIT
   DATAFILES test_geometry.fcl
   TEST_ARGS -- ./test_geometry.fcl
@@ -147,10 +144,10 @@ cet_test(driftvolumes_test
   SOURCES driftvolumes_test.cxx
   DATAFILES test_geometry.fcl
   TEST_ARGS ./test_geometry.fcl
-  LIBRARIES larcorealg_Geometry
+  LIBRARIES larcorealg::Geometry
             messagefacility::MF_MessageLogger
             fhiclcpp::fhiclcpp
-            cetlib::cetlib cetlib_except
+            cetlib::cetlib cetlib_except::cetlib_except
   USE_BOOST_UNIT
   )
 
@@ -171,7 +168,7 @@ cet_test(readoutdatacontainers_test
 
 cet_test(topology_test
   SOURCES topology_test.cxx
-  LIBRARIES larcorealg_Geometry
+  LIBRARIES larcorealg::Geometry
   USE_BOOST_UNIT
   )
 
diff --git a/test/TestUtils/CMakeLists.txt b/test/TestUtils/CMakeLists.txt
index d0babc9..698b441 100644
--- a/test/TestUtils/CMakeLists.txt
+++ b/test/TestUtils/CMakeLists.txt
@@ -2,7 +2,7 @@
 cet_test(NameSelector_test
   SOURCES NameSelector_test.cxx
   LIBRARIES
-    larcorealg_TestUtils
+    larcorealg::TestUtils
     canvas::canvas
   USE_BOOST_UNIT
 )
