diff --git a/CMakeLists.txt b/CMakeLists.txt
index b53a1e9..b908b15 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,11 +10,11 @@ cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
 
 project(larrecodnn VERSION 09.09.09.01 LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
-find_package(Threads)
+find_package( cetmodules REQUIRED )
+find_package( Threads )
 
-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules $ENV{ART_DIR}/Modules)
+find_package( canvas_root_io )
+find_package( art )
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -28,25 +28,27 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( larcoreobj )
-find_ups_product( larcorealg )
-find_ups_product( larcore )
-find_ups_product( lardataobj )
-find_ups_product( lardataalg )
-find_ups_product( lardata )
-find_ups_product( larevt )
-find_ups_product( larsim )
-find_ups_product( larreco )
-find_ups_product( nutools )
-find_ups_product( nug4 )
-find_ups_product( nurandom )
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product( postgresql )
-find_ups_product( range )
-find_ups_product( eigen )
-find_ups_product( tbb )
-find_ups_root()
+find_package( larcoreobj )
+find_package( larcorealg )
+find_package( larcore )
+find_package( lardataobj )
+find_package( lardataalg )
+find_package( lardata )
+find_package( larevt )
+find_package( larsim )
+find_package( larreco )
+find_package( nutools )
+find_package( nug4 )
+find_package( nurandom )
+find_package( nusimdata )
+find_package( art )
+find_package( art_root_io )
+find_package( Eigen3 )
+find_package( PostgreSQL )
+find_package( ROOT )
+find_package( range3 )
+find_package( TBB ) 
+find_package( CURL )
 
 # macros for dictionary and simple_plugin
 include(ArtDictionary)
@@ -54,20 +56,20 @@ include(ArtMake)
 include(BuildPlugins)
 
 if(( DEFINED ENV{TENSORFLOW_DIR} ) OR ( DEFINED ENV{TRITON_DIR} ))
-  find_ups_product( protobuf )
-  cet_find_library(PROTOBUF NAMES protobuf PATHS ENV PROTOBUF_LIB NO_DEFAULT_PATH )
+  find_package( protobuf )
+  cet_find_library(PROTOBUF NAMES protobuf PATHS ENV PROTOBUF_LIB  )
 endif ()
 # tensorflow is optional
 if( DEFINED ENV{TENSORFLOW_DIR} )
   # find tensorflow library
-  find_ups_product( tensorflow )
-  cet_find_library(TENSORFLOW NAMES tensorflow PATHS ENV TENSORFLOW_LIB NO_DEFAULT_PATH )
-  cet_find_library(FOR_TENSORFLOW NAMES tensorflow_framework PATHS ENV TENSORFLOW_LIB NO_DEFAULT_PATH )
+  find_package( tensorflow )
+  cet_find_library(TENSORFLOW NAMES tensorflow PATHS ENV TENSORFLOW_LIB  )
+  cet_find_library(FOR_TENSORFLOW NAMES tensorflow_framework PATHS ENV TENSORFLOW_LIB  )
 endif ()
 # trtis_clients is optional
 if( DEFINED ENV{TRITON_DIR} )
-  find_ups_product(triton)
-  find_ups_product(grpc)
+  find_package(TRITON)
+  find_package(grpc)
 endif ()
 
 include_directories( $ENV{EIGEN_INC} )
@@ -76,7 +78,10 @@ include_directories( $ENV{EIGEN_INC} )
 add_subdirectory(larrecodnn)
 
 # ups - table and config files
+if(WANT_UPS)
 add_subdirectory(ups)
+endif()
 
 # packaging utility
-include(UseCPack)
+#include(UseCPack)
+cet_cmake_config()
diff --git a/larrecodnn/ImagePatternAlgs/CMakeLists.txt b/larrecodnn/ImagePatternAlgs/CMakeLists.txt
index 8de6661..7c5ef32 100644
--- a/larrecodnn/ImagePatternAlgs/CMakeLists.txt
+++ b/larrecodnn/ImagePatternAlgs/CMakeLists.txt
@@ -1,4 +1,4 @@
-if( DEFINED ENV{TENSORFLOW_DIR} )
+if( DEFINED $ENV{TENSORFLOW_DIR} )
   add_subdirectory(Tensorflow)
 endif ()
 if( DEFINED ENV{TRITON_DIR} )
diff --git a/larrecodnn/ImagePatternAlgs/Keras/CMakeLists.txt b/larrecodnn/ImagePatternAlgs/Keras/CMakeLists.txt
index 645f837..999cbf5 100644
--- a/larrecodnn/ImagePatternAlgs/Keras/CMakeLists.txt
+++ b/larrecodnn/ImagePatternAlgs/Keras/CMakeLists.txt
@@ -1,4 +1,4 @@
-art_make(LIB_LIBRARIES ${TBB})
+art_make(LIB_LIBRARIES TBB::tbb)
 
 add_subdirectory(Tools)
 
diff --git a/larrecodnn/ImagePatternAlgs/Keras/Tools/CMakeLists.txt b/larrecodnn/ImagePatternAlgs/Keras/Tools/CMakeLists.txt
index 1ad8490..afd09dc 100644
--- a/larrecodnn/ImagePatternAlgs/Keras/Tools/CMakeLists.txt
+++ b/larrecodnn/ImagePatternAlgs/Keras/Tools/CMakeLists.txt
@@ -1,11 +1,12 @@
 art_make(
           TOOL_LIBRARIES
-          larreco_RecoAlg_ImagePatternAlgs_DataProvider
-          larrecodnn_ImagePatternAlgs_Keras
-          larcorealg_Geometry
+          larreco::RecoAlg_ImagePatternAlgs_DataProvider
+          larrecodnn::ImagePatternAlgs_Keras
+          larcorealg::Geometry
           ROOT::Core
           canvas::canvas
           messagefacility::MF_MessageLogger
+          TBB::tbb
         )
 
 install_source()
diff --git a/larrecodnn/ImagePatternAlgs/Modules/CMakeLists.txt b/larrecodnn/ImagePatternAlgs/Modules/CMakeLists.txt
index b5fd6f0..de51b46 100644
--- a/larrecodnn/ImagePatternAlgs/Modules/CMakeLists.txt
+++ b/larrecodnn/ImagePatternAlgs/Modules/CMakeLists.txt
@@ -1,28 +1,31 @@
 art_make( MODULE_LIBRARIES
-        larreco_RecoAlg_ImagePatternAlgs_DataProvider
-	lardata_ArtDataHelper
-        lardataobj_RawData
-        art_Framework_Services_System_TriggerNamesService_service
-        larsim_MCCheater_ParticleInventoryService_service
-                       larreco_Calorimetry
-                       larcorealg_Geometry
-                       lardataalg_DetectorInfo
-                       lardataobj_RecoBase
-                       nusimdata_SimulationBase
-                       nurandom_RandomUtils_NuRandomService_service
+        larreco::RecoAlg_ImagePatternAlgs_DataProvider
+	lardata::ArtDataHelper
+        lardataobj::RawData
+        art::Framework_Services_System_TriggerNamesService_service
+        larsim::MCCheater_ParticleInventoryService_service
+                       larreco::Calorimetry
+                       larcorealg::Geometry
+                       lardataalg::DetectorInfo
+                       lardataobj::RecoBase
+                       nusimdata::SimulationBase
+                       nurandom::RandomUtils_NuRandomService_service
                        art::Framework_Core
                        art::Framework_Principal
                        art::Framework_Services_Registry
                        art_root_io::tfile_support
                        art_root_io::TFileService_service
-                       ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
+                       art::Framework_Services_System_FloatingPointControl_service
                        art::Persistency_Common
                        art::Persistency_Provenance
                        art::Utilities
                        canvas::canvas
                        messagefacility::MF_MessageLogger
+                       ROOT::Physics
+                       ROOT::Hist
+                       ROOT::MathCore
+                       ROOT::Tree
                        CLHEP::CLHEP
-                       ${ROOT_BASIC_LIB_LIST}
                        ROOT::Core
 )
 
diff --git a/larrecodnn/ImagePatternAlgs/Tensorflow/Modules/CMakeLists.txt b/larrecodnn/ImagePatternAlgs/Tensorflow/Modules/CMakeLists.txt
index 18c31a0..71c88ef 100644
--- a/larrecodnn/ImagePatternAlgs/Tensorflow/Modules/CMakeLists.txt
+++ b/larrecodnn/ImagePatternAlgs/Tensorflow/Modules/CMakeLists.txt
@@ -1,9 +1,9 @@
 art_make( MODULE_LIBRARIES
-	lardata_ArtDataHelper
-        lardataobj_RawData
+	lardata::ArtDataHelper
+        lardataobj::RawData
 	larrecodnn_ImagePatternAlgs_Tensorflow_PointIdAlg
-        art_Framework_Services_System_TriggerNamesService_service
-        larsim_MCCheater_ParticleInventoryService_service
+        art::Framework_Services_System_TriggerNamesService_service
+        larsim::MCCheater_ParticleInventoryService_service
 )
 
 install_headers()
diff --git a/larrecodnn/ImagePatternAlgs/Tensorflow/PointIdAlg/CMakeLists.txt b/larrecodnn/ImagePatternAlgs/Tensorflow/PointIdAlg/CMakeLists.txt
index 0125f3a..2a763f0 100644
--- a/larrecodnn/ImagePatternAlgs/Tensorflow/PointIdAlg/CMakeLists.txt
+++ b/larrecodnn/ImagePatternAlgs/Tensorflow/PointIdAlg/CMakeLists.txt
@@ -1,20 +1,20 @@
 include_directories($ENV{TENSORFLOW_INC}/absl)
 
-art_make(LIB_LIBRARIES larreco_RecoAlg_ImagePatternAlgs_DataProvider
-                       larrecodnn_ImagePatternAlgs_Keras
-                       larrecodnn_ImagePatternAlgs_Tensorflow_TF
-                       larcore_Geometry_Geometry_service
-                       larcorealg_Geometry
-                       lardataalg_DetectorInfo
-                       lardataobj_RecoBase
-                       larreco_Calorimetry
-                       nusimdata_SimulationBase
+art_make(LIB_LIBRARIES larreco::RecoAlg_ImagePatternAlgs_DataProvider
+                       larrecodnn::ImagePatternAlgs_Keras
+                       larrecodnn::ImagePatternAlgs_Tensorflow_TF
+                       larcore::Geometry_Geometry_service
+                       larcorealg::Geometry
+                       lardataalg::DetectorInfo
+                       lardataobj::RecoBase
+                       larreco::Calorimetry
+                       nusimdata::SimulationBase
                        art::Framework_Core
                        art::Framework_Principal
                        art::Framework_Services_Registry
                        art_root_io::tfile_support
                        art_root_io::TFileService_service
-                       ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
+                       art::Framework_Services_System_FloatingPointControl_service
                        art::Persistency_Common
                        art::Persistency_Provenance
                        art::Utilities
@@ -22,7 +22,6 @@ art_make(LIB_LIBRARIES larreco_RecoAlg_ImagePatternAlgs_DataProvider
                        messagefacility::MF_MessageLogger
                        fhiclcpp::fhiclcpp
                        cetlib::cetlib cetlib_except
-                       ${ROOT_BASIC_LIB_LIST}
                        ROOT::Core
                        ROOT::Minuit
                        ROOT::Minuit2
diff --git a/larrecodnn/ImagePatternAlgs/Tensorflow/Tools/CMakeLists.txt b/larrecodnn/ImagePatternAlgs/Tensorflow/Tools/CMakeLists.txt
index 2457899..8767a36 100644
--- a/larrecodnn/ImagePatternAlgs/Tensorflow/Tools/CMakeLists.txt
+++ b/larrecodnn/ImagePatternAlgs/Tensorflow/Tools/CMakeLists.txt
@@ -2,9 +2,9 @@ include_directories( $ENV{TENSORFLOW_INC}/absl )
 
 art_make(
           TOOL_LIBRARIES
-          larreco_RecoAlg_ImagePatternAlgs_DataProvider
-          larrecodnn_ImagePatternAlgs_Tensorflow_TF
-          larcorealg_Geometry
+          larreco::RecoAlg_ImagePatternAlgs_DataProvider
+          larrecodnn::ImagePatternAlgs_Tensorflow_TF
+          larcorealg::Geometry
           ROOT::Core
           canvas::canvas
           messagefacility::MF_MessageLogger
diff --git a/larrecodnn/ImagePatternAlgs/Triton/Tools/CMakeLists.txt b/larrecodnn/ImagePatternAlgs/Triton/Tools/CMakeLists.txt
index 0432c24..35ef93e 100644
--- a/larrecodnn/ImagePatternAlgs/Triton/Tools/CMakeLists.txt
+++ b/larrecodnn/ImagePatternAlgs/Triton/Tools/CMakeLists.txt
@@ -1,7 +1,7 @@
 art_make(
           TOOL_LIBRARIES
-          larreco_RecoAlg_ImagePatternAlgs_DataProvider
-          larcorealg_Geometry
+          larreco::RecoAlg_ImagePatternAlgs_DataProvider
+          larcorealg::Geometry
           larrecodnn_ImagePatternAlgs_NuSonic_Triton
           ROOT::Core
           canvas::canvas
