diff --git a/CMakeLists.txt b/CMakeLists.txt
index 854c5dc..79d813e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,10 +11,11 @@
 #  make install
 #  make package (builds distribution tarfile)
 # ======================================================================
+cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
 
 
-cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
 
+find_package(cetmodules 2.29.02 REQUIRED)
 project(sbnobj VERSION 09.12.09 LANGUAGES CXX)
 
 message(STATUS
@@ -22,10 +23,9 @@ message(STATUS
   "\n-- === *** Start of configuration of ${PROJECT_NAME} ***"
   "\n-- ===")
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
+# cetbuildtools contains our cmake modules ### MIGRATE-ACTION-REQUIRED: remove
 
-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules)
+list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules) ### MIGRATE-ACTION-REQUIRED: remove
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -37,25 +37,26 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
   EXTRA_CXX_FLAGS -Wno-unused-local-typedefs
 )
 
-cet_report_compiler_flags()
+cet_report_compiler_flags(REPORT_THRESHOLD VERBOSE)
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( canvas )
-find_ups_product( larcoreobj )
-find_ups_product( lardataobj )
-find_ups_product( larcorealg )
-find_ups_product( lardataalg )
-find_ups_product( nusimdata )
-find_ups_product(clhep)
-find_ups_boost( )
-find_ups_root()
+find_package( messagefacility )
+find_package( canvas ) ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
+find_package( larcoreobj ) ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
+find_package( lardataobj ) ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
+find_package( larcorealg ) ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
+find_package( lardataalg ) ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
+find_package( nusimdata ) ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
+find_package( Boost COMPONENTS system)
+find_package( ROOT )
+find_package( CLHEP COMPONENTS All)
 
 # macros for dictionary and simple_plugin
 include(ArtDictionary)
 include(CetMake)
 include(BasicPlugin)
 
-# add cet_find_library commands here when needed
+# add cet_find_library commands here when needed ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
 
 # ADD SOURCE CODE SUBDIRECTORIES HERE
 add_subdirectory(sbnobj)
@@ -65,10 +66,8 @@ add_subdirectory(test)
 
 # ups - table and config files
 # must be AFTER all other subdirectories
-add_subdirectory(ups)
 
 # packaging utility - must be last
-include(UseCPack)
 
 
 message(STATUS
@@ -77,3 +76,4 @@ message(STATUS
   "\n-- ============================================================================="
   "\n-- "
   )
+cet_cmake_config()
diff --git a/sbnobj/Common/Analysis/CMakeLists.txt b/sbnobj/Common/Analysis/CMakeLists.txt
index 0fd15e3..23e6a8a 100644
--- a/sbnobj/Common/Analysis/CMakeLists.txt
+++ b/sbnobj/Common/Analysis/CMakeLists.txt
@@ -1,10 +1,10 @@
-cet_make( 
+cet_make(  ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
   LIBRARIES
-    cetlib_except
+    cetlib_except::cetlib_except
   NO_DICTIONARY
   )
 
-art_dictionary(DICTIONARY_LIBRARIES sbnobj_Common_Analysis)
+art_dictionary(DICTIONARY_LIBRARIES sbnobj::Common_Analysis)
 
 install_headers()
 install_source()
diff --git a/sbnobj/Common/CRT/CMakeLists.txt b/sbnobj/Common/CRT/CMakeLists.txt
index fc4b0f3..2454a3c 100644
--- a/sbnobj/Common/CRT/CMakeLists.txt
+++ b/sbnobj/Common/CRT/CMakeLists.txt
@@ -1,10 +1,11 @@
-cet_make( 
+cet_make(  ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
   LIBRARIES
-    cetlib_except
+    cetlib_except::cetlib_except
+    lardataobj::Simulation
   NO_DICTIONARY
   )
 
-art_dictionary(DICTIONARY_LIBRARIES sbnobj_Common_CRT)
+art_dictionary(DICTIONARY_LIBRARIES sbnobj::Common_CRT)
 
 install_headers()
 install_source()
diff --git a/sbnobj/Common/EventGen/MeVPrtl/CMakeLists.txt b/sbnobj/Common/EventGen/MeVPrtl/CMakeLists.txt
index d81f1da..9c8e559 100644
--- a/sbnobj/Common/EventGen/MeVPrtl/CMakeLists.txt
+++ b/sbnobj/Common/EventGen/MeVPrtl/CMakeLists.txt
@@ -1,11 +1,15 @@
-cet_make( 
+cet_find_library( DK2NU_TREE NAMES dk2nuTree PATHS ENV DK2NUDATA_LIB NO_DEFAULT_PATH ) ### MIGRATE-ACTION-RECOMMENDED: use of ENV ... may be UPS-dependent ### MIGRATE-ACTION-RECOMME ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
+include_directories( $ENV{DK2NUDATA_INC} ) ### MIGRATE-ACTION-RECOMMENDED: use target_link_directories() with target semantics
+
+cet_make(  ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
   LIBRARIES
-    cetlib_except
+    cetlib_except::cetlib_except
+    nusimdata::SimulationBase
     ${ROOT_BASIC_LIB_LIST}
   NO_DICTIONARY
   )
 
-art_dictionary(DICTIONARY_LIBRARIES sbnobj_Common_EventGen_MeVPrtl)
+art_dictionary(DICTIONARY_LIBRARIES sbnobj::Common_EventGen_MeVPrtl nusimdata::SimulationBase)
 
 install_headers()
 install_source()
diff --git a/sbnobj/Common/PMT/Data/CMakeLists.txt b/sbnobj/Common/PMT/Data/CMakeLists.txt
index 4208c34..f1c07e4 100644
--- a/sbnobj/Common/PMT/Data/CMakeLists.txt
+++ b/sbnobj/Common/PMT/Data/CMakeLists.txt
@@ -1,9 +1,10 @@
-cet_make(
+cet_make( ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
   LIBRARIES
+  lardataobj::RawData
   NO_DICTIONARY
   )
 
-art_dictionary(DICTIONARY_LIBRARIES sbnobj_Common_PMT_Data)
+art_dictionary(DICTIONARY_LIBRARIES sbnobj::Common_PMT_Data)
 
 install_headers()
 install_source()
diff --git a/sbnobj/Common/POTAccounting/CMakeLists.txt b/sbnobj/Common/POTAccounting/CMakeLists.txt
index e62a34c..a0dfc05 100644
--- a/sbnobj/Common/POTAccounting/CMakeLists.txt
+++ b/sbnobj/Common/POTAccounting/CMakeLists.txt
@@ -1,7 +1,7 @@
 cet_make( 
   LIBRARIES
-    cetlib_except
-    ${ROOT_BASIC_LIB_LIST}
+    cetlib_except::cetlib_except
+    ROOT::Core
   NO_DICTIONARY
   )
 
diff --git a/sbnobj/Common/Reco/CMakeLists.txt b/sbnobj/Common/Reco/CMakeLists.txt
index eb5077f..6032f15 100644
--- a/sbnobj/Common/Reco/CMakeLists.txt
+++ b/sbnobj/Common/Reco/CMakeLists.txt
@@ -1,12 +1,14 @@
-cet_make( 
+cet_make(  ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
   LIBRARIES
-    cetlib_except
-    ${MF_MESSAGELOGGER}
+    cetlib_except::cetlib_except
+    messagefacility::MF_MessageLogger
+    larcorealg::Geometry
+    lardataobj::AnalysisBase
     ${ROOT_BASIC_LIB_LIST}
   NO_DICTIONARY
   )
 
-art_dictionary(DICTIONARY_LIBRARIES sbnobj_Common_Reco)
+art_dictionary(DICTIONARY_LIBRARIES sbnobj::Common_Reco)
 
 install_headers()
 install_source()
diff --git a/sbnobj/Common/SBNEventWeight/CMakeLists.txt b/sbnobj/Common/SBNEventWeight/CMakeLists.txt
index 555921e..fe555cc 100644
--- a/sbnobj/Common/SBNEventWeight/CMakeLists.txt
+++ b/sbnobj/Common/SBNEventWeight/CMakeLists.txt
@@ -1,15 +1,20 @@
-cet_make( 
+cet_find_library( DK2NU_TREE NAMES dk2nuTree PATHS ENV DK2NUDATA_LIB NO_DEFAULT_PATH ) ### MIGRATE-ACTION-RECOMMENDED: use of ENV ... may be UPS-dependent ### MIGRATE-ACTION-RECOMME ### MIGRATE-ACTION-RECOMMENDED: use cet_find_package() with target semantics for linking
+include_directories( $ENV{DK2NUDATA_INC} ) ### MIGRATE-ACTION-RECOMMENDED: use target_link_directories() with target semantics
+
+cet_make(  ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
   NO_DICTIONARY
   LIBRARIES
-    cetlib_except
-    ${ROOT_BASIC_LIB_LIST}
-    ${CLHEP}
+    cetlib_except::cetlib_except
+    ROOT::Matrix
+    ROOT::Core
+    CLHEP::CLHEP
+    nusimdata::SimulationBase
     ${ART_FRAMEWORK_SERVICES_REGISTRY}
     ${ART_FRAMEWORK_SERVICES_OPTIONAL}
     ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
 )
 
-art_dictionary(DICTIONARY_LIBRARIES sbnobj_Common_SBNEventWeight)
+art_dictionary(DICTIONARY_LIBRARIES sbnobj::Common_SBNEventWeight)
 
 install_headers()
 install_source()
diff --git a/sbnobj/ICARUS/CRT/CMakeLists.txt b/sbnobj/ICARUS/CRT/CMakeLists.txt
index f8960f0..1b17f2e 100644
--- a/sbnobj/ICARUS/CRT/CMakeLists.txt
+++ b/sbnobj/ICARUS/CRT/CMakeLists.txt
@@ -1,10 +1,11 @@
-cet_make( 
+cet_make(  ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
   LIBRARIES
-    cetlib_except
+    cetlib_except::cetlib_except
+    lardataobj::Simulation
   NO_DICTIONARY
   )
 
-art_dictionary(DICTIONARY_LIBRARIES sbnobj_ICARUS_CRT)
+art_dictionary(DICTIONARY_LIBRARIES sbnobj::ICARUS_CRT)
 
 install_headers()
 install_source()
diff --git a/sbnobj/ICARUS/PMT/Data/CMakeLists.txt b/sbnobj/ICARUS/PMT/Data/CMakeLists.txt
index 73f4279..54ae671 100644
--- a/sbnobj/ICARUS/PMT/Data/CMakeLists.txt
+++ b/sbnobj/ICARUS/PMT/Data/CMakeLists.txt
@@ -1,9 +1,9 @@
-cet_make(
+cet_make( ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
   LIBRARIES
   NO_DICTIONARY
   )
 
-art_dictionary(DICTIONARY_LIBRARIES sbnobj_ICARUS_PMT_Data)
+art_dictionary(DICTIONARY_LIBRARIES sbnobj::ICARUS_PMT_Data lardataobj::RawData)
 
 install_headers()
 install_source()
diff --git a/sbnobj/ICARUS/PMT/Trigger/Data/CMakeLists.txt b/sbnobj/ICARUS/PMT/Trigger/Data/CMakeLists.txt
index e394eb3..61866b3 100644
--- a/sbnobj/ICARUS/PMT/Trigger/Data/CMakeLists.txt
+++ b/sbnobj/ICARUS/PMT/Trigger/Data/CMakeLists.txt
@@ -1,11 +1,13 @@
-cet_make( 
+cet_make(  ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
   LIBRARIES
-    lardataobj_RawData
-    cetlib_except
+    lardataobj::RawData
+    larcorealg::CoreUtils
+    lardataalg::DetectorInfo
+    cetlib_except::cetlib_except
   NO_DICTIONARY
   )
 
-art_dictionary(DICTIONARY_LIBRARIES sbnobj_ICARUS_PMT_Trigger_Data)
+art_dictionary(DICTIONARY_LIBRARIES sbnobj::ICARUS_PMT_Trigger_Data)
 
 install_headers()
 install_source()
diff --git a/sbnobj/SBND/CRT/CMakeLists.txt b/sbnobj/SBND/CRT/CMakeLists.txt
index 3de9938..7652999 100644
--- a/sbnobj/SBND/CRT/CMakeLists.txt
+++ b/sbnobj/SBND/CRT/CMakeLists.txt
@@ -1,10 +1,11 @@
-cet_make( 
+cet_make(  ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
   LIBRARIES
-    cetlib_except
+    cetlib_except::cetlib_except
+    lardataobj::Simulation
   NO_DICTIONARY
   )
 
-art_dictionary(DICTIONARY_LIBRARIES sbnobj_SBND_CRT)
+art_dictionary(DICTIONARY_LIBRARIES sbnobj::SBND_CRT)
 
 install_headers()
 install_source()
diff --git a/sbnobj/SBND/Commissioning/CMakeLists.txt b/sbnobj/SBND/Commissioning/CMakeLists.txt
index d83ec1f..d67e27a 100644
--- a/sbnobj/SBND/Commissioning/CMakeLists.txt
+++ b/sbnobj/SBND/Commissioning/CMakeLists.txt
@@ -1,10 +1,10 @@
-cet_make( 
+cet_make(  ### MIGRATE-ACTION-RECOMMENDED: use cet_make_library(), build_dictionary(), basic_plugin() with explicit source lists
   LIBRARIES
-    cetlib_except
+    cetlib_except::cetlib_except
   NO_DICTIONARY
   )
 
-art_dictionary(DICTIONARY_LIBRARIES sbnobj_SBND_Commissioning)
+art_dictionary(DICTIONARY_LIBRARIES sbnobj_SBND_Commissioning lardataobj::RecoBase)
 
 install_headers()
 install_source()
diff --git a/sbnobj/SBND/Trigger/CMakeLists.txt b/sbnobj/SBND/Trigger/CMakeLists.txt
index 90adbd4..5df630a 100644
--- a/sbnobj/SBND/Trigger/CMakeLists.txt
+++ b/sbnobj/SBND/Trigger/CMakeLists.txt
@@ -1,6 +1,6 @@
 cet_make( 
   LIBRARIES
-    cetlib_except
+    cetlib_except::cetlib_except
   NO_DICTIONARY
   )
 
