diff --git a/CMakeLists.txt b/CMakeLists.txt
index ebadeeb..ba5414f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,10 +10,9 @@ cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
 
 project(lareventdisplay VERSION 09.02.05 LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
-
-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules $ENV{ART_DIR}/Modules)
+find_package( cetmodules REQUIRED )
+find_package( canvas_root_io )
+find_package( art )
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -27,23 +26,24 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( eigen )
-find_ups_product( larcoreobj )
-find_ups_product( larcorealg )
-find_ups_product( larcore )
-find_ups_product( lardataobj )
-find_ups_product( lardataalg )
-find_ups_product( lardata )
-find_ups_product( larevt )
-find_ups_product( larsim )
-find_ups_product( larreco )
-find_ups_product( nuevdb )
-find_ups_product( nug4 )
-find_ups_product( nugen )
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product( postgresql )
-find_ups_boost()
+find_package( Eigen3 )
+find_package( larcoreobj )
+find_package( larcorealg )
+find_package( larcore )
+find_package( lardataobj )
+find_package( lardataalg )
+find_package( lardata )
+find_package( larevt )
+find_package( larsim )
+find_package( larreco )
+find_package( nuevdb )
+find_package( nug4 )
+find_package( nugen )
+find_package( nusimdata )
+find_package( art )
+find_package( art_root_io )
+find_package( PostgreSQL )
+find_package( Boost COMPONENTS system )
 
 # use the system zlib
 find_package( ZLIB REQUIRED )
@@ -54,13 +54,13 @@ include(ArtDictionary)
 include(ArtMake)
 include(BuildPlugins)
 
-include_directories( $ENV{EIGEN_INC} )
+include_directories( ${EIGEN3_INCLUDE_DIR} )
 
 # source
 add_subdirectory(lareventdisplay)
-
 # ups - table and config files
-add_subdirectory(ups)
+#add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
+#include(UseCPack)
+cet_cmake_config()
diff --git a/lareventdisplay/EventDisplay/3DDrawers/CMakeLists.txt b/lareventdisplay/EventDisplay/3DDrawers/CMakeLists.txt
index 279b6ef..5141a05 100644
--- a/lareventdisplay/EventDisplay/3DDrawers/CMakeLists.txt
+++ b/lareventdisplay/EventDisplay/3DDrawers/CMakeLists.txt
@@ -3,11 +3,13 @@ cet_enable_asserts()
 art_make(TOOL_LIBRARIES
            art::Framework_Principal
            canvas::canvas
-           larcorealg_Geometry
-           lardataalg_DetectorInfo
-           lardataobj_RecoBase
-           lareventdisplay_EventDisplay_ColorDrawingOptions_service
-           lareventdisplay_EventDisplay_RecoDrawingOptions_service)
+           larcorealg::Geometry
+           lardataalg::DetectorInfo
+           lardataobj::RecoBase
+           lareventdisplay::EventDisplay_ColorDrawingOptions_service
+           lareventdisplay::EventDisplay_RecoDrawingOptions_service
+           ${PandoraMonitoring}
+           nuevdb::EventDisplayBase)
 
 install_headers()
 install_source()
diff --git a/lareventdisplay/EventDisplay/CMakeLists.txt b/lareventdisplay/EventDisplay/CMakeLists.txt
index dba76a9..f115e19 100644
--- a/lareventdisplay/EventDisplay/CMakeLists.txt
+++ b/lareventdisplay/EventDisplay/CMakeLists.txt
@@ -17,7 +17,7 @@ art_make_library(
   LIBRARIES
     art::Framework_Principal
     art::Framework_Services_Registry
-    ${ART_PERSISTENCY_PROVENANCE}
+    art::Persistency_Provenance
     messagefacility::MF_MessageLogger
     ROOT::Core
     ROOT::EG
@@ -30,49 +30,51 @@ art_make_library(
     ROOT::RIO
     canvas::canvas
     cetlib_except::cetlib_except
-    lardataalg_DetectorInfo
-    larcorealg_Geometry
-    lardata_ArtDataHelper
-    lardata_Utilities
-    lardataobj_RawData
-    lareventdisplay_EventDisplay_AnalysisDrawingOptions_service
-    lareventdisplay_EventDisplay_ColorDrawingOptions_service
-    lareventdisplay_EventDisplay_EvdLayoutOptions_service
-    lareventdisplay_EventDisplay_InfoTransfer_service
-    lareventdisplay_EventDisplay_RawDrawingOptions_service
-    lareventdisplay_EventDisplay_RecoDrawingOptions_service
-    lareventdisplay_EventDisplay_SimulationDrawingOptions_service
-    larsim_MCCheater_ParticleInventoryService_service
-    larsim_Simulation
-    nusimdata_SimulationBase
-    nuevdb_EventDisplayBase
+    ROOT::TreePlayer
+    lardataalg::DetectorInfo
+    larcorealg::Geometry
+    lardata::ArtDataHelper
+    lardata::Utilities
+    lardataobj::RawData
+    lardataobj::RecoBase
+    lareventdisplay::EventDisplay_AnalysisDrawingOptions_service
+    lareventdisplay::EventDisplay_ColorDrawingOptions_service
+    lareventdisplay::EventDisplay_EvdLayoutOptions_service
+    lareventdisplay::EventDisplay_InfoTransfer_service
+    lareventdisplay::EventDisplay_RawDrawingOptions_service
+    lareventdisplay::EventDisplay_RecoDrawingOptions_service
+    lareventdisplay::EventDisplay_SimulationDrawingOptions_service
+    larsim::MCCheater_ParticleInventoryService_service
+    larsim::Simulation
+    nusimdata::SimulationBase
+    nuevdb::EventDisplayBase
 )
 
 simple_plugin(GraphCluster "module" lareventdisplay_EventDisplay)
 simple_plugin(EVD "module" lareventdisplay_EventDisplay)
 
-simple_plugin(AnalysisDrawingOptions "service" nuevdb_EventDisplayBase)
-simple_plugin(EvdLayoutOptions "service" nuevdb_EventDisplayBase)
-simple_plugin(RawDrawingOptions "service" nuevdb_EventDisplayBase)
-simple_plugin(RecoDrawingOptions "service" nuevdb_EventDisplayBase)
-simple_plugin(SimulationDrawingOptions "service" nuevdb_EventDisplayBase)
+simple_plugin(AnalysisDrawingOptions "service" nuevdb::EventDisplayBase)
+simple_plugin(EvdLayoutOptions "service" nuevdb::EventDisplayBase)
+simple_plugin(RawDrawingOptions "service" nuevdb::EventDisplayBase)
+simple_plugin(RecoDrawingOptions "service" nuevdb::EventDisplayBase)
+simple_plugin(SimulationDrawingOptions "service" nuevdb::EventDisplayBase)
 
 simple_plugin(ColorDrawingOptions "service"
-              ${ART_PERSISTENCY_PROVENANCE}
+              art::Persistency_Provenance
               messagefacility::MF_MessageLogger
               ROOT::Core
               ROOT::Graf3d
               ROOT::Hist
               ROOT::MathCore
-              larcorealg_Geometry
-              lardataobj_RecoBase
-              nuevdb_EventDisplayBase)
+              larcorealg::Geometry
+              lardataobj::RecoBase
+              nuevdb::EventDisplayBase)
 
 simple_plugin(InfoTransfer "service"
-              larcorealg_Geometry
-              nuevdb_EventDisplayBase
+              larcorealg::Geometry
+              nuevdb::EventDisplayBase
               art::Framework_Principal
-              ${ART_PERSISTENCY_PROVENANCE}
+              art::Persistency_Provenance
               messagefacility::MF_MessageLogger
               ROOT::Core)
 
diff --git a/lareventdisplay/EventDisplay/ExptDrawers/CMakeLists.txt b/lareventdisplay/EventDisplay/ExptDrawers/CMakeLists.txt
index eaeeeba..88ccb22 100644
--- a/lareventdisplay/EventDisplay/ExptDrawers/CMakeLists.txt
+++ b/lareventdisplay/EventDisplay/ExptDrawers/CMakeLists.txt
@@ -6,9 +6,9 @@ art_make(
     ROOT::Core
     ROOT::Physics
     canvas::canvas
-    larcorealg_Geometry
-    lareventdisplay_EventDisplay_RawDrawingOptions_service
-    nuevdb_EventDisplayBase
+    larcorealg::Geometry
+    lareventdisplay::EventDisplay_RawDrawingOptions_service
+    nuevdb::EventDisplayBase
   )
 
 install_headers()
diff --git a/lareventdisplay/EventDisplay/SimDrawers/CMakeLists.txt b/lareventdisplay/EventDisplay/SimDrawers/CMakeLists.txt
index 586d898..55e49cf 100644
--- a/lareventdisplay/EventDisplay/SimDrawers/CMakeLists.txt
+++ b/lareventdisplay/EventDisplay/SimDrawers/CMakeLists.txt
@@ -9,13 +9,13 @@ art_make(
     ROOT::EG
     ROOT::Physics
     canvas::canvas
-    larcorealg_Geometry
+    larcorealg::Geometry
     lareventdisplay_EventDisplay
     lareventdisplay_EventDisplay_ColorDrawingOptions_service
     lareventdisplay_EventDisplay_SimulationDrawingOptions_service
-    larsim_Simulation
-    nusimdata_SimulationBase
-    nuevdb_EventDisplayBase
+    larsim::Simulation
+    nusimdata::SimulationBase
+    nuevdb::EventDisplayBase
 )
 
 install_headers()
diff --git a/lareventdisplay/EventDisplay/wfHitDrawers/CMakeLists.txt b/lareventdisplay/EventDisplay/wfHitDrawers/CMakeLists.txt
index abaebe1..b7ce361 100644
--- a/lareventdisplay/EventDisplay/wfHitDrawers/CMakeLists.txt
+++ b/lareventdisplay/EventDisplay/wfHitDrawers/CMakeLists.txt
@@ -4,14 +4,17 @@ art_make(
   TOOL_LIBRARIES
     art::Framework_Principal
     art::Framework_Services_Registry
+    ROOT::Hist
     ROOT::Core
     canvas::canvas
-    lardata_ArtDataHelper
-    lardataobj_RawData
-    lareventdisplay_EventDisplay_ColorDrawingOptions_service
-    lareventdisplay_EventDisplay_RawDrawingOptions_service
-    lareventdisplay_EventDisplay_RecoDrawingOptions_service
-    nuevdb_EventDisplayBase
+    lardata::ArtDataHelper
+    lardataobj::RawData
+    lardataobj::RecoBase
+    larcorealg::Geometry
+    lareventdisplay::EventDisplay_ColorDrawingOptions_service
+    lareventdisplay::EventDisplay_RawDrawingOptions_service
+    lareventdisplay::EventDisplay_RecoDrawingOptions_service
+    nuevdb::EventDisplayBase
 )
 
 install_headers()
