--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,7 +11,7 @@
 project(larvecutils VERSION 09.00.01 LANGUAGES CXX)
 
 # cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
+find_package(cetmodules REQUIRED)
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -36,11 +36,12 @@
 add_subdirectory(larvecutils)
 
 # tests
-add_subdirectory(test)
+#add_subdirectory(test)
 
 # ups - table and config files
 # must be AFTER all other subdirectories
-add_subdirectory(ups)
+#add_subdirectory(ups)
 
 # packaging utility - must be last
-include(UseCPack)
+#include(UseCPack)
+cet_cmake_config()
