diff --git a/CMakeLists.txt b/CMakeLists.txt
index a55337a..7394db5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,10 +15,9 @@
 
 cmake_minimum_required(VERSION 3.18.2 FATAL_ERROR)
 
-project(nug4 LANGUAGES CXX)
+project(nug4 VERSION 1.07.01 LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
+find_package(cetmodules REQUIRED)
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -32,18 +31,41 @@ cet_set_compiler_flags(DIAGS CAUTIOUS WERROR
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product(canvas_root_io)
-find_ups_product( nusimdata )
-find_ups_product( clhep )
-find_ups_product( cetlib )
-find_ups_product( cetlib_except )
-find_ups_geant4( )
-find_ups_root()
+find_package(art_root_io)
+find_package(canvas_root_io)
+find_package( nusimdata )
+find_package( clhep )
+find_package( cetlib )
+find_package( cetlib_except )
+find_package( ROOT REQUIRED COMPONENTS MathCore Geom GeomPainter GX11 EG EGPythia6 Gui)
+find_package( art )
+find_package( Geant4 REQUIRED CONFIG OPTIONAL_COMPONENTS 
+    Geant4::G4FR
+    Geant4::G4GMocren
+    Geant4::G4RayTracer
+    Geant4::G4Tree
+    Geant4::G4VRML
+    Geant4::G4event
+    Geant4::G4geometry
+    Geant4::G4global
+    Geant4::G4intercoms
+    Geant4::G4materials
+    Geant4::G4modeling
+    Geant4::G4particles
+    Geant4::G4persistency
+    Geant4::G4physicslists
+    Geant4::G4processes
+    Geant4::G4readout
+    Geant4::G4run
+    Geant4::G4tracking
+    Geant4::G4visHepRep
+    Geant4::G4vis_management
+)
 
 message(STATUS "ROOTSYS is ${ROOTSYS}")
 
+find_package(messagefacility)
+
 # macros
 include(ArtMake)
 include(BuildPlugins)
@@ -51,9 +73,12 @@ include(CetRootCint)
 
 # source
 add_subdirectory (nug4)
+add_subdirectory (cmakeConfig)
 
 # ups - table and config files
+if(WANT_UPS)
 add_subdirectory(ups)
+endif()
 
 # packaging utility
 include(UseCPack)
diff --git a/cmakeConfig/CMakeLists.txt b/cmakeConfig/CMakeLists.txt
new file mode 100644
index 0000000..4c8b1af
--- /dev/null
+++ b/cmakeConfig/CMakeLists.txt
@@ -0,0 +1,3 @@
+
+# create package configuration and version files
+cet_cmake_config(CONFIG_FRAGMENTS nug4-fragment.cmake.in)
diff --git a/cmakeConfig/nug4-fragment.cmake.in b/cmakeConfig/nug4-fragment.cmake.in
new file mode 100644
index 0000000..d022213
--- /dev/null
+++ b/cmakeConfig/nug4-fragment.cmake.in
@@ -0,0 +1,8 @@
+find_package( ROOT REQUIRED COMPONENTS MathCore Geom GeomPainter GX11 EG EGPythia6 Gui)
+find_package( art REQUIRED)
+find_package( art_root_io REQUIRED)
+find_package( cetlib REQUIRED)
+find_package( cetlib_except REQUIRED)
+find_package( nusimdata REQUIRED )
+find_package( Geant4 REQUIRED)
+
diff --git a/nug4/G4Base/CMakeLists.txt b/nug4/G4Base/CMakeLists.txt
index 7cb487c..4b538e3 100644
--- a/nug4/G4Base/CMakeLists.txt
+++ b/nug4/G4Base/CMakeLists.txt
@@ -1,35 +1,35 @@
 
 art_make( LIB_LIBRARIES nusimdata::SimulationBase
-                        nug4_MagneticFieldServices_MagneticFieldServiceStandard_service
-                        nug4_MagneticField
-			                  canvas::canvas
-                        ART_FRAMEWORK_SERVICES_REGISTRY
-                        MF_MESSAGELOGGER
-                        FHICLCPP
+                        nug4::MagneticFieldServices_MagneticFieldServiceStandard_service
+                        nug4::MagneticField
+			canvas::canvas
+                        art::Framework_Services_Registry
+                        messagefacility::MF_MessageLogger
+                        messagefacility::headers
+                        fhiclcpp::fhiclcpp fhiclcpp::types
                         cetlib::cetlib
                         cetlib_except::cetlib_except
-                        G4FR
-                        G4GMOCREN
-                        G4RAYTRACER
-                        G4TREE
-                        G4VRML
-                        G4EVENT
-                        G4GEOMETRY
-                        G4GLOBAL
-                        G4INTERCOMS
-                        G4MATERIALS
-                        G4MODELING
-                        G4PARTICLES
-                        G4PERSISTENCY
-                        G4PHYSICSLISTS
-                        G4PROCESSES
-                        G4READOUT
-                        G4RUN
-                        G4TRACKING
-                        G4VISHEPREP
-                        G4VIS_MANAGEMENT
-                        XERCESC
-                        CLHEP
+                        Geant4::G4FR
+                        Geant4::G4GMocren
+                        Geant4::G4RayTracer
+                        Geant4::G4Tree
+                        Geant4::G4VRML
+                        Geant4::G4event
+                        Geant4::G4geometry
+                        Geant4::G4global
+                        Geant4::G4intercoms
+                        Geant4::G4materials
+                        Geant4::G4modeling
+                        Geant4::G4particles
+                        Geant4::G4persistency
+                        Geant4::G4physicslists
+                        Geant4::G4processes
+                        Geant4::G4readout
+                        Geant4::G4run
+                        Geant4::G4tracking
+                        Geant4::G4visHepRep
+                        Geant4::G4vis_management
+                        ${XERCESC}
                         )
 
 install_headers()
diff --git a/nug4/MagneticField/CMakeLists.txt b/nug4/MagneticField/CMakeLists.txt
index 6fca3f3..6c11c1c 100644
--- a/nug4/MagneticField/CMakeLists.txt
+++ b/nug4/MagneticField/CMakeLists.txt
@@ -2,17 +2,14 @@ include_directories( ${CMAKE_CURRENT_SOURCE_DIR} )
 
 art_make(NO_PLUGINS
           LIB_LIBRARIES
-          G4GEOMETRY
           messagefacility::MF_MessageLogger
-          
           canvas::canvas
-          fhiclcpp::fhiclcpp
+          fhiclcpp::fhiclcpp fhiclcpp::types
           cetlib::cetlib
           cetlib_except::cetlib_except
-          CLHEP
-          ${ROOT_BASIC_LIB_LIST}
+          ${ROOT_LIBRARIES}
           ROOT::Geom
-          PUBLIC G4GEOMETRY
+          PUBLIC Geant4::G4geometry
          )
          
 install_headers()
diff --git a/nug4/MagneticFieldServices/CMakeLists.txt b/nug4/MagneticFieldServices/CMakeLists.txt
index c624dfd..c379ef3 100644
--- a/nug4/MagneticFieldServices/CMakeLists.txt
+++ b/nug4/MagneticFieldServices/CMakeLists.txt
@@ -1,13 +1,12 @@
 include_directories( ${CMAKE_CURRENT_SOURCE_DIR} )
 
 simple_plugin( MagneticFieldServiceStandard "service"
+            LIBRARIES
             messagefacility::MF_MessageLogger
-            
+            messagefacility::headers
             cetlib_except::cetlib_except
             nug4_MagneticField
-            G4GEOMETRY
-            CLHEP
-            ${ROOT_BASIC_LIB_LIST}
+            ${ROOT_LIBRARIES}
             ROOT::Geom )
 
 install_headers()
diff --git a/nug4/ParticleNavigation/CMakeLists.txt b/nug4/ParticleNavigation/CMakeLists.txt
index 8a90f1b..cf416a1 100644
--- a/nug4/ParticleNavigation/CMakeLists.txt
+++ b/nug4/ParticleNavigation/CMakeLists.txt
@@ -11,11 +11,11 @@ art_make( NO_PLUGINS
 			art::Persistency_Provenance
 			art::Utilities
 			canvas::canvas
+                        messagefacility::MF_MessageLogger
+                        cetlib::cetlib
+                        cetlib_except::cetlib_except
 			messagefacility::MF_MessageLogger
-			
-        		cetlib::cetlib
-			cetlib_except::cetlib_except
-			${ROOT_BASIC_LIB_LIST}
+			${ROOT_LIBRARIES}
         )
 
 install_headers()
