commit b7b2cb39de7cd40f3e68a2334444cb17285c444d
Author: Patrick Gartung <gartung@fnal.gov>
Date:   Thu Apr 2 14:40:37 2020 -0500

    Cetmodules

diff --git a/.spack_patched b/.spack_patched
new file mode 100755
index 0000000..e69de29
diff --git a/CMakeLists.txt b/CMakeLists.txt
index a53afe1..43db905 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,12 +13,13 @@
 #  make package (builds distribution tarfile)
 # ======================================================================
 
-cmake_minimum_required(VERSION 3.11)  # use cmake 3.11 or later
+cmake_minimum_required(VERSION 3.12)  # use cmake 3.12 or later
 
-project(nug4)
+project(nug4 VERSION 1.04.01 LANGUAGES CXX)
 
 # cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
+set(fcl_dir job)
+find_package(cetmodules REQUIRED)
 
 list(APPEND CMAKE_MODULE_PATH
   $ENV{CANVAS_ROOT_IO_DIR}/Modules
@@ -27,40 +28,29 @@ list(APPEND CMAKE_MODULE_PATH
 include(CetCMakeEnv)
 cet_cmake_env()
 
-# -D_GLIBCXX_USE_NANOSLEEP is needed to make std::this_thread::sleep_for available. (IFDatabase)
-cet_set_compiler_flags(DIAGS CAUTIOUS WERROR
-                       NO_UNDEFINED
-                       EXTRA_FLAGS -pedantic
-                       EXTRA_CXX_FLAGS -D_GLIBCXX_USE_NANOSLEEP
-                                       -Wno-unused-local-typedefs -Wno-variadic-macros)
-cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product( nusimdata )
-find_ups_product( cetlib )
-find_ups_product( cetlib_except )
-find_ups_geant4( )
-find_ups_product( cetbuildtools )
-
-message(STATUS "ROOTSYS is ${ROOTSYS}")
-
+find_package( art )
+find_package( art_root_io)
+find_package( nusimdata )
+find_package( cetlib )
+find_package( cetlib_except )
+find_package( Geant4 )
+include(${Geant4_USE_FILE})
 # macros
 include(ArtMake)
 include(BuildPlugins)
 include(CetRootCint)
 
 # include search path
-include_directories ( $ENV{G4INCLUDE} )
+include_directories ( ${Geant4_INCLUDE_DIRS} )
 include_directories ( $ENV{XERCES_C_INC} )
 
 # source
 add_subdirectory (nug4)
 
 # ups - table and config files
-add_subdirectory(ups)
+#add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
-
+add_subdirectory(cmakeConfig)
diff --git a/cmakeConfig/CMakeLists.txt b/cmakeConfig/CMakeLists.txt
new file mode 100644
index 0000000..4c8b1af
--- /dev/null
+++ b/cmakeConfig/CMakeLists.txt
@@ -0,0 +1,3 @@
+
+# create package configuration and version files
+cet_cmake_config(CONFIG_FRAGMENTS nug4-fragment.cmake.in)
diff --git a/cmakeConfig/nug4-fragment.cmake.in b/cmakeConfig/nug4-fragment.cmake.in
new file mode 100644
index 0000000..d022213
--- /dev/null
+++ b/cmakeConfig/nug4-fragment.cmake.in
@@ -0,0 +1,8 @@
+find_package( ROOT REQUIRED COMPONENTS MathCore Geom GeomPainter GX11 EG EGPythia6 Gui)
+find_package( art REQUIRED)
+find_package( art_root_io REQUIRED)
+find_package( cetlib REQUIRED)
+find_package( cetlib_except REQUIRED)
+find_package( nusimdata REQUIRED )
+find_package( Geant4 REQUIRED)
+
diff --git a/nug4/G4Base/CMakeLists.txt b/nug4/G4Base/CMakeLists.txt
index c0e4e71..af9a3d1 100644
--- a/nug4/G4Base/CMakeLists.txt
+++ b/nug4/G4Base/CMakeLists.txt
@@ -6,30 +6,9 @@ art_make( LIB_LIBRARIES nusimdata_SimulationBase
                         ${MF_UTILITIES}
                         ${FHICLCPP}
                         cetlib
-                        ${G4FR}
-                        ${G4GMOCREN}
-                        ${G4RAYTRACER}
-                        ${G4TREE}
-                        ${G4VRML}
-                        ${G4EVENT}
-                        ${G4GEOMETRY}
-                        ${G4GLOBAL}
-                        ${G4INTERCOMS}
-                        ${G4MATERIALS}
-                        ${G4MODELING}
-                        ${G4PARTICLES}
-                        ${G4PERSISTENCY}
-                        ${G4PHYSICSLISTS}
-                        ${G4PROCESSES}
-                        ${G4READOUT}
-                        ${G4RUN}
-                        ${G4TRACKING}
-                        ${G4VISHEPREP}
-                        ${G4VIS_MANAGEMENT}
-                        ${XERCESC}
-                        ${CLHEP}
+                        ${CLHEP_LIBRARIES}
+                        ${Geant4_LIBRARIES}
                         )
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/nug4/MagneticField/CMakeLists.txt b/nug4/MagneticField/CMakeLists.txt
index 5e74909..3f293e6 100644
--- a/nug4/MagneticField/CMakeLists.txt
+++ b/nug4/MagneticField/CMakeLists.txt
@@ -4,9 +4,8 @@ simple_plugin( MagneticField service
                ${MF_UTILITIES}
 	       cetlib_except
                ${CLHEP}
-               ${ROOT_BASIC_LIB_LIST}
-               ${ROOT_GEOM} )
+               ${ROOT_LIBRARIES}
+               ROOT::Geom )
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/nug4/ParticleNavigation/CMakeLists.txt b/nug4/ParticleNavigation/CMakeLists.txt
index fc27314..2c0f43a 100644
--- a/nug4/ParticleNavigation/CMakeLists.txt
+++ b/nug4/ParticleNavigation/CMakeLists.txt
@@ -15,9 +15,8 @@ art_make( NO_PLUGINS
 			${MF_UTILITIES}
         		cetlib
 			cetlib_except
-			${ROOT_BASIC_LIB_LIST}
+			${ROOT_LIBRARIES}
         )
 
 install_headers()
 install_fhicl()
-install_source()
