diff --git a/CMakeLists.txt b/CMakeLists.txt
index aeac6a1..82abba4 100644
--- a/CMakeLists.txt	2020-02-26 16:35:06.000000000 -0600
+++ b/CMakeLists.txt	2020-02-26 16:36:32.000000000 -0600
@@ -15,10 +15,10 @@
 
 cmake_minimum_required(VERSION 3.11)  # use cmake 3.11 or later
 
-project(nug4)
+project(nug4 1.04.01 LANGUAGE CXX)
 
 # cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
+find_package(cetmodules REQUIRED)
 
 list(APPEND CMAKE_MODULE_PATH
   $ENV{CANVAS_ROOT_IO_DIR}/Modules
@@ -36,12 +36,12 @@
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product( nusimdata )
-find_ups_product( cetlib )
-find_ups_product( cetlib_except )
-find_ups_geant4( )
+find_package( art )
+find_package( art_root_io)
+find_package( nusimdata )
+find_package( cetlib )
+find_package( cetlib_except )
+find_package( geant4 )
 find_ups_product( cetbuildtools )
 
 message(STATUS "ROOTSYS is ${ROOTSYS}")
@@ -59,8 +59,7 @@
 add_subdirectory (nug4)
 
 # ups - table and config files
-add_subdirectory(ups)
+#add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
-
+cet_cmake_config()
