diff --git a/CMakeLists.txt b/CMakeLists.txt
index 552faa9..936704b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,50 +12,52 @@
 #  make install
 #  make package (builds distribution tarfile)
 # ======================================================================
+cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
 
-cmake_minimum_required(VERSION 3.18.2 FATAL_ERROR)
 
+find_package(cetmodules 2.29.06 REQUIRED)
 project(nug4 VERSION 1.11.01 LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
-
 include(CetCMakeEnv)
 cet_cmake_env()
 
+# Find our own CMake modules, and tell everyone else where to find them.
+cet_cmake_module_directories(Modules)
+
 # -D_GLIBCXX_USE_NANOSLEEP is needed to make std::this_thread::sleep_for available. (IFDatabase)
 cet_set_compiler_flags(DIAGS CAUTIOUS WERROR
                        NO_UNDEFINED
-                       EXTRA_FLAGS -pedantic
-                       EXTRA_CXX_FLAGS -D_GLIBCXX_USE_NANOSLEEP
-                                       -Wno-unused-local-typedefs -Wno-variadic-macros)
-cet_report_compiler_flags()
-
-# these are minimum required versions, not the actual product versions
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product(canvas_root_io)
-find_ups_product( nusimdata )
-find_ups_product( clhep )
-find_ups_product( cetlib )
-find_ups_product( cetlib_except )
-find_ups_product( pythia8 )
-find_ups_geant4( )
-find_ups_root()
-
-message(STATUS "ROOTSYS is ${ROOTSYS}")
+                       EXTRA_FLAGS -pedantic 
+                                   $<$<COMPILE_LANGUAGE:CXX>:-D_GLIBCXX_USE_NANOSLEEP 
+                                   -Wno-unused-local-typedefs -Wno-variadic-macros>)
+cet_report_compiler_flags(REPORT_THRESHOLD VERBOSE)
+
+find_package(art REQUIRED)
+find_package(art_root_io REQUIRED)
+find_package(canvas_root_io REQUIRED)
+find_package(nusimdata REQUIRED)
+find_package(cetlib REQUIRED)
+find_package(cetlib_except REQUIRED)
+find_package(CLHEP REQUIRED)
+find_package(Boost REQUIRED)
+find_package(Geant4 REQUIRED)
+find_package(ROOT COMPONENTS Core Physics Geom REQUIRED PUBLIC)
 
 # macros
 include(ArtMake)
 include(BuildPlugins)
 include(CetRootCint)
 
+# headers
+# allow the use of #include "Geant4/.."
+include_directories("${Geant4_INCLUDE_DIRS}/../")
+
 # source
 add_subdirectory (nug4)
+add_subdirectory (Modules)
 
 # ups - table and config files
-add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
 
+cet_cmake_config()
diff --git a/nug4/AdditionalG4Physics/CMakeLists.txt b/nug4/AdditionalG4Physics/CMakeLists.txt
index d0f4b1a..1edbb36 100644
--- a/nug4/AdditionalG4Physics/CMakeLists.txt
+++ b/nug4/AdditionalG4Physics/CMakeLists.txt
@@ -1,23 +1,23 @@
-cet_find_library( PYTHIA8 NAMES pythia8 PATHS ENV PYTHIA8_LIB NO_DEFAULT_PATH )
 
-art_make( NO_PLUGINS
-          LIB_LIBRARIES messagefacility::MF_MessageLogger
-                        # cetlib::cetlib
-                        # cetlib_except::cetlib_except
-                        G4EVENT
-                        G4GEOMETRY
-                        G4GLOBAL
-                        G4INTERCOMS
-                        G4MATERIALS
-                        G4MODELING
-                        G4PARTICLES
-                        G4PHYSICSLISTS
-                        G4PROCESSES
-                        G4RUN
-                        G4TRACKING
-                        # XERCESC
-                        CLHEP
-                        PYTHIA8
+find_package(Pythia8 REQUIRED)
+
+art_make_library(SOURCE Py8CharmDecayerPhysics.cc Py8Decayer.cc
+                         Py8TauDecayerPhysics.cc RegisterAddG4PhysicsShorthands.cc
+                 LIBRARIES PRIVATE
+                        messagefacility::MF_MessageLogger
+                        Geant4::G4event
+                        Geant4::G4geometry
+                        Geant4::G4global
+                        Geant4::G4intercoms
+                        Geant4::G4materials
+                        Geant4::G4modeling
+                        Geant4::G4particles
+                        Geant4::G4physicslists
+                        Geant4::G4processes
+                        Geant4::G4run
+                        Geant4::G4tracking
+                        CLHEP::CLHEP
+                        Pythia8::pythia8
                         )
 
 install_headers()
diff --git a/nug4/G4Base/CMakeLists.txt b/nug4/G4Base/CMakeLists.txt
index 7cb487c..beb9d82 100644
--- a/nug4/G4Base/CMakeLists.txt
+++ b/nug4/G4Base/CMakeLists.txt
@@ -1,35 +1,36 @@
 
-art_make( LIB_LIBRARIES nusimdata::SimulationBase
-                        nug4_MagneticFieldServices_MagneticFieldServiceStandard_service
-                        nug4_MagneticField
-			                  canvas::canvas
-                        ART_FRAMEWORK_SERVICES_REGISTRY
-                        MF_MESSAGELOGGER
-                        FHICLCPP
+art_make_library( LIBRARIES PRIVATE
+                        nug4::MagneticFieldServices_MagneticFieldServiceStandard_service
+                        nug4::MagneticField
+                        nusimdata::SimulationBase
+                        canvas::canvas
+                        art::Framework_Services_Registry
+                        messagefacility::MF_MessageLogger
+                        fhiclcpp::fhiclcpp
                         cetlib::cetlib
                         cetlib_except::cetlib_except
-                        G4FR
-                        G4GMOCREN
-                        G4RAYTRACER
-                        G4TREE
-                        G4VRML
-                        G4EVENT
-                        G4GEOMETRY
-                        G4GLOBAL
-                        G4INTERCOMS
-                        G4MATERIALS
-                        G4MODELING
-                        G4PARTICLES
-                        G4PERSISTENCY
-                        G4PHYSICSLISTS
-                        G4PROCESSES
-                        G4READOUT
-                        G4RUN
-                        G4TRACKING
-                        G4VISHEPREP
-                        G4VIS_MANAGEMENT
-                        XERCESC
-                        CLHEP
+                        Geant4::G4FR
+                        Geant4::G4GMocren
+                        Geant4::G4RayTracer
+                        Geant4::G4Tree
+                        Geant4::G4VRML
+                        Geant4::G4event
+                        Geant4::G4geometry
+                        Geant4::G4global
+                        Geant4::G4intercoms
+                        Geant4::G4materials
+                        Geant4::G4modeling
+                        Geant4::G4particles
+                        Geant4::G4persistency
+                        Geant4::G4physicslists
+                        Geant4::G4processes
+                        Geant4::G4readout
+                        Geant4::G4run
+                        Geant4::G4tracking
+                        Geant4::G4visHepRep
+                        Geant4::G4vis_management
+                        XercesC::XercesC
+                        CLHEP::CLHEP
                         )
 
 install_headers()
diff --git a/nug4/MagneticField/CMakeLists.txt b/nug4/MagneticField/CMakeLists.txt
index 6fca3f3..045d28a 100644
--- a/nug4/MagneticField/CMakeLists.txt
+++ b/nug4/MagneticField/CMakeLists.txt
@@ -1,18 +1,18 @@
-include_directories( ${CMAKE_CURRENT_SOURCE_DIR} )
 
-art_make(NO_PLUGINS
-          LIB_LIBRARIES
-          G4GEOMETRY
+cet_make_library( SOURCE MagneticFieldStandard.cxx
+          HEADERS_TARGET
+          LIBRARIES PRIVATE
+          Geant4::G4geometry
           messagefacility::MF_MessageLogger
-          
           canvas::canvas
           fhiclcpp::fhiclcpp
           cetlib::cetlib
           cetlib_except::cetlib_except
-          CLHEP
-          ${ROOT_BASIC_LIB_LIST}
+          CLHEP::CLHEP
+          PUBLIC
+          ROOT::Core
           ROOT::Geom
-          PUBLIC G4GEOMETRY
+          ROOT::Physics
          )
          
 install_headers()
diff --git a/nug4/MagneticFieldServices/CMakeLists.txt b/nug4/MagneticFieldServices/CMakeLists.txt
index c624dfd..27df5ac 100644
--- a/nug4/MagneticFieldServices/CMakeLists.txt
+++ b/nug4/MagneticFieldServices/CMakeLists.txt
@@ -1,13 +1,13 @@
-include_directories( ${CMAKE_CURRENT_SOURCE_DIR} )
 
-simple_plugin( MagneticFieldServiceStandard "service"
+cet_build_plugin( MagneticFieldServiceStandard art::service
+      LIBRARIES PRIVATE
             messagefacility::MF_MessageLogger
-            
             cetlib_except::cetlib_except
-            nug4_MagneticField
-            G4GEOMETRY
-            CLHEP
-            ${ROOT_BASIC_LIB_LIST}
+            Geant4::G4geometry
+            CLHEP::CLHEP
+            PUBLIC
+            nug4::MagneticField
+            ROOT::Core
             ROOT::Geom )
 
 install_headers()
diff --git a/nug4/ParticleNavigation/CMakeLists.txt b/nug4/ParticleNavigation/CMakeLists.txt
index 8a90f1b..4ca0b0e 100644
--- a/nug4/ParticleNavigation/CMakeLists.txt
+++ b/nug4/ParticleNavigation/CMakeLists.txt
@@ -1,9 +1,8 @@
 
-# for classes_def.xml!!
-include_directories( ${CMAKE_CURRENT_SOURCE_DIR} )
-
-art_make( NO_PLUGINS
-          LIB_LIBRARIES nusimdata::SimulationBase
+art_make_library( SOURCE EmEveIdCalculator.cxx  EveIdCalculator.cxx
+                         ParticleHistory.cxx  ParticleList.cxx
+                  LIBRARIES PRIVATE
+                        nusimdata::SimulationBase
                         art::Framework_Core
 			art::Framework_Principal
 			art::Framework_Services_Registry
@@ -12,10 +11,9 @@ art_make( NO_PLUGINS
 			art::Utilities
 			canvas::canvas
 			messagefacility::MF_MessageLogger
-			
         		cetlib::cetlib
 			cetlib_except::cetlib_except
-			${ROOT_BASIC_LIB_LIST}
+                        ROOT::Core
         )
 
 install_headers()
