diff --git a/CMakeLists.txt b/CMakeLists.txt
index bb973c2..cc247c8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,21 +13,14 @@
 # ======================================================================
 
 
-# use cmake 2.8 or later
-cmake_minimum_required (VERSION 2.8)
+# use cmake 3.12 or later
+cmake_minimum_required (VERSION 3.12)
 
-project(larevt)
-
-# cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
-IF (NOT CETBUILDTOOLS_VERSION)
-    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
-ENDIF()
-
-set(CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules
-		      $ENV{CETBUILDTOOLS_DIR}/Modules
-		      ${CMAKE_MODULE_PATH})
+project(larevt VERSION 8.10.03 LANGUAGES CXX)
 
+set(fcl_dir job)
+set(gdml_dir gdml)
+find_package( cetmodules REQUIRED )
 include(CetCMakeEnv)
 cet_cmake_env()
 
@@ -40,20 +33,9 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( larcoreobj )
-find_ups_product( larcorealg )
-find_ups_product( larcore )
-find_ups_product( lardataobj )
-find_ups_product( lardataalg )
-find_ups_product( lardata )
-find_ups_product( nusimdata )
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product( messagefacility )
-find_ups_root()
-find_ups_product( postgresql )
-find_ups_product( libwda )
-find_ups_product( cetbuildtools )
+find_package( PostgreSQL REQUIRED )
+find_package( ROOT REQUIRED COMPONENTS FFTW )
+find_package( lardata REQUIRED )
 
 # macros for dictionary and simple_plugin
 include(ArtDictionary)
@@ -65,9 +47,9 @@ add_subdirectory(larevt)
 
 # tests
 add_subdirectory(test)
-
+add_subdirectory(cmakeConfig)
 # ups - table and config files
-add_subdirectory(ups)
+#add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
+#include(UseCPack)
diff --git a/cmakeConfig/CMakeLists.txt b/cmakeConfig/CMakeLists.txt
new file mode 100644
index 0000000..4d15a17
--- /dev/null
+++ b/cmakeConfig/CMakeLists.txt
@@ -0,0 +1,3 @@
+
+# create package configuration and version files
+cet_cmake_config(CONFIG_FRAGMENTS larevt-fragment.cmake.in)
diff --git a/cmakeConfig/larevt-fragment.cmake.in b/cmakeConfig/larevt-fragment.cmake.in
new file mode 100644
index 0000000..95d36a1
--- /dev/null
+++ b/cmakeConfig/larevt-fragment.cmake.in
@@ -0,0 +1,3 @@
+find_package( PostgreSQL REQUIRED )
+find_package( ROOT REQUIRED COMPONENTS FFTW )
+find_package( lardata REQUIRED )
diff --git a/larevt/CalData/CMakeLists.txt b/larevt/CalData/CMakeLists.txt
index e9bc78f..aefe678 100644
--- a/larevt/CalData/CMakeLists.txt
+++ b/larevt/CalData/CMakeLists.txt
@@ -7,9 +7,9 @@ simple_plugin(CalWireAna "module"
               ${ART_ROOT_IO_TFILESERVICE_SERVICE}
               ${ART_ROOT_IO_TFILE_SUPPORT}
 	      ${MF_MESSAGELOGGER}
-              ${ROOT_CORE}
-              ${ROOT_HIST}
-              ${ROOT_MATHCORE})
+              ROOT::Core
+              ROOT::Hist
+              ROOT::MathCore)
 
 simple_plugin(CalWire "module"
               larcorealg_Geometry
@@ -17,10 +17,10 @@ simple_plugin(CalWire "module"
               lardata_ArtDataHelper
               ${ART_FRAMEWORK_SERVICES_REGISTRY}
               ${MF_MESSAGELOGGER}
-              ${ROOT_CORE}
-              ${ROOT_HIST}
-              ${ROOT_MATHCORE}
-              ${ROOT_RIO})
+              ROOT::Core
+              ROOT::Hist
+              ROOT::MathCore
+              ROOT::RIO)
 
 simple_plugin(CalWireT962 "module"
               larcorealg_Geometry
@@ -28,13 +28,12 @@ simple_plugin(CalWireT962 "module"
               lardata_ArtDataHelper
               ${ART_FRAMEWORK_SERVICES_REGISTRY}
               ${MF_MESSAGELOGGER}
-              ${ROOT_CORE}
-              ${ROOT_HIST}
-              ${ROOT_MATHCORE}
-              ${ROOT_RIO})
+              ROOT::Core
+              ROOT::Hist
+              ROOT::MathCore
+              ROOT::RIO)
 
 install_headers()
 install_fhicl()
-install_source()
 
 add_subdirectory(test)
diff --git a/larevt/CalData/test/CMakeLists.txt b/larevt/CalData/test/CMakeLists.txt
index 2d755fd..9b0fe74 100644
--- a/larevt/CalData/test/CMakeLists.txt
+++ b/larevt/CalData/test/CMakeLists.txt
@@ -1,20 +1,22 @@
 cet_enable_asserts()
 
 simple_plugin(FFTTest "module"
-			lardata_Utilities_LArFFT_service
-			${ART_FRAMEWORK_SERVICES_REGISTRY}
-			${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
-			${ART_ROOT_IO_TFILESERVICE_SERVICE}
-			art_Persistency_Common
-			art_Persistency_Provenance
-	 		art_Utilities
-                        canvas
-			${MF_MESSAGELOGGER}
-			
-	                ${FHICLCPP}
-			cetlib cetlib_except
-			${ROOT_BASIC_LIB_LIST}
-              
+              lardata_Utilities_LArFFT_service
+              ${ART_FRAMEWORK_SERVICES_REGISTRY}
+              ${ART_ROOT_IO_TFILE_SUPPORT}
+              ${ART_ROOT_IO_TFILESERVICE_SERVICE}
+              art_Persistency_Common
+              art_Persistency_Provenance
+              art_Utilities
+              canvas
+              ${MF_MESSAGELOGGER}
+              ${FHICLCPP}
+              cetlib cetlib_except
+              ROOT::Core
+              ROOT::MathCore
+              ROOT::FFTW
+              ROOT::Hist
+              ROOT::RIO
 )
 
 install_headers()
diff --git a/larevt/CalibrationDBI/LArBackend/job/CMakeLists.txt b/larevt/CalibrationDBI/LArBackend/job/CMakeLists.txt
index 520ed80..3997a04 100644
--- a/larevt/CalibrationDBI/LArBackend/job/CMakeLists.txt
+++ b/larevt/CalibrationDBI/LArBackend/job/CMakeLists.txt
@@ -4,5 +4,3 @@ install_fhicl()
 
 # Also put a copy in the source tree.
 
-FILE(GLOB fcl_files *.fcl)
-install_source( EXTRAS ${fcl_files} )
diff --git a/larevt/CalibrationDBI/Providers/CMakeLists.txt b/larevt/CalibrationDBI/Providers/CMakeLists.txt
index 33a0419..f16b6a2 100644
--- a/larevt/CalibrationDBI/Providers/CMakeLists.txt
+++ b/larevt/CalibrationDBI/Providers/CMakeLists.txt
@@ -1,6 +1,6 @@
-cet_find_library(LIBWDA NAMES wda PATHS ENV LIBWDA_LIB NO_DEFAULT_PATH)
+cet_find_library(LIBWDA NAMES wda )
 
-include_directories($ENV{LIBWDA_FQ_DIR}/include)
+include_directories(${LIBWDA_INC_DIR})
 
 art_make(LIB_LIBRARIES
            larevt_CalibrationDBI_IOVData
@@ -12,8 +12,7 @@ art_make(LIB_LIBRARIES
            cetlib_except
            larcorealg_Geometry
            ${ART_FRAMEWORK_SERVICES_REGISTRY}
-           ${ROOT_CORE}
+           ROOT::Core
         )
 
 install_headers()
-install_source()
diff --git a/larevt/CalibrationDBI/Services/CMakeLists.txt b/larevt/CalibrationDBI/Services/CMakeLists.txt
index 5977112..d992194 100644
--- a/larevt/CalibrationDBI/Services/CMakeLists.txt
+++ b/larevt/CalibrationDBI/Services/CMakeLists.txt
@@ -6,4 +6,3 @@ art_make(SERVICE_LIBRARIES
          )
 
 install_headers()
-install_source()
diff --git a/larevt/Filters/CMakeLists.txt b/larevt/Filters/CMakeLists.txt
index d6489fd..f1b0f21 100644
--- a/larevt/Filters/CMakeLists.txt
+++ b/larevt/Filters/CMakeLists.txt
@@ -12,18 +12,17 @@ art_make(LIB_LIBRARIES
            ${ART_ROOT_IO_TFILESERVICE_SERVICE}
            ${ART_ROOT_IO_TFILE_SUPPORT}
            ${MF_MESSAGELOGGER}
-           ${ROOT_CORE}
-           ${ROOT_HIST}
-           ${ROOT_PHYSICS}
+           ROOT::Core
+           ROOT::Hist
+           ROOT::Physics
            larcorealg_Geometry
            lardataobj_RawData
            nusimdata_SimulationBase
          SERVICE_LIBRARIES
            larevt_Filters
            ${MF_MESSAGELOGGER}
-           ${ROOT_CORE}
+           ROOT::Core
          )
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larevt/SpaceCharge/CMakeLists.txt b/larevt/SpaceCharge/CMakeLists.txt
index e9a338c..31d1504 100644
--- a/larevt/SpaceCharge/CMakeLists.txt
+++ b/larevt/SpaceCharge/CMakeLists.txt
@@ -2,9 +2,9 @@ art_make(NO_PLUGINS
          LIB_LIBRARIES
            canvas
            ${FHICLCPP}
-           ${ROOT_CORE}
-           ${ROOT_HIST}
-           ${ROOT_RIO}
+           ROOT::Core
+           ROOT::Hist
+           ROOT::RIO
            cetlib
            cetlib_except
          )
diff --git a/larevt/SpaceChargeServices/CMakeLists.txt b/larevt/SpaceChargeServices/CMakeLists.txt
index ff7e80e..e33760b 100644
--- a/larevt/SpaceChargeServices/CMakeLists.txt
+++ b/larevt/SpaceChargeServices/CMakeLists.txt
@@ -1,7 +1,7 @@
 simple_plugin(SpaceChargeServiceStandard "service"
               larevt_SpaceCharge
               ${ART_FRAMEWORK_PRINCIPAL}
-              ${ROOT_CORE}
+              ROOT::Core
             )
 
 install_headers()
diff --git a/test/Filters/CMakeLists.txt b/test/Filters/CMakeLists.txt
index 05ea49d..d77656f 100644
--- a/test/Filters/CMakeLists.txt
+++ b/test/Filters/CMakeLists.txt
@@ -3,10 +3,9 @@ cet_enable_asserts()
 simple_plugin ( SimpleChannelStatusTest "module"
                     larevt_Filters
                     ${MF_MESSAGELOGGER}
-                    
                     ${FHICLCPP}
                     cetlib cetlib_except
-                    ${ROOT_BASIC_LIB_LIST}
+                    ROOT::Core
               )
 
 cet_test(ChannelStatusServiceTest HANDBUILT
