diff --git a/CMakeLists.txt b/CMakeLists.txt
index 09ac39a..fb152fd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,10 +10,9 @@ cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
 
 project(larevt VERSION 09.03.05 LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
-
-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules $ENV{ART_DIR}/Modules)
+find_package( cetmodules REQUIRED )
+find_package( canvas_root_io )
+find_package( art )
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -27,21 +26,20 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( larcoreobj )
-find_ups_product( larcorealg )
-find_ups_product( larcore )
-find_ups_product( lardataobj )
-find_ups_product( lardataalg )
-find_ups_product( lardata )
-find_ups_product( nusimdata )
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product( messagefacility )
-find_ups_root()
-find_ups_product( postgresql )
-find_ups_product( libwda )
-find_ups_product(clhep)
-find_ups_product(sqlite)
+find_package( larcoreobj )
+find_package( larcorealg )
+find_package( larcore )
+find_package( lardataobj )
+find_package( lardataalg )
+find_package( lardata )
+find_package( nusimdata )
+find_package( art )
+find_package( art_root_io )
+find_package( messagefacility )
+find_package( ROOT )
+find_package( PostgreSQL )
+find_package( libwda )
+find_package( CLHEP )
 
 # macros for dictionary and simple_plugin
 include(ArtDictionary)
@@ -53,9 +51,8 @@ add_subdirectory(larevt)
 
 # tests
 add_subdirectory(test)
-
 # ups - table and config files
-add_subdirectory(ups)
-
+#add_subdirectory(ups)
 # packaging utility
-include(UseCPack)
+#include(UseCPack)
+cet_cmake_config()
diff --git a/larevt/CalData/CMakeLists.txt b/larevt/CalData/CMakeLists.txt
index dca57a7..1cda338 100644
--- a/larevt/CalData/CMakeLists.txt
+++ b/larevt/CalData/CMakeLists.txt
@@ -1,8 +1,8 @@
-simple_plugin(CalWireAna "module"
-              lardataobj_RecoBase
-              lardata_Utilities_LArFFT_service
-              lardata_Utilities
-              larcorealg_Geometry
+cet_build_plugin(CalWireAna art::module LIBRARIES
+              lardataobj::RecoBase
+              lardata::Utilities_LArFFT_service
+              lardata::Utilities
+              larcorealg::Geometry
               art::Framework_Services_Registry
               art_root_io::TFileService_service
               art_root_io::tfile_support
@@ -11,10 +11,10 @@ simple_plugin(CalWireAna "module"
               ROOT::Hist
               ROOT::MathCore)
 
-simple_plugin(CalWire "module"
-              larcorealg_Geometry
-              lardataobj_RawData
-              lardata_ArtDataHelper
+cet_build_plugin(CalWire art::module LIBRARIES
+              larcorealg::Geometry
+              lardataobj::RawData
+              lardata::ArtDataHelper
               art::Framework_Services_Registry
               messagefacility::MF_MessageLogger
               ROOT::Core
@@ -22,10 +22,10 @@ simple_plugin(CalWire "module"
               ROOT::MathCore
               ROOT::RIO)
 
-simple_plugin(CalWireT962 "module"
-              larcorealg_Geometry
-              lardataobj_RawData
-              lardata_ArtDataHelper
+cet_build_plugin(CalWireT962 art::module LIBRARIES
+              larcorealg::Geometry
+              lardataobj::RawData
+              lardata::ArtDataHelper
               art::Framework_Services_Registry
               messagefacility::MF_MessageLogger
               ROOT::Core
diff --git a/larevt/CalData/test/CMakeLists.txt b/larevt/CalData/test/CMakeLists.txt
index 1fa472e..39481b9 100644
--- a/larevt/CalData/test/CMakeLists.txt
+++ b/larevt/CalData/test/CMakeLists.txt
@@ -1,7 +1,8 @@
 cet_enable_asserts()
 
-simple_plugin(FFTTest "module"
-			lardata_Utilities_LArFFT_service
+cet_build_plugin(FFTTest art::module 
+                        LIBRARIES
+			lardata::Utilities_LArFFT_service
 			art::Framework_Services_Registry
 			art_root_io::tfile_support ROOT::Core
 			art_root_io::TFileService_service
@@ -10,10 +11,8 @@ simple_plugin(FFTTest "module"
 	 		art::Utilities
                         canvas::canvas
 			messagefacility::MF_MessageLogger
-
 	                fhiclcpp::fhiclcpp
-			cetlib::cetlib cetlib_except
-
+			cetlib::cetlib cetlib_except::cetlib_except
 )
 
 install_headers()
diff --git a/larevt/CalibrationDBI/IOVData/CMakeLists.txt b/larevt/CalibrationDBI/IOVData/CMakeLists.txt
index 90d153f..cb1e34b 100644
--- a/larevt/CalibrationDBI/IOVData/CMakeLists.txt
+++ b/larevt/CalibrationDBI/IOVData/CMakeLists.txt
@@ -1,3 +1,3 @@
-art_make()
+cet_make()
 
 install_headers()
diff --git a/larevt/CalibrationDBI/LArBackend/job/CMakeLists.txt b/larevt/CalibrationDBI/LArBackend/job/CMakeLists.txt
index 520ed80..3997a04 100644
--- a/larevt/CalibrationDBI/LArBackend/job/CMakeLists.txt
+++ b/larevt/CalibrationDBI/LArBackend/job/CMakeLists.txt
@@ -4,5 +4,3 @@ install_fhicl()
 
 # Also put a copy in the source tree.
 
-FILE(GLOB fcl_files *.fcl)
-install_source( EXTRAS ${fcl_files} )
diff --git a/larevt/CalibrationDBI/Providers/CMakeLists.txt b/larevt/CalibrationDBI/Providers/CMakeLists.txt
index 0fc0eeb..bcd2bb4 100644
--- a/larevt/CalibrationDBI/Providers/CMakeLists.txt
+++ b/larevt/CalibrationDBI/Providers/CMakeLists.txt
@@ -1,18 +1,18 @@
-cet_find_library(LIBWDA NAMES wda PATHS ENV LIBWDA_LIB NO_DEFAULT_PATH)
-cet_find_library(SQLITE NAMES sqlite3_ups PATHS ENV SQLITE_LIB NO_DEFAULT_PATH)
+find_library(LIBWDA NAMES wda PATHS $ENV{LIBWDA_DIR}/lib)
+find_library(SQLITE NAMES sqlite3 REQUIRED PATHS $ENV{SQLITE_LIB})
 
-include_directories($ENV{LIBWDA_FQ_DIR}/include)
+include_directories($ENV{LIBWDA_DIR}/include)
 
-art_make(LIB_LIBRARIES
-           larevt_CalibrationDBI_IOVData
-           LIBWDA
-           SQLITE
+cet_make(LIB_LIBRARIES
+           larevt::CalibrationDBI_IOVData
+           ${LIBWDA}
+           ${SQLITE}
            messagefacility::MF_MessageLogger
            fhiclcpp::fhiclcpp
            canvas::canvas
            cetlib::cetlib
            cetlib_except::cetlib_except
-           larcorealg_Geometry
+           larcorealg::Geometry
            art::Framework_Services_Registry
            ROOT::Core
            art::Utilities
diff --git a/larevt/CalibrationDBI/Services/CMakeLists.txt b/larevt/CalibrationDBI/Services/CMakeLists.txt
index 9058096..67fe7b2 100644
--- a/larevt/CalibrationDBI/Services/CMakeLists.txt
+++ b/larevt/CalibrationDBI/Services/CMakeLists.txt
@@ -1,9 +1,30 @@
-art_make(SERVICE_LIBRARIES
-           larevt_CalibrationDBI_Providers
-           larevt_CalibrationDBI_IOVData
+cet_make(SERVICE_LIBRARIES
+           larevt::CalibrationDBI_Providers
+           larevt::CalibrationDBI_IOVData
            art::Framework_Services_Registry
            art::Framework_Principal
          )
 
+
+simple_plugin(SIOVChannelStatusService "service"
+           larevt::CalibrationDBI_Providers
+           larevt::CalibrationDBI_IOVData
+           art::Framework_Services_Registry
+           art::Framework_Principal)
+simple_plugin(SIOVDetPedestalService "service"
+           larevt::CalibrationDBI_Providers
+           larevt::CalibrationDBI_IOVData
+           art::Framework_Services_Registry
+           art::Framework_Principal)
+simple_plugin(SIOVElectronicsCalibService "service"
+           larevt::CalibrationDBI_Providers
+           larevt::CalibrationDBI_IOVData
+           art::Framework_Services_Registry
+           art::Framework_Principal)
+simple_plugin(SIOVPmtGainService "service"
+           larevt::CalibrationDBI_Providers
+           larevt::CalibrationDBI_IOVData
+           art::Framework_Services_Registry
+           art::Framework_Principal)
+
 install_headers()
-install_source()
diff --git a/larevt/Filters/CMakeLists.txt b/larevt/Filters/CMakeLists.txt
index 9ea553d..149d984 100644
--- a/larevt/Filters/CMakeLists.txt
+++ b/larevt/Filters/CMakeLists.txt
@@ -1,13 +1,15 @@
-art_make(LIB_LIBRARIES
-           larcorealg_Geometry
-           lardataobj_RecoBase
+cet_make(LIB_LIBRARIES
+           larcorealg::Geometry
+           lardataobj::RecoBase
            art::Framework_Services_Registry
            canvas::canvas
            messagefacility::MF_MessageLogger
            fhiclcpp::fhiclcpp
            cetlib_except::cetlib_except
-         MODULE_LIBRARIES
-           lardata_Utilities
+         )
+
+set(MODULE_LIBRARIES
+           lardata::Utilities
            art::Framework_Services_Registry
            art_root_io::TFileService_service
            art_root_io::tfile_support
@@ -15,16 +17,29 @@ art_make(LIB_LIBRARIES
            ROOT::Core
            ROOT::Hist
            ROOT::Physics
-           lardataalg_DetectorInfo
-           larcorealg_Geometry
-           lardataobj_RawData
-           nusimdata_SimulationBase
-         SERVICE_LIBRARIES
-           larevt_Filters
+           lardataalg::DetectorInfo
+           larcorealg::Geometry
+           lardataobj::RawData
+           nusimdata::SimulationBase
+         )
+
+cet_build_plugin(ADCFilter art::module LIBRARIES ${MODULE_LIBRARIES})
+cet_build_plugin(EventFilter art::module LIBRARIES ${MODULE_LIBRARIES})
+cet_build_plugin(EventTimeFilter art::module LIBRARIES ${MODULE_LIBRARIES})
+cet_build_plugin(FinalStateParticleFilter art::module LIBRARIES ${MODULE_LIBRARIES})
+cet_build_plugin(LArG4ParticleFilter art::module LIBRARIES ${MODULE_LIBRARIES})
+cet_build_plugin(MuonFilter art::module LIBRARIES ${MODULE_LIBRARIES})
+cet_build_plugin(SmallClusterFilter art::module LIBRARIES ${MODULE_LIBRARIES})
+
+
+set(SERVICE_LIBRARIES
+           larevt::Filters
            messagefacility::MF_MessageLogger
            ROOT::Core
          )
 
+cet_build_plugin(SimpleChannelStatusService art::service LIBRARIES ${SERVICE_LIBRARIES})
+
 install_headers()
 install_fhicl()
 install_source()
diff --git a/larevt/SpaceCharge/CMakeLists.txt b/larevt/SpaceCharge/CMakeLists.txt
index 24032c1..c8192c6 100644
--- a/larevt/SpaceCharge/CMakeLists.txt
+++ b/larevt/SpaceCharge/CMakeLists.txt
@@ -1,5 +1,6 @@
-art_make(NO_PLUGINS
-         LIB_LIBRARIES
+cet_make_library(SOURCE SpaceChargeStandard.cxx
+         HEADERS_TARGET
+         LIBRARIES
            canvas::canvas
            fhiclcpp::fhiclcpp
            ROOT::Core
diff --git a/larevt/SpaceChargeServices/CMakeLists.txt b/larevt/SpaceChargeServices/CMakeLists.txt
index aab385c..bf987cd 100644
--- a/larevt/SpaceChargeServices/CMakeLists.txt
+++ b/larevt/SpaceChargeServices/CMakeLists.txt
@@ -1,5 +1,6 @@
-simple_plugin(SpaceChargeServiceStandard "service"
-              larevt_SpaceCharge
+cet_build_plugin(SpaceChargeServiceStandard art::service
+              LIBRARIES
+              larevt::SpaceCharge
               art::Framework_Principal
               ROOT::Core
             )
diff --git a/test/Filters/CMakeLists.txt b/test/Filters/CMakeLists.txt
index 8fff392..83372ec 100644
--- a/test/Filters/CMakeLists.txt
+++ b/test/Filters/CMakeLists.txt
@@ -1,11 +1,11 @@
 cet_enable_asserts()
 
-simple_plugin ( SimpleChannelStatusTest "module"
-                    larevt_Filters
+cet_build_plugin ( SimpleChannelStatusTest art::module
+                   LIBRARIES
+                    larevt::Filters
                     messagefacility::MF_MessageLogger
-                    
                     fhiclcpp::fhiclcpp
-                    cetlib::cetlib cetlib_except
+                    cetlib::cetlib cetlib_except::cetlib_except
               )
 
 cet_test(ChannelStatusServiceTest HANDBUILT
@@ -15,9 +15,9 @@ cet_test(ChannelStatusServiceTest HANDBUILT
 )
 
 cet_test(SimpleChannelStatus_test
-  SOURCES SimpleChannelStatus_test.cxx
-  LIBRARIES larevt_Filters
-            larevt_CalibrationDBI_IOVData
+  SOURCE SimpleChannelStatus_test.cxx
+  LIBRARIES larevt::Filters
+            larevt::CalibrationDBI_IOVData
             fhiclcpp::fhiclcpp
   USE_BOOST_UNIT
 )
