commit c6ccf39462d609e5ef1aa48d5748e947a463e62d
Author: Patrick Gartung <gartung@fnal.gov>
Date:   Fri Apr 3 14:26:24 2020 -0500

    Cetmodules

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b33cd0c..3538a5c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,47 +13,43 @@
 #  make package (builds distribution tarfile)
 # ======================================================================
 
-cmake_minimum_required(VERSION 3.11)  # use cmake 3.11 or later
 
-project(nugen)
-
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
-
-list(APPEND CMAKE_MODULE_PATH
-  $ENV{CANVAS_ROOT_IO_DIR}/Modules
-  $ENV{ART_DIR}/Modules)
+# use cmake 3.12 or later
+cmake_minimum_required (VERSION 3.12)
 
+project(nugen VERSION 1.10.02 LANGUAGES CXX)
+set(fcl_dir job)
+find_package(cetmodules)
 include(CetCMakeEnv)
 cet_cmake_env()
 
+# -D_GLIBCXX_USE_NANOSLEEP is needed to make std::this_thread::sleep_for available.
 cet_set_compiler_flags(DIAGS CAUTIOUS WERROR
                        NO_UNDEFINED
                        EXTRA_FLAGS -pedantic
-                       EXTRA_CXX_FLAGS -Wno-unused-local-typedefs -Wno-variadic-macros)
+                       EXTRA_CXX_FLAGS -D_GLIBCXX_USE_NANOSLEEP
+                                       -Wno-unused-local-typedefs -Wno-variadic-macros)
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product( genie )
-find_ups_product( dk2nudata )
-find_ups_product( dk2nugenie )
-find_ups_product( libwda )
-find_ups_product( postgresql )
-find_ups_product( nusimdata )
-find_ups_product( cetlib )
-find_ups_product( cetlib_except )
-find_ups_product( cetbuildtools )
+find_package( PostgreSQL REQUIRED)
+find_package( Boost REQUIRED COMPONENTS filesystem system date_time thread regex)
+find_package( ROOT REQUIRED COMPONENTS MathCore Geom GeomPainter GX11 EG EGPythia6 Gui)
+set(HAVE_ROOT6 BOOL:True)
+set(ROOTCLING ROOT::rootcling)
+find_package( art_root_io REQUIRED)
+find_package( nusimdata REQUIRED )
+
 
 message(STATUS "ROOTSYS is ${ROOTSYS}")
 
 ## find_library directives
 # other dependencies
-cet_find_library( PYTHIA6 NAMES Pythia6 PATHS ENV PYLIB NO_DEFAULT_PATH )
-cet_find_library( LHAPDF NAMES LHAPDF PATHS ENV LHAPDF_LIB NO_DEFAULT_PATH )
-cet_find_library( LOG4CPP NAMES log4cpp PATHS ENV LOG4CPP_LIB NO_DEFAULT_PATH )
-cet_find_library( XML2 NAMES xml2 PATHS $ENV{LIBXML2_FQ_DIR}/lib NO_DEFAULT_PATH )
+cet_find_library( PYTHIA6 NAMES Pythia6 PATHS ${CMAKE_INSTALL_RPATH}  NO_DEFAULT_PATH )
+cet_find_library( LHAPDF NAMES LHAPDF PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+cet_find_library( LOG4CPP NAMES log4cpp PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+cet_find_library( XML2 NAMES xml2 PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+cet_find_library( CRY NAMES CRY PATHS ${CMAKE_INSTALL_RPATH}  NO_DEFAULT_PATH )
 
 
 # genie-config for R-2 series didn't support --version (returns blank)
@@ -66,63 +62,67 @@ cet_find_library( XML2 NAMES xml2 PATHS $ENV{LIBXML2_FQ_DIR}/lib NO_DEFAULT_PATH
 #if( ${GENIE_MAJOR_VERSION} LESS 3 )
 
 # use GENIE UPS version to decide ...
-parse_ups_version( ${GENIE_VERSION} )
+#set(GENIE_VERSION $ENV{GENIE_VERSION})
+#include(ParseVersion)
+#parse_underscored_version( ${GENIE_VERSION} )
+set(VMAJ 3)
+set(VMIN 0)
 if( ${VMAJ} LESS 3 )
   add_definitions(-DGENIE_PRE_R3)
 #  MESSAGE("--NUGEN-- GENIE_VERSION=${VMAJ} ${VMIN} -DGENIE_PRE_R3")
 # these are the pre-R-3 library names
-  cet_find_library( GALGORITHM NAMES GAlgorithm PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GBARYONRESONANCE NAMES GBaryonResonance PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GBASE NAMES GBase PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GBODEKYANG NAMES GBodekYang PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GCHARM NAMES GCharm PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GCOH NAMES GCoh PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GDFRC NAMES GDfrc PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GDIS NAMES GDIS PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GCROSSSECTIONS NAMES GCrossSections PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GDECAY NAMES GDecay PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GELAS NAMES GElas PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GELFF NAMES GElFF PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GHEP NAMES GHEP PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GEVGCORE NAMES GEVGCore  PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GEVGMODULES NAMES GEVGModules PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GEVGDRIVERS NAMES GEVGDrivers PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GGIBUU NAMES GGiBUU PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GHADRONTRANSP NAMES GHadronTransp PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFRAGMENTATION NAMES GFragmentation PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GINTERACTION NAMES GInteraction PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GLLEWELLYNSMITH NAMES GLlewellynSmith  PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GMEC NAMES GMEC PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GMESSENGER NAMES GMessenger PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GNUGAMMA NAMES GNuGamma PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GNUE NAMES GNuE PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GNTUPLE NAMES GNtuple PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GNUCLEAR NAMES GNuclear PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GNUMERICAL NAMES GNumerical PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GQPM NAMES GQPM PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPDG NAMES GPDG PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPDF NAMES GPDF PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GQEL NAMES GQEL PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GRES NAMES GRES PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GREGISTRY NAMES GRegistry PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
+  cet_find_library( GALGORITHM NAMES GAlgorithm PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GBARYONRESONANCE NAMES GBaryonResonance PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GBASE NAMES GBase PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GBODEKYANG NAMES GBodekYang PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GCHARM NAMES GCharm PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GCOH NAMES GCoh PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GDFRC NAMES GDfrc PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GDIS NAMES GDIS PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GCROSSSECTIONS NAMES GCrossSections PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GDECAY NAMES GDecay PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GELAS NAMES GElas PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GELFF NAMES GElFF PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GHEP NAMES GHEP PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GEVGCORE NAMES GEVGCore  PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GEVGMODULES NAMES GEVGModules PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GEVGDRIVERS NAMES GEVGDrivers PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GGIBUU NAMES GGiBUU PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GHADRONTRANSP NAMES GHadronTransp PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GFRAGMENTATION NAMES GFragmentation PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GINTERACTION NAMES GInteraction PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GLLEWELLYNSMITH NAMES GLlewellynSmith  PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GMEC NAMES GMEC PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GMESSENGER NAMES GMessenger PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GNUGAMMA NAMES GNuGamma PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GNUE NAMES GNuE PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GNTUPLE NAMES GNtuple PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GNUCLEAR NAMES GNuclear PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GNUMERICAL NAMES GNumerical PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GQPM NAMES GQPM PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPDG NAMES GPDG PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPDF NAMES GPDF PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GQEL NAMES GQEL PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GRES NAMES GRES PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GREGISTRY NAMES GRegistry PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
 
   # prior to GENIE R-2_9_0 Sehgal's name was spelled incorrectly
   # so if minor version # is 8 or less use GReinSeghal, otherwise GReinSehgal
   # parse_ups_version returns VMAJ, VMIN, and VPRJ
-  parse_ups_version( ${GENIE_VERSION} )
+  #parse_ups_version( $ENV{GENIE_VERSION} )
   set ( GRSSPELLING GReinSehgal )
   if ( ${VMAJ} EQUAL 2 AND ${VMIN} LESS 9 )
     set ( GRSSPELLING GReinSeghal )
   endif ( ${VMAJ} EQUAL 2 AND ${VMIN} LESS 9 )
   message(STATUS "GENIE version ${GENIE_VERSION} major ${VMAJ} minor ${VMIN} ${GRSSPELLING}")
-  cet_find_library( GREINSEHGAL NAMES ${GRSSPELLING} PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
+  cet_find_library( GREINSEHGAL NAMES ${GRSSPELLING} PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
 
-  cet_find_library( GUTILS NAMES GUtils PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GGEO NAMES GGeo PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFLUXDRIVERS NAMES GFluxDrivers PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GMUELOSS NAMES GMuELoss PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GREWEIGHT NAMES GReWeight PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GNUCLEONDECAY NAMES GNucleonDecay PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
+  cet_find_library( GUTILS NAMES GUtils PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GGEO NAMES GGeo PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GFLUXDRIVERS NAMES GFluxDrivers PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GMUELOSS NAMES GMuELoss PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GREWEIGHT NAMES GReWeight PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GNUCLEONDECAY NAMES GNucleonDecay PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
   set(GENIE_LIB_LIST  ${GALGORITHM}
                       ${GBARYONRESONANCE}
                       ${GBASE}
@@ -169,60 +169,60 @@ else()
 # these are the POST-R-3 library names
 #  MESSAGE("--NUGEN-- GENIE_VERSION=${VMAJ} ${VMIN} not defining -DGENIE_PRE_R3")
 
-  cet_find_library( GFWMSG       NAMES GFwMsg       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFWREG       NAMES GFwReg       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFWALG       NAMES GFwAlg       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFWINT       NAMES GFwInt       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFWGHEP      NAMES GFwGHEP      PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFWNUM       NAMES GFwNum       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFWUTL       NAMES GFwUtl       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFWPARDAT    NAMES GFwParDat    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFWEG        NAMES GFwEG        PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFWNTP       NAMES GFwNtp       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHXSIG      NAMES GPhXSIg      PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHPDF       NAMES GPhPDF       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHNUCLST    NAMES GPhNuclSt    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHCMN       NAMES GPhCmn       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHDCY       NAMES GPhDcy       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHHADTRANSP NAMES GPhHadTransp PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHHADNZ     NAMES GPhHadnz     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHDEEX      NAMES GPhDeEx      PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHAMNGXS    NAMES GPhAMNGXS    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHAMNGEG    NAMES GPhAMNGEG    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHCHMXS     NAMES GPhChmXS     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHCOHXS     NAMES GPhCohXS     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHCOHEG     NAMES GPhCohEG     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHDISXS     NAMES GPhDISXS     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHDISEG     NAMES GPhDISEG     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHDFRCXS    NAMES GPhDfrcXS    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHDFRCEG    NAMES GPhDfrcEG    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHGLWRESXS  NAMES GPhGlwResXS  PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHGLWRESEG  NAMES GPhGlwResEG  PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHIBDXS     NAMES GPhIBDXS     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHIBDEG     NAMES GPhIBDEG     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHMNUCXS    NAMES GPhMNucXS    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHMNUCEG    NAMES GPhMNucEG    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHMEL       NAMES GPhMEL       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHNUELXS    NAMES GPhNuElXS    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHNUELEG    NAMES GPhNuElEG    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHQELXS     NAMES GPhQELXS     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHQELEG     NAMES GPhQELEG     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHRESXS     NAMES GPhResXS     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHRESEG     NAMES GPhResEG     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHSTRXS     NAMES GPhStrXS     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHSTREG     NAMES GPhStrEG     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHNDCY      NAMES GPhNDcy      PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GTLGEO       NAMES GTlGeo       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GTLFLX       NAMES GTlFlx       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
+  cet_find_library( GFWMSG       NAMES GFwMsg       PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GFWREG       NAMES GFwReg       PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GFWALG       NAMES GFwAlg       PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GFWINT       NAMES GFwInt       PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GFWGHEP      NAMES GFwGHEP      PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GFWNUM       NAMES GFwNum       PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GFWUTL       NAMES GFwUtl       PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GFWPARDAT    NAMES GFwParDat    PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GFWEG        NAMES GFwEG        PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GFWNTP       NAMES GFwNtp       PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHXSIG      NAMES GPhXSIg      PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHPDF       NAMES GPhPDF       PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHNUCLST    NAMES GPhNuclSt    PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHCMN       NAMES GPhCmn       PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHDCY       NAMES GPhDcy       PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHHADTRANSP NAMES GPhHadTransp PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHHADNZ     NAMES GPhHadnz     PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHDEEX      NAMES GPhDeEx      PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHAMNGXS    NAMES GPhAMNGXS    PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHAMNGEG    NAMES GPhAMNGEG    PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHCHMXS     NAMES GPhChmXS     PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHCOHXS     NAMES GPhCohXS     PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHCOHEG     NAMES GPhCohEG     PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHDISXS     NAMES GPhDISXS     PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHDISEG     NAMES GPhDISEG     PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHDFRCXS    NAMES GPhDfrcXS    PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHDFRCEG    NAMES GPhDfrcEG    PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHGLWRESXS  NAMES GPhGlwResXS  PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHGLWRESEG  NAMES GPhGlwResEG  PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHIBDXS     NAMES GPhIBDXS     PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHIBDEG     NAMES GPhIBDEG     PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHMNUCXS    NAMES GPhMNucXS    PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHMNUCEG    NAMES GPhMNucEG    PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHMEL       NAMES GPhMEL       PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHNUELXS    NAMES GPhNuElXS    PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHNUELEG    NAMES GPhNuElEG    PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHQELXS     NAMES GPhQELXS     PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHQELEG     NAMES GPhQELEG     PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHRESXS     NAMES GPhResXS     PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHRESEG     NAMES GPhResEG     PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHSTRXS     NAMES GPhStrXS     PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHSTREG     NAMES GPhStrEG     PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GPHNDCY      NAMES GPhNDcy      PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GTLGEO       NAMES GTlGeo       PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GTLFLX       NAMES GTlFlx       PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
 
   # if GENIE's Reweight gets spun into separate product
   # then the following must change for GENIE_LIB to ? GENIERW_LIB
-  cet_find_library( GRWFWK       NAMES GRwFwk       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GRWIO        NAMES GRwIO        PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GRWCLC       NAMES GRwClc       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
+  cet_find_library( GRWFWK       NAMES GRwFwk       PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GRWIO        NAMES GRwIO        PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+  cet_find_library( GRWCLC       NAMES GRwClc       PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
 
   # GENIE pre-R-3 used GSL via ROOT's Math libs, now makes direct call not available via that route
-  cet_find_library( GSL NAMES gsl PATHS ENV GSL_LIB NO_DEFAULT_PATH )
+  cet_find_library( GSL NAMES gsl PATHS ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
 
 # this is the order that GENIE "recommends"
 # genie-config --libs
@@ -289,35 +289,43 @@ endif()
 #MESSAGE("--NUGEN-- GENIE ${VMAJ} messenger ${GMESSENGER} gfwmsg ${GFWMSG} GENIE_LIB_LIST=${GENIE_LIB_LIST}")
 
 # dk2nu
-cet_find_library( DK2NU_TREE  NAMES dk2nuTree  PATHS ENV DK2NUDATA_LIB NO_DEFAULT_PATH )
-cet_find_library( DK2NU_GENIE NAMES dk2nuGenie PATHS ENV DK2NUGENIE_LIB NO_DEFAULT_PATH )
-include_directories( $ENV{DK2NUDATA_INC} $ENV{DK2NUGENIE_INC} )
+cet_find_library( DK2NU_TREE  NAMES dk2nuTree  PATHS  ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+cet_find_library( DK2NU_GENIE NAMES dk2nuGenie PATHS  ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+include_directories( ${DK2NUDATA_INC} ${DK2NUGENIE_INC} )
 
 # ifdhc
-cet_find_library( IFDH NAMES ifdh PATHS ENV IFDHC_LIB NO_DEFAULT_PATH )
-include_directories( $ENV{IFDHC_FQ_DIR}/inc )
+cet_find_library( IFDH NAMES ifdh PATHS  ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+include_directories( ${IFDHC_INC} )
 
 # ifdhc_art  # as a service
-cet_find_library( IFDH_SERVICE NAMES IFDH_service PATHS ENV IFDH_ART_LIB NO_DEFAULT_PATH )
-include_directories( $ENV{IFDH_ART_FQ_DIR}/inc )
+cet_find_library( IFDH_SERVICE NAMES IFDH_service PATHS  ${CMAKE_INSTALL_RPATH} NO_DEFAULT_PATH )
+include_directories( ${IFDH_ART_INC} )
 
 # macros
 include(ArtMake)
+include(ArtDictionary)
 include(BuildPlugins)
+include(CetRootCint)
 
 # include search path
-include_directories ( $ENV{GENIE_INC}/GENIE )
-include_directories ( $ENV{GENIE_REWEIGHT}/src )
-include_directories ( $ENV{LIBXML2_INC} )
-include_directories ( $ENV{LOG4CPP_INC} )
+include_directories ( ${GENIE_INC}/GENIE )
+include_directories ( ${GENIE_REWEIGHT}/src )
+include_directories ( ${LIBXML2_INC} )
+include_directories ( ${CRYHOME})
+include_directories ( ${LOG4CPP_INC} )
+
+# included for Mac OSX using XQuartz for X11
+if(APPLE)
+  include_directories ( /opt/X11/include )
+endif()
 
 # source
 add_subdirectory (nugen)
 add_subdirectory (tools)
-
+add_subdirectory(cmakeConfig)
 # ups - table and config files
-add_subdirectory(ups)
+#add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
+#include(UseCPack)
 
diff --git a/cmakeConfig/CMakeLists.txt b/cmakeConfig/CMakeLists.txt
new file mode 100644
index 0000000..3629050
--- /dev/null
+++ b/cmakeConfig/CMakeLists.txt
@@ -0,0 +1,3 @@
+
+# create package configuration and version files
+cet_cmake_config(CONFIG_FRAGMENTS nugen-fragment.cmake.in)
diff --git a/cmakeConfig/nugen-fragment.cmake.in b/cmakeConfig/nugen-fragment.cmake.in
new file mode 100644
index 0000000..18ff201
--- /dev/null
+++ b/cmakeConfig/nugen-fragment.cmake.in
@@ -0,0 +1,6 @@
+find_package( ROOT REQUIRED COMPONENTS MathCore Geom GeomPainter GX11 EG EGPythia6 Gui)
+find_package( art REQUIRED)
+find_package( art_root_io REQUIRED)
+find_package( cetlib REQUIRED)
+find_package( cetlib_except REQUIRED)
+find_package( nusimdata REQUIRED )
diff --git a/nugen/EventGeneratorBase/CMakeLists.txt b/nugen/EventGeneratorBase/CMakeLists.txt
index 257459f..6bb33aa 100644
--- a/nugen/EventGeneratorBase/CMakeLists.txt
+++ b/nugen/EventGeneratorBase/CMakeLists.txt
@@ -1,7 +1,6 @@
 
 install_headers()
 install_fhicl()
-install_source()
 
 add_subdirectory(GENIE)
 add_subdirectory(GiBUU)
diff --git a/nugen/EventGeneratorBase/GENIE/CMakeLists.txt b/nugen/EventGeneratorBase/GENIE/CMakeLists.txt
index eb3601b..08873f0 100644
--- a/nugen/EventGeneratorBase/GENIE/CMakeLists.txt
+++ b/nugen/EventGeneratorBase/GENIE/CMakeLists.txt
@@ -21,13 +21,12 @@ art_make( LIB_LIBRARIES nusimdata_SimulationBase
                         ${GENIE_LIB_LIST}
                         ${DK2NU_TREE}
                         ${DK2NU_GENIE}
-                        ${CLHEP}
+                        ${CLHEP_LIBRARIES}
                         ${IFDH}
-                        ${ROOT_GEOM}
-                        ${ROOT_GEOMPAINTER}
-                        ${ROOT_BASIC_LIB_LIST} )
+                        ROOT::Geom
+                        ROOT::GeomPainter
+                        ${ROOT_LIBRARIES} )
 
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/nugen/EventGeneratorBase/GiBUU/CMakeLists.txt b/nugen/EventGeneratorBase/GiBUU/CMakeLists.txt
index 3d3ef73..32bafd6 100644
--- a/nugen/EventGeneratorBase/GiBUU/CMakeLists.txt
+++ b/nugen/EventGeneratorBase/GiBUU/CMakeLists.txt
@@ -13,10 +13,9 @@ art_make( LIB_LIBRARIES nugen_EventGeneratorBase_GENIE
                         ${LHAPDF}
                         ${GENIE_LIB_LIST}
                         ${IFDH}
-                        ${ROOT_GEOM}
-                        ${ROOT_GEOMPAINTER}
-                        ${ROOT_BASIC_LIB_LIST} )
+                        ROOT::Geom
+                        ROOT::GeomPainter
+                        ${ROOT_LIBRARIES} )
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/nugen/EventGeneratorBase/test/CMakeLists.txt b/nugen/EventGeneratorBase/test/CMakeLists.txt
index 5463ec1..7d5804d 100644
--- a/nugen/EventGeneratorBase/test/CMakeLists.txt
+++ b/nugen/EventGeneratorBase/test/CMakeLists.txt
@@ -22,15 +22,13 @@ art_make( EXCLUDE EventGeneratorTest_module.cc
                         ${GENIE_LIB_LIST}
                         ${CLHEP}
                         ${DK2NU_TREE}
-                        ${ROOT_BASIC_LIB_LIST}
-                        ${ROOT_EG}
-                        ${ROOT_TREEPLAYER}
-                        ${ROOT_FFTW}
-                        ${ROOT_REFLEX}
-                        ${ROOTSYS}/lib/libEGPythia6.so
-                        ${ROOT_GUI}
+                        ${ROOT_LIBRARIES}
+                        ROOT::EG
+                        ROOT::TreePlayer
+                        ROOT::FFTW
+                        ROOT::EGPythia6
+                        ROOT::Gui
                         )
 
 install_headers()
 install_fhicl()
-install_source( EXTRAS README )
diff --git a/nugen/NuReweight/CMakeLists.txt b/nugen/NuReweight/CMakeLists.txt
index 7609061..899fb20 100644
--- a/nugen/NuReweight/CMakeLists.txt
+++ b/nugen/NuReweight/CMakeLists.txt
@@ -5,12 +5,11 @@ art_make(
     ${PYTHIA6}
     ${LHAPDF}
     ${GENIE_LIB_LIST}
-    ${ROOT_BASIC_LIB_LIST}
+    ${ROOT_LIBRARIES}
 )
 
 install_headers()
 install_fhicl()
-install_source()
 
 add_subdirectory(art)
 
diff --git a/nugen/NuReweight/art/CMakeLists.txt b/nugen/NuReweight/art/CMakeLists.txt
index 9d3c8bf..bc69a32 100644
--- a/nugen/NuReweight/art/CMakeLists.txt
+++ b/nugen/NuReweight/art/CMakeLists.txt
@@ -21,13 +21,12 @@ set( NURW_LIBS ${ART_FRAMEWORK_SERVICES_REGISTRY}
                ${PYTHIA6}
                ${LHAPDF}
                ${GENIE_LIB_LIST}
-               ${ROOT_BASIC_LIB_LIST}
-               ${ROOT_EG}
-               ${ROOT_TREEPLAYER}
-               ${ROOT_FFTW}
-               ${ROOT_REFLEX}
-               ${ROOTSYS}/lib/libEGPythia6.so
-               ${ROOT_GUI}
+               ${ROOT_LIBRARIES}
+               ROOT::EG
+               ROOT::TreePlayer
+               ROOT::FFTW
+               ROOT::EGPythia6
+               ROOT::Gui
            )
 
 art_make_library( SOURCE NuReweight.cxx
@@ -42,4 +41,3 @@ simple_plugin( ReweightAna module nugen_NuReweight_art
 
 install_headers()
 install_fhicl()
-install_source()
