diff --git a/CMakeLists.txt b/CMakeLists.txt
index dc8bab2..346f8b2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,18 +13,13 @@
 # ======================================================================
 
 # use cmake 2.8 or later
-cmake_minimum_required (VERSION 2.8)
+cmake_minimum_required (VERSION 3.11)
+
+project(ifdh_art VERSION 2.10.02 LANGUAGES CXX )
 
 ##SET ( CMAKE_VERBOSE_MAKEFILE true )
-# cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
-IF (NOT CETBUILDTOOLS_VERSION)
-    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
-ENDIF()
 
-set(CMAKE_MODULE_PATH $ENV{ART_DIR}/Modules
-		      $ENV{CETBUILDTOOLS_DIR}/Modules
-		      ${CMAKE_MODULE_PATH})
+find_package(cetmodules)
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -37,46 +32,30 @@ cet_set_compiler_flags(DIAGS VIGILANT
 
 cet_report_compiler_flags()
 
-find_ups_product( art )
-find_ups_product( ifdhc  )
-find_ups_product( ifbeam  )
-find_ups_product( nucondb  )
-find_ups_product( libwda  )
-find_ups_product( cetbuildtools  )
-find_ups_product( cetlib_except )
+cet_find_simple_package( ifbeam  )
+cet_find_simple_package( nucondb  )
+cet_find_simple_package( wda  )
+find_package( Boost REQUIRED COMPONENTS filesystem system )
+find_package( messagefacility )
+find_package( cetlib_except )
+find_package( art )
 
-# include search path
-# decide if we have IFDHC_FQ_DIR or IFDHC_DIR
-if ( $ENV{IFDHC_FQ_DIR} MATCHES "ifdhc" )
-  message(STATUS "using FQ_DIR")
-  include_directories($ENV{IFDHC_FQ_DIR}/inc)
-  include_directories($ENV{LIBWDA_FQ_DIR}/include)
-  include_directories($ENV{IFBEAM_FQ_DIR}/include)
-  include_directories($ENV{NUCONDB_FQ_DIR}/include)
-else()
-  message(STATUS "using DIR")
-  include_directories($ENV{IFDHC_DIR}/inc)
-  include_directories($ENV{LIBWDA_DIR}/include)
-  include_directories($ENV{IFBEAM_DIR}/include)
-  include_directories($ENV{NUCONDB_DIR}/include)
-endif()
 
 #ifdhc
-cet_find_library( IFDH    NAMES ifdh    PATHS ENV IFDHC_LIB   NO_DEFAULT_PATH )
-cet_find_library( IFBEAM  NAMES ifbeam  PATHS ENV IFBEAM_LIB  NO_DEFAULT_PATH )
-cet_find_library( NUCONDB NAMES nucondb PATHS ENV NUCONDB_LIB NO_DEFAULT_PATH )
-cet_find_library( WDA     NAMES wda     PATHS ENV LIBWDA_LIB  NO_DEFAULT_PATH )
+cet_find_library( IFDHC  NAMES libidfh.so  )
+string(REPLACE "lib/libifdh.so" "inc" IFDHC_INC ${IFDHC})
+cet_find_library( IFBEAM  NAMES ifbeam  )
+cet_find_library( NUCONDB NAMES nucondb )
+cet_find_library( WDA     NAMES wda     )
 
 # SQLite
-cet_find_library(SQLITE3 NAMES sqlite3 PATHS ENV SQLITE_LIB NO_DEFAULT_PATH)
+cet_find_library(SQLITE3 NAMES sqlite3 )
 
 # macros
-include(ArtMake)
 include(BuildPlugins)
 
 add_subdirectory (src)
-#add_subdirectory (tests)
-add_subdirectory (ups)
 
 # packaging utility
-include(UseCPack)
+cet_cmake_config()
+
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ec4769d..b52a789 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,6 +1,5 @@
 
 macro( install_ifdh_headers   )
-  _cet_check_inc_directory()
   set(header_install_dir ${${product}_inc_dir} )
   message( STATUS "install_ifdh_headers: headers will be installed in ${header_install_dir}" )
   FILE(GLOB headers [^.]*.h [^.]*.hh [^.]*.icc )
diff --git a/src/IFBeamService/CMakeLists.txt b/src/IFBeamService/CMakeLists.txt
index 66adae2..a502840 100644
--- a/src/IFBeamService/CMakeLists.txt
+++ b/src/IFBeamService/CMakeLists.txt
@@ -2,6 +2,7 @@
 
 
 simple_plugin(IFBeam "service" 
+                           ${ART_UTILITIES}
 			   ${IFBEAM} 
                            ${WDA}
                            ${IFDH}
@@ -9,8 +10,8 @@ simple_plugin(IFBeam "service"
 			   ${ART_UTILITIES}
 			   ${FHICLCPP}
                            ${CETLIB}
-                           ${Boost_FILESYSTEM_LIBRARY}
-                           ${Boost_SYSTEM_LIBRARY}
+                           Boost::filesystem
+                           Boost::system
               BASENAME_ONLY
          )
 
diff --git a/src/IFCatalogInterface/CMakeLists.txt b/src/IFCatalogInterface/CMakeLists.txt
index ea6098f..5dff4bf 100644
--- a/src/IFCatalogInterface/CMakeLists.txt
+++ b/src/IFCatalogInterface/CMakeLists.txt
@@ -7,9 +7,9 @@ simple_plugin(IFCatalogInterface "service"
                            ${MF_UTILITIES}
 			   ${FHICLCPP}
                            ${CETLIB} ${CETLIB_EXCEPT}
-                           ${Boost_FILESYSTEM_LIBRARY}
-                           ${Boost_SYSTEM_LIBRARY}
-                           ${ROOT_BASIC_LIB_LIST}
+                           Boost::filesystem
+                           Boost::system
+                           ${ROOT_LIBRARIES}
               BASENAME_ONLY
 		)
 include_directories(../IFDHService)
diff --git a/src/IFDHService/CMakeLists.txt b/src/IFDHService/CMakeLists.txt
index 6154868..bb283c0 100644
--- a/src/IFDHService/CMakeLists.txt
+++ b/src/IFDHService/CMakeLists.txt
@@ -7,8 +7,8 @@ simple_plugin(IFDH service
                            ${MF_UTILITIES}
 			   ${FHICLCPP}
                            ${CETLIB} ${CETLIB_EXCEPT}
-                           ${Boost_FILESYSTEM_LIBRARY}
-                           ${Boost_SYSTEM_LIBRARY}
+                           Boost::filesystem
+                           Boost::system
               BASENAME_ONLY
              )
 
diff --git a/src/IFFileTransfer/CMakeLists.txt b/src/IFFileTransfer/CMakeLists.txt
index 856b8c8..4a06e19 100644
--- a/src/IFFileTransfer/CMakeLists.txt
+++ b/src/IFFileTransfer/CMakeLists.txt
@@ -10,8 +10,8 @@ simple_plugin(IFFileTransfer "service"
 			   ${FHICLCPP}
                            ${CETLIB} ${CETLIB_EXCEPT}
                            ${ROOT_BASIC_LIB_LIST}
-                           ${Boost_FILESYSTEM_LIBRARY}
-                           ${Boost_SYSTEM_LIBRARY}
+                           Boost::filesystem
+                           Boost::system
               BASENAME_ONLY
            )
 
diff --git a/src/nucondbService/CMakeLists.txt b/src/nucondbService/CMakeLists.txt
index 1d2c403..22d55ae 100644
--- a/src/nucondbService/CMakeLists.txt
+++ b/src/nucondbService/CMakeLists.txt
@@ -8,8 +8,8 @@ simple_plugin(nucondb "service"
                            ${MF_MESSAGELOGGER}
 			   ${FHICLCPP}
                            ${CETLIB}
-                           ${Boost_FILESYSTEM_LIBRARY}
-                           ${Boost_SYSTEM_LIBRARY}
+                           Boost::filesystem
+                           Boost::system
               BASENAME_ONLY
         )
 
