diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7f63de8..017aa30 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,18 +13,13 @@
 # ======================================================================
 
 # use cmake 2.8 or later
-cmake_minimum_required (VERSION 2.8)
+cmake_minimum_required (VERSION 3.11)
 
-##SET ( CMAKE_VERBOSE_MAKEFILE true )
-# cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
-IF (NOT CETBUILDTOOLS_VERSION)
-    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
-ENDIF()
 
 set(CMAKE_MODULE_PATH $ENV{ART_DIR}/Modules
 		      $ENV{CETBUILDTOOLS_DIR}/Modules
 		      ${CMAKE_MODULE_PATH})
+find_package( cetmodules  )
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -37,29 +32,17 @@
 
 cet_report_compiler_flags()
 
-find_ups_product( art )
-find_ups_product( ifdhc  )
-find_ups_product( ifbeam  )
-find_ups_product( nucondb  )
-find_ups_product( libwda  )
-find_ups_product( cetbuildtools  )
-find_ups_product( cetlib_except )
-
-# include search path
-# decide if we have IFDHC_FQ_DIR or IFDHC_DIR
-if ( $ENV{IFDHC_FQ_DIR} MATCHES "ifdhc" )
-  message(STATUS "using FQ_DIR")
-  include_directories($ENV{IFDHC_FQ_DIR}/inc)
-  include_directories($ENV{LIBWDA_FQ_DIR}/include)
-  include_directories($ENV{IFBEAM_FQ_DIR}/include)
-  include_directories($ENV{NUCONDB_FQ_DIR}/include)
-else()
-  message(STATUS "using DIR")
-  include_directories($ENV{IFDHC_DIR}/inc)
-  include_directories($ENV{LIBWDA_DIR}/include)
-  include_directories($ENV{IFBEAM_DIR}/include)
-  include_directories($ENV{NUCONDB_DIR}/include)
-endif()
+find_package( art )
+find_package( ifdhc  )
+find_package( ifbeam  )
+find_package( nucondb  )
+find_package( libwda  )
+find_package( cetlib_except )
+
+include_directories($ENV{IFDHC_FQ_DIR}/inc)
+include_directories($ENV{LIBWDA_FQ_DIR}/include)
+include_directories($ENV{IFBEAM_FQ_DIR}/include)
+include_directories($ENV{NUCONDB_FQ_DIR}/include)
 
 #ifdhc
 cet_find_library( IFDH    NAMES ifdh    PATHS ENV IFDHC_LIB   NO_DEFAULT_PATH )
@@ -76,7 +59,7 @@
 
 add_subdirectory (src)
 #add_subdirectory (tests)
-add_subdirectory (ups)
+#add_subdirectory (ups)
 
 # packaging utility
-include(UseCPack)
+cet_cmake_config()
