diff --git a/CMakeLists.txt b/CMakeLists.txt
index 334455e..28d7d07 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,12 +13,12 @@
 #  make package (builds distribution tarfile)
 # ======================================================================
 
-cmake_minimum_required(VERSION 3.11)  # use cmake 3.11 or later
+cmake_minimum_required(VERSION 3.12)  # use cmake 3.12 or later
 
-project(nuevdb)
+project(nuevdb VERSION 1.02.02 LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
+set(fcl_dir job)
+find_package(cetmodules REQUIRED)
 
 list(APPEND CMAKE_MODULE_PATH
   $ENV{CANVAS_ROOT_IO_DIR}/Modules
@@ -36,14 +36,10 @@ cet_set_compiler_flags(DIAGS CAUTIOUS WERROR
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product( libwda )
-find_ups_product( postgresql )
-find_ups_product( nusimdata )
-find_ups_product( cetlib )
-find_ups_product( cetlib_except )
-find_ups_product( cetbuildtools )
+find_package( art_root_io REQUIRED )
+find_package( libwda )
+find_package( PostgreSQL REQUIRED)
+find_package( nusimdata REQUIRED)
 
 message(STATUS "ROOTSYS is ${ROOTSYS}")
 
@@ -53,7 +49,7 @@ include(BuildPlugins)
 include(CetRootCint)
 
 # include search path
-include_directories ( $ENV{CRYHOME}/src )
+include_directories ( ${CRYHOME}/src )
 
 # included for Mac OSX using XQuartz for X11
 if(APPLE)
@@ -62,10 +58,10 @@ endif()
 
 # source
 add_subdirectory (nuevdb)
-
+add_subdirectory(cmakeConfig)
 # ups - table and config files
-add_subdirectory(ups)
+#add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
+#include(UseCPack)
 
diff --git a/cmakeConfig/CMakeLists.txt b/cmakeConfig/CMakeLists.txt
new file mode 100644
index 0000000..3c96fe3
--- /dev/null
+++ b/cmakeConfig/CMakeLists.txt
@@ -0,0 +1,3 @@
+
+# create package configuration and version files
+cet_cmake_config(CONFIG_FRAGMENTS nuevdb-fragment.cmake.in)
diff --git a/cmakeConfig/nuevdb-fragment.cmake.in b/cmakeConfig/nuevdb-fragment.cmake.in
new file mode 100644
index 0000000..cedc2d1
--- /dev/null
+++ b/cmakeConfig/nuevdb-fragment.cmake.in
@@ -0,0 +1,4 @@
+find_package( art-root-io REQUIRED )
+find_package( libwda )
+find_package( PostgreSQL REQUIRED)
+find_package( nusimdata REQUIRED)
diff --git a/nuevdb/EventDisplayBase/CMakeLists.txt b/nuevdb/EventDisplayBase/CMakeLists.txt
index 7b39bbd..aaaa97c 100644
--- a/nuevdb/EventDisplayBase/CMakeLists.txt
+++ b/nuevdb/EventDisplayBase/CMakeLists.txt
@@ -17,29 +17,27 @@ set( EVD_LIBS ${ART_FRAMEWORK_SERVICES_REGISTRY}
               cetlib
               cetlib_except
               ${Boost_SYSTEM_LIBRARY}
-              ${ROOT_EVE}
-              ${ROOT_EG}
-              ${ROOT_GX11}
-              ${ROOT_TREEPLAYER}
-              ${ROOT_GEOM}
-              ${ROOT_GED}
-              ${ROOT_RGL}
-              ${ROOT_GUI}
-              ${ROOT_CORE}
-              ${ROOT_CINT}
-              ${ROOT_RIO}
-              ${ROOT_NET}
-              ${ROOT_HIST}
-              ${ROOT_GRAF}
-              ${ROOT_GRAF3D}
-              ${ROOT_GPAD}
-              ${ROOT_TREE}
-              ${ROOT_RINT}
-              ${ROOT_POSTSCRIPT}
-              ${ROOT_MATRIX}
-              ${ROOT_PHYSICS}
-              ${ROOT_MATHCORE}
-              ${ROOT_THREAD} )
+              ROOT::EG
+              ROOT::GX11
+              ROOT::TreePlayer
+              ROOT::Geom
+              ROOT::Ged
+              ROOT::RGL
+              ROOT::Gui
+              ROOT::Core
+              ROOT::RIO
+              ROOT::Rint
+              ROOT::Net
+              ROOT::Hist
+              ROOT::Graf
+              ROOT::Graf3d
+              ROOT::Gpad
+              ROOT::Tree
+              ROOT::Postscript
+              ROOT::Matrix
+              ROOT::Physics
+              ROOT::MathCore
+              ROOT::Thread )
 
 cet_rootcint( ${PACKAGE} )
 
diff --git a/nuevdb/IFDatabase/CMakeLists.txt b/nuevdb/IFDatabase/CMakeLists.txt
index b078e9d..00a9f0c 100644
--- a/nuevdb/IFDatabase/CMakeLists.txt
+++ b/nuevdb/IFDatabase/CMakeLists.txt
@@ -1,13 +1,12 @@
 set( PACKAGE IFDatabase )
 FILE( GLOB src_files *.cpp )
 
-cet_find_library( LIBWDA NAMES wda PATHS ENV LIBWDA_LIB NO_DEFAULT_PATH )
-cet_find_library( PQ NAMES pq PATHS ENV POSTGRESQL_LIBRARIES NO_DEFAULT_PATH )
+cet_find_library( LIBWDA NAMES wda PATHS ${CMAKE_PREFIX_PATH})
 find_library( CURL NAMES curl )
 
-include_directories( $ENV{LIBWDA_FQ_DIR}/include )
+include_directories( ${LIBWDA_DIR}/include )
 
-art_make_library( SOURCE ${src_files} #${CMAKE_CURRENT_BINARY_DIR}/${PACKAGE}Cint.cc
+art_make_library( SOURCE ${src_files}
                   LIBRARIES     ${LIBWDA}
                                 ${BOOST_DATE_TIME}
                                 ${BOOST_SYSTEM}
@@ -15,7 +14,7 @@ art_make_library( SOURCE ${src_files} #${CMAKE_CURRENT_BINARY_DIR}/${PACKAGE}Cin
                                 ${BOOST_THREAD}
                                 ${BOOST_REGEX}
                                 ${XCLIB}
-                                ${PQ}
+                                ${PostgreSQL_LIBRARIES}
                                 ${CURL}
                  )
 
@@ -56,4 +55,3 @@ simple_plugin( DBI "service"
 
 install_headers()
 install_fhicl()
-install_source()
