diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5856efe..1f8c6ff 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,36 +1,39 @@
 
 cmake_minimum_required (VERSION 3.12...3.18 FATAL_ERROR)
 
-find_package(cetbuildtools REQUIRED)
-project(artg4tk)
+project(artg4tk VERSION 10.02.01 LANGUAGES CXX )
+set( fcl_dir  fcl )
+set( gdml_dir gdml )
+ 
+find_package(cetmodules REQUIRED)
 
-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules $ENV{ART_DIR}/Modules)
+find_package(canvas_root_io)
 
 include(CetCMakeEnv)
 cet_cmake_env()
 
 cet_set_compiler_flags(DIAGS CAUTIOUS WERROR NO_UNDEFINED EXTRA_FLAGS -pedantic)
 
-#if (NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-#set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined,--no-allow-shlib-undefined,--as-needed")
-#set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULELINKER_FLAGS} -Wl,--no-undefined,--no-allow-shlib-undefined,--as-needed")
-#set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--no-undefined,--no-allow-shlib-undefined,--as-needed")
-#endif()
 
 cet_report_compiler_flags()
 
-find_ups_product(art)
-find_ups_product(art_root_io)
-find_ups_boost()
-find_ups_product(canvas)
-find_ups_product(canvas_root_io)
-find_ups_product(cetlib)
-find_ups_product(cetlib_except)
-find_ups_product(clhep)
-find_ups_product(hep_concurrency)
-find_ups_product(fhiclcpp)
-find_ups_geant4()
-find_ups_product(messagefacility)
+# These are needed for Art stuff
+find_package(art REQUIRED)
+find_package(art_root_io REQUIRED)
+find_package(canvas_root_io REQUIRED)
+find_package(canvas REQUIRED)
+find_package(fhiclcpp REQUIRED)
+find_package(messagefacility REQUIRED)
+find_package(cetlib REQUIRED)
+find_package(cetlib_except REQUIRED)
+find_package(hep_concurrency REQUIRED)
+find_package(CLHEP REQUIRED)
+find_package(Geant4 REQUIRED COMPONENTS ui_all vis_all OPTIONAL_COMPONENTS G4processes G4geometry G4materials  G4intercoms G4graphics_reps G4persistency G4global G4physicslists G4run G4particles G4track )
+find_package(ROOT REQUIRED COMPONENTS)
+find_package(Boost REQUIRED COMPONENTS system filesystem)
+include(${Geant4_USE_FILE})
+get_target_property(BoostIncludes Boost::filesystem INTERFACE_INCLUDE_DIRECTORIES)
+include_directories(${BoostIncludes})
 
 include(ArtDictionary)
 include(ArtMake)
@@ -39,7 +42,9 @@ include(BuildPlugins)
 add_subdirectory(artg4tk)
 add_subdirectory(examples)
 add_subdirectory(fcl)
+if(WANT_UPS)
 add_subdirectory(ups)
+endif()
 
 # packaging utility - must be included last
 include(UseCPack)
diff --git a/artg4tk/CMakeLists.txt b/artg4tk/CMakeLists.txt
index 40e2874..3eb86d0 100644
--- a/artg4tk/CMakeLists.txt
+++ b/artg4tk/CMakeLists.txt
@@ -1,6 +1,4 @@
-
-include_directories( $ENV{GEANT4_FQ_DIR}/include )
-
+set(CMAKE_CXX_FLAGS "${Geant4_CXX_FLAGS} -Wno-shadow -Wno-unused-parameter -Wno-ignored-qualifiers -Wno-misleading-indentation -Wno-maybe-uninitialized -Wno-overloaded-virtual")
 add_subdirectory(actionBase)
 add_subdirectory(Analysis)
 add_subdirectory(Core)
diff --git a/artg4tk/Core/CMakeLists.txt b/artg4tk/Core/CMakeLists.txt
index ef0a109..775f77d 100644
--- a/artg4tk/Core/CMakeLists.txt
+++ b/artg4tk/Core/CMakeLists.txt
@@ -3,7 +3,7 @@ art_make(
     artg4tk_geantInit
     artg4tk_services_DetectorHolder_service
     artg4tk_services_PhysicsListHolder_service
-    ${G4INTERFACES}
+    ${Geant4_LIBRARIES}
 )
 
 install_headers()
diff --git a/artg4tk/DataProducts/EventGenerators/CMakeLists.txt b/artg4tk/DataProducts/EventGenerators/CMakeLists.txt
index 35ab057..d1e2780 100644
--- a/artg4tk/DataProducts/EventGenerators/CMakeLists.txt
+++ b/artg4tk/DataProducts/EventGenerators/CMakeLists.txt
@@ -1,3 +1,3 @@
-art_make(LIB_LIBRARIES canvas ${CLHEP})
+art_make(LIB_LIBRARIES canvas ${CLHEP_LIBRARIES})
 
 install_headers()
diff --git a/artg4tk/DataProducts/G4DetectorHits/CMakeLists.txt b/artg4tk/DataProducts/G4DetectorHits/CMakeLists.txt
index 0a83369..a796e2f 100644
--- a/artg4tk/DataProducts/G4DetectorHits/CMakeLists.txt
+++ b/artg4tk/DataProducts/G4DetectorHits/CMakeLists.txt
@@ -1,3 +1,3 @@
-art_make(LIB_LIBRARIES ${G4PARTICLES})
+art_make(LIB_LIBRARIES G4particles)
 
 install_headers()
diff --git a/artg4tk/EventGenerators/CMakeLists.txt b/artg4tk/EventGenerators/CMakeLists.txt
index f105fee..0d0085f 100644
--- a/artg4tk/EventGenerators/CMakeLists.txt
+++ b/artg4tk/EventGenerators/CMakeLists.txt
@@ -1,5 +1,5 @@
 art_make(
   MODULE_LIBRARIES
   artg4tk_DataProducts_EventGenerators
-  ${CLHEP}
+  ${CLHEP_LIBRARIES}
 )
diff --git a/artg4tk/ExpDataAccess/CMakeLists.txt b/artg4tk/ExpDataAccess/CMakeLists.txt
index f9ce49d..317fdc8 100644
--- a/artg4tk/ExpDataAccess/CMakeLists.txt
+++ b/artg4tk/ExpDataAccess/CMakeLists.txt
@@ -8,7 +8,7 @@ endif()
 art_make(
   LIB_LIBRARIES
     ${CURL_LIBRARIES}
-    ${ROOT_CORE}
-    ${ROOT_HIST}
+    ROOT::Core
+    ROOT::Hist
     MF_MessageLogger
 )
diff --git a/artg4tk/G4PhysModelParamStudy/AnalysisBase/CMakeLists.txt b/artg4tk/G4PhysModelParamStudy/AnalysisBase/CMakeLists.txt
index afba9d2..245544e 100644
--- a/artg4tk/G4PhysModelParamStudy/AnalysisBase/CMakeLists.txt
+++ b/artg4tk/G4PhysModelParamStudy/AnalysisBase/CMakeLists.txt
@@ -1,10 +1,3 @@
-find_program(ROOT_CONFIG_EXECUTABLE root-config PATHS ${ROOTSYS}/bin $ENV{ROOTSYS}/bin)
-
-execute_process(
-  COMMAND ${ROOT_CONFIG_EXECUTABLE} --libs
-  OUTPUT_VARIABLE ROOT_LIBRARIES
-  OUTPUT_STRIP_TRAILING_WHITESPACE
-)
 
 art_make(
   LIB_LIBRARIES
@@ -12,6 +5,6 @@ art_make(
     artg4tk_ExpDataAccess
     artg4tk_G4PhysModelParamStudy_AnalysisProducts
     artg4tk_G4PhysModelParamStudy_AnalysisProducts_dict
-    ${G4_LIB_LIST}
+    ${Geant4_LIBRARIES}
     ${ROOT_LIBRARIES}
 )
diff --git a/artg4tk/G4PhysModelParamStudy/AnalysisModules/CMakeLists.txt b/artg4tk/G4PhysModelParamStudy/AnalysisModules/CMakeLists.txt
index 30b17c1..dd0ead4 100644
--- a/artg4tk/G4PhysModelParamStudy/AnalysisModules/CMakeLists.txt
+++ b/artg4tk/G4PhysModelParamStudy/AnalysisModules/CMakeLists.txt
@@ -3,5 +3,5 @@ art_make(
     art_root_io_TFileService_service
     artg4tk_G4PhysModelParamStudy_AnalysisBase
     artg4tk_G4PhysModelParamStudy_StatAnalysisTools
-    ${G4_LIB_LIST}
+    ${Geant4_LIBRARIES}
 )
diff --git a/artg4tk/G4PhysModelParamStudy/ApplicationModules/CMakeLists.txt b/artg4tk/G4PhysModelParamStudy/ApplicationModules/CMakeLists.txt
index 4752bdd..d614991 100644
--- a/artg4tk/G4PhysModelParamStudy/ApplicationModules/CMakeLists.txt
+++ b/artg4tk/G4PhysModelParamStudy/ApplicationModules/CMakeLists.txt
@@ -4,5 +4,5 @@ art_make(
     artg4tk_G4PhysModelParamStudy_G4Components
     artg4tk_geantInit
     artg4tk_pluginDetectors_gdml
-    ${G4_LIB_LIST}
+    ${Geant4_LIBRARIES}
 )
diff --git a/artg4tk/G4PhysModelParamStudy/G4Components/CMakeLists.txt b/artg4tk/G4PhysModelParamStudy/G4Components/CMakeLists.txt
index d6f9fdc..53b441a 100644
--- a/artg4tk/G4PhysModelParamStudy/G4Components/CMakeLists.txt
+++ b/artg4tk/G4PhysModelParamStudy/G4Components/CMakeLists.txt
@@ -1,4 +1,4 @@
 art_make(
   LIB_LIBRARIES
-    ${G4_LIB_LIST}
+  ${Geant4_LIBRARIES}
 )
diff --git a/artg4tk/G4PhysModelParamStudy/G4Services/CMakeLists.txt b/artg4tk/G4PhysModelParamStudy/G4Services/CMakeLists.txt
index 6ca7679..4477233 100644
--- a/artg4tk/G4PhysModelParamStudy/G4Services/CMakeLists.txt
+++ b/artg4tk/G4PhysModelParamStudy/G4Services/CMakeLists.txt
@@ -2,5 +2,5 @@ art_make(
   SERVICE_LIBRARIES
     artg4tk_G4PhysModelParamStudy_G4Components
     artg4tk_geantInit
-    ${G4_LIB_LIST}
+    ${Geant4_LIBRARIES}
 )
diff --git a/artg4tk/G4PhysModelParamStudy/RooMUTools/CMakeLists.txt b/artg4tk/G4PhysModelParamStudy/RooMUTools/CMakeLists.txt
index ef31b28..a78fd66 100644
--- a/artg4tk/G4PhysModelParamStudy/RooMUTools/CMakeLists.txt
+++ b/artg4tk/G4PhysModelParamStudy/RooMUTools/CMakeLists.txt
@@ -1,7 +1,6 @@
 find_library(ROOMULIB NAMES plotutils PATHS $ENV{ROOMU_SYS}/lib NO_DEFAULT_PATH)
 
 include_directories($ENV{ROOMU_SYS})
-
 art_make(
   MODULE_LIBRARIES
     art_root_io_TFileService_service
diff --git a/artg4tk/actionBase/CMakeLists.txt b/artg4tk/actionBase/CMakeLists.txt
index 90d153f..9d79864 100644
--- a/artg4tk/actionBase/CMakeLists.txt
+++ b/artg4tk/actionBase/CMakeLists.txt
@@ -1,3 +1,4 @@
+include_directories(${Geant4_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS})
 art_make()
 
 install_headers()
diff --git a/artg4tk/geantInit/CMakeLists.txt b/artg4tk/geantInit/CMakeLists.txt
index 3f9394d..a30ccd0 100644
--- a/artg4tk/geantInit/CMakeLists.txt
+++ b/artg4tk/geantInit/CMakeLists.txt
@@ -1,15 +1,16 @@
+include_directories(${CLHEP_INCLUDE_DIRS})
 art_make(
   LIB_LIBRARIES
     artg4tk_services_ActionHolder_service
     artg4tk_services_DetectorHolder_service
     canvas
     cetlib_except
-    ${CLHEP}
-    ${G4EVENT}
-    ${G4GLOBAL}
-    ${G4INTERCOMS}
-    ${G4RUN}
-    ${G4TRACKING}
+    ${CLHEP_LIBRARIES}
+    G4event
+    G4global
+    G4intercoms
+    G4run
+    G4tracking
 )
 
 install_headers()
diff --git a/artg4tk/lists/CMakeLists.txt b/artg4tk/lists/CMakeLists.txt
index 0ebd139..ff5c12e 100644
--- a/artg4tk/lists/CMakeLists.txt
+++ b/artg4tk/lists/CMakeLists.txt
@@ -11,18 +11,18 @@ art_make(
     ${ROOT_GEOM}
     ${ROOT_MATHCORE}
     ${CLHEP}
-    ${G4DIGITS_HITS}
-    ${G4EVENT}
-    ${G4GEOMETRY}
-    ${G4GLOBAL}
-    ${G4GRAPHICS_REPS}
-    ${G4MATERIALS}
-    ${G4INTERCOMS}
-    ${G4PARTICLES}
-    ${G4PHYSICSLISTS}
-    ${G4PROCESSES}
-    ${G4TRACK}
-    ${G4RUN}
+    #${G4DIGITS_HITS}
+    #Geant4::G4Event
+    Geant4::G4geometry
+    Geant4::G4global
+    Geant4::G4graphics_reps
+    Geant4::G4materials
+    Geant4::G4intercoms
+    Geant4::G4particles
+    Geant4::G4physicslists
+    Geant4::G4processes
+    Geant4::G4track
+    Geant4::G4run
   SERVICE_LIBRARIES
     art::Framework_Core
 )
diff --git a/artg4tk/pluginActions/PrimaryEvent/CMakeLists.txt b/artg4tk/pluginActions/PrimaryEvent/CMakeLists.txt
index 8f8af84..876478c 100644
--- a/artg4tk/pluginActions/PrimaryEvent/CMakeLists.txt
+++ b/artg4tk/pluginActions/PrimaryEvent/CMakeLists.txt
@@ -3,8 +3,8 @@ art_make(
     art::Framework_Principal
     artg4tk_actionBase
     artg4tk_services_ActionHolder_service
-    ${CLHEP}
-    ${G4GLOBAL}
-    ${G4PARTICLES}
+    ${CLHEP_LIBRARIES}
+    G4global
+    G4particles
     messagefacility::MF_MessageLogger
 )
diff --git a/artg4tk/pluginActions/general/CMakeLists.txt b/artg4tk/pluginActions/general/CMakeLists.txt
index 36f62fc..2e2df80 100644
--- a/artg4tk/pluginActions/general/CMakeLists.txt
+++ b/artg4tk/pluginActions/general/CMakeLists.txt
@@ -2,9 +2,9 @@ art_make(
   SERVICE_LIBRARIES
     artg4tk_actionBase
     artg4tk_services_ActionHolder_service
-    ${CLHEP}
-    ${G4EVENT}
-    ${G4GLOBAL}
-    ${G4PARTICLES}
-    ${G4RUN}
+    ${CLHEP_LIBRARIES}
+    G4event
+    G4global
+    G4particles
+    G4run
 )
diff --git a/artg4tk/pluginActions/myKiller/CMakeLists.txt b/artg4tk/pluginActions/myKiller/CMakeLists.txt
index 503b116..ebf7565 100644
--- a/artg4tk/pluginActions/myKiller/CMakeLists.txt
+++ b/artg4tk/pluginActions/myKiller/CMakeLists.txt
@@ -2,6 +2,6 @@ art_make(
   SERVICE_LIBRARIES
     artg4tk_actionBase
     artg4tk_services_ActionHolder_service
-    ${CLHEP}
+    ${CLHEP_LIBRARIES}
     messagefacility::MF_MessageLogger
 )
diff --git a/artg4tk/pluginActions/myparticleGun/CMakeLists.txt b/artg4tk/pluginActions/myparticleGun/CMakeLists.txt
index 5eb4c57..62bf1e9 100644
--- a/artg4tk/pluginActions/myparticleGun/CMakeLists.txt
+++ b/artg4tk/pluginActions/myparticleGun/CMakeLists.txt
@@ -2,7 +2,7 @@ art_make(
   SERVICE_LIBRARIES
     artg4tk_actionBase
     artg4tk_services_ActionHolder_service
-    ${G4EVENT}
-    ${G4PARTICLES}
+    G4event
+    G4particles
     messagefacility::MF_MessageLogger
 )
diff --git a/artg4tk/pluginActions/myparticleGun/HepevtInputAction_service.cc b/artg4tk/pluginActions/myparticleGun/HepevtInputAction_service.cc
index 1c7cfee..9edbe26 100644
--- a/artg4tk/pluginActions/myparticleGun/HepevtInputAction_service.cc
+++ b/artg4tk/pluginActions/myparticleGun/HepevtInputAction_service.cc
@@ -29,6 +29,8 @@
 // Date: August 2012
 //=============================================================================
 
+#include "art/Framework/Services/Registry/ServiceDefinitionMacros.h"
+
 // Framework includes
 #include "fhiclcpp/ParameterSet.h"
 
diff --git a/artg4tk/pluginActions/myparticleGun/myParticleGunAction_service.cc b/artg4tk/pluginActions/myparticleGun/myParticleGunAction_service.cc
index a790352..eafd918 100644
--- a/artg4tk/pluginActions/myparticleGun/myParticleGunAction_service.cc
+++ b/artg4tk/pluginActions/myparticleGun/myParticleGunAction_service.cc
@@ -29,7 +29,7 @@
 // }
 // Author: Hans Wenzel (Fermilab)
 //=============================================================================
-
+#include "art/Framework/Services/Registry/ServiceDefinitionMacros.h"
 // art includes
 #include "fhiclcpp/ParameterSet.h"
 
diff --git a/artg4tk/pluginActions/physicsList/CMakeLists.txt b/artg4tk/pluginActions/physicsList/CMakeLists.txt
index 2ab7307..34cd0c5 100644
--- a/artg4tk/pluginActions/physicsList/CMakeLists.txt
+++ b/artg4tk/pluginActions/physicsList/CMakeLists.txt
@@ -2,8 +2,8 @@ art_make(
   SERVICE_LIBRARIES
     artg4tk_services_PhysicsListHolder_service
     artg4tk_lists
-    ${G4PHYSICSLISTS}
-    ${G4RUN}
+    G4physicslists
+    G4run
 )
 
 install_headers()
diff --git a/artg4tk/pluginDetectors/gdml/CMakeLists.txt b/artg4tk/pluginDetectors/gdml/CMakeLists.txt
index 00a9b11..5ece289 100644
--- a/artg4tk/pluginDetectors/gdml/CMakeLists.txt
+++ b/artg4tk/pluginDetectors/gdml/CMakeLists.txt
@@ -2,14 +2,14 @@ art_make(
   LIB_LIBRARIES
     artg4tk_DataProducts_G4DetectorHits
     ${CLHEP}
-    ${G4DIGITS_HITS}
-    ${G4EVENT}
-    ${G4GEOMETRY}
-    ${G4GLOBAL}
-    ${G4GRAPHICS_REPS}
-    ${G4PARTICLES}
-    ${G4PERSISTENCY}
-    ${G4TRACK}
+    #${G4DIGITS_HITS}
+    #Geant4::G4Event
+    Geant4::G4geometry
+    Geant4::G4global
+    Geant4::G4graphics_reps
+    Geant4::G4particles
+    Geant4::G4persistency
+    Geant4::G4track
     ${XERCESC}
   SERVICE_LIBRARIES
     art::Framework_Core
@@ -17,7 +17,7 @@ art_make(
     artg4tk_pluginDetectors_gdml
     artg4tk_services_DetectorHolder_service
     art::Persistency_Provenance
-    ${G4MATERIALS}
+    Geant4::G4materials
 )
 
 install_headers()
diff --git a/artg4tk/util/CMakeLists.txt b/artg4tk/util/CMakeLists.txt
index 29a2e9f..b5979f1 100644
--- a/artg4tk/util/CMakeLists.txt
+++ b/artg4tk/util/CMakeLists.txt
@@ -1,3 +1,7 @@
-art_make(LIB_LIBRARIES ${G4GRAPHICS_REPS} cetlib_except)
+art_make(
+  LIB_LIBRARIES
+    G4graphics_reps
+    ${Geant4_LIBRARIES}
+    cetlib_except )
 
 install_headers()
diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index dd654d5..f6df0ed 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -1,9 +1,7 @@
 file(GLOB fcl_files *.fcl *.gdml *.xml *.xsd *.C *.mac)
 
-install(
-  FILES ${fcl_files}
-  DESTINATION ${product}/${version}/${PROJECT_NAME}/examples
-)
+install( FILES ${fcl_files}
+         DESTINATION ${CMAKE_INSTALL_PREFIX}/examples )
 
 foreach(aFile ${fcl_files})
   get_filename_component(basename ${aFile} NAME)
diff --git a/macros/CMakeLists.txt b/macros/CMakeLists.txt
index 7c4b927..dedbdc5 100644
--- a/macros/CMakeLists.txt
+++ b/macros/CMakeLists.txt
@@ -1,9 +1,7 @@
 file(GLOB mac_files *.mac)
 
-install(
-  FILES ${mac_files}
-  DESTINATION ${product}/${version}/${PROJECT_NAME}/macros
-)
+install( FILES ${mac_files}
+         DESTINATION ${CMAKE_INSTALL_PREFIX}/macros )
 
 foreach(aFile ${mac_files})
   get_filename_component(basename ${aFile} NAME)
