--- spack-src/CMakeLists.txt.orig	2021-09-08 17:32:38.000000000 -0500
+++ spack-src/CMakeLists.txt	2021-09-10 10:09:33.575500724 -0500
@@ -1,35 +1,37 @@
 
 cmake_minimum_required (VERSION 3.12...3.18 FATAL_ERROR)
 
-project(artg4tk)
-
-find_package(cetbuildtools REQUIRED)
-
-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules $ENV{ART_DIR}/Modules)
-
+project(artg4tk VERSION 10.02.01 LANGUAGES CXX )
+set( fcl_dir  fcl )
+set( gdml_dir gdml )
+ 
+find_package(cetmodules REQUIRED)
+ 
+find_package(canvas_root_io)
+ 
 include(CetCMakeEnv)
+
 cet_cmake_env()
 
 cet_set_compiler_flags(DIAGS CAUTIOUS WERROR NO_UNDEFINED EXTRA_FLAGS -pedantic)
 
-#if (NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-#set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined,--no-allow-shlib-undefined,--as-needed")
-#set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULELINKER_FLAGS} -Wl,--no-undefined,--no-allow-shlib-undefined,--as-needed")
-#set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--no-undefined,--no-allow-shlib-undefined,--as-needed")
-#endif()
-
-cet_report_compiler_flags()
-
-find_ups_product(art)
-find_ups_product(art_root_io)
-find_ups_boost()
-find_ups_product(canvas)
-find_ups_product(canvas_root_io)
-find_ups_product(cetlib_except)
-find_ups_product(clhep)
-find_ups_product(fhiclcpp)
-find_ups_geant4()
-find_ups_product(messagefacility)
+# These are needed for Art stuff
+find_package(art REQUIRED)
+find_package(art_root_io REQUIRED)
+find_package(canvas_root_io REQUIRED)
+find_package(canvas REQUIRED)
+find_package(fhiclcpp REQUIRED)
+find_package(messagefacility REQUIRED)
+find_package(cetlib REQUIRED)
+find_package(cetlib_except REQUIRED)
+find_package(hep_concurrency REQUIRED)
+find_package(CLHEP REQUIRED)
+find_package(Geant4 REQUIRED COMPONENTS ui_all vis_all OPTIONAL_COMPONENTS G4processes G4geometry G4materials  G4intercoms G4graphics_reps G4persistency G4global G4physicslists G4run G4particles G4track )
+find_package(ROOT REQUIRED COMPONENTS)
+find_package(Boost REQUIRED COMPONENTS system filesystem)
+include(${Geant4_USE_FILE})
+get_target_property(BoostIncludes Boost::filesystem INTERFACE_INCLUDE_DIRECTORIES)
+include_directories(${BoostIncludes})
 
 include(ArtDictionary)
 include(ArtMake)
@@ -38,7 +40,10 @@
 add_subdirectory(artg4tk)
 add_subdirectory(examples)
 add_subdirectory(fcl)
+
+if(WANT_UPS)
 add_subdirectory(ups)
+endif()
 
 # packaging utility - must be included last
 include(UseCPack)
