commit 5b3a5d319cb9dc1c60c9b14966fbb4327a127a35
Author: Patrick Gartung <gartung@fnal.gov>
Date:   Sat Apr 4 20:29:40 2020 -0500

    Cetmodules

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fd742a3..9b40ef6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,21 +13,16 @@
 # ======================================================================
 
 
-# use cmake 2.8 or later
-cmake_minimum_required (VERSION 2.8)
+# use cmake 3.11 or later
+cmake_minimum_required (VERSION 3.11)
 
-project(larana)
+project(larana VERSION 08.10.03 LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
-IF (NOT CETBUILDTOOLS_VERSION)
-    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
-ENDIF()
-
-set(CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules
-		      $ENV{CETBUILDTOOLS_DIR}/Modules
-		      ${CMAKE_MODULE_PATH})
+set(fcl_dir job)
+set(fw_dir config_data)
+set(gdml_dir gdml)
 
+find_package(cetmodules REQUIRED)
 include(CetCMakeEnv)
 cet_cmake_env()
 
@@ -40,23 +35,8 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( larcoreobj )
-find_ups_product( larcorealg )
-find_ups_product( larcore )
-find_ups_product( lardataobj )
-find_ups_product( lardataalg )
-find_ups_product( lardata )
-find_ups_product( larevt )
-find_ups_product( larsim )
-find_ups_product( larreco )
-find_ups_product( nutools )
-find_ups_product( nug4 )
-find_ups_product( nurandom )
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product( cetbuildtools )
-find_ups_product( postgresql )
-find_ups_product( eigen )
+find_package( larreco REQUIRED )
+
 
 # macros for dictionary and simple_plugin
 include(ArtDictionary)
@@ -68,10 +48,10 @@ add_subdirectory(larana)
 
 # tests
 add_subdirectory(test)
-
+add_subdirectory(cmakeConfig)
 # ups - table and config files
-add_subdirectory(ups)
+#add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
+#include(UseCPack)
 
diff --git a/cmakeConfig/CMakeLists.txt b/cmakeConfig/CMakeLists.txt
new file mode 100644
index 0000000..ab17cf6
--- /dev/null
+++ b/cmakeConfig/CMakeLists.txt
@@ -0,0 +1,4 @@
+
+# create package configuration and version files
+
+cet_cmake_config()
diff --git a/cmakeConfig/larana-fragment.cmake.in b/cmakeConfig/larana-fragment.cmake.in
new file mode 100644
index 0000000..9f34616
--- /dev/null
+++ b/cmakeConfig/larana-fragment.cmake.in
@@ -0,0 +1 @@
+find_package( larreco REQUIRED )
diff --git a/larana/CosmicRemoval/CMakeLists.txt b/larana/CosmicRemoval/CMakeLists.txt
index b75eea5..1a76d77 100644
--- a/larana/CosmicRemoval/CMakeLists.txt
+++ b/larana/CosmicRemoval/CMakeLists.txt
@@ -3,9 +3,9 @@ add_subdirectory(TrackContainment)
 art_make(
   LIB_LIBRARIES
     ${FHICLCPP}
-    ${ROOT_CORE}
-    ${ROOT_HIST}
-    ${ROOT_PHYSICS}
+    ROOT::Core
+    ROOT::HisT
+    ROOT::Physics
     cetlib_except
     larcorealg_Geometry
     lardataobj_AnalysisBase
@@ -17,9 +17,9 @@ art_make(
     ${ART_ROOT_IO_TFILESERVICE_SERVICE}
     ${ART_ROOT_IO_TFILE_SUPPORT}
     ${MF_MESSAGELOGGER}
-    ${ROOT_CORE}
-    ${ROOT_PHYSICS}
-    ${ROOT_TREE}
+    ROOT::Core
+    ROOT::Physics
+    ROOT::Tree
     larana_CosmicRemoval
     larana_CosmicRemoval_TrackContainment
     larcorealg_Geometry
@@ -31,4 +31,3 @@ art_make(
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larana/CosmicRemoval/TrackContainment/CMakeLists.txt b/larana/CosmicRemoval/TrackContainment/CMakeLists.txt
index 4e4801f..239b807 100644
--- a/larana/CosmicRemoval/TrackContainment/CMakeLists.txt
+++ b/larana/CosmicRemoval/TrackContainment/CMakeLists.txt
@@ -1,7 +1,7 @@
 art_make(
   LIB_LIBRARIES
     ${FHICLCPP}
-    ${ROOT_CORE}
+    ROOT::Core
     cetlib_except
     larcorealg_Geometry
     lardataobj_AnalysisBase
@@ -11,4 +11,3 @@ art_make(
 
 
 install_headers()
-install_source()
diff --git a/larana/OpticalDetector/CMakeLists.txt b/larana/OpticalDetector/CMakeLists.txt
index 67825fe..72f5d6c 100644
--- a/larana/OpticalDetector/CMakeLists.txt
+++ b/larana/OpticalDetector/CMakeLists.txt
@@ -7,14 +7,14 @@ art_make(
     ${ART_FRAMEWORK_SERVICES_REGISTRY}
     ${ART_ROOT_IO_TFILESERVICE_SERVICE}
     ${ART_ROOT_IO_TFILE_SUPPORT}
-    ${CLHEP}
+    ${CLHEP_LIBRARIES}
     ${FHICLCPP}
     ${MF_MESSAGELOGGER}
-    ${ROOT_CORE}
-    ${ROOT_HIST}
-    ${ROOT_PHYSICS}
-    ${ROOT_RIO}
-    ${ROOT_TREE}
+    ROOT::Core
+    ROOT::Hist
+    ROOT::Physics
+    ROOT::RIO
+    ROOT::Tree
     canvas
     cetlib_except
     larana_OpticalDetector_OpDigiProperties_service
@@ -34,12 +34,11 @@ art_make(
   SERVICE_LIBRARIES
     ${CLHEP}
     ${MF_MESSAGELOGGER}
-    ${ROOT_CORE}
-    ${ROOT_HIST}
+    ROOT::Core
+    ROOT::Hist
     larcorealg_Geometry
   )
 
 install_headers()
 install_fhicl()
-install_source()
 install_fw(LIST toyWaveform.txt SUBDIRNAME OpticalDetector)
diff --git a/larana/OpticalDetector/OpHitFinder/CMakeLists.txt b/larana/OpticalDetector/OpHitFinder/CMakeLists.txt
index fb7a8b8..e40038e 100644
--- a/larana/OpticalDetector/OpHitFinder/CMakeLists.txt
+++ b/larana/OpticalDetector/OpHitFinder/CMakeLists.txt
@@ -6,9 +6,8 @@ art_make(
     ${MF_MESSAGELOGGER}
     ${FHICLCPP}
     cetlib_except
-    ${ROOT_CORE}
-    ${ROOT_HIST}
+    ROOT::Core
+    ROOT::Hist
   )
 
 install_headers()
-install_source()
diff --git a/larana/ParticleIdentification/CMakeLists.txt b/larana/ParticleIdentification/CMakeLists.txt
index 812a1da..88a1a4f 100644
--- a/larana/ParticleIdentification/CMakeLists.txt
+++ b/larana/ParticleIdentification/CMakeLists.txt
@@ -8,13 +8,13 @@ art_make(
     art_Utilities
     ${FHICLCPP}
     ${MF_MESSAGELOGGER}
-    ${ROOT_CORE}
-    ${ROOT_HIST}
-    ${ROOT_MATHCORE}
-    ${ROOT_PHYSICS}
-    ${ROOT_RIO}
-    ${ROOT_TMVA}
-    ${ROOT_TREE}
+    ROOT::Core
+    ROOT::Hist
+    ROOT::MathCore
+    ROOT::Physics
+    ROOT::RIO
+    ROOT::TMVA
+    ROOT::Tree
     canvas
     cetlib
     cetlib_except
@@ -28,6 +28,5 @@ art_make(
 
 install_headers()
 install_fhicl()
-install_source()
 
 add_subdirectory(scripts)
diff --git a/larana/ParticleIdentification/scripts/CMakeLists.txt b/larana/ParticleIdentification/scripts/CMakeLists.txt
index 76ebd68..1335578 100644
--- a/larana/ParticleIdentification/scripts/CMakeLists.txt
+++ b/larana/ParticleIdentification/scripts/CMakeLists.txt
@@ -1,2 +1 @@
 install_fhicl()
-install_source()
diff --git a/larana/T0Finder/AssociationsTools/CMakeLists.txt b/larana/T0Finder/AssociationsTools/CMakeLists.txt
index c79a0bf..18f1c62 100644
--- a/larana/T0Finder/AssociationsTools/CMakeLists.txt
+++ b/larana/T0Finder/AssociationsTools/CMakeLists.txt
@@ -6,7 +6,7 @@ art_make(
     ${ART_FRAMEWORK_SERVICES_REGISTRY}
     ${ART_PERSISTENCY_PROVENANCE}
     ${MF_MESSAGELOGGER}
-    ${ROOT_CORE}
+    ROOT::Core
     canvas
     larcorealg_Geometry
     larsim_MCCheater_BackTrackerService_service
diff --git a/larana/T0Finder/CMakeLists.txt b/larana/T0Finder/CMakeLists.txt
index f585fda..d2892b4 100644
--- a/larana/T0Finder/CMakeLists.txt
+++ b/larana/T0Finder/CMakeLists.txt
@@ -6,10 +6,10 @@ art_make(
     ${ART_ROOT_IO_TFILESERVICE_SERVICE}
     ${ART_ROOT_IO_TFILE_SUPPORT}
     ${MF_MESSAGELOGGER}
-    ${ROOT_CORE}
-    ${ROOT_HIST}
-    ${ROOT_PHYSICS}
-    ${ROOT_TREE}
+    ROOT::Core
+    ROOT::Hist
+    ROOT::Physics
+    ROOT::Tree
     larcorealg_Geometry
     lardataobj_AnalysisBase
     lardataobj_RecoBase
@@ -20,5 +20,4 @@ art_make(
 
 install_headers()
 install_fhicl()
-install_source()
 install_scripts()
diff --git a/larana/TruncatedMean/Algorithm/CMakeLists.txt b/larana/TruncatedMean/Algorithm/CMakeLists.txt
index a7fbe37..a8c375c 100644
--- a/larana/TruncatedMean/Algorithm/CMakeLists.txt
+++ b/larana/TruncatedMean/Algorithm/CMakeLists.txt
@@ -2,4 +2,3 @@ art_make()
 
 install_headers()
 install_fhicl()
-install_source()
