diff --git a/CMakeLists.txt b/CMakeLists.txt
index 06fb8e7..655985d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,10 +10,11 @@ cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
 
 project(larana VERSION 09.03.06 LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
+set(larana_FW_DIR fw)
 
-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules $ENV{ART_DIR}/Modules)
+find_package(cetmodules REQUIRED)
+find_package(canvas_root_io)
+find_package(art)
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -27,39 +28,39 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( larcoreobj )
-find_ups_product( larcorealg )
-find_ups_product( larcore )
-find_ups_product( lardataobj )
-find_ups_product( lardataalg )
-find_ups_product( lardata )
-find_ups_product( larevt )
-find_ups_product( larsim )
-find_ups_product( larreco )
-find_ups_product( nutools )
-find_ups_product( nug4 )
-find_ups_product( nurandom )
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product( postgresql )
-find_ups_product( eigen )
+find_package( larcoreobj )
+find_package( larcorealg )
+find_package( larcore )
+find_package( lardataobj )
+find_package( lardataalg )
+find_package( lardata )
+find_package( larevt )
+find_package( larsim )
+find_package( larreco )
+find_package( nutools )
+find_package( nug4 )
+find_package( nurandom )
+find_package( nusimdata )
+find_package( art )
+find_package( art_root_io )
+find_package( Eigen3 )
+find_package( PostgreSQL )
 
 # macros for dictionary and simple_plugin
 include(ArtDictionary)
 include(ArtMake)
 include(BuildPlugins)
 
-include_directories( $ENV{EIGEN_INC} )
+include_directories( ${EIGEN3_INCLUDE_DIR} )
 
 # source
 add_subdirectory(larana)
 
 # tests
 add_subdirectory(test)
-
 # ups - table and config files
-add_subdirectory(ups)
+#add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
-
+#include(UseCPack)
+cet_cmake_config()
diff --git a/larana/CosmicRemoval/CMakeLists.txt b/larana/CosmicRemoval/CMakeLists.txt
index 0c0a17b..2c77aff 100644
--- a/larana/CosmicRemoval/CMakeLists.txt
+++ b/larana/CosmicRemoval/CMakeLists.txt
@@ -2,17 +2,17 @@ add_subdirectory(TrackContainment)
 
 art_make(
   LIB_LIBRARIES
-    ${ART_UTILITIES}
-    fhiclcpp::fhiclcpp
+    art::Utilities
+    fhiclcpp::fhiclcpp fhiclcpp::types
     ROOT::Core
     ROOT::Hist
     ROOT::Physics
     cetlib_except::cetlib_except
-    larcorealg_Geometry
-    lardataobj_AnalysisBase
-    lardataobj_RecoBase
-    larsim_PhotonPropagation_PhotonVisibilityService_service
-    nusimdata_SimulationBase
+    larcorealg::Geometry
+    lardataobj::AnalysisBase
+    lardataobj::RecoBase
+    larsim::PhotonPropagation_PhotonVisibilityService_service
+    nusimdata::SimulationBase
   MODULE_LIBRARIES
     art::Framework_Services_Registry
     art_root_io::TFileService_service
@@ -21,14 +21,14 @@ art_make(
     ROOT::Core
     ROOT::Physics
     ROOT::Tree
-    larana_CosmicRemoval
-    larana_CosmicRemoval_TrackContainment
-    lardataalg_DetectorInfo
-    larcorealg_Geometry
-    lardata_ArtDataHelper
-    lardataobj_AnalysisBase
-    lardataobj_RecoBase
-    larreco_RecoAlg_Cluster3DAlgs
+    larana::CosmicRemoval
+    larana::CosmicRemoval_TrackContainment
+    lardataalg::DetectorInfo
+    larcorealg::Geometry
+    lardata::ArtDataHelper
+    lardataobj::AnalysisBase
+    lardataobj::RecoBase
+    larreco::RecoAlg_Cluster3DAlgs
   )
 
 install_headers()
diff --git a/larana/CosmicRemoval/TrackContainment/CMakeLists.txt b/larana/CosmicRemoval/TrackContainment/CMakeLists.txt
index c390032..dbe6a07 100644
--- a/larana/CosmicRemoval/TrackContainment/CMakeLists.txt
+++ b/larana/CosmicRemoval/TrackContainment/CMakeLists.txt
@@ -3,9 +3,9 @@ art_make(
     fhiclcpp::fhiclcpp
     ROOT::Core
     cetlib_except::cetlib_except
-    larcorealg_Geometry
-    lardataobj_AnalysisBase
-    lardataobj_RecoBase
+    larcorealg::Geometry
+    lardataobj::AnalysisBase
+    lardataobj::RecoBase
   )
 
 
diff --git a/larana/OpticalDetector/CMakeLists.txt b/larana/OpticalDetector/CMakeLists.txt
index 415d556..a2acaaa 100644
--- a/larana/OpticalDetector/CMakeLists.txt
+++ b/larana/OpticalDetector/CMakeLists.txt
@@ -8,7 +8,7 @@ art_make(
     art_root_io::TFileService_service
     art_root_io::tfile_support
     CLHEP::CLHEP
-    fhiclcpp::fhiclcpp
+    fhiclcpp::fhiclcpp fhiclcpp::types
     messagefacility::MF_MessageLogger
     ROOT::Core
     ROOT::Hist
@@ -17,29 +17,28 @@ art_make(
     ROOT::Tree
     canvas::canvas
     cetlib_except::cetlib_except
-    larana_OpticalDetector_OpDigiProperties_service
-    larana_OpticalDetector_OpHitFinder
-    larcorealg_Geometry
-    lardataobj_RecoBase
-    larsim_MCSTReco
-    larsim_PhotonPropagation_PhotonVisibilityService_service
-    nusimdata_SimulationBase
+    larana::OpticalDetector_OpDigiProperties_service
+    larana::OpticalDetector_OpHitFinder
+    larcorealg::Geometry
+    lardataobj::RecoBase
+    larsim::MCSTReco
+    larsim::PhotonPropagation_PhotonVisibilityService_service
+    nusimdata::SimulationBase
   MODULE_LIBRARIES
     art::Framework_Services_Registry
-    larana_OpticalDetector
-    lardataobj_RawData
-    larsim_MCCheater_ParticleInventoryService_service
-    larsim_Simulation
-    nurandom_RandomUtils_NuRandomService_service
+    larana::OpticalDetector
+    lardataobj::RawData
+    larsim::MCCheater_ParticleInventoryService_service
+    larsim::Simulation
+    nurandom::RandomUtils_NuRandomService_service
   SERVICE_LIBRARIES
     CLHEP::CLHEP
     messagefacility::MF_MessageLogger
     ROOT::Core
     ROOT::Hist
-    larcorealg_Geometry
+    larcorealg::Geometry
   )
 
 install_headers()
 install_fhicl()
-install_source()
 install_fw(LIST toyWaveform.txt SUBDIRNAME OpticalDetector)
diff --git a/larana/OpticalDetector/OpHitFinder/CMakeLists.txt b/larana/OpticalDetector/OpHitFinder/CMakeLists.txt
index fdfc309..9340adb 100644
--- a/larana/OpticalDetector/OpHitFinder/CMakeLists.txt
+++ b/larana/OpticalDetector/OpHitFinder/CMakeLists.txt
@@ -1,10 +1,10 @@
 # OpFlashFinder_module.cc is not built - it seems specific to uboone
 art_make(
   LIB_LIBRARIES
-    lardataobj_RecoBase
-    larcorealg_Geometry
+    lardataobj::RecoBase
+    larcorealg::Geometry
     messagefacility::MF_MessageLogger
-    fhiclcpp::fhiclcpp
+    fhiclcpp::fhiclcpp fhiclcpp::types
     cetlib_except::cetlib_except
     ROOT::Core
     ROOT::Hist
diff --git a/larana/ParticleIdentification/CMakeLists.txt b/larana/ParticleIdentification/CMakeLists.txt
index de7f397..d5611c3 100644
--- a/larana/ParticleIdentification/CMakeLists.txt
+++ b/larana/ParticleIdentification/CMakeLists.txt
@@ -2,11 +2,11 @@ art_make(
   LIB_LIBRARIES
     art::Framework_Principal
     art::Framework_Services_Registry
-    ${ART_PERSISTENCY_PROVENANCE}
+    art::Persistency_Provenance
     art_root_io::TFileService_service
     art_root_io::tfile_support
     art::Utilities
-    fhiclcpp::fhiclcpp
+    fhiclcpp::fhiclcpp fhiclcpp::types
     messagefacility::MF_MessageLogger
     ROOT::Core
     ROOT::Hist
@@ -18,16 +18,15 @@ art_make(
     canvas::canvas
     cetlib::cetlib
     cetlib_except::cetlib_except
-    larcorealg_Geometry
-    lardataobj_AnalysisBase
-    larreco_Calorimetry
-    nusimdata_SimulationBase
+    larcorealg::Geometry
+    lardataobj::AnalysisBase
+    larreco::Calorimetry
+    nusimdata::SimulationBase
   MODULE_LIBRARIES
-   larana_ParticleIdentification
+   larana::ParticleIdentification
   )
 
 install_headers()
 install_fhicl()
-install_source()
 
 add_subdirectory(scripts)
diff --git a/larana/T0Finder/AssociationsTools/CMakeLists.txt b/larana/T0Finder/AssociationsTools/CMakeLists.txt
index bc372b3..0b85a00 100644
--- a/larana/T0Finder/AssociationsTools/CMakeLists.txt
+++ b/larana/T0Finder/AssociationsTools/CMakeLists.txt
@@ -4,12 +4,12 @@ art_make(
   TOOL_LIBRARIES
     art::Framework_Principal
     art::Framework_Services_Registry
-    ${ART_PERSISTENCY_PROVENANCE}
+    art::Persistency_Provenance
     messagefacility::MF_MessageLogger
     ROOT::Core
     canvas::canvas
-    larcorealg_Geometry
-    larsim_MCCheater_BackTrackerService_service
+    larcorealg::Geometry
+    larsim::MCCheater_BackTrackerService_service
 )
 
 install_headers()
diff --git a/larana/T0Finder/CMakeLists.txt b/larana/T0Finder/CMakeLists.txt
index d9ab07d..762a5cc 100644
--- a/larana/T0Finder/CMakeLists.txt
+++ b/larana/T0Finder/CMakeLists.txt
@@ -9,13 +9,13 @@ art_make(MODULE_LIBRARIES
            ROOT::Hist
            ROOT::Physics
            ROOT::Tree
-           larcorealg_Geometry
-           lardataalg_DetectorInfo
-           lardataobj_AnalysisBase
-           lardataobj_RecoBase
-           larsim_MCCheater_BackTrackerService_service
-           larsim_MCCheater_ParticleInventoryService_service
-           nusimdata_SimulationBase)
+           larcorealg::Geometry
+           lardataalg::DetectorInfo
+           lardataobj::AnalysisBase
+           lardataobj::RecoBase
+           larsim::MCCheater_BackTrackerService_service
+           larsim::MCCheater_ParticleInventoryService_service
+           nusimdata::SimulationBase)
 
 install_headers()
 install_fhicl()
diff --git a/test/OpticalDetector/CMakeLists.txt b/test/OpticalDetector/CMakeLists.txt
index daf35f9..4757be5 100644
--- a/test/OpticalDetector/CMakeLists.txt
+++ b/test/OpticalDetector/CMakeLists.txt
@@ -8,5 +8,5 @@ include(CetTest)
 cet_enable_asserts()
 
 cet_test(OpFlashAlg_test USE_BOOST_UNIT
-                         LIBRARIES larana_OpticalDetector
+			 LIBRARIES larana::OpticalDetector
 )
