diff --git a/CMakeLists.txt b/CMakeLists.txt
index b127e2a..1426500 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,9 +10,11 @@ cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
 
 project(larg4 VERSION 09.06.02.01 LANGUAGES CXX)
 
-find_package(cetbuildtools REQUIRED)
+set(larg4_FW_DIR fw)
 
-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules $ENV{ART_DIR}/Modules)
+find_package(cetmodules REQUIRED)
+find_package(canvas_root_io)
+find_package(art)
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -21,24 +23,27 @@ cet_set_compiler_flags(DIAGS CAUTIOUS WERROR NO_UNDEFINED EXTRA_FLAGS -pedantic)
 
 cet_report_compiler_flags()
 
-find_ups_product(art)
-find_ups_product(art_root_io)
-find_package(artg4tk REQUIRED)
-find_ups_product(canvas)
-find_ups_product(canvas_root_io)
-find_ups_product(cetlib)
-find_ups_product(clhep)
-find_ups_product(fhiclcpp)
-find_ups_geant4()
-find_ups_product(larcore)
-find_ups_product(larcoreobj)
-find_ups_product(larcorealg)
-find_ups_product(lardataalg)
-find_ups_product(lardataobj)
-find_ups_product(messagefacility)
-find_ups_product(nusimdata)
-find_ups_product(nug4)
-find_ups_product(nurandom)
+find_package( art )
+find_package( art_root_io )
+find_package( artg4tk )
+find_package( canvas )
+find_package( canvas_root_io )
+find_package( cetlib )
+find_package( CLHEP )
+find_package( fhiclcpp )
+find_package( larcore )
+find_package( larcoreobj )
+find_package( larcorealg )
+find_package( lardataalg )
+find_package( lardataobj )
+find_package( messagefacility )
+find_package( nusimdata )
+find_package( nug4 )
+find_package( nurandom )
+find_package( Boost COMPONENTS ALL REQUIRED )
+find_package( Geant4 REQUIRED )
+include(${Geant4_USE_FILE})
+
 find_package(Range REQUIRED)
 
 include(ArtDictionary)
@@ -49,8 +54,11 @@ add_subdirectory(fcl)
 add_subdirectory(gdml)
 add_subdirectory(larg4)
 add_subdirectory(macros)
+if(WANT_UPS)
 add_subdirectory(ups)
+endif()
 
 # packaging utility
 # this has to be last
 include(UseCPack)
+cet_cmake_config()
diff --git a/larg4/Analysis/CMakeLists.txt b/larg4/Analysis/CMakeLists.txt
index 2480860..9d31e26 100644
--- a/larg4/Analysis/CMakeLists.txt
+++ b/larg4/Analysis/CMakeLists.txt
@@ -1,4 +1,5 @@
-art_make(
+set(CMAKE_CXX_FLAGS "${Geant4_CXX_FLAGS} -Wno-shadow -Wno-unused-parameter")
+set(
   MODULE_LIBRARIES
     art_root_io::TFileService_service
     art::Framework_Services_Registry
@@ -9,9 +10,16 @@ art_make(
     canvas::canvas
     cetlib_except::cetlib_except
     ${G4ZLIB}
-    nusimdata_SimulationBase
+    nusimdata::SimulationBase
     ROOT::Core
     ROOT::Hist
     ROOT::Tree
 )
+
+cet_build_plugin(CheckAuxDetHit art::module LIBRARIES  ${MODULE_LIBRARIES})
+
+cet_build_plugin (CheckMCParticle art::module LIBRARIES ${MODULE_LIBRARIES})
+
+cet_build_plugin(CheckSimEnergyDeposit art::module LIBRARIES ${MODULE_LIBRARIES})
+
 install_source()
diff --git a/larg4/Core/CMakeLists.txt b/larg4/Core/CMakeLists.txt
index a1967b9..58c6efa 100644
--- a/larg4/Core/CMakeLists.txt
+++ b/larg4/Core/CMakeLists.txt
@@ -1,4 +1,5 @@
-art_make(
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${Geant4_CXX_FLAGS} -Wno-shadow -Wno-unused-parameter")
+set(
   MODULE_LIBRARIES
     art::Framework_Principal
     art::Framework_Services_Registry
@@ -13,20 +14,22 @@ art_make(
     cetlib::cetlib
     cetlib_except::cetlib_except
     CLHEP::CLHEP
-    fhiclcpp
-    ${G4EVENT}
-    ${G4INTERCOMS}
-    ${G4INTERFACES}
-    ${G4RUN}
-    ${G4TRACKING}
-    larg4_pluginActions_MCTruthEventAction_service
-    larg4_pluginActions_ParticleListAction_service
-    nurandom_RandomUtils_NuRandomService_service
+    fhiclcpp::fhiclcpp
+    Geant4::G4event
+    Geant4::G4intercoms
+    Geant4::G4interfaces
+    Geant4::G4run
+    Geant4::G4tracking
+    larg4::pluginActions_MCTruthEventAction_service
+    larg4::pluginActions_ParticleListAction_service
+    nurandom::RandomUtils_NuRandomService_service
     nug4::AdditionalG4Physics
-    MF_MessageLogger
+    messagefacility::MF_MessageLogger
     ROOT::Core
     ROOT::Physics
 )
 
+cet_build_plugin(larg4Main art::module LIBRARIES ${MODULE_LIBRARIES})
+
 install_headers()
 install_source()
diff --git a/larg4/Services/CMakeLists.txt b/larg4/Services/CMakeLists.txt
index deb24fc..5276efe 100644
--- a/larg4/Services/CMakeLists.txt
+++ b/larg4/Services/CMakeLists.txt
@@ -1,10 +1,12 @@
-simple_plugin(
-  LArG4Detector service
+set(CMAKE_CXX_FLAGS "${Geant4_CXX_FLAGS} -Wno-shadow -Wno-unused-parameter")
+message("XercesC_LIBRARIES = ${XercesC_LIBRARIES}")
+message("CLHEP_LIBRARIES = ${CLHEP_LIBRARIES}")
+cet_build_plugin(LArG4Detector art::service
   SOURCE
     LArG4Detector_service.cc
     SimEnergyDepositSD.cc
     AuxDetSD.cc
-  NOP
+  NOP LIBRARIES
     art::Framework_Core
     art::Framework_Principal
     art::Framework_Services_Registry
@@ -16,18 +18,18 @@ simple_plugin(
     cetlib::cetlib
     cetlib_except::cetlib_except
     CLHEP::CLHEP
-    fhiclcpp
-    G4DIGITS_HITS
-    G4EVENT
-    G4GEOMETRY
-    G4GLOBAL
-    G4MATERIALS
-    G4PERSISTENCY
-    larcorealg_Geometry
+    fhiclcpp::fhiclcpp
+    Geant4::G4digits_hits
+    Geant4::G4event
+    Geant4::G4geometry
+    Geant4::G4global
+    Geant4::G4materials
+    Geant4::G4persistency
+    larcorealg::Geometry
+    messagefacility::MF_MessageLogger
     larg4_pluginActions_ParticleListAction_service
-    MF_MessageLogger
     ROOT::Core
-    XERCESC
+    XercesC::XercesC
 )
 
 install_headers()
diff --git a/larg4/pluginActions/CMakeLists.txt b/larg4/pluginActions/CMakeLists.txt
index c76932e..3d978f6 100644
--- a/larg4/pluginActions/CMakeLists.txt
+++ b/larg4/pluginActions/CMakeLists.txt
@@ -1,6 +1,6 @@
-simple_plugin(
-  MCTruthEventAction service
-NOP
+set(CMAKE_CXX_FLAGS "${Geant4_CXX_FLAGS} -Wno-shadow -Wno-unused-parameter")
+cet_build_plugin(MCTruthEventAction art::service 
+NOP LIBRARIES
   art::Framework_Principal
   art::Framework_Services_Registry
   artg4tk::actionBase
@@ -9,21 +9,20 @@ NOP
   canvas::canvas
   cetlib_except::cetlib_except
   CLHEP::CLHEP
-  fhiclcpp
-  ${G4GLOBAL}
-  ${G4PARTICLES}
-  MF_MessageLogger
-  nusimdata_SimulationBase
-  nug4_G4Base
+  fhiclcpp::fhiclcpp
+  Geant4::G4global
+  Geant4::G4particles
+  messagefacility::MF_MessageLogger
+  nusimdata::SimulationBase
+  nug4::G4Base
   ROOT::Core
   ROOT::Physics
 SOURCE
   MCTruthEventAction_service.cc
 )
 
-simple_plugin(
-  ParticleListAction service
-NOP
+cet_build_plugin(ParticleListAction art::service
+NOP LIBRARIES
   art::Framework_Services_Registry
   canvas::canvas
   art::Framework_Core
@@ -32,11 +31,11 @@ NOP
   artg4tk::services_ActionHolder_service
   art::Persistency_Provenance
   CLHEP::CLHEP
-  ${G4PARTICLES}
-  MF_MessageLogger
+  Geant4::G4particles
+  messagefacility::MF_MessageLogger
   nusimdata_SimulationBase
-  nug4_G4Base
-  nug4_ParticleNavigation
+  nug4::G4Base
+  nug4::ParticleNavigation
   ROOT::Core
   ROOT::Physics
   Range::Range
