diff --git a/CMakeLists.txt b/CMakeLists.txt
index ddca84b..e3286f9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,17 +13,12 @@
 #  make package (builds distribution tarfile)
 # ======================================================================
 
-cmake_minimum_required(VERSION 3.11)  # use cmake 3.11 or later
+cmake_minimum_required(VERSION 3.12)  # use cmake 3.12 or later
 
-project(nutools)
-
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
-
-list(APPEND CMAKE_MODULE_PATH
-  $ENV{CANVAS_ROOT_IO_DIR}/Modules
-  $ENV{ART_DIR}/Modules)
+project(nutools VERSION 3.06.01 LANGUAGES CXX)
 
+set(fcl_dir job)
+find_package(cetmodules)
 include(CetCMakeEnv)
 cet_cmake_env()
 
@@ -36,18 +31,12 @@ cet_set_compiler_flags(DIAGS CAUTIOUS WERROR
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product( nusimdata )
-find_ups_product( cetlib )
-find_ups_product( cetlib_except )
-find_ups_product( cetbuildtools )
-
-message(STATUS "ROOTSYS is ${ROOTSYS}")
-
-## find_library directives
-# other dependencies
-cet_find_library( CRY NAMES CRY PATHS ENV CRY_LIB NO_DEFAULT_PATH )
+find_package( ROOT REQUIRED COMPONENTS MathCore Geom GeomPainter GX11 EG EGPythia6 Gui)
+find_package( art REQUIRED)
+find_package( art_root_io REQUIRED)
+find_package( cetlib REQUIRED)
+find_package( cetlib_except REQUIRED)
+find_package( nusimdata REQUIRED )
 
 # macros
 include(ArtMake)
@@ -55,21 +44,28 @@ include(BuildPlugins)
 include(CetRootCint)
 
 # include search path
-include_directories ( $ENV{CRYHOME}/src )
+include_directories ( ${CRYHOME}/src )
 
 # included for Mac OSX using XQuartz for X11
 if(APPLE)
   include_directories ( /opt/X11/include )
 endif()
 
+
+
+## find_library directives
+# other dependencies
+cet_find_library( CRY NAMES CRY PATHS ENV CRY_LIB NO_DEFAULT_PATH )
+
 # source
 add_subdirectory (nutools)
-add_subdirectory (tools)
-add_subdirectory (bundle)
-
+#add_subdirectory (tools)
+#add_subdirectory (test)
+#add_subdirectory (bundle)
+add_subdirectory(cmakeConfig)
 # ups - table and config files
-add_subdirectory(ups)
+#add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
+#include(UseCPack)
 
diff --git a/cmakeConfig/CMakeLists.txt b/cmakeConfig/CMakeLists.txt
new file mode 100644
index 0000000..1dff491
--- /dev/null
+++ b/cmakeConfig/CMakeLists.txt
@@ -0,0 +1,3 @@
+
+# create package configuration and version files
+cet_cmake_config(CONFIG_FRAGMENTS nutools-fragment.cmake.in)
diff --git a/cmakeConfig/nutools-fragment.cmake.in b/cmakeConfig/nutools-fragment.cmake.in
new file mode 100644
index 0000000..18ff201
--- /dev/null
+++ b/cmakeConfig/nutools-fragment.cmake.in
@@ -0,0 +1,6 @@
+find_package( ROOT REQUIRED COMPONENTS MathCore Geom GeomPainter GX11 EG EGPythia6 Gui)
+find_package( art REQUIRED)
+find_package( art_root_io REQUIRED)
+find_package( cetlib REQUIRED)
+find_package( cetlib_except REQUIRED)
+find_package( nusimdata REQUIRED )
diff --git a/nutools/EventGeneratorBase/CMakeLists.txt b/nutools/EventGeneratorBase/CMakeLists.txt
index 4ac5f89..36855a8 100644
--- a/nutools/EventGeneratorBase/CMakeLists.txt
+++ b/nutools/EventGeneratorBase/CMakeLists.txt
@@ -1,7 +1,6 @@
 
 install_headers()
 install_fhicl()
-install_source()
 
 add_subdirectory(CRY)
 #add_subdirectory(GENIE)
diff --git a/nutools/EventGeneratorBase/CRY/CMakeLists.txt b/nutools/EventGeneratorBase/CRY/CMakeLists.txt
index 9fb9755..83efed9 100644
--- a/nutools/EventGeneratorBase/CRY/CMakeLists.txt
+++ b/nutools/EventGeneratorBase/CRY/CMakeLists.txt
@@ -5,11 +5,11 @@ art_make( LIB_LIBRARIES nusimdata_SimulationBase
                         ${MF_UTILITIES}
                         ${FHICLCPP}
                         cetlib
-			cetlib_except
-                        ${ROOT_GEOM}
-                        ${ROOT_GEOMPAINTER}
-			${ROOT_EG}
-                        ${ROOT_BASIC_LIB_LIST} )
+                        cetlib_except
+                        ${ROOT_Geom_LIBRARY}
+                        ${ROOT_GeomPainter_LIBRARY}
+                        ${ROOT_EG_LIBRARY}
+                        ${ROOT_LIBRARIES} )
 
 
 install_headers()
diff --git a/tools/refactor/CMakeLists.txt b/tools/refactor/CMakeLists.txt
index ae9a7f6..e903a9b 100644
--- a/tools/refactor/CMakeLists.txt
+++ b/tools/refactor/CMakeLists.txt
@@ -1,4 +1,4 @@
 _nu_install_tools( DIRNAME refactor
-                   SCRIPT_LIST fix_cmake_nutools.pl 
-		               fix-header-locs-nutools.pl 
+                   SCRIPT_LIST fix_cmake_nutools.pl
+		               fix-header-locs-nutools.pl
 			       update_for_nutools.sh )
