diff --git a/CMakeLists.txt b/CMakeLists.txt
index ae7e8f8..3e053a6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,47 +1,36 @@
 # ======================================================================
 #  nusimdata main build file
 #
-#  cd .../path/to/build/directory
-#  source .../path/to/nusimdata/ups/setup_for_development
-#  cmake [-DCMAKE_INSTALL_PREFIX=/install/path]
-#        -DCMAKE_BUILD_TYPE=$CETPKG_TYPE
-#        $CETPKG_SOURCE
-#        
-#  make
-#  make test
-#  make install
-#  make package (builds distribution tarfile)
 # ======================================================================
-
-
 # use cmake 3.3 or later
 cmake_minimum_required(VERSION 3.18.2 FATAL_ERROR)
 
 project(nusimdata VERSION 1.24.06 LANGUAGES CXX)
 
-find_package(cetbuildtools)
-
-list(APPEND CMAKE_MODULE_PATH
-  $ENV{CANVAS_ROOT_IO_DIR}/Modules)
+find_package(cetmodules 2.29.06 REQUIRED)
+project(nusimdata VERSION 1.25.00 LANGUAGES CXX)
 
 include(CetCMakeEnv)
 cet_cmake_env()
 
+# Find our own CMake modules, and tell everyone else where to find them.
+cet_cmake_module_directories(Modules)
+
 # -D_GLIBCXX_USE_NANOSLEEP is needed to make std::this_thread::sleep_for available.
 cet_set_compiler_flags(DIAGS CAUTIOUS WERROR 
                        NO_UNDEFINED
                        EXTRA_FLAGS -pedantic 
-                       EXTRA_CXX_FLAGS -D_GLIBCXX_USE_NANOSLEEP 
-                                       -Wno-unused-local-typedefs -Wno-variadic-macros)
-cet_report_compiler_flags()
+                                   $<$<COMPILE_LANGUAGE:CXX>:-D_GLIBCXX_USE_NANOSLEEP 
+                                   -Wno-unused-local-typedefs -Wno-variadic-macros>)
+cet_report_compiler_flags(REPORT_THRESHOLD VERBOSE)
 
-# these are minimum required versions, not the actual product versions
-find_ups_product( canvas )
-find_ups_product( canvas_root_io ) # Needed for art_dictionary().
-find_ups_product( dk2nudata )
-find_ups_root(  )
+find_package(cetlib_except REQUIRED)
+find_package(cetlib REQUIRED)
+find_package(canvas PUBLIC REQUIRED)
+find_package(canvas_root_io REQUIRED) # Needed for art_dictionary().
+find_package(dk2nudata PUBLIC REQUIRED)
 
-# macros 
+# macros
 include(CetMake)
 include(ArtDictionary)
 include(BasicPlugin)
@@ -49,8 +38,8 @@ include(BasicPlugin)
 # source
 add_subdirectory (nusimdata)
 add_subdirectory (test)
+add_subdirectory (Modules)
 
 # ups - table and config files
-add_subdirectory(ups)
 # packaging utility
-include(UseCPack)
+cet_cmake_config()
diff --git a/nusimdata/SimulationBase/CMakeLists.txt b/nusimdata/SimulationBase/CMakeLists.txt
index ca37b9f..ce91e02 100644
--- a/nusimdata/SimulationBase/CMakeLists.txt
+++ b/nusimdata/SimulationBase/CMakeLists.txt
@@ -1,24 +1,16 @@
 
-cet_find_library( DK2NU_TREE NAMES dk2nuTree PATHS ENV DK2NUDATA_LIB NO_DEFAULT_PATH )
-include_directories( $ENV{DK2NUDATA_INC} )
+find_package(ROOT PUBLIC COMPONENTS Core Physics RIO Tree REQUIRED)
+find_package(ROOT PRIVATE COMPONENTS EG REQUIRED)
 
-# ROOT_BASIC_LIB_LIST is:
-#   Core RIO Net Imt Hist Graf Graf3d Gpad Tree Rint Postscript
-#   Matrix Physics MathCore Thread
-
-set( SIMB_LIBS
+cet_make_library(SOURCE GTruth.cxx  MCFlux.cxx  MCNeutrino.cxx
+                        MCParticle.cxx  MCTrajectory.cxx  MCTruth.cxx
+               LIBRARIES PRIVATE
                canvas::canvas
-               fhiclcpp
-               DK2NU_TREE  # additional data objects assoc w/ MCTruth
-               cetlib::cetlib
-               cetlib_except::cetlib_except
-               ${ROOT_BASIC_LIB_LIST}
-               ROOT::EG )  # EG needed for TParticleDB
-
-cet_make( LIBRARIES ${SIMB_LIBS}
-          NO_DICTIONARY )
+               ROOT::EG # EG needed for TParticleDB
+               PUBLIC
+               ROOT::Physics)
 
-art_dictionary( DICTIONARY_LIBRARIES nusimdata::SimulationBase )
+art_dictionary(DICTIONARY_LIBRARIES nusimdata::SimulationBase dk2nu::Tree)
 
 install_headers()
 install_fhicl()
diff --git a/test/SimulationBase/CMakeLists.txt b/test/SimulationBase/CMakeLists.txt
index 2638ecd..2686a9e 100644
--- a/test/SimulationBase/CMakeLists.txt
+++ b/test/SimulationBase/CMakeLists.txt
@@ -1,5 +1,6 @@
+find_package(ROOT PRIVATE COMPONENTS Physics REQUIRED)
 cet_test(MCParticle_test USE_BOOST_UNIT
-  SOURCES MCParticle_test.cxx
-  LIBRARIES nusimdata::SimulationBase
+  SOURCE MCParticle_test.cxx
+  LIBRARIES PRIVATE nusimdata::SimulationBase cetlib::headers ROOT::Physics
   )
 
