diff --git a/CMakeLists.txt b/CMakeLists.txt
index d9cae99..c617d91 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,18 +16,12 @@
 # use cmake 3.10 or later
 cmake_minimum_required (VERSION 3.10)
 
-project(larreco)
-
-# cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
-IF (NOT CETBUILDTOOLS_VERSION)
-    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
-ENDIF()
-
-set(CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules
-		      $ENV{CETBUILDTOOLS_DIR}/Modules
-		      ${CMAKE_MODULE_PATH})
+project(larreco VERSION 08.00.00 LANGUAGES CXX)
 
+set(fw_dir config_data)
+set(fcl_dir job)
+set(gdml_dir gdml)
+find_package( cetmodules )
 include(CetCMakeEnv)
 cet_cmake_env()
 
@@ -40,23 +34,8 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( larcoreobj )
-find_ups_product( larcorealg )
-find_ups_product( larcore )
-find_ups_product( lardataobj )
-find_ups_product( lardataalg )
-find_ups_product( lardata )
-find_ups_product( larevt )
-find_ups_product( larsim )
-find_ups_product( nutools )
-find_ups_product( nug4 )
-find_ups_product( nurandom )
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product( cetbuildtools )
-find_ups_product( postgresql )
-find_ups_product( range )
-find_ups_product( eigen )
+find_package( ROOT REQUIRED COMPONENTS GenVector Geom Minuit Minuit2 XMLIO Gdml TMVA EG VMC)
+find_package( larsim REQUIRED  )
 
 # macros for dictionary and simple_plugin
 include(ArtDictionary)
@@ -64,13 +43,13 @@ include(ArtMake)
 include(BuildPlugins)
 
 # not ready for c2 yet
-if( DEFINED ENV{TENSORFLOW_DIR} )
+if( DEFINED TENSORFLOW_DIR )
   # find tensorflow library
-  find_ups_product( protobuf )
-  find_ups_product( tensorflow )
-  cet_find_library(TENSORFLOW NAMES tensorflow-core PATHS ENV TENSORFLOW_LIB NO_DEFAULT_PATH )
-  cet_find_library(PROTOBUF NAMES protobuf PATHS ENV PROTOBUF_LIB NO_DEFAULT_PATH )
-  # This is a hack for a broken tensorflow link, which 
+  find_package( Protobuf )
+  find_package( Tensorflow )
+  cet_find_library(TENSORFLOW NAMES tensorflow-core )
+  cet_find_library(PROTOBUF NAMES protobuf )
+  # This is a hack for a broken tensorflow link, which
   # requires -lrt for glibc earlier than 2.17 (i.e. SLF6)
   # We take advantage of OSTYPE defined by cetbuildtools
   if ( "${OSTYPE}" MATCHES "slf6" )
@@ -83,9 +62,9 @@ add_subdirectory(larreco)
 
 # tests
 add_subdirectory(test)
-
+cet_cmake_config()
 # ups - table and config files
-add_subdirectory(ups)
+#add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
+#include(UseCPack)
