commit d8ef532029f2d65da1c970f768df3908c383f368
Author: Patrick Gartung <gartung@fnal.gov>
Date:   Sat Apr 4 18:43:46 2020 -0500

    Cetmodules

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d9cae99..ecc6cab 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,50 +13,29 @@
 # ======================================================================
 
 
-# use cmake 3.10 or later
-cmake_minimum_required (VERSION 3.10)
+# use cmake 3.12 or later
+cmake_minimum_required (VERSION 3.12)
 
-project(larreco)
-
-# cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
-IF (NOT CETBUILDTOOLS_VERSION)
-    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
-ENDIF()
-
-set(CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules
-		      $ENV{CETBUILDTOOLS_DIR}/Modules
-		      ${CMAKE_MODULE_PATH})
+project(larreco VERSION 08.28.00 LANGUAGES CXX)
 
+set(fw_dir config_data)
+set(fcl_dir job)
+set(gdml_dir gdml)
+find_package( cetmodules )
 include(CetCMakeEnv)
 cet_cmake_env()
 
 cet_set_compiler_flags(DIAGS CAUTIOUS
   WERROR
   NO_UNDEFINED
-  EXTRA_FLAGS -pedantic -Wno-unused-local-typedefs
+  EXTRA_FLAGS -pedantic -Wno-unused-local-typedefs -Wno-maybe-uninitialized -Wno-shadow -Wno-unused-parameter
 )
 
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( larcoreobj )
-find_ups_product( larcorealg )
-find_ups_product( larcore )
-find_ups_product( lardataobj )
-find_ups_product( lardataalg )
-find_ups_product( lardata )
-find_ups_product( larevt )
-find_ups_product( larsim )
-find_ups_product( nutools )
-find_ups_product( nug4 )
-find_ups_product( nurandom )
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product( cetbuildtools )
-find_ups_product( postgresql )
-find_ups_product( range )
-find_ups_product( eigen )
+find_package( ROOT REQUIRED COMPONENTS GenVector Geom Minuit Minuit2 XMLIO Gdml TMVA EG )
+find_package( larsim REQUIRED  )
 
 # macros for dictionary and simple_plugin
 include(ArtDictionary)
@@ -64,13 +43,13 @@ include(ArtMake)
 include(BuildPlugins)
 
 # not ready for c2 yet
-if( DEFINED ENV{TENSORFLOW_DIR} )
+if( DEFINED TENSORFLOW_DIR )
   # find tensorflow library
-  find_ups_product( protobuf )
-  find_ups_product( tensorflow )
-  cet_find_library(TENSORFLOW NAMES tensorflow-core PATHS ENV TENSORFLOW_LIB NO_DEFAULT_PATH )
-  cet_find_library(PROTOBUF NAMES protobuf PATHS ENV PROTOBUF_LIB NO_DEFAULT_PATH )
-  # This is a hack for a broken tensorflow link, which 
+  find_package( Protobuf )
+  find_package( Tensorflow )
+  cet_find_library(TENSORFLOW NAMES tensorflow-core )
+  cet_find_library(PROTOBUF NAMES protobuf )
+  # This is a hack for a broken tensorflow link, which
   # requires -lrt for glibc earlier than 2.17 (i.e. SLF6)
   # We take advantage of OSTYPE defined by cetbuildtools
   if ( "${OSTYPE}" MATCHES "slf6" )
@@ -83,9 +62,9 @@ add_subdirectory(larreco)
 
 # tests
 add_subdirectory(test)
-
+add_subdirectory(cmakeConfig)
 # ups - table and config files
-add_subdirectory(ups)
+#add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
+#include(UseCPack)
diff --git a/cmakeConfig/CMakeLists.txt b/cmakeConfig/CMakeLists.txt
new file mode 100644
index 0000000..0070c86
--- /dev/null
+++ b/cmakeConfig/CMakeLists.txt
@@ -0,0 +1,4 @@
+
+# create package configuration and version files
+
+cet_cmake_config(CONFIG_FRAGMENTS larreco-fragment.cmake.in)
diff --git a/cmakeConfig/larreco-fragment.cmake.in b/cmakeConfig/larreco-fragment.cmake.in
new file mode 100644
index 0000000..8fdb424
--- /dev/null
+++ b/cmakeConfig/larreco-fragment.cmake.in
@@ -0,0 +1,2 @@
+find_package( ROOT REQUIRED COMPONENTS GenVector Geom Minuit Minuit2 XMLIO Gdml TMVA EG VMC)
+find_package( larsim REQUIRED  )
diff --git a/larreco/Calibrator/CMakeLists.txt b/larreco/Calibrator/CMakeLists.txt
index f226347..0cf708e 100644
--- a/larreco/Calibrator/CMakeLists.txt
+++ b/larreco/Calibrator/CMakeLists.txt
@@ -2,4 +2,3 @@ simple_plugin(PhotonCalibratorServiceStandard "service")
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/Calorimetry/CMakeLists.txt b/larreco/Calorimetry/CMakeLists.txt
index c9764a9..92b1424 100644
--- a/larreco/Calorimetry/CMakeLists.txt
+++ b/larreco/Calorimetry/CMakeLists.txt
@@ -8,17 +8,16 @@ art_make(LIB_LIBRARIES
            ${MF_MESSAGELOGGER}
            ${FHICLCPP}
            cetlib_except
-           ${ROOT_CORE}
-           ${ROOT_PHYSICS}
+           ROOT::Core
+           ROOT::Physics
          MODULE_LIBRARIES
            larcorealg_Geometry
            larreco_Calorimetry
-           ${ROOT_CORE}
-           ${ROOT_HIST}
-           ${ROOT_PHYSICS}
+           ROOT::Core
+           ROOT::Hist
+           ROOT::Physics
            ${MF_MESSAGELOGGER}
          )
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/ClusterFinder/CMakeLists.txt b/larreco/ClusterFinder/CMakeLists.txt
index 108e513..59970b2 100644
--- a/larreco/ClusterFinder/CMakeLists.txt
+++ b/larreco/ClusterFinder/CMakeLists.txt
@@ -13,15 +13,14 @@ art_make(LIB_LIBRARIES
            nusimdata_SimulationBase
            ${ART_FRAMEWORK_SERVICES_REGISTRY}
            ${ART_ROOT_IO_TFILE_SUPPORT}
-           ${ROOT_CORE}
-           ${ROOT_PHYSICS}
+           ROOT::Core
+           ROOT::Physics
            ${ART_ROOT_IO_TFILESERVICE_SERVICE}
            ${MF_MESSAGELOGGER}
          )
 
 install_headers()
 install_fhicl()
-install_source()
 
 add_subdirectory(RStarTree)
 add_subdirectory(TrackShowerSplitter)
diff --git a/larreco/ClusterFinder/RStarTree/CMakeLists.txt b/larreco/ClusterFinder/RStarTree/CMakeLists.txt
index eec96b5..2ee79f7 100644
--- a/larreco/ClusterFinder/RStarTree/CMakeLists.txt
+++ b/larreco/ClusterFinder/RStarTree/CMakeLists.txt
@@ -1,2 +1 @@
 install_headers()
-install_source()
diff --git a/larreco/ClusterFinder/TrackShowerSplitter/CMakeLists.txt b/larreco/ClusterFinder/TrackShowerSplitter/CMakeLists.txt
index b498f05..c72ebb1 100644
--- a/larreco/ClusterFinder/TrackShowerSplitter/CMakeLists.txt
+++ b/larreco/ClusterFinder/TrackShowerSplitter/CMakeLists.txt
@@ -1,30 +1,26 @@
 add_subdirectory(Segmentation2D)
 
 simple_plugin(TrackShowerHits   "module"
-			lardataobj_RawData
-		 	lardataobj_RecoBase
-			larreco_RecoAlg
-			larreco_ClusterFinder_TrackShowerSplitter_Segmentation2D
-		 	larcorealg_Geometry
-			larcore_Geometry_Geometry_service
-			lardata_Utilities
-               ${ART_FRAMEWORK_CORE}
-			${ART_FRAMEWORK_PRINCIPAL}
-			${ART_FRAMEWORK_SERVICES_REGISTRY}
-         		${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
-    			${ART_ROOT_IO_TFILESERVICE_SERVICE}
- 			art_Persistency_Common
-			art_Persistency_Provenance
-			art_Utilities
-			canvas
-			${MF_MESSAGELOGGER}
-			
-			cetlib cetlib_except
-			${ROOT_BASIC_LIB_LIST}
-              
+                              lardataobj_RawData
+                              lardataobj_RecoBase
+                              larreco_RecoAlg
+                              larreco_ClusterFinder_TrackShowerSplitter_Segmentation2D
+                              larcorealg_Geometry
+                              larcore_Geometry_Geometry_service
+                              lardata_Utilities
+                              ${ART_FRAMEWORK_CORE}
+                              ${ART_FRAMEWORK_PRINCIPAL}
+                              ${ART_FRAMEWORK_SERVICES_REGISTRY}
+                              ${ART_ROOT_IO_TFILE_SUPPORT}
+                              ${ART_ROOT_IO_TFILESERVICE_SERVICE}
+                              art_Persistency_Common
+                              art_Persistency_Provenance
+                              art_Utilities
+                              canvas
+                              ${MF_MESSAGELOGGER}
+                              cetlib cetlib_except
+                              ${ROOT_LIBRARIES}
 )
 
 install_headers()
 install_fhicl()
-install_source()
-install_scripts()
diff --git a/larreco/ClusterFinder/TrackShowerSplitter/Segmentation2D/CMakeLists.txt b/larreco/ClusterFinder/TrackShowerSplitter/Segmentation2D/CMakeLists.txt
index ef1a5a9..5fdee7c 100644
--- a/larreco/ClusterFinder/TrackShowerSplitter/Segmentation2D/CMakeLists.txt
+++ b/larreco/ClusterFinder/TrackShowerSplitter/Segmentation2D/CMakeLists.txt
@@ -1,7 +1,7 @@
 art_make(LIB_LIBRARIES
            larreco_RecoAlg_PMAlg
-           ${ROOT_CORE}
-           ${ROOT_PHYSICS}
+           ROOT::Core
+           ROOT::Physics
            canvas
            ${MF_MESSAGELOGGER}
            ${FHICLCPP}
@@ -10,4 +10,3 @@ art_make(LIB_LIBRARIES
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/DirOfGamma/CMakeLists.txt b/larreco/DirOfGamma/CMakeLists.txt
index 95ed09c..c25d842 100644
--- a/larreco/DirOfGamma/CMakeLists.txt
+++ b/larreco/DirOfGamma/CMakeLists.txt
@@ -2,8 +2,8 @@ art_make(LIB_LIBRARIES
            larcorealg_Geometry
            larreco_RecoAlg_PMAlg
            ${ART_FRAMEWORK_SERVICES_REGISTRY}
-           ${ROOT_CORE}
-           ${ROOT_PHYSICS}
+           ROOT::Core
+           ROOT::Physics
            canvas
            ${MF_MESSAGELOGGER}
            cetlib_except
@@ -13,11 +13,10 @@ art_make(LIB_LIBRARIES
            larreco_Calorimetry
            larreco_DirOfGamma
            ${ART_FRAMEWORK_SERVICES_REGISTRY}
-           ${ROOT_CORE}
-           ${ROOT_PHYSICS}
+           ROOT::Core
+           ROOT::Physics
            ${MF_MESSAGELOGGER}
          )
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/EventFinder/CMakeLists.txt b/larreco/EventFinder/CMakeLists.txt
index 25b6237..ec61f4f 100644
--- a/larreco/EventFinder/CMakeLists.txt
+++ b/larreco/EventFinder/CMakeLists.txt
@@ -1,9 +1,8 @@
 art_make(MODULE_LIBRARIES
            lardataobj_RecoBase
-           ${ROOT_CORE}
+           ROOT::Core
            ${MF_MESSAGELOGGER}
          )
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/Genfit/CMakeLists.txt b/larreco/Genfit/CMakeLists.txt
index ee893ea..ffee56f 100644
--- a/larreco/Genfit/CMakeLists.txt
+++ b/larreco/Genfit/CMakeLists.txt
@@ -5,16 +5,15 @@ art_make(LIB_LIBRARIES
            canvas
            ${MF_MESSAGELOGGER}
            cetlib_except
-           ${ROOT_GEOM}
-           ${ROOT_GRAF3D}
-           ${ROOT_CORE}
-           ${ROOT_MATHCORE}
-           ${ROOT_MATRIX}
-           ${ROOT_PHYSICS}
-           ${ROOT_EG}
-           ${ROOT_VMC}
+           ROOT::Geom
+           ROOT::Graf3d
+           ROOT::Core
+           ROOT::MathCore
+           ROOT::Matrix
+           ROOT::Physics
+           ROOT::EG
+           #ROOT::VMC
          )
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/HitFinder/CMakeLists.txt b/larreco/HitFinder/CMakeLists.txt
index 76f3393..22a21ec 100644
--- a/larreco/HitFinder/CMakeLists.txt
+++ b/larreco/HitFinder/CMakeLists.txt
@@ -7,7 +7,7 @@ art_make(LIB_LIBRARIES
            ${MF_MESSAGELOGGER}
            ${FHICLCPP}
            cetlib_except
-           ${ROOT_CORE}
+           ROOT::Core
          MODULE_LIBRARIES
            larreco_HitFinder
            larsim_MCCheater_BackTrackerService_service
@@ -17,15 +17,14 @@ art_make(LIB_LIBRARIES
            lardata_ArtDataHelper
            ${ART_FRAMEWORK_SERVICES_REGISTRY}
            ${ART_ROOT_IO_TFILE_SUPPORT}
-           ${ROOT_CORE}
            ${ART_ROOT_IO_TFILESERVICE_SERVICE}
            art_Framework_Services_System_TriggerNamesService_service
            ${MF_MESSAGELOGGER}
-           ${ROOT_HIST}
-           ${ROOT_MATHCORE}
-           ${ROOT_PHYSICS}
+           ROOT::Core
+           ROOT::Hist
+           ROOT::MathCore
+           ROOT::Physics
          )
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/HitFinder/HitFinderTools/CMakeLists.txt b/larreco/HitFinder/HitFinderTools/CMakeLists.txt
index 731fcb0..4aa4ec4 100644
--- a/larreco/HitFinder/HitFinderTools/CMakeLists.txt
+++ b/larreco/HitFinder/HitFinderTools/CMakeLists.txt
@@ -13,20 +13,19 @@ set( hitfinder_tool_lib_list
 			art_Persistency_Common
 			art_Persistency_Provenance
 			${ART_FRAMEWORK_SERVICES_BASIC}
-			${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+			${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE_LIBRARY}
             ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
 			${ART_ROOT_IO_TFILESERVICE_SERVICE}
 			${ART_FRAMEWORK_SERVICES_SYSTEM_FILECATALOGMETADATA_SERVICE}
 			art_Utilities
 			canvas
 			${MF_MESSAGELOGGER}
-			
 			${FHICLCPP}
 			${CETLIB}
 			${Boost_FILESYSTEM_LIBRARY}
 			${Boost_SYSTEM_LIBRARY}
-            ${CLHEP}
-			${ROOT_BASIC_LIB_LIST}
+			${CLHEP_LIBRARIES}
+			${ROOT_LIBRARIES}
     )
 
 include(FindOpenMP)
@@ -46,5 +45,4 @@ endif(OPENMP_FOUND)
 art_make(TOOL_LIBRARIES ${hitfinder_tool_lib_list} )
 
 install_headers()
-install_source()
 install_fhicl()
diff --git a/larreco/MCComp/CMakeLists.txt b/larreco/MCComp/CMakeLists.txt
index f524e75..571361c 100644
--- a/larreco/MCComp/CMakeLists.txt
+++ b/larreco/MCComp/CMakeLists.txt
@@ -5,14 +5,13 @@ art_make(LIB_LIBRARIES
            ${ART_FRAMEWORK_SERVICES_REGISTRY}
            canvas
            cetlib_except
-           ${ROOT_CORE}
+           ROOT::Core
          MODULE_LIBRARIES
            larreco_MCComp
            larcorealg_Geometry
            ${ART_FRAMEWORK_SERVICES_REGISTRY}
-           ${ROOT_CORE}
+           ROOT::Core
          )
 
 # install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/QuadVtx/CMakeLists.txt b/larreco/QuadVtx/CMakeLists.txt
index 1b98464..0568fdb 100644
--- a/larreco/QuadVtx/CMakeLists.txt
+++ b/larreco/QuadVtx/CMakeLists.txt
@@ -1,8 +1,8 @@
 include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 
 art_make(LIB_LIBRARIES
-           ${ROOT_CORE}
-           ${ROOT_HIST}
+           ROOT::Core
+           ROOT::Hist
          MODULE_LIBRARIES
            larcorealg_Geometry
            lardataobj_RecoBase
@@ -13,13 +13,12 @@ art_make(LIB_LIBRARIES
            ${ART_ROOT_IO_TFILE_SUPPORT}
            ${ART_FRAMEWORK_SERVICES_REGISTRY}
            ${MF_MESSAGELOGGER}
-           ${ROOT_CORE}
-           ${ROOT_HIST}
-           ${ROOT_MATRIX}
-           ${ROOT_PHYSICS}
-           ${ROOT_TREE}
+           ROOT::Core
+           ROOT::Hist
+           ROOT::Matrix
+           ROOT::Physics
+           ROOT::Tree
         )
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/RecoAlg/CMTool/CMTAlgMatch/CMakeLists.txt b/larreco/RecoAlg/CMTool/CMTAlgMatch/CMakeLists.txt
index 2cd7b05..b8118bf 100644
--- a/larreco/RecoAlg/CMTool/CMTAlgMatch/CMakeLists.txt
+++ b/larreco/RecoAlg/CMTool/CMTAlgMatch/CMakeLists.txt
@@ -2,13 +2,12 @@ art_make(LIB_LIBRARIES
            larreco_RecoAlg_CMTool_CMToolBase
            larcorealg_Geometry
            ${ART_FRAMEWORK_SERVICES_REGISTRY}
-           ${ROOT_CORE}
-           ${ROOT_RIO}
-           ${ROOT_TREE}
+           ROOT::Core
+           ROOT::RIO
+           ROOT::Tree
            canvas
            cetlib_except
         )
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/RecoAlg/CMTool/CMTAlgMerge/CMakeLists.txt b/larreco/RecoAlg/CMTool/CMTAlgMerge/CMakeLists.txt
index 586fe7a..67a8c88 100644
--- a/larreco/RecoAlg/CMTool/CMTAlgMerge/CMakeLists.txt
+++ b/larreco/RecoAlg/CMTool/CMTAlgMerge/CMakeLists.txt
@@ -2,4 +2,3 @@ art_make(LIB_LIBRARIES larreco_RecoAlg_CMTool_CMToolBase)
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/RecoAlg/CMTool/CMTAlgPriority/CMakeLists.txt b/larreco/RecoAlg/CMTool/CMTAlgPriority/CMakeLists.txt
index 586fe7a..67a8c88 100644
--- a/larreco/RecoAlg/CMTool/CMTAlgPriority/CMakeLists.txt
+++ b/larreco/RecoAlg/CMTool/CMTAlgPriority/CMakeLists.txt
@@ -2,4 +2,3 @@ art_make(LIB_LIBRARIES larreco_RecoAlg_CMTool_CMToolBase)
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/RecoAlg/CMTool/CMToolApp/CMakeLists.txt b/larreco/RecoAlg/CMTool/CMToolApp/CMakeLists.txt
index 586fe7a..67a8c88 100644
--- a/larreco/RecoAlg/CMTool/CMToolApp/CMakeLists.txt
+++ b/larreco/RecoAlg/CMTool/CMToolApp/CMakeLists.txt
@@ -2,4 +2,3 @@ art_make(LIB_LIBRARIES larreco_RecoAlg_CMTool_CMToolBase)
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/RecoAlg/CMTool/CMToolBase/CMakeLists.txt b/larreco/RecoAlg/CMTool/CMToolBase/CMakeLists.txt
index 61d3f15..b78f78f 100644
--- a/larreco/RecoAlg/CMTool/CMToolBase/CMakeLists.txt
+++ b/larreco/RecoAlg/CMTool/CMToolBase/CMakeLists.txt
@@ -2,4 +2,3 @@ art_make(LIB_LIBRARIES larreco_RecoAlg_ClusterRecoUtil)
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/RecoAlg/CMakeLists.txt b/larreco/RecoAlg/CMakeLists.txt
index 743e773..aeea019 100644
--- a/larreco/RecoAlg/CMakeLists.txt
+++ b/larreco/RecoAlg/CMakeLists.txt
@@ -5,10 +5,10 @@ art_make(LIB_LIBRARIES
            larreco_RecoAlg_ClusterRecoUtil
            larreco_RecoAlg_CMTool_CMToolBase
            larreco_RecoAlg_ImagePatternAlgs_DataProvider
-           ${ROOT_CORE}
-           ${ROOT_PHYSICS}
-           ${ROOT_MATRIX}
-           ${ROOT_MINUIT}
+           ROOT::Core
+           ROOT::Physics
+           ROOT::Matrix
+           ROOT::Minuit
            canvas
            ${FHICLCPP}
            cetlib_except
@@ -24,4 +24,3 @@ add_subdirectory(xml)
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/RecoAlg/Cluster3DAlgs/CMakeLists.txt b/larreco/RecoAlg/Cluster3DAlgs/CMakeLists.txt
index f3670d0..506134b 100644
--- a/larreco/RecoAlg/Cluster3DAlgs/CMakeLists.txt
+++ b/larreco/RecoAlg/Cluster3DAlgs/CMakeLists.txt
@@ -1,5 +1,5 @@
 cet_enable_asserts()
-
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-int-in-bool-context")
 add_subdirectory(ConvexHull)
 add_subdirectory(Voronoi)
 add_subdirectory(PathFinding)
@@ -13,12 +13,12 @@ art_make(LIB_LIBRARIES
            ${ART_FRAMEWORK_PRINCIPAL}
            ${ART_FRAMEWORK_SERVICES_REGISTRY}
            ${ART_ROOT_IO_TFILE_SUPPORT}
-           ${ROOT_CORE}
-           ${ROOT_GPAD}
-           ${ROOT_HIST}
-           ${ROOT_MATRIX}
-           ${ROOT_PHYSICS}
-           ${ROOT_TREE}
+           ROOT::Core
+           ROOT::Gpad
+           ROOT::Hist
+           ROOT::Matrix
+           ROOT::Physics
+           ROOT::Tree
            ${ART_ROOT_IO_TFILESERVICE_SERVICE}
            art_Persistency_Provenance
            canvas
diff --git a/larreco/RecoAlg/Cluster3DAlgs/ConvexHull/CMakeLists.txt b/larreco/RecoAlg/Cluster3DAlgs/ConvexHull/CMakeLists.txt
index 8692bd1..b1e2245 100644
--- a/larreco/RecoAlg/Cluster3DAlgs/ConvexHull/CMakeLists.txt
+++ b/larreco/RecoAlg/Cluster3DAlgs/ConvexHull/CMakeLists.txt
@@ -4,4 +4,3 @@ art_make()
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/RecoAlg/Cluster3DAlgs/PathFinding/CMakeLists.txt b/larreco/RecoAlg/Cluster3DAlgs/PathFinding/CMakeLists.txt
index 23bcfe2..e0966ee 100644
--- a/larreco/RecoAlg/Cluster3DAlgs/PathFinding/CMakeLists.txt
+++ b/larreco/RecoAlg/Cluster3DAlgs/PathFinding/CMakeLists.txt
@@ -4,7 +4,7 @@ art_make(TOOL_LIBRARIES
            larreco_RecoAlg_Cluster3DAlgs
            larreco_RecoAlg_Cluster3DAlgs_Voronoi
            ${ART_ROOT_IO_TFILE_SUPPORT}
-           ${ROOT_CORE}
+           ${ROOT_Core_LIBRARY}
            ${ART_ROOT_IO_TFILESERVICE_SERVICE}
            canvas
            ${MF_MESSAGELOGGER}
@@ -12,4 +12,3 @@ art_make(TOOL_LIBRARIES
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/RecoAlg/Cluster3DAlgs/Voronoi/CMakeLists.txt b/larreco/RecoAlg/Cluster3DAlgs/Voronoi/CMakeLists.txt
index c0241c8..36bdc90 100644
--- a/larreco/RecoAlg/Cluster3DAlgs/Voronoi/CMakeLists.txt
+++ b/larreco/RecoAlg/Cluster3DAlgs/Voronoi/CMakeLists.txt
@@ -1,7 +1,5 @@
 cet_enable_asserts()
 
 art_make(LIB_LIBRARIES larreco_RecoAlg_Cluster3DAlgs_ConvexHull)
-
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/RecoAlg/ClusterRecoUtil/CMakeLists.txt b/larreco/RecoAlg/ClusterRecoUtil/CMakeLists.txt
index d4a33ac..d6f5d9a 100644
--- a/larreco/RecoAlg/ClusterRecoUtil/CMakeLists.txt
+++ b/larreco/RecoAlg/ClusterRecoUtil/CMakeLists.txt
@@ -1,8 +1,7 @@
 art_make(LIB_LIBRARIES
            lardata_Utilities
-           ${ROOT_BASIC_LIB_LIST}
+           ${ROOT_LIBRARIES}
         )
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/RecoAlg/ImagePatternAlgs/CMakeLists.txt b/larreco/RecoAlg/ImagePatternAlgs/CMakeLists.txt
index d25d137..0baf5a2 100644
--- a/larreco/RecoAlg/ImagePatternAlgs/CMakeLists.txt
+++ b/larreco/RecoAlg/ImagePatternAlgs/CMakeLists.txt
@@ -1,5 +1,5 @@
 # not ready for c2 yet
-if( DEFINED ENV{TENSORFLOW_DIR} )
+if( TENSORFLOW_DIR )
   add_subdirectory(Tensorflow)
 endif ()
 add_subdirectory(DataProvider)
diff --git a/larreco/RecoAlg/ImagePatternAlgs/DataProvider/CMakeLists.txt b/larreco/RecoAlg/ImagePatternAlgs/DataProvider/CMakeLists.txt
index 662a083..6b5c27c 100644
--- a/larreco/RecoAlg/ImagePatternAlgs/DataProvider/CMakeLists.txt
+++ b/larreco/RecoAlg/ImagePatternAlgs/DataProvider/CMakeLists.txt
@@ -2,13 +2,12 @@ art_make(LIB_LIBRARIES
            larcorealg_Geometry
            larreco_Calorimetry
            ${ART_FRAMEWORK_SERVICES_REGISTRY}
-           ${ROOT_CORE}
+           ROOT::Core
            canvas
            ${MF_MESSAGELOGGER}
            cetlib_except
-           ${CLHEP}
+           ${CLHEP_LIBRARIES}
         )
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/RecoAlg/ImagePatternAlgs/Tensorflow/CMakeLists.txt b/larreco/RecoAlg/ImagePatternAlgs/Tensorflow/CMakeLists.txt
index 0b3e11c..ea832f5 100644
--- a/larreco/RecoAlg/ImagePatternAlgs/Tensorflow/CMakeLists.txt
+++ b/larreco/RecoAlg/ImagePatternAlgs/Tensorflow/CMakeLists.txt
@@ -1,4 +1,4 @@
-include_directories ( $ENV{TENSORFLOW_INC}/eigen )
+include_directories ( ${TENSORFLOW_INC}/eigen )
 
 add_subdirectory(PointIdAlg)
 add_subdirectory(TF)
diff --git a/larreco/RecoAlg/ImagePatternAlgs/Tensorflow/Modules/CMakeLists.txt b/larreco/RecoAlg/ImagePatternAlgs/Tensorflow/Modules/CMakeLists.txt
index e5e7ea9..4b661ab 100644
--- a/larreco/RecoAlg/ImagePatternAlgs/Tensorflow/Modules/CMakeLists.txt
+++ b/larreco/RecoAlg/ImagePatternAlgs/Tensorflow/Modules/CMakeLists.txt
@@ -18,7 +18,7 @@ art_make( MODULE_LIBRARIES
 		canvas
 		${MF_MESSAGELOGGER}	
 		cetlib cetlib_except
-		${ROOT_BASIC_LIB_LIST}
+		${ROOT_LIBRARIES}
 )
 
 install_fhicl()
diff --git a/larreco/RecoAlg/ImagePatternAlgs/Tensorflow/PointIdAlg/CMakeLists.txt b/larreco/RecoAlg/ImagePatternAlgs/Tensorflow/PointIdAlg/CMakeLists.txt
index 37c0b4c..0e1d7b4 100644
--- a/larreco/RecoAlg/ImagePatternAlgs/Tensorflow/PointIdAlg/CMakeLists.txt
+++ b/larreco/RecoAlg/ImagePatternAlgs/Tensorflow/PointIdAlg/CMakeLists.txt
@@ -1,34 +1,32 @@
 art_make(
-          LIB_LIBRARIES larreco_RecoAlg_ImagePatternAlgs_DataProvider 
-			larreco_RecoAlg_ImagePatternAlgs_Keras
+          LIB_LIBRARIES larreco_RecoAlg_ImagePatternAlgs_DataProvider
+                        larreco_RecoAlg_ImagePatternAlgs_Keras
                         larreco_RecoAlg_ImagePatternAlgs_Tensorflow_TF
                         larcore_Geometry_Geometry_service
                         larcorealg_Geometry
-			lardataobj_RecoBase
-			larreco_Calorimetry
-			nusimdata_SimulationBase
-			${ART_FRAMEWORK_CORE}
-			${ART_FRAMEWORK_PRINCIPAL}
-			${ART_FRAMEWORK_SERVICES_REGISTRY}
-			${ART_FRAMEWORK_SERVICES_BASIC}
-			${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
-			${ART_ROOT_IO_TFILESERVICE_SERVICE}
-			${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
-			art_Persistency_Common
-			art_Persistency_Provenance
-			art_Utilities
-			canvas
-			${MF_MESSAGELOGGER}
-			
-			${FHICLCPP}
-			cetlib cetlib_except
-			${CLHEP}
-			${ROOT_BASIC_LIB_LIST}
-			${ROOT_MINUIT}
-			${ROOT_MINUIT2}	
-			${Boost_SYSTEM_LIBRARY}
+                        lardataobj_RecoBase
+                        larreco_Calorimetry
+                        nusimdata_SimulationBase
+                        ${ART_FRAMEWORK_CORE}
+                        ${ART_FRAMEWORK_PRINCIPAL}
+                        ${ART_FRAMEWORK_SERVICES_REGISTRY}
+                        ${ART_FRAMEWORK_SERVICES_BASIC}
+                        ${ART_ROOT_IO_TFILE_SUPPORT}
+                        ${ART_ROOT_IO_TFILESERVICE_SERVICE}
+                        ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
+                        art_Persistency_Common
+                        art_Persistency_Provenance
+                        art_Utilities
+                        canvas
+                        ${MF_MESSAGELOGGER}
+                        ${FHICLCPP}
+                        cetlib cetlib_except
+                        ${CLHEP_LIBRARIES}
+                        ${ROOT_LIBRARIES}
+                        ROOT::Minuit
+                        ROOT::Minuit2
+                        ${Boost_SYSTEM_LIBRARY}
         )
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/RecoAlg/ImagePatternAlgs/Tensorflow/TF/CMakeLists.txt b/larreco/RecoAlg/ImagePatternAlgs/Tensorflow/TF/CMakeLists.txt
index c9c35d8..770461c 100644
--- a/larreco/RecoAlg/ImagePatternAlgs/Tensorflow/TF/CMakeLists.txt
+++ b/larreco/RecoAlg/ImagePatternAlgs/Tensorflow/TF/CMakeLists.txt
@@ -1,13 +1,10 @@
-art_make( 
-          LIB_LIBRARIES 
-			${FHICLCPP}
-			cetlib cetlib_except
-			${TENSORFLOW}
-			${PROTOBUF}
-			${FOR_TENSORFLOW}
+art_make( LIB_LIBRARIES
+          ${FHICLCPP}
+          cetlib cetlib_except
+          ${TENSORFLOW}
+          ${PROTOBUF}
+          ${FOR_TENSORFLOW}
       #rt #workaround for larsoft issue 17960
         )
 
 install_headers()
-install_source()
-
diff --git a/larreco/RecoAlg/ImagePatternAlgs/job/CMakeLists.txt b/larreco/RecoAlg/ImagePatternAlgs/job/CMakeLists.txt
index fc867fc..8ac298f 100644
--- a/larreco/RecoAlg/ImagePatternAlgs/job/CMakeLists.txt
+++ b/larreco/RecoAlg/ImagePatternAlgs/job/CMakeLists.txt
@@ -2,4 +2,3 @@ install_fhicl()
 
 # copy in the source tree
 FILE(GLOB fcl_files *.fcl)
-install_source( EXTRAS ${fcl_files} )
diff --git a/larreco/RecoAlg/PMAlg/CMakeLists.txt b/larreco/RecoAlg/PMAlg/CMakeLists.txt
index 60e545c..9bfefca 100644
--- a/larreco/RecoAlg/PMAlg/CMakeLists.txt
+++ b/larreco/RecoAlg/PMAlg/CMakeLists.txt
@@ -1,15 +1,14 @@
 art_make(LIB_LIBRARIES
            larcorealg_Geometry
            ${ART_FRAMEWORK_SERVICES_REGISTRY}
-           ${ROOT_CORE}
-           ${ROOT_MATRIX}
-           ${ROOT_PHYSICS}
+           ROOT::Core
+           ROOT::Matrix
+           ROOT::Physics
+           ROOT::Minuit2
            canvas
            ${MF_MESSAGELOGGER}
            cetlib_except
-           ${ROOT_MINUIT2}
         )
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/RecoAlg/TCAlg/CMakeLists.txt b/larreco/RecoAlg/TCAlg/CMakeLists.txt
index 3b1fe95..2f5c837 100644
--- a/larreco/RecoAlg/TCAlg/CMakeLists.txt
+++ b/larreco/RecoAlg/TCAlg/CMakeLists.txt
@@ -15,17 +15,17 @@ art_make(LIB_LIBRARIES
            canvas
            cetlib
            cetlib_except
-           ${ROOT_CORE}
-           ${ROOT_GENVECTOR}
-           ${ROOT_GPAD}
-           ${ROOT_GRAF}
-           ${ROOT_GRAF3D}
-           ${ROOT_HIST}
-           ${ROOT_MATHCORE}
-           ${ROOT_MATRIX}
-           ${ROOT_RIO}
-           ${ROOT_TREE}
-           ${ROOT_TMVA}
+           ROOT::Core
+           ROOT::GenVector
+           ROOT::Gpad
+           ROOT::Graf
+           ROOT::Graf3d
+           ROOT::Hist
+           ROOT::MathCore
+           ROOT::Matrix
+           ROOT::RIO
+           ROOT::Tree
+           ROOT::TMVA
         )
 
 add_subdirectory(TCDebugTools)
diff --git a/larreco/RecoAlg/TCAlg/TCDebugTools/CMakeLists.txt b/larreco/RecoAlg/TCAlg/TCDebugTools/CMakeLists.txt
index 9f44bce..1d2c42a 100644
--- a/larreco/RecoAlg/TCAlg/TCDebugTools/CMakeLists.txt
+++ b/larreco/RecoAlg/TCAlg/TCDebugTools/CMakeLists.txt
@@ -2,5 +2,5 @@
 cet_make_exec( rsf_evd
                SOURCE showertree.C
                LIBRARIES
-               ${ROOT_BASIC_LIB_LIST}
+               ${ROOT_LIBRARIES}
                )
diff --git a/larreco/ShowerFinder/CMakeLists.txt b/larreco/ShowerFinder/CMakeLists.txt
index f351364..dfad314 100644
--- a/larreco/ShowerFinder/CMakeLists.txt
+++ b/larreco/ShowerFinder/CMakeLists.txt
@@ -6,12 +6,11 @@ art_make(MODULE_LIBRARIES
           larreco_RecoAlg
           lardataobj_RecoBase
           ${ART_FRAMEWORK_SERVICES_REGISTRY}
-          ${ROOT_CORE}
-          ${ROOT_PHYSICS}
+          ROOT::Core
+          ROOT::Physics
           ${ART_ROOT_IO_TFILESERVICE_SERVICE}
           ${MF_MESSAGELOGGER}
         )
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/ShowerFinder/ShowerReco3D/CMakeLists.txt b/larreco/ShowerFinder/ShowerReco3D/CMakeLists.txt
index 9721460..b052ca2 100644
--- a/larreco/ShowerFinder/ShowerReco3D/CMakeLists.txt
+++ b/larreco/ShowerFinder/ShowerReco3D/CMakeLists.txt
@@ -7,7 +7,7 @@ art_make(LIB_LIBRARIES
            larreco_RecoAlg_CMTool_CMTAlgPriority
            larreco_RecoAlg_CMTool_CMTAlgMatch
            ${ART_FRAMEWORK_SERVICES_REGISTRY}
-           ${ROOT_CORE}
+           ROOT::Core
            canvas
            cetlib_except
          MODULE_LIBRARIES
@@ -16,8 +16,8 @@ art_make(LIB_LIBRARIES
            ${ART_FRAMEWORK_SERVICES_REGISTRY}
            ${ART_ROOT_IO_TFILESERVICE_SERVICE}
            ${ART_ROOT_IO_TFILE_SUPPORT}
-           ${ROOT_CORE}
-           ${ROOT_PHYSICS}
+           ROOT::Core
+           ROOT::Physics
            ${MF_MESSAGELOGGER}
          )
 
diff --git a/larreco/ShowerFinder/ShowerReco3D/job/CMakeLists.txt b/larreco/ShowerFinder/ShowerReco3D/job/CMakeLists.txt
index 520ed80..ee86b91 100644
--- a/larreco/ShowerFinder/ShowerReco3D/job/CMakeLists.txt
+++ b/larreco/ShowerFinder/ShowerReco3D/job/CMakeLists.txt
@@ -5,4 +5,3 @@ install_fhicl()
 # Also put a copy in the source tree.
 
 FILE(GLOB fcl_files *.fcl)
-install_source( EXTRAS ${fcl_files} )
diff --git a/larreco/ShowerFinder/ShowerTools/CMakeLists.txt b/larreco/ShowerFinder/ShowerTools/CMakeLists.txt
index ebd885e..b81a870 100644
--- a/larreco/ShowerFinder/ShowerTools/CMakeLists.txt
+++ b/larreco/ShowerFinder/ShowerTools/CMakeLists.txt
@@ -7,5 +7,4 @@ art_make(TOOL_LIBRARIES
 )
 
 install_headers()
-install_source()
 install_fhicl()
diff --git a/larreco/ShowerFinder/ShowerTools/Cheating/CMakeLists.txt b/larreco/ShowerFinder/ShowerTools/Cheating/CMakeLists.txt
index 9f8060a..a364a17 100644
--- a/larreco/ShowerFinder/ShowerTools/Cheating/CMakeLists.txt
+++ b/larreco/ShowerFinder/ShowerTools/Cheating/CMakeLists.txt
@@ -3,5 +3,4 @@ art_make(TOOL_LIBRARIES
   )
 
 install_headers()
-install_source()
 install_fhicl()
diff --git a/larreco/ShowerFinder/job/CMakeLists.txt b/larreco/ShowerFinder/job/CMakeLists.txt
index 520ed80..ee86b91 100644
--- a/larreco/ShowerFinder/job/CMakeLists.txt
+++ b/larreco/ShowerFinder/job/CMakeLists.txt
@@ -5,4 +5,3 @@ install_fhicl()
 # Also put a copy in the source tree.
 
 FILE(GLOB fcl_files *.fcl)
-install_source( EXTRAS ${fcl_files} )
diff --git a/larreco/SpacePointFinder/CMakeLists.txt b/larreco/SpacePointFinder/CMakeLists.txt
index e45d650..c08428b 100644
--- a/larreco/SpacePointFinder/CMakeLists.txt
+++ b/larreco/SpacePointFinder/CMakeLists.txt
@@ -1,4 +1,5 @@
-include_directories ( $ENV{G4INCLUDE} )
+include(${Geant4_USE_FILE})
+include_directories ( ${Geant4_INCLUDE_DIRS} )
 
 art_make(MODULE_LIBRARIES
            larreco_RecoAlg
@@ -6,4 +7,3 @@ art_make(MODULE_LIBRARIES
          )
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/SpacePointSolver/CMakeLists.txt b/larreco/SpacePointSolver/CMakeLists.txt
index eff59b7..7eb86d6 100644
--- a/larreco/SpacePointSolver/CMakeLists.txt
+++ b/larreco/SpacePointSolver/CMakeLists.txt
@@ -5,8 +5,8 @@ art_make(BASENAME_ONLY
          LIB_LIBRARIES
            larcorealg_Geometry
            ${ART_FRAMEWORK_SERVICES_REGISTRY}
-           ${ROOT_CORE}
-           ${ROOT_PHYSICS}
+           ROOT::Core
+           ROOT::Physics
            canvas
            cetlib_except
          MODULE_LIBRARIES
@@ -16,10 +16,9 @@ art_make(BASENAME_ONLY
            lardata_ArtDataHelper
            larreco_SpacePointSolver
            ${ART_FRAMEWORK_SERVICES_REGISTRY}
-           ${ROOT_CORE}
-           ${ROOT_HIST}
+           ROOT::Core
+           ROOT::Hist
         )
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/SpacePointSolver/HitReaders/CMakeLists.txt b/larreco/SpacePointSolver/HitReaders/CMakeLists.txt
index 479e01c..d573e71 100644
--- a/larreco/SpacePointSolver/HitReaders/CMakeLists.txt
+++ b/larreco/SpacePointSolver/HitReaders/CMakeLists.txt
@@ -7,4 +7,3 @@ art_make(TOOL_LIBRARIES
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/TrackFinder/CMakeLists.txt b/larreco/TrackFinder/CMakeLists.txt
index 37d9ec3..805f095 100644
--- a/larreco/TrackFinder/CMakeLists.txt
+++ b/larreco/TrackFinder/CMakeLists.txt
@@ -1,4 +1,4 @@
-include_directories ($ENV{G4INCLUDE})
+include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 
 art_make(EXCLUDE
            Track3DKalman_module.cc
@@ -8,7 +8,7 @@ art_make(EXCLUDE
          MODULE_LIBRARIES
            larreco_RecoAlg
            nug4_MagneticField_MagneticField_service
-           ${ROOT_CORE}
+           ${ROOT_Core_LIBRARY}
            ${MF_MESSAGELOGGER}
          )
 
@@ -21,11 +21,11 @@ simple_plugin(Track3DKalman "module"
                 ${ART_FRAMEWORK_SERVICES_REGISTRY}
                 ${ART_ROOT_IO_TFILESERVICE_SERVICE}
                 ${ART_ROOT_IO_TFILE_SUPPORT}
-                ${CLHEP}
-                ${ROOT_CORE}
-                ${ROOT_MATRIX}
-                ${ROOT_PHYSICS}
-                ${ROOT_TREE}
+                ${CLHEP_LIBRARIES}
+                ROOT::Core
+                ROOT::Matrix
+                ROOT::Physics
+                ROOT::Tree
                 ${MF_MESSAGELOGGER}
               )
 
@@ -38,11 +38,11 @@ simple_plugin(Track3DKalmanSPS "module"
                 ${ART_ROOT_IO_TFILESERVICE_SERVICE}
                 ${ART_ROOT_IO_TFILE_SUPPORT}
                 ${MF_MESSAGELOGGER}
-                ${ROOT_CORE}
-                ${ROOT_HIST}
-                ${ROOT_MATRIX}
-                ${ROOT_PHYSICS}
-                ${ROOT_TREE}
+                ROOT::Core
+                ROOT::Hist
+                ROOT::Matrix
+                ROOT::Physics
+                ROOT::Tree
               )
 
 simple_plugin(KalmanFilterFitTrackMaker "tool"
@@ -53,4 +53,3 @@ simple_plugin(KalmanFilterFitTrackMaker "tool"
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/VertexFinder/CMakeLists.txt b/larreco/VertexFinder/CMakeLists.txt
index 53da87e..56d40b4 100644
--- a/larreco/VertexFinder/CMakeLists.txt
+++ b/larreco/VertexFinder/CMakeLists.txt
@@ -9,9 +9,9 @@ art_make(EXCLUDE FeatureVertexFinder_module.cc
            ${ART_FRAMEWORK_SERVICES_REGISTRY}
            ${ART_ROOT_IO_TFILESERVICE_SERVICE}
            ${ART_ROOT_IO_TFILE_SUPPORT}
-           ${ROOT_CORE}
-           ${ROOT_HIST}
-           ${ROOT_PHYSICS}
+           ROOT::Core
+           ROOT::Hist
+           ${ROOT_Physics_LIBRARY}
            ${MF_MESSAGELOGGER}
          )
 
@@ -19,11 +19,10 @@ art_make(EXCLUDE FeatureVertexFinder_module.cc
 simple_plugin(FeatureVertexFinder "module"
                 ${MF_MESSAGELOGGER}
                 ${ART_FRAMEWORK_SERVICES_REGISTRY}
-                ${ROOT_CORE}
-                ${ROOT_HIST}
+                ROOT::Core
+                ROOT::Hist
                 larcorealg_Geometry
                 lardataobj_RecoBase
               )
 
-install_fhicl()
-install_source()
+install_fhicl(_LIBRARY)
diff --git a/larreco/WireCell/CMakeLists.txt b/larreco/WireCell/CMakeLists.txt
index 3c921e0..9251fe2 100644
--- a/larreco/WireCell/CMakeLists.txt
+++ b/larreco/WireCell/CMakeLists.txt
@@ -4,15 +4,14 @@ art_make(MODULE_LIBRARIES
            lardataobj_RecoBase
            nusimdata_SimulationBase
            ${ART_FRAMEWORK_SERVICES_REGISTRY}
-           ${ROOT_CORE}
-           ${ROOT_RIO}
-           ${ROOT_TREE}
            ${MF_MESSAGELOGGER}
-           ${ROOT_EG}
-           ${ROOT_HIST}
-           ${ROOT_PHYSICS}
+           ROOT::Core
+           ROOT::RIO
+           ROOT::Tree
+           ROOT::EG
+           ROOT::Hist
+           ROOT::Physics
         )
 
 # install_headers()
 install_fhicl()
-install_source()
diff --git a/test/HitFinder/CMakeLists.txt b/test/HitFinder/CMakeLists.txt
index 04eb516..d358dcd 100644
--- a/test/HitFinder/CMakeLists.txt
+++ b/test/HitFinder/CMakeLists.txt
@@ -8,7 +8,7 @@ include(CetTest)
 cet_enable_asserts()
 
 cet_test(HitAnaAlg_test USE_BOOST_UNIT
-			LIBRARIES larreco_HitFinder
+                        LIBRARIES larreco_HitFinder
 )
 
 #cet_test(standalone_test)
