diff --git a/CMakeLists.txt b/CMakeLists.txt
index fdc3d21..0aa05a3 100644
@@ -10,10 +10,11 @@ cmake_minimum_required(VERSION 3.19 FATAL_ERROR)

 project(larreco VERSION 09.07.08.01 LANGUAGES CXX)

-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
+set(larreco_FW_DIR fw)

-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules $ENV{ART_DIR}/Modules)
+find_package( cetmodules REQUIRED )
+find_package( canvas_root_io )
+find_package( art )

 include(CetCMakeEnv)
 cet_cmake_env()
@@ -27,34 +28,36 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()

 # these are minimum required versions, not the actual product versions
-find_ups_product( eigen )
-find_ups_product( larcoreobj )
-find_ups_product( larcorealg )
-find_ups_product( larcore )
-find_ups_product( lardataobj )
-find_ups_product( lardataalg )
-find_ups_product( lardata )
-find_ups_product( larevt )
-find_ups_product( larsim )
-find_ups_product( larvecutils )
-find_ups_product( nutools )
-find_ups_product( nug4 )
-find_ups_product( nurandom )
-find_ups_product( nusimdata )
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product(messagefacility)
-find_ups_product( postgresql )
-find_ups_product( range )
-find_ups_boost()
-find_ups_root()
+find_package( Eigen3 )
+find_package( larcoreobj )
+find_package( larcorealg )
+find_package( larcore )
+find_package( lardataobj )
+find_package( lardataalg )
+find_package( lardata )
+find_package( larevt )
+find_package( larsim )
+find_package( larvecutils )
+find_package( nutools )
+find_package( nug4 )
+find_package( nurandom )
+find_package( nusimdata )
+find_package( art )
+find_package( art_root_io )
+find_package( messagefacility )
+find_package( cetlib_except )
+find_package( PostgreSQL )
+find_package( RangeV3 )
+find_package( ROOT )
+find_package( Geant4 )
+find_package( CLHEP COMPONENTS Random REQUIRED )

 # macros for dictionary and simple_plugin
 include(ArtDictionary)
 include(ArtMake)
 include(BuildPlugins)

-include_directories( $ENV{EIGEN_INC} )
+include_directories( ${EIGEN3_INCLUDE_DIRS} )

 # source
 add_subdirectory(larreco)
@@ -63,7 +66,8 @@ add_subdirectory(larreco)
 add_subdirectory(test)

 # ups - table and config files
-add_subdirectory(ups)
+#add_subdirectory(ups)

 # packaging utility
-include(UseCPack)
+#include(UseCPack)
+cet_cmake_config()
diff --git a/larreco/Calibrator/CMakeLists.txt b/larreco/Calibrator/CMakeLists.txt
index f226347..0cf708e 100644
--- a/larreco/Calibrator/CMakeLists.txt
+++ b/larreco/Calibrator/CMakeLists.txt
@@ -2,4 +2,3 @@ simple_plugin(PhotonCalibratorServiceStandard "service")
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larreco/Calorimetry/CMakeLists.txt b/larreco/Calorimetry/CMakeLists.txt
index e3b68ea..4f67805 100644
--- a/larreco/Calorimetry/CMakeLists.txt
+++ b/larreco/Calorimetry/CMakeLists.txt
@@ -1,25 +1,45 @@
-art_make(LIB_LIBRARIES
-           lardataalg_DetectorInfo
-           lardataobj_RecoBase
-           lardata_ArtDataHelper
-           larcorealg_Geometry
-           lardataobj_AnalysisBase
+set(LIB_LIBRARIES
+           lardataalg::DetectorInfo
+           lardataobj::RecoBase
+           lardata::ArtDataHelper
+           larcorealg::Geometry
+           lardataobj::AnalysisBase
            art::Framework_Services_Registry
-           ${ART_UTILITIES}
-           ${CANVAS}
-           ${CETLIB_EXCEPT}
            fhiclcpp::fhiclcpp
+           cetlib::cetlib
+           cetlib_except::cetlib_except
+           art::Utilities
+           fhiclcpp::fhiclcpp fhiclcpp::types
            messagefacility::MF_MessageLogger
            ROOT::Core
            ROOT::Physics
-         MODULE_LIBRARIES
-           larcorealg_Geometry
-           larreco_Calorimetry
+)
+set( MODULE_LIBRARIES
+           larcorealg::Geometry
+           larreco::Calorimetry
            ROOT::Core
            ROOT::Hist
            ROOT::Physics
            messagefacility::MF_MessageLogger)
 
+cet_make_library(SOURCE CalorimetryAlg.cxx  TrackCalorimetryAlg.cxx
+                 HEADERS_TARGET
+                 LIBRARIES ${LIB_LIBRARIES})
+
+cet_build_plugin(CaloChecker art::module LIBRARIES ${MODULE_LIBRARIES})
+
+cet_build_plugin(Calorimetry art::module LIBRARIES ${MODULE_LIBRARIES})
+
+cet_build_plugin(GnocchiCalorimetry art::module LIBRARIES ${MODULE_LIBRARIES})
+
+cet_build_plugin(PrintCalorimetry art::module LIBRARIES ${MODULE_LIBRARIES})
+
+cet_build_plugin(ShowerCalorimetry art::module LIBRARIES ${MODULE_LIBRARIES})
+
+cet_build_plugin(TrackCalorimetry art::module LIBRARIES ${MODULE_LIBRARIES}) 
+
+
+
 install_headers()
 install_fhicl()
 install_source()
diff --git a/larreco/ClusterFinder/CMakeLists.txt b/larreco/ClusterFinder/CMakeLists.txt
index 59e1987..e0fc369 100644
--- a/larreco/ClusterFinder/CMakeLists.txt
+++ b/larreco/ClusterFinder/CMakeLists.txt
@@ -1,16 +1,16 @@
 art_make(LIB_LIBRARIES
-           lardataobj_RecoBase
+           lardataobj::RecoBase
          MODULE_LIBRARIES
-           larreco_Calorimetry
-           larreco_RecoAlg_Cluster3DAlgs
-           larreco_RecoAlg_TCAlg
-           larreco_RecoAlg
-           larreco_ClusterFinder
-           larsim_MCCheater_ParticleInventoryService_service
-           lardataobj_AnalysisBase
-           lardataobj_RecoBase
-           nurandom_RandomUtils_NuRandomService_service
-           nusimdata_SimulationBase
+           larreco::Calorimetry
+           larreco::RecoAlg_Cluster3DAlgs
+           larreco::RecoAlg_TCAlg
+           larreco::RecoAlg
+           larreco::ClusterFinder
+           larsim::MCCheater_ParticleInventoryService_service
+           lardataobj::AnalysisBase
+           lardataobj::RecoBase
+           nurandom::RandomUtils_NuRandomService_service
+           nusimdata::SimulationBase
            art::Framework_Services_Registry
            art_root_io::tfile_support
            ROOT::Core
diff --git a/larreco/ClusterFinder/TrackShowerSplitter/CMakeLists.txt b/larreco/ClusterFinder/TrackShowerSplitter/CMakeLists.txt
index 0d6a681..4243e36 100644
--- a/larreco/ClusterFinder/TrackShowerSplitter/CMakeLists.txt
+++ b/larreco/ClusterFinder/TrackShowerSplitter/CMakeLists.txt
@@ -1,13 +1,13 @@
 add_subdirectory(Segmentation2D)
 
 simple_plugin(TrackShowerHits   "module"
-			lardataobj_RawData
-		 	lardataobj_RecoBase
+			lardataobj::RawData
+		 	lardataobj::RecoBase
 			larreco_RecoAlg
 			larreco_ClusterFinder_TrackShowerSplitter_Segmentation2D
-		 	larcorealg_Geometry
-			larcore_Geometry_Geometry_service
-			lardata_Utilities
+		 	larcorealg::Geometry
+			larcore::Geometry_Geometry_service
+			lardata::Utilities
                art::Framework_Core
 			art::Framework_Principal
 			art::Framework_Services_Registry
@@ -18,9 +18,7 @@ simple_plugin(TrackShowerHits   "module"
 			art::Utilities
 			canvas::canvas
 			messagefacility::MF_MessageLogger
-
-			cetlib::cetlib cetlib_except
-
+			cetlib::cetlib cetlib_except::cetlib_except
 )
 
 install_headers()
diff --git a/larreco/ClusterFinder/TrackShowerSplitter/Segmentation2D/CMakeLists.txt b/larreco/ClusterFinder/TrackShowerSplitter/Segmentation2D/CMakeLists.txt
index e0cde62..85499a3 100644
--- a/larreco/ClusterFinder/TrackShowerSplitter/Segmentation2D/CMakeLists.txt
+++ b/larreco/ClusterFinder/TrackShowerSplitter/Segmentation2D/CMakeLists.txt
@@ -1,5 +1,5 @@
 art_make(LIB_LIBRARIES
-           larreco_RecoAlg_PMAlg
+           larreco::RecoAlg_PMAlg
            ROOT::Core
            ROOT::Physics
            canvas::canvas
diff --git a/larreco/DirOfGamma/CMakeLists.txt b/larreco/DirOfGamma/CMakeLists.txt
index bc8dbd6..8e8cd24 100644
--- a/larreco/DirOfGamma/CMakeLists.txt
+++ b/larreco/DirOfGamma/CMakeLists.txt
@@ -1,6 +1,6 @@
 art_make(LIB_LIBRARIES
-           larcorealg_Geometry
-           larreco_RecoAlg_PMAlg
+           larcorealg::Geometry
+           larreco::RecoAlg_PMAlg
            art::Framework_Services_Registry
            ROOT::Core
            ROOT::Physics
@@ -8,10 +8,10 @@ art_make(LIB_LIBRARIES
            messagefacility::MF_MessageLogger
            cetlib_except::cetlib_except
          MODULE_LIBRARIES
-           larreco_RecoAlg
-           larreco_RecoAlg_PMAlg
-           larreco_Calorimetry
-           larreco_DirOfGamma
+           larreco::RecoAlg
+           larreco::RecoAlg_PMAlg
+           larreco::Calorimetry
+           larreco::DirOfGamma
            art::Framework_Services_Registry
            ROOT::Core
            ROOT::Physics
diff --git a/larreco/EventFinder/CMakeLists.txt b/larreco/EventFinder/CMakeLists.txt
index 2159ff7..a06a411 100644
--- a/larreco/EventFinder/CMakeLists.txt
+++ b/larreco/EventFinder/CMakeLists.txt
@@ -1,5 +1,5 @@
 art_make(MODULE_LIBRARIES
-           lardataobj_RecoBase
+           lardataobj::RecoBase
            ROOT::Core
            messagefacility::MF_MessageLogger
          )
diff --git a/larreco/HitFinder/CMakeLists.txt b/larreco/HitFinder/CMakeLists.txt
index 2a77d3c..02f5539 100644
--- a/larreco/HitFinder/CMakeLists.txt
+++ b/larreco/HitFinder/CMakeLists.txt
@@ -1,21 +1,21 @@
 add_subdirectory(HitFinderTools)
 
 art_make(LIB_LIBRARIES
-           lardataobj_RawData
-           lardataobj_RecoBase
-           larcorealg_Geometry
+           lardataobj::RawData
+           lardataobj::RecoBase
+           larcorealg::Geometry
            messagefacility::MF_MessageLogger
            fhiclcpp::fhiclcpp
            cetlib_except::cetlib_except
            ROOT::Core
-           ${ART_UTILITIES}
+           art::Utilities
          MODULE_LIBRARIES
            larreco_HitFinder
-           larsim_MCCheater_BackTrackerService_service
-           larreco_RecoAlg
-           lardataobj_RecoBase
-           larcorealg_Geometry
-           lardata_ArtDataHelper
+           larsim::MCCheater_BackTrackerService_service
+           larreco::RecoAlg
+           lardataobj::RecoBase
+           larcorealg::Geometry
+           lardata::ArtDataHelper
            art::Framework_Services_Registry
            art_root_io::tfile_support
            ROOT::Core
diff --git a/larreco/HitFinder/HitFinderTools/CMakeLists.txt b/larreco/HitFinder/HitFinderTools/CMakeLists.txt
index 3e9df1d..d914cb3 100644
--- a/larreco/HitFinder/HitFinderTools/CMakeLists.txt
+++ b/larreco/HitFinder/HitFinderTools/CMakeLists.txt
@@ -2,25 +2,22 @@ cet_enable_asserts()

 set( hitfinder_tool_lib_list
                         larreco_RecoAlg
-                        larcorealg_Geometry
-                        lardataobj_RecoBase
-                        larcore_Geometry_Geometry_service
-                        lardata_Utilities
-                        nurandom_RandomUtils_NuRandomService_service
-                        larvecutils_MarqFitAlg
+                        larcorealg::Geometry
+                        lardataobj::RecoBase
+                        larcore::Geometry_Geometry_service
+                        lardata::Utilities
+                        nurandom::RandomUtils_NuRandomService_service
+                        larvecutils::MarqFitAlg
                         art::Framework_Core
                         art::Framework_Principal
                         art::Framework_Services_Registry
                         art::Persistency_Common
                         art::Persistency_Provenance
                         art_root_io::tfile_support ROOT::Core
-            ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
                         art_root_io::TFileService_service
-                        ${ART_FRAMEWORK_SERVICES_SYSTEM_FILECATALOGMETADATA_SERVICE}
                         art::Utilities
                         canvas::canvas
                         messagefacility::MF_MessageLogger
-
                         fhiclcpp::fhiclcpp
                         cetlib::cetlib
                         Boost::filesystem
diff --git a/larreco/MCComp/CMakeLists.txt b/larreco/MCComp/CMakeLists.txt
index 4025dba..9236ccb 100644
--- a/larreco/MCComp/CMakeLists.txt
+++ b/larreco/MCComp/CMakeLists.txt
@@ -1,15 +1,15 @@
 #add_subdirectory(job)
 
 art_make(LIB_LIBRARIES
-           larcorealg_Geometry
+           larcorealg::Geometry
            art::Framework_Services_Registry
            canvas::canvas
            cetlib_except::cetlib_except
            ROOT::Core
-           ${ART_UTILITIES}
+           art::Utilities
          MODULE_LIBRARIES
-           larreco_MCComp
-           larcorealg_Geometry
+           larreco::MCComp
+           larcorealg::Geometry
            art::Framework_Services_Registry
            ROOT::Core
          )
diff --git a/larreco/QuadVtx/CMakeLists.txt b/larreco/QuadVtx/CMakeLists.txt
index 05e40ef..e329ce6 100644
--- a/larreco/QuadVtx/CMakeLists.txt
+++ b/larreco/QuadVtx/CMakeLists.txt
@@ -4,12 +4,12 @@ art_make(LIB_LIBRARIES
            ROOT::Core
            ROOT::Hist
          MODULE_LIBRARIES
-           larcorealg_Geometry
-           lardataalg_DetectorInfo
-           lardataobj_RecoBase
-           larreco_QuadVtx
-           nusimdata_SimulationBase
-           lardata_ArtDataHelper
+           larcorealg::Geometry
+           lardataalg::DetectorInfo
+           lardataobj::RecoBase
+           larreco::QuadVtx
+           nusimdata::SimulationBase
+           lardata::ArtDataHelper
            art_root_io::TFileService_service
            art_root_io::tfile_support
            art::Framework_Services_Registry
diff --git a/larreco/QuadVtx/QuadVtx_module.cc b/larreco/QuadVtx/QuadVtx_module.cc
index e472588..417ff7d 100644
--- a/larreco/QuadVtx/QuadVtx_module.cc
+++ b/larreco/QuadVtx/QuadVtx_module.cc
@@ -6,6 +6,7 @@
 #include <iostream>
 #include <random>
 #include <string>
+#include <cmath>
 
 // framework libraries
 #include "art/Framework/Core/EDProducer.h"
@@ -158,7 +159,7 @@ namespace quad {
         for (unsigned int j = i + offset + 1; j < pts.size(); j += stride) {
           const Line2D l(pts[i], pts[j]);
 
-          if (isinf(l.m) || isnan(l.m) || isinf(l.c) || isnan(l.c)) continue;
+          if (std::isinf(l.m) || std::isnan(l.m) || std::isinf(l.c) || std::isnan(l.c)) continue;
 
           if (R > 0) {
             float z1, z2;
diff --git a/larreco/RecoAlg/CMTool/CMTAlgMatch/CMakeLists.txt b/larreco/RecoAlg/CMTool/CMTAlgMatch/CMakeLists.txt
index 5ff0dda..de10cb6 100644
--- a/larreco/RecoAlg/CMTool/CMTAlgMatch/CMakeLists.txt
+++ b/larreco/RecoAlg/CMTool/CMTAlgMatch/CMakeLists.txt
@@ -1,8 +1,8 @@
 art_make(LIB_LIBRARIES
-           larreco_RecoAlg_CMTool_CMToolBase
-           larcorealg_Geometry
+           larreco::RecoAlg_CMTool_CMToolBase
+           larcorealg::Geometry
            art::Framework_Services_Registry
-           ${ART_UTILITIES}
+           art::Utilities
            ROOT::Core
            ROOT::RIO
            ROOT::Tree
diff --git a/larreco/RecoAlg/CMTool/CMTAlgPriority/CMakeLists.txt b/larreco/RecoAlg/CMTool/CMTAlgPriority/CMakeLists.txt
index 586fe7a..80290f9 100644
--- a/larreco/RecoAlg/CMTool/CMTAlgPriority/CMakeLists.txt
+++ b/larreco/RecoAlg/CMTool/CMTAlgPriority/CMakeLists.txt
@@ -1,4 +1,4 @@
-art_make(LIB_LIBRARIES larreco_RecoAlg_CMTool_CMToolBase)
+art_make(LIB_LIBRARIES larreco::RecoAlg_CMTool_CMToolBase)
 
 install_headers()
 install_fhicl()
diff --git a/larreco/RecoAlg/CMTool/CMToolApp/CMakeLists.txt b/larreco/RecoAlg/CMTool/CMToolApp/CMakeLists.txt
index 586fe7a..80290f9 100644
--- a/larreco/RecoAlg/CMTool/CMToolApp/CMakeLists.txt
+++ b/larreco/RecoAlg/CMTool/CMToolApp/CMakeLists.txt
@@ -1,4 +1,4 @@
-art_make(LIB_LIBRARIES larreco_RecoAlg_CMTool_CMToolBase)
+art_make(LIB_LIBRARIES larreco::RecoAlg_CMTool_CMToolBase)
 
 install_headers()
 install_fhicl()
diff --git a/larreco/RecoAlg/CMTool/CMToolBase/CMakeLists.txt b/larreco/RecoAlg/CMTool/CMToolBase/CMakeLists.txt
index 61d3f15..df67d2b 100644
--- a/larreco/RecoAlg/CMTool/CMToolBase/CMakeLists.txt
+++ b/larreco/RecoAlg/CMTool/CMToolBase/CMakeLists.txt
@@ -1,4 +1,4 @@
-art_make(LIB_LIBRARIES larreco_RecoAlg_ClusterRecoUtil)
+art_make(LIB_LIBRARIES larreco::RecoAlg_ClusterRecoUtil)
 
 install_headers()
 install_fhicl()
diff --git a/larreco/RecoAlg/CMakeLists.txt b/larreco/RecoAlg/CMakeLists.txt
index 1a64b99..9162a0e 100644
--- a/larreco/RecoAlg/CMakeLists.txt
+++ b/larreco/RecoAlg/CMakeLists.txt
@@ -1,12 +1,12 @@
 
 art_make(LIB_LIBRARIES
-           lardataalg_DetectorInfo
-           larcorealg_Geometry
-           larreco_RecoAlg_TCAlg
-           larreco_RecoAlg_PMAlg
-           larreco_RecoAlg_ClusterRecoUtil
-           larreco_RecoAlg_CMTool_CMToolBase
-           larreco_RecoAlg_ImagePatternAlgs_DataProvider
+           lardataalg::DetectorInfo
+           larcorealg::Geometry
+           larreco::RecoAlg_TCAlg
+           larreco::RecoAlg_PMAlg
+           larreco::RecoAlg_ClusterRecoUtil
+           larreco::RecoAlg_CMTool_CMToolBase
+           larreco::RecoAlg_ImagePatternAlgs_DataProvider
            ROOT::Core
            ROOT::Physics
            ROOT::Matrix
diff --git a/larreco/RecoAlg/Cluster3DAlgs/CMakeLists.txt b/larreco/RecoAlg/Cluster3DAlgs/CMakeLists.txt
index 5a33050..7af7f46 100644
--- a/larreco/RecoAlg/Cluster3DAlgs/CMakeLists.txt
+++ b/larreco/RecoAlg/Cluster3DAlgs/CMakeLists.txt
@@ -5,16 +5,16 @@ add_subdirectory(Voronoi)
 add_subdirectory(PathFinding)
 
 art_make(LIB_LIBRARIES
-           lardataalg_DetectorInfo
-           lardataobj_RecoBase
-           larcorealg_Geometry
-           lardata_ArtDataHelper
-           lardata_RecoObjects
+           lardataalg::DetectorInfo
+           lardataobj::RecoBase
+           larcorealg::Geometry
+           lardata::ArtDataHelper
+           lardata::RecoObjects
            art::Framework_Core
            art::Framework_Principal
            art::Framework_Services_Registry
            art_root_io::tfile_support
-           ${ART_UTILITIES}
+           art::Utilities
            ROOT::Core
            ROOT::Gpad
            ROOT::Hist
@@ -28,7 +28,7 @@ art_make(LIB_LIBRARIES
            fhiclcpp::fhiclcpp
            cetlib::cetlib
            cetlib_except::cetlib_except
-          TOOL_LIBRARIES larreco_RecoAlg_Cluster3DAlgs
+          TOOL_LIBRARIES larreco::RecoAlg_Cluster3DAlgs
         )
 
 install_headers()
diff --git a/larreco/RecoAlg/Cluster3DAlgs/PathFinding/CMakeLists.txt b/larreco/RecoAlg/Cluster3DAlgs/PathFinding/CMakeLists.txt
index 15657f2..7a962e6 100644
--- a/larreco/RecoAlg/Cluster3DAlgs/PathFinding/CMakeLists.txt
+++ b/larreco/RecoAlg/Cluster3DAlgs/PathFinding/CMakeLists.txt
@@ -1,8 +1,8 @@
 cet_enable_asserts()
 
 art_make(TOOL_LIBRARIES
-           larreco_RecoAlg_Cluster3DAlgs
-           larreco_RecoAlg_Cluster3DAlgs_Voronoi
+           larreco::RecoAlg_Cluster3DAlgs
+           larreco::RecoAlg_Cluster3DAlgs_Voronoi
            art_root_io::tfile_support
            ROOT::Core
            art_root_io::TFileService_service
diff --git a/larreco/RecoAlg/Cluster3DAlgs/Voronoi/CMakeLists.txt b/larreco/RecoAlg/Cluster3DAlgs/Voronoi/CMakeLists.txt
index c0241c8..676e38a 100644
--- a/larreco/RecoAlg/Cluster3DAlgs/Voronoi/CMakeLists.txt
+++ b/larreco/RecoAlg/Cluster3DAlgs/Voronoi/CMakeLists.txt
@@ -1,6 +1,6 @@
 cet_enable_asserts()
 
-art_make(LIB_LIBRARIES larreco_RecoAlg_Cluster3DAlgs_ConvexHull)
+art_make(LIB_LIBRARIES larreco::RecoAlg_Cluster3DAlgs_ConvexHull)
 
 install_headers()
 install_fhicl()
diff --git a/larreco/RecoAlg/ClusterRecoUtil/CMakeLists.txt b/larreco/RecoAlg/ClusterRecoUtil/CMakeLists.txt
index 45b939b..9d37ff4 100644
--- a/larreco/RecoAlg/ClusterRecoUtil/CMakeLists.txt
+++ b/larreco/RecoAlg/ClusterRecoUtil/CMakeLists.txt
@@ -1,5 +1,5 @@
 art_make(LIB_LIBRARIES
-           lardata_Utilities
+           lardata::Utilities
            ROOT::Graf
            ROOT::Gpad
         )
diff --git a/larreco/RecoAlg/ImagePatternAlgs/DataProvider/CMakeLists.txt b/larreco/RecoAlg/ImagePatternAlgs/DataProvider/CMakeLists.txt
index 4b9e64b..a3313c8 100644
--- a/larreco/RecoAlg/ImagePatternAlgs/DataProvider/CMakeLists.txt
+++ b/larreco/RecoAlg/ImagePatternAlgs/DataProvider/CMakeLists.txt
@@ -1,11 +1,12 @@
 art_make(LIB_LIBRARIES
-           larcorealg_Geometry
-           larreco_Calorimetry
+           larcorealg::Geometry
+           larreco::Calorimetry
            art::Framework_Services_Registry
            ROOT::Core
            canvas::canvas
            messagefacility::MF_MessageLogger
            cetlib_except::cetlib_except
+           CLHEP::Random
         )
 
 install_headers()
diff --git a/larreco/RecoAlg/PMAlg/CMakeLists.txt b/larreco/RecoAlg/PMAlg/CMakeLists.txt
index 1308f95..1de6853 100644
--- a/larreco/RecoAlg/PMAlg/CMakeLists.txt
+++ b/larreco/RecoAlg/PMAlg/CMakeLists.txt
@@ -1,10 +1,8 @@
 art_make(LIB_LIBRARIES
-           larcorealg_Geometry
-           lardataalg_DetectorInfo
+           larcorealg::Geometry
+           lardataalg::DetectorInfo
            art::Framework_Services_Registry
-           ${ART_UTILITIES}
-           ${CANVAS}
-           ${CETLIB_EXCEPT}
+           art::Utilities
            messagefacility::MF_MessageLogger
            ROOT::Core
            ROOT::Matrix
diff --git a/larreco/RecoAlg/TCAlg/CMakeLists.txt b/larreco/RecoAlg/TCAlg/CMakeLists.txt
index dfb6ee8..4bd2995 100644
--- a/larreco/RecoAlg/TCAlg/CMakeLists.txt
+++ b/larreco/RecoAlg/TCAlg/CMakeLists.txt
@@ -1,21 +1,20 @@
 art_make(LIB_LIBRARIES
-           larsim_MCCheater_BackTrackerService_service
-           larsim_MCCheater_ParticleInventoryService_service
-           nug4_ParticleNavigation
-           nusimdata_SimulationBase
-           lardata_RecoObjects
-           lardataalg_DetectorInfo
-           larcorealg_Geometry
+           larsim::MCCheater_BackTrackerService_service
+           larsim::MCCheater_ParticleInventoryService_service
+           nug4::ParticleNavigation
+           nusimdata::SimulationBase
+           lardata::RecoObjects
+           lardataalg::DetectorInfo
+           larcorealg::Geometry
            larreco_Calorimetry
            art::Framework_Principal
            art::Framework_Services_Registry
-           ${ART_PERSISTENCY_PROVENANCE}
            art_root_io::TFileService_service
            art_root_io::tfile_support
-           ${ART_UTILITIES}
-           ${CANVAS}
-           ${CETLIB_EXCEPT}
+           art::Persistency_Provenance
+           art::Utilities
            cetlib::cetlib
+           cetlib_except::cetlib_except
            ROOT::Core
            ROOT::GenVector
            ROOT::Gpad
diff --git a/larreco/SCECorrections/CMakeLists.txt b/larreco/SCECorrections/CMakeLists.txt
index 9a3f1e1..13237f2 100644
--- a/larreco/SCECorrections/CMakeLists.txt
+++ b/larreco/SCECorrections/CMakeLists.txt
@@ -1,7 +1,7 @@
 art_make(MODULE_LIBRARIES
-  lardataalg_DetectorInfo
-  larcorealg_Geometry
-  lardataobj_RecoBase
+  lardataalg::DetectorInfo
+  larcorealg::Geometry
+  lardataobj::RecoBase
   art::Framework_Services_Registry
   ROOT::Core
   ROOT::Physics
diff --git a/larreco/ShowerFinder/CMakeLists.txt b/larreco/ShowerFinder/CMakeLists.txt
index 2270928..615bc4f 100644
--- a/larreco/ShowerFinder/CMakeLists.txt
+++ b/larreco/ShowerFinder/CMakeLists.txt
@@ -2,8 +2,8 @@ add_subdirectory(ShowerReco3D)
 add_subdirectory(job)
 
 art_make(MODULE_LIBRARIES
-          larreco_RecoAlg
-          lardataobj_RecoBase
+          larreco::RecoAlg
+          lardataobj::RecoBase
           art::Framework_Services_Registry
           ROOT::Core
           ROOT::Physics
diff --git a/larreco/ShowerFinder/ShowerReco3D/CMakeLists.txt b/larreco/ShowerFinder/ShowerReco3D/CMakeLists.txt
index 30aebe0..f4c0ebc 100644
--- a/larreco/ShowerFinder/ShowerReco3D/CMakeLists.txt
+++ b/larreco/ShowerFinder/ShowerReco3D/CMakeLists.txt
@@ -1,18 +1,18 @@
 add_subdirectory(job)
 
 art_make(LIB_LIBRARIES
-           lardata_Utilities
-           larreco_Calorimetry
-           larreco_RecoAlg_CMTool_CMToolBase
-           larreco_RecoAlg_CMTool_CMTAlgPriority
-           larreco_RecoAlg_CMTool_CMTAlgMatch
+           lardata::Utilities
+           larreco::Calorimetry
+           larreco::RecoAlg_CMTool_CMToolBase
+           larreco::RecoAlg_CMTool_CMTAlgPriority
+           larreco::RecoAlg_CMTool_CMTAlgMatch
            art::Framework_Services_Registry
            ROOT::Core
            canvas::canvas
            cetlib_except::cetlib_except
          MODULE_LIBRARIES
-           larreco_MCComp
-           larreco_ShowerFinder_ShowerReco3D
+           larreco::MCComp
+           larreco::ShowerFinder_ShowerReco3D
            art::Framework_Services_Registry
            art_root_io::TFileService_service
            art_root_io::tfile_support
diff --git a/larreco/SpacePointFinder/CMakeLists.txt b/larreco/SpacePointFinder/CMakeLists.txt
index 3809bad..43f8caa 100644
--- a/larreco/SpacePointFinder/CMakeLists.txt
+++ b/larreco/SpacePointFinder/CMakeLists.txt
@@ -1,7 +1,7 @@
 include_directories ( $ENV{G4INCLUDE} )
 
 art_make(MODULE_LIBRARIES
-           larreco_RecoAlg
+           larreco::RecoAlg
            messagefacility::MF_MessageLogger
          )
 install_headers()
diff --git a/larreco/SpacePointSolver/CMakeLists.txt b/larreco/SpacePointSolver/CMakeLists.txt
index df6c5e4..efef150 100644
--- a/larreco/SpacePointSolver/CMakeLists.txt
+++ b/larreco/SpacePointSolver/CMakeLists.txt
@@ -2,20 +2,18 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 add_subdirectory(HitReaders)
 
 art_make(LIB_LIBRARIES
-           larcorealg_Geometry
-           lardataalg_DetectorInfo
+           larcorealg::Geometry
+           lardataalg::DetectorInfo
            art::Framework_Services_Registry
-           ${ART_UTILITIES}
-           ${CANVAS}
-           ${CETLIB_EXCEPT}
+           art::Utilities
            ROOT::Core
            ROOT::Physics
          MODULE_LIBRARIES
-           larcorealg_Geometry
-           lardataobj_RecoBase
-           larsim_MCCheater_BackTrackerService_service
-           lardata_ArtDataHelper
-           larreco_SpacePointSolver
+           larcorealg::Geometry
+           lardataobj::RecoBase
+           larsim::MCCheater_BackTrackerService_service
+           lardata::ArtDataHelper
+           larreco::SpacePointSolver
            art::Framework_Services_Registry
            ROOT::Core
            ROOT::Hist
diff --git a/larreco/SpacePointSolver/HitReaders/HitsICARUS_tool.cc b/larreco/SpacePointSolver/HitReaders/HitsICARUS_tool.cc
index 0ce4193..3ec6b35 100755
--- a/larreco/SpacePointSolver/HitReaders/HitsICARUS_tool.cc
+++ b/larreco/SpacePointSolver/HitReaders/HitsICARUS_tool.cc
@@ -16,6 +16,7 @@
 // std includes
 #include <ostream>
 #include <vector>
+#include <cmath>
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 // implementation follows
@@ -78,7 +79,7 @@ bool HitsICARUS::readHits(const std::vector<art::Ptr<recob::Hit>>& inputHits,
 {
     for(auto& hit: inputHits)
     {
-        if(hit->Integral() < 0 || isnan(hit->Integral()) || isinf(hit->Integral()))
+        if(hit->Integral() < 0 || std::isnan(hit->Integral()) || std::isinf(hit->Integral()))
         {
             mf::LogWarning("Hits_ICARUS") << "WARNING: bad recob::Hit::Integral() = "
             << hit->Integral()
diff --git a/larreco/SpacePointSolver/HitReaders/HitsStandard_tool.cc b/larreco/SpacePointSolver/HitReaders/HitsStandard_tool.cc
index 90dc1aa..1401657 100755
--- a/larreco/SpacePointSolver/HitReaders/HitsStandard_tool.cc
+++ b/larreco/SpacePointSolver/HitReaders/HitsStandard_tool.cc
@@ -16,7 +16,7 @@
 // std includes
 #include <ostream>
 #include <vector>
-
+#include <cmath>
 //------------------------------------------------------------------------------------------------------------------------------------------
 // implementation follows
 
@@ -82,7 +82,7 @@ bool HitsStandard::readHits(const std::vector<art::Ptr<recob::Hit>>& inputHits,
 
     for(auto& hit: inputHits)
     {
-        if(hit->Integral() < 0 || isnan(hit->Integral()) || isinf(hit->Integral()))
+        if(hit->Integral() < 0 || std::isnan(hit->Integral()) || std::isinf(hit->Integral()))
         {
             mf::LogWarning("HitsStandard") << "WARNING: bad recob::Hit::Integral() = "
             << hit->Integral()
diff --git a/larreco/SpacePointSolver/SpacePointSolver_module.cc b/larreco/SpacePointSolver/SpacePointSolver_module.cc
index 1e37a48..e365dfe 100644
--- a/larreco/SpacePointSolver/SpacePointSolver_module.cc
+++ b/larreco/SpacePointSolver/SpacePointSolver_module.cc
@@ -5,6 +5,7 @@
 // C/C++ standard libraries
 #include <iostream>
 #include <string>
+#include <cmath>
 
 // framework libraries
 #include "art/Framework/Core/EDProducer.h"
@@ -221,7 +222,7 @@ namespace reco3d {
           const double coupling = exp(-sqrt(dist2) / 2);
           sc1->fNeighbours.emplace_back(sc2, coupling);
 
-          if (isnan(1 / sqrt(dist2)) || isinf(1 / sqrt(dist2))) {
+          if (std::isnan(1 / sqrt(dist2)) || std::isinf(1 / sqrt(dist2))) {
             std::cout << dist2 << " " << sc1->fX << " " << sc2->fX << " " << sc1->fY << " "
                       << sc2->fY << " " << sc1->fZ << " " << sc2->fZ << std::endl;
             abort();
diff --git a/larreco/TrackFinder/CMakeLists.txt b/larreco/TrackFinder/CMakeLists.txt
index f0010b8..0ee3005 100644
--- a/larreco/TrackFinder/CMakeLists.txt
+++ b/larreco/TrackFinder/CMakeLists.txt
@@ -6,18 +6,18 @@ art_make(EXCLUDE
            DumpTracks_module.cc
            KalmanFilterFitTrackMaker_tool.cc
          MODULE_LIBRARIES
-           larreco_RecoAlg
-           nug4_MagneticFieldServices_MagneticFieldServiceStandard_service
+           larreco::RecoAlg
+           nug4::MagneticFieldServices_MagneticFieldServiceStandard_service
            ROOT::Core
            messagefacility::MF_MessageLogger
          )
 
 simple_plugin(Track3DKalman "module"
-                larreco_Genfit
-                lardataobj_RecoBase
-                larcorealg_Geometry
-                nurandom_RandomUtils_NuRandomService_service
-                nusimdata_SimulationBase
+                larreco::Genfit
+                lardataobj::RecoBase
+                larcorealg::Geometry
+                nurandom::RandomUtils_NuRandomService_service
+                nusimdata::SimulationBase
                 art::Framework_Services_Registry
                 art_root_io::TFileService_service
                 art_root_io::tfile_support
@@ -29,10 +29,10 @@ simple_plugin(Track3DKalman "module"
               )
 
 simple_plugin(Track3DKalmanSPS "module"
-                larreco_Genfit
-                lardataobj_RecoBase
-                larcorealg_Geometry
-                nusimdata_SimulationBase
+                larreco::Genfit
+                lardataobj::RecoBase
+                larcorealg::Geometry
+                nusimdata::SimulationBase
                 art::Framework_Services_Registry
                 art_root_io::TFileService_service
                 art_root_io::tfile_support
@@ -46,8 +46,8 @@ simple_plugin(Track3DKalmanSPS "module"
 
 simple_plugin(KalmanFilterFitTrackMaker "tool"
                 canvas::canvas
-                lardataobj_RecoBase
-                larreco_RecoAlg
+                lardataobj::RecoBase
+                larreco::RecoAlg
               )
 
 install_headers()
diff --git a/larreco/VertexFinder/CMakeLists.txt b/larreco/VertexFinder/CMakeLists.txt
index bbbd746..9402e2b 100644
--- a/larreco/VertexFinder/CMakeLists.txt
+++ b/larreco/VertexFinder/CMakeLists.txt
@@ -1,11 +1,11 @@
 art_make(EXCLUDE FeatureVertexFinder_module.cc
          MODULE_LIBRARIES
-           larsim_MCCheater_ParticleInventoryService_service
-           lardataobj_RecoBase
-           larcorealg_Geometry
+           larsim::MCCheater_ParticleInventoryService_service
+           lardataobj::RecoBase
+           larcorealg::Geometry
            larreco_RecoAlg
-           nusimdata_SimulationBase
-           lardata_RecoObjects
+           nusimdata::SimulationBase
+           lardata::RecoObjects
            art::Framework_Services_Registry
            art_root_io::TFileService_service
            art_root_io::tfile_support
@@ -20,9 +20,9 @@ simple_plugin(FeatureVertexFinder "module"
                 art::Framework_Services_Registry
                 ROOT::Core
                 ROOT::Hist
-                larcorealg_Geometry
-                lardataalg_DetectorInfo
-                lardataobj_RecoBase
+                larcorealg::Geometry
+                lardataalg::DetectorInfo
+                lardataobj::RecoBase
               )
 
 install_fhicl()
diff --git a/larreco/WireCell/CMakeLists.txt b/larreco/WireCell/CMakeLists.txt
index 6f364f4..7ff1ee3 100644
--- a/larreco/WireCell/CMakeLists.txt
+++ b/larreco/WireCell/CMakeLists.txt
@@ -1,8 +1,8 @@
 art_make(MODULE_LIBRARIES
-           larcorealg_Geometry
-           lardataobj_RawData
-           lardataobj_RecoBase
-           nusimdata_SimulationBase
+           larcorealg::Geometry
+           lardataobj::RawData
+           lardataobj::RecoBase
+           nusimdata::SimulationBase
            art::Framework_Services_Registry
            ROOT::Core
            ROOT::RIO
