diff --git a/src/Physics/Charm/XSection/SlowRsclCharmDISPXSecLO.h b/src/Physics/Charm/XSection/SlowRsclCharmDISPXSecLO.h
index 989fbc39b..6a0e64c85 100644
--- a/src/Physics/Charm/XSection/SlowRsclCharmDISPXSecLO.h
+++ b/src/Physics/Charm/XSection/SlowRsclCharmDISPXSecLO.h
@@ -21,6 +21,7 @@
 #ifndef _SLOW_RESCALING_CHARM_DIS_PARTIAL_XSEC_LO_H_
 #define _SLOW_RESCALING_CHARM_DIS_PARTIAL_XSEC_LO_H_
 
+#include "Physics/XSectionIntegration/XSecIntegratorI.h"
 #include "Framework/EventGen/XSecAlgorithmI.h"
 
 namespace genie {
