--- CMakeLists.txt.orig	2020-02-25 14:32:04.000000000 -0600
+++ CMakeLists.txt	2020-02-25 14:35:11.000000000 -0600
@@ -16,13 +16,9 @@
 # use cmake 3.10 or later
 cmake_minimum_required (VERSION 3.10)
 
-project(icarusutil)
+project(icarusutil VERSION 8.43.00 LANGUAGE CXX)
 
-# cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
-IF (NOT CETBUILDTOOLS_VERSION)
-    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
-ENDIF()
+find_package(cetmodules REQURIED)
 
 set(CMAKE_MODULE_PATH $ENV{CANVAS_DIR}/Modules
 		      $ENV{CETBUILDTOOLS_DIR}/Modules
@@ -42,11 +38,8 @@
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( art )
-find_ups_product(art_root_io)
-find_ups_product( cetbuildtools )
-
-
+find_package( art )
+find_package( art_root_io)
 
 # macros for dictionary and simple_plugin
 include(ArtDictionary)
@@ -58,9 +51,6 @@
 # subdirectory for test code
 # add_subdirectory(test)
 
-# ups - table and config files
-add_subdirectory(ups)
-
 # XML project files.
 # add_subdirectory(xml)
 
@@ -92,4 +82,4 @@
 add_subdirectory(scripts)
 
 # packaging utility
-include(UseCPack)
+cet_cmake_config()
