diff --git a/google/cloud/bigtable/internal/logging_instance_admin_client_test.cc b/google/cloud/bigtable/internal/logging_instance_admin_client_test.cc
index c34e737..e944cf0 100644
--- a/google/cloud/bigtable/internal/logging_instance_admin_client_test.cc
+++ b/google/cloud/bigtable/internal/logging_instance_admin_client_test.cc
@@ -18,6 +18,7 @@
 #include "google/cloud/bigtable/testing/mock_response_reader.h"
 #include "google/cloud/testing_util/assert_ok.h"
 #include "google/cloud/testing_util/capture_log_lines_backend.h"
+#include "absl/memory/memory.h"
 #include <gmock/gmock.h>
 
 namespace google {
diff --git a/google/cloud/storage/tests/object_integration_test.cc b/google/cloud/storage/tests/object_integration_test.cc
index 43325d2..5734806 100644
--- a/google/cloud/storage/tests/object_integration_test.cc
+++ b/google/cloud/storage/tests/object_integration_test.cc
@@ -22,6 +22,7 @@
 #include "google/cloud/testing_util/contains_once.h"
 #include "google/cloud/testing_util/expect_exception.h"
 #include "google/cloud/testing_util/status_matchers.h"
+#include "absl/memory/memory.h"
 #include <gmock/gmock.h>
 #include <sys/types.h>
 #include <algorithm>
