diff --git a/CMakeLists.txt b/CMakeLists.txt
index d3698de..9fae3e4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,17 +21,18 @@ message(STATUS
   "\n-- === *** Start of configuration of icaruscode ***"
   "\n-- ===")
 
-project(icaruscode)
+project(icaruscode VERSION 08.43.00 LANGUAGES CXX)
 
+find_package(cetmodules)
 # cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
-IF (NOT CETBUILDTOOLS_VERSION)
-    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
-ENDIF()
-
-set(CMAKE_MODULE_PATH $ENV{CANVAS_DIR}/Modules
-		      $ENV{CETBUILDTOOLS_DIR}/Modules
-		      ${CMAKE_MODULE_PATH})
+#SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
+#IF (NOT CETBUILDTOOLS_VERSION)
+#    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
+#ENDIF()
+#
+#set(CMAKE_MODULE_PATH $ENV{CANVAS_DIR}/Modules
+#		      $ENV{CETBUILDTOOLS_DIR}/Modules
+#		      ${CMAKE_MODULE_PATH})
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -45,30 +46,38 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( larcore v1_00_00 )
-find_ups_product( larcoreobj )
-find_ups_product( lardata v1_00_00 )
-find_ups_product( lardataobj )
-find_ups_product( larevt v1_00_00 )
-find_ups_product( larsim v1_00_00 )
-find_ups_product( larreco v1_00_00 )
-find_ups_product( larpandora v1_00_00 )
-find_ups_product( larana v1_00_00 )
-find_ups_product( larpandora v1_00_00 )
-find_ups_product( nutools v1_00_00 )
-find_ups_product( nug4 )
-find_ups_product( nurandom )
-find_ups_product( art v1_09_00 )
-find_ups_product(art_root_io)
-find_ups_product( cetbuildtools v3_10_00 )
-find_ups_product( postgresql v9_1_5 )
-find_ups_product( eigen )
-find_ups_boost( v1_53_0 )
-find_ups_product( ifdh_art )
-find_ups_product( guideline_sl )
-
-find_ups_product( artdaq_core )
-find_ups_product( sbndaq_artdaq_core )
+find_package( larcore )
+find_package( larcoreobj )
+find_package( lardata )
+find_package( lardataobj )
+find_package( larevt )
+find_package( larsim )
+find_package( larreco )
+find_package( larpandora )
+find_package( larana )
+find_package( nutools )
+find_package( nug4 )
+find_package( nurandom )
+find_package( art )
+find_package(art_root_io)
+find_package(CLHEP REQUIRED COMPONENTS Matrix Vector Random)
+find_package( eigen )
+find_package( ifdh_art )
+find_package( guideline_sl )
+find_package( artdaq_core )
+find_package( sbndaq_artdaq_core )
+find_package( ROOT REQUIRED COMPONENTS Core Tree Geom XMLIO Gdml)
+get_target_property(RootIncludes ROOT::Tree INTERFACE_INCLUDE_DIRECTORIES)
+include_directories(${RootIncludes})
+find_package( Boost COMPONENTS boost unit_test_framework )
+get_target_property(BoostIncludes Boost::filesystem INTERFACE_INCLUDE_DIRECTORIES)
+include_directories(${BoostIncludes})
+find_package(TBB COMPONENTS tbb)
+get_target_property(TBBIncludes TBB::tbb INTERFACE_INCLUDE_DIRECTORIES)
+include_directories(${TBBIncludes})
+set(fcl_dir job)
+set(gdml_dir gdml)
+set(fw_dir fw)
 
-cet_find_library( PQ NAMES pq PATHS ENV POSTGRESQL_LIBRARIES NO_DEFAULT_PATH )
+cet_find_library( PQ NAMES pq )
 cet_find_library( NURANDOM_RANDOMUTILS_NURANDOMSERVICE_SERVICE 
@@ -90,10 +93,9 @@ add_subdirectory(test)
 add_subdirectory(fcl)
 
 # ups - table and config files
-add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
+cet_cmake_config()
 
 message(STATUS
   "\n-- ==="
diff --git a/icaruscode/Geometry/CMakeLists.txt b/icaruscode/Geometry/CMakeLists.txt
index 4c1ad54..1727256 100644
--- a/icaruscode/Geometry/CMakeLists.txt
+++ b/icaruscode/Geometry/CMakeLists.txt
@@ -9,7 +9,7 @@ art_make(
                         ${ART_FRAMEWORK_SERVICES_REGISTRY}
                         ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
                         ${ART_ROOT_IO_TFILESERVICE_SERVICE}
-			${BOOST_LIB}
+			Boost::boost
                         ${FHICLCPP}
 	  LIB_LIBRARIES larcorealg_Geometry
                         ${MF_MESSAGELOGGER}
diff --git a/icaruscode/gallery/galleryAnalysis/CMakeLists.txt b/icaruscode/gallery/galleryAnalysis/CMakeLists.txt
index bcad54a..e5f05ff 100644
--- a/icaruscode/gallery/galleryAnalysis/CMakeLists.txt
+++ b/icaruscode/gallery/galleryAnalysis/CMakeLists.txt
@@ -55,7 +55,8 @@ set(CMAKE_INSTALL_PREFIX ${CMAKE_CURRENT_SOURCE_DIR})
 #
 # include directories
 #
-include_directories($ENV{BOOST_INC})
+get_target_property(BoostIncludes Boost::filesystem INTERFACE_INCLUDE_DIRECTORIES)
+include_directories(${BoostIncludes})
 include_directories($ENV{ROOT_INC} $ENV{CLHEP_INC})
 include_directories($ENV{CETLIB_INC} $ENV{CETLIB_EXCEPT_INC} $ENV{CANVAS_INC} $ENV{CANVAS_ROOT_IO_INC})
 include_directories($ENV{FHICLCPP_INC} $ENV{MESSAGEFACILITY_INC})
diff --git a/icaruscode/gallery/CMakeLists.txt b/icaruscode/gallery/CMakeLists.txt
--- a/icaruscode/gallery/CMakeLists.txt
+++ b/icaruscode/gallery/CMakeLists.txt
@@ -4,14 +4,6 @@
 
 foreach(ExampleDir IN ITEMS galleryAnalysis )
   
-  # take *all* the files from the listed directory
-  file(GLOB_RECURSE GalleryExamples RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} ${ExampleDir}/* )
-  
-  # cet_install_files() will reproduce the intermediate subdirectories from the
-  # source list into the destination directory
-  cet_install_files(
-    LIST ${GalleryExamples}
-    DIRNAME examples/gallery/${ExampleDir}
-    )
+ install(DIRECTORY ${ExampleDir}  DESTINATION examples/gallery/${ExampleDir})
   
 endforeach()

diff --git a/icaruscode/RecoUtils/CMakeLists.txt b/icaruscode/RecoUtils/CMakeLists.txt
--- a/icaruscode/RecoUtils/CMakeLists.txt
+++ b/icaruscode/RecoUtils/CMakeLists.txt
@@ -15,7 +15,7 @@
                            ${ART_FRAMEWORK_CORE}
                            ${ART_FRAMEWORK_PRINCIPAL}
                            ${ART_FRAMEWORK_SERVICES_REGISTRY}
-                           ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+                           ${ART_ROOT_IO_TFILE_SUPPORT}
                            ${ART_ROOT_IO_TFILESERVICE_SERVICE}
                            art_Persistency_Common canvas
                            art_Persistency_Provenance canvas
@@ -23,10 +23,10 @@
                            ${MF_MESSAGELOGGER}
                            ${MF_UTILITIES}
                            ${FHICLCPP}
-                           ${ROOT_GEOM}
-                           ${ROOT_XMLIO}
-                           ${ROOT_GDML}
-                           ${ROOT_BASIC_LIB_LIST}
+                           Root::Geom
+                           Root::XMLIO
+                           Root::Gdml
+                           Root::Core
                            ${Boost_SYSTEM_LIBRARY}
         )
 
