diff --git a/CMakeLists.txt b/CMakeLists.txt
index d3698de..69980a4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,17 +21,18 @@
   "\n-- === *** Start of configuration of icaruscode ***"
   "\n-- ===")
 
-project(icaruscode)
+project(icaruscode VERSION 08.43.00 LANGUAGES CXX)
 
+find_package(cetmodules)
 # cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
-IF (NOT CETBUILDTOOLS_VERSION)
-    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
-ENDIF()
-
-set(CMAKE_MODULE_PATH $ENV{CANVAS_DIR}/Modules
-		      $ENV{CETBUILDTOOLS_DIR}/Modules
-		      ${CMAKE_MODULE_PATH})
+#SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
+#IF (NOT CETBUILDTOOLS_VERSION)
+#    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
+#ENDIF()
+#
+#set(CMAKE_MODULE_PATH $ENV{CANVAS_DIR}/Modules
+#		      $ENV{CETBUILDTOOLS_DIR}/Modules
+#		      ${CMAKE_MODULE_PATH})
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -45,40 +46,46 @@
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( larcore v1_00_00 )
-find_ups_product( larcoreobj )
-find_ups_product( lardata v1_00_00 )
-find_ups_product( lardataobj )
-find_ups_product( larevt v1_00_00 )
-find_ups_product( larsim v1_00_00 )
-find_ups_product( larreco v1_00_00 )
-find_ups_product( larpandora v1_00_00 )
-find_ups_product( larana v1_00_00 )
-find_ups_product( larpandora v1_00_00 )
-find_ups_product( nutools v1_00_00 )
-find_ups_product( nug4 )
-find_ups_product( nurandom )
-find_ups_product( art v1_09_00 )
-find_ups_product( art_root_io)
-find_ups_product( cetbuildtools v3_10_00 )
-find_ups_product( libwda )
-find_ups_product( postgresql v9_1_5 )
-find_ups_product( eigen )
-find_ups_product( fftw )
-find_ups_boost( v1_53_0 )
-find_ups_product( ifdh_art )
-find_ups_product( guideline_sl )
-find_ups_product( icarus_signal_processing )
-find_ups_product( icarusutil )
-find_ups_product( icarus_data )
+find_package( larcore )
+find_package( larcoreobj )
+find_package( lardata )
+find_package( lardataobj )
+find_package( larevt )
+find_package( larsim )
+find_package( larreco )
+find_package( larpandora )
+find_package( larana )
+find_package( nutools )
+find_package( nug4 )
+find_package( nurandom )
+find_package( art )
+find_package(art_root_io)
+find_package(CLHEP REQUIRED COMPONENTS Matrix Vector Random RandomObjects)
+find_package( Eigen3 )
+find_package( ifdh_art )
+find_path(GSL_INC NAMES gsl/span PATH_SUFFIXES include)
+message("=== got GSL_INC ${GSL_INC}")
+find_package( artdaq-core )
+find_package( sbndaq-artdaq-core )
+cet_find_simple_package(sqlite3)
+find_package( range-v3 REQUIRED COMPONENTS range-v3 meta)
+cet_find_simple_package( trace INCPATH_VAR TRACE_INC )
+find_package( ROOT REQUIRED COMPONENTS Core Tree Geom XMLIO Gdml RGL Physics EG Minuit Minuit2 GenVector MathCore)
+get_target_property(RootIncludes ROOT::Tree INTERFACE_INCLUDE_DIRECTORIES)
+include_directories(${RootIncludes})
+find_package( Boost COMPONENTS boost unit_test_framework )
+get_target_property(BoostIncludes Boost::filesystem INTERFACE_INCLUDE_DIRECTORIES)
+include_directories(${BoostIncludes})
+find_package(TBB COMPONENTS tbb)
+get_target_property(TBBIncludes TBB::tbb INTERFACE_INCLUDE_DIRECTORIES)
+include_directories(${TBBIncludes})
+set(fcl_dir job)
+set(gdml_dir gdml)
+set(fw_dir fw)
 
-find_ups_product( artdaq_core )
-find_ups_product( sbndaq_artdaq_core )
-
-cet_find_library( PQ NAMES pq PATHS ENV POSTGRESQL_LIBRARIES NO_DEFAULT_PATH )
+cet_find_library( PQ NAMES pq )
 cet_find_library( NURANDOM_RANDOMUTILS_NURANDOMSERVICE_SERVICE 
-                  NAMES nurandom_RandomUtils_NuRandomService_service
-		  PATHS ENV NURANDOM_LIB NO_DEFAULT_PATH )
+                  NAMES nurandom_RandomUtils_NuRandomService_service)
 
 # deal with fftw
 cet_find_library(ICARUS_FFTW3_LIBRARY  NAMES fftw3  PATHS $ENV{FFTW_LIBRARY} )
@@ -103,10 +110,9 @@
 add_subdirectory(fcl)
 
 # ups - table and config files
-add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
+cet_cmake_config()
 
 message(STATUS
   "\n-- ==="
diff --git a/icaruscode/Analysis/CMakeLists.txt b/icaruscode/Analysis/CMakeLists.txt
index e29358a..5b1238b 100644
--- a/icaruscode/Analysis/CMakeLists.txt
+++ b/icaruscode/Analysis/CMakeLists.txt
@@ -14,7 +14,7 @@ art_make( MODULE_LIBRARIES larcorealg_Geometry
                            ${ART_FRAMEWORK_CORE}
                            ${ART_FRAMEWORK_PRINCIPAL}
                            ${ART_FRAMEWORK_SERVICES_REGISTRY}
-                           ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+                           ${ART_ROOT_IO_TFILE_SUPPORT} ROOT::Core
                            ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
                            ${ART_ROOT_IO_TFILESERVICE_SERVICE}
                            art_Persistency_Common
@@ -27,10 +27,13 @@ art_make( MODULE_LIBRARIES larcorealg_Geometry
                            ${CETLIB}
                            cetlib_except
                            ${CLHEP}
-                           ${ROOT_GEOM}
-                           ${ROOT_XMLIO}
-                           ${ROOT_GDML}
-                           ${ROOT_BASIC_LIB_LIST}
+                           Eigen3::Eigen
+                           ROOT::Geom
+                           ROOT::XMLIO
+                           ROOT::Gdml
+                           ROOT::RGL
+                           ROOT::Physics
+                           ROOT::Core
                            icaruscode_RecoUtils
         )
 
diff --git a/icaruscode/Analysis/tools/CMakeLists.txt b/icaruscode/Analysis/tools/CMakeLists.txt
index 88080f6..0e215e8 100644
--- a/icaruscode/Analysis/tools/CMakeLists.txt
+++ b/icaruscode/Analysis/tools/CMakeLists.txt
@@ -15,7 +15,7 @@ art_make( TOOL_LIBRARIES lardataobj_RecoBase
                          ${ART_FRAMEWORK_PRINCIPAL}
                          ${ART_FRAMEWORK_SERVICES_REGISTRY}
                          ${ART_FRAMEWORK_SERVICES_BASIC}
-                         ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+                         ${ART_ROOT_IO_TFILE_SUPPORT} ROOT::Core
                          ${ART_ROOT_IO_TFILESERVICE_SERVICE}
                          ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
                          art_Persistency_Common
@@ -30,8 +30,16 @@ art_make( TOOL_LIBRARIES lardataobj_RecoBase
                          ${Boost_FILESYSTEM_LIBRARY}
                          ${Boost_SYSTEM_LIBRARY}
                          ${CLHEP}
-                         ${ROOT_BASIC_LIB_LIST}
-                         ${ROOT_MINUIT}
+			 ROOT::ROOTDataFrame
+			 ROOT::Physics
+			 ROOT::Geom
+			 ROOT::FFTW
+			 ROOT::XMLIO
+			 ROOT::Gdml
+			 ROOT::Tree
+			 ROOT::Core
+
+                         ROOT::Minuit
         )
 
 install_headers()
diff --git a/icaruscode/Analysis/tools/MCTruth/CMakeLists.txt b/icaruscode/Analysis/tools/MCTruth/CMakeLists.txt
index 5a2cb50..b63ccc0 100644
--- a/icaruscode/Analysis/tools/MCTruth/CMakeLists.txt
+++ b/icaruscode/Analysis/tools/MCTruth/CMakeLists.txt
@@ -13,7 +13,7 @@ art_make( MODULE_LIBRARIES lardataobj_RecoBase
                            ${ART_FRAMEWORK_PRINCIPAL}
                            ${ART_FRAMEWORK_SERVICES_REGISTRY}
                            ${ART_FRAMEWORK_SERVICES_BASIC}
-                           ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+                           ${ART_ROOT_IO_TFILE_SUPPORT} ROOT::Core
                            ${ART_ROOT_IO_TFILESERVICE_SERVICE}
                            ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
                            art_Persistency_Common
@@ -25,10 +25,13 @@ art_make( MODULE_LIBRARIES lardataobj_RecoBase
                            ${FHICLCPP}
                            ${CETLIB}
                            cetlib_except
-                           ${Boost_FILESYSTEM_LIBRARY}
-                           ${Boost_SYSTEM_LIBRARY}
                            ${CLHEP}
-                           ${ROOT_BASIC_LIB_LIST}
+                           ROOT::ROOTDataFrame
+                           ROOT::Physics
+                           ROOT::Geom
+                           ROOT::FFTW
+                           ROOT::Tree
+                           ROOT::Core
 
           TOOL_LIBRARIES   lardataobj_RecoBase
                            larcorealg_Geometry
@@ -43,7 +46,7 @@ art_make( MODULE_LIBRARIES lardataobj_RecoBase
                            ${ART_FRAMEWORK_PRINCIPAL}
                            ${ART_FRAMEWORK_SERVICES_REGISTRY}
                            ${ART_FRAMEWORK_SERVICES_BASIC}
-                           ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+                           ${ART_ROOT_IO_TFILE_SUPPORT} ROOT::Core
                            ${ART_ROOT_IO_TFILESERVICE_SERVICE}
                            ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
                            art_Persistency_Common
@@ -55,10 +58,14 @@ art_make( MODULE_LIBRARIES lardataobj_RecoBase
                            ${FHICLCPP}
                            ${CETLIB}
                            cetlib_except
-                           ${Boost_FILESYSTEM_LIBRARY}
-                           ${Boost_SYSTEM_LIBRARY}
                            ${CLHEP}
-                           ${ROOT_BASIC_LIB_LIST}
+
+                           ROOT::ROOTDataFrame
+                           ROOT::Physics
+                           ROOT::Geom
+                           ROOT::FFTW
+                           ROOT::Tree
+                           ROOT::Core
         )
 
 install_headers()
diff --git a/icaruscode/CRT/CMakeLists.txt b/icaruscode/CRT/CMakeLists.txt
index 47bde23..5feb0f3 100644
--- a/icaruscode/CRT/CMakeLists.txt
+++ b/icaruscode/CRT/CMakeLists.txt
@@ -28,8 +28,12 @@ art_make(
         ${MF_MESSAGELOGGER}
         ${MF_UTILITIES}
         ${FHICLCPP}
-        ${ROOT_BASIC_LIB_LIST}
-        ${ROOT_GEOM}
+        ROOT::ROOTDataFrame
+	ROOT::Physics
+	ROOT::Geom
+	ROOT::FFTW
+	ROOT::Tree
+	ROOT::Core
         ${CETLIB}
 )
 
@@ -44,8 +48,12 @@ art_make_library(
         ${MF_MESSAGELOGGER}
         ${MF_UTILITIES}
         ${FHICLCPP}
-        ${ROOT_BASIC_LIB_LIST}
-        ${ROOT_GEOM}
+        ROOT::ROOTDataFrame
+	ROOT::Physics
+	ROOT::Geom
+	ROOT::FFTW
+	ROOT::Tree
+	ROOT::Core
         ${CETLIB}
 )
 
@@ -61,7 +69,12 @@ simple_plugin( CRTGeometryHelper service
         ${ART_FRAMEWORK_SERVICES_REGISTRY}
         ${MF_MESSAGELOGGER}
         ${MF_UTILITIES}
-        ${ROOT_BASIC_LIB_LIST}
+        ROOT::ROOTDataFrame
+	ROOT::Physics
+	ROOT::Geom
+	ROOT::FFTW
+	ROOT::Tree
+	ROOT::Core
         ${CETLIB}
 )
 
@@ -83,8 +96,14 @@ simple_plugin( CRTDetSim module
         ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
         ${MF_MESSAGELOGGER}
         ${MF_UTILITIES}
-        ${ROOT_BASIC_LIB_LIST}
+        ROOT::ROOTDataFrame
+	ROOT::Physics
+	ROOT::Geom
+	ROOT::FFTW
+	ROOT::Tree
+	ROOT::Core
         ${CLHEP}
+        CLHEP::RandomObjects
         ${CETLIB}
 )
 
@@ -107,8 +126,14 @@ simple_plugin( CRTSimHitProducer module
         ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
         ${MF_MESSAGELOGGER}
         ${MF_UTILITIES}
-        ${ROOT_BASIC_LIB_LIST}
+        ROOT::ROOTDataFrame
+	ROOT::Physics
+	ROOT::Geom
+	ROOT::FFTW
+	ROOT::Tree
+	ROOT::Core
         ${CLHEP}
+        CLHEP::RandomObjects
         ${CETLIB}
 )
 
@@ -131,8 +156,14 @@ simple_plugin(CRTTzeroProducer module
         ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
         ${MF_MESSAGELOGGER}
         ${MF_UTILITIES}
-        ${ROOT_BASIC_LIB_LIST}
+        ROOT::ROOTDataFrame
+	ROOT::Physics
+	ROOT::Geom
+	ROOT::FFTW
+	ROOT::Tree
+	ROOT::Core
         ${CLHEP}
+        CLHEP::RandomObjects
         ${CETLIB}
 )
 
@@ -155,8 +186,14 @@ simple_plugin(CRTTrackProducer module
         ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
         ${MF_MESSAGELOGGER}
         ${MF_UTILITIES}
-        ${ROOT_BASIC_LIB_LIST}
+        ROOT::ROOTDataFrame
+	ROOT::Physics
+	ROOT::Geom
+	ROOT::FFTW
+	ROOT::Tree
+	ROOT::Core
         ${CLHEP}
+        CLHEP::RandomObjects
         ${CETLIB}
 )
 
@@ -179,11 +216,14 @@ simple_plugin( CRTAnalysis module
                         ${MF_MESSAGELOGGER}
                         ${MF_UTILITIES}
                         cetlib cetlib_except
-                        ${ROOT_BASIC_LIB_LIST}
-                        ${ROOT_GEOM}
-                        ${ROOT_XMLIO}
-                        ${ROOT_GDML}              
-
+			ROOT::ROOTDataFrame
+			ROOT::Physics
+			ROOT::Geom
+			ROOT::FFTW
+			ROOT::Tree
+                        ROOT::XMLIO
+                        ROOT::Gdml              
+			ROOT::Core
 )
 
 
diff --git a/icaruscode/CRT/CRTUtils/CMakeLists.txt b/icaruscode/CRT/CRTUtils/CMakeLists.txt
index 8273e7a..ea26dd6 100644
--- a/icaruscode/CRT/CRTUtils/CMakeLists.txt
+++ b/icaruscode/CRT/CRTUtils/CMakeLists.txt
@@ -24,11 +24,15 @@ art_make(    LIBRARY_NAME icaruscode_CRTUtils
                            ${MF_MESSAGELOGGER}
                            
                            ${FHICLCPP}
-                           ${ROOT_GEOM}
-                           ${ROOT_XMLIO}
-                           ${ROOT_GDML}
-                           ${ROOT_BASIC_LIB_LIST}
-                           ${Boost_SYSTEM_LIBRARY}
+                           ROOT::ROOTDataFrame
+                           ROOT::Physics
+                           ROOT::Geom
+                           ROOT::FFTW
+                           ROOT::XMLIO
+                           ROOT::Gdml
+                           ROOT::Tree
+                           ROOT::Core
+                           Boost::system
                            icaruscode_CRT
                            icaruscode_CRTProducts
                            icaruscode_CRTData
diff --git a/icaruscode/Filters/CMakeLists.txt b/icaruscode/Filters/CMakeLists.txt
index 6bdf003..0018173 100644
--- a/icaruscode/Filters/CMakeLists.txt
+++ b/icaruscode/Filters/CMakeLists.txt
@@ -14,7 +14,7 @@ art_make(
                         ${ART_FRAMEWORK_CORE}
                         ${ART_FRAMEWORK_PRINCIPAL}
                         ${ART_FRAMEWORK_SERVICES_REGISTRY}
-                        ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+                        ${ART_ROOT_IO_TFILE_SUPPORT} ROOT::Core
                         ${ART_ROOT_IO_TFILESERVICE_SERVICE}
                         art_Persistency_Common 
                         art_Persistency_Provenance 
@@ -23,10 +23,12 @@ art_make(
                         ${MF_UTILITIES}
                         ${FHICLCPP}
                         ${CETLIB}
-                        ${ROOT_GEOM}
-                        ${ROOT_XMLIO}
-                        ${ROOT_GDML}
-                        ${ROOT_BASIC_LIB_LIST}
+                        ROOT::ROOTDataFrame
+                        ROOT::Physics
+                        ROOT::Geom
+                        ROOT::FFTW
+                        ROOT::Tree
+                        ROOT::Core
         )
 
 # install_headers()
diff --git a/icaruscode/Generators/CMakeLists.txt b/icaruscode/Generators/CMakeLists.txt
index 1e6e573..b847eba 100644
--- a/icaruscode/Generators/CMakeLists.txt
+++ b/icaruscode/Generators/CMakeLists.txt
@@ -21,7 +21,7 @@ art_make(
             ${ART_FRAMEWORK_CORE}
             ${ART_FRAMEWORK_PRINCIPAL}
             ${ART_FRAMEWORK_SERVICES_REGISTRY}
-            ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+            ${ART_ROOT_IO_TFILE_SUPPORT} ROOT::Core
             ${ART_ROOT_IO_TFILESERVICE_SERVICE}
             art_Persistency_Common
             art_Persistency_Provenance
@@ -31,11 +31,19 @@ art_make(
             ${MF_UTILITIES}
             ${FHICLCPP}
             ${CETLIB}
-            ${ROOT_FFTW}
-            ${ROOT_BASIC_LIB_LIST}
-	    ${ROOT_EG}
+	ROOT::ROOTDataFrame
+	ROOT::Physics
+	ROOT::Geom
+	ROOT::FFTW
+	ROOT::XMLIO
+	ROOT::Gdml
+	ROOT::Tree
+	ROOT::Core
+	ROOT::EG
+        CLHEP::RandomObjects
         )
 
+
 install_headers()
 install_fhicl()
 install_source()
diff --git a/icaruscode/Geometry/CMakeLists.txt b/icaruscode/Geometry/CMakeLists.txt
index 4c1ad54..a4a43aa 100644
--- a/icaruscode/Geometry/CMakeLists.txt
+++ b/icaruscode/Geometry/CMakeLists.txt
@@ -7,27 +7,35 @@ art_make(
 			${ART_FRAMEWORK_MODULES}
                         ${ART_FRAMEWORK_PRINCIPAL}
                         ${ART_FRAMEWORK_SERVICES_REGISTRY}
-                        ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+                        ${ART_ROOT_IO_TFILE_SUPPORT} ROOT::Core
                         ${ART_ROOT_IO_TFILESERVICE_SERVICE}
-			${BOOST_LIB}
+			Boost::boost
                         ${FHICLCPP}
 	  LIB_LIBRARIES larcorealg_Geometry
                         ${MF_MESSAGELOGGER}
                         ${CETLIB}
 			${CETLIB_EXCEPT}
                         ${FHICLCPP}
-                        ${ROOT_GEOM}
-                        ${ROOT_BASIC_LIB_LIST}
+			ROOT::ROOTDataFrame
+			ROOT::Physics
+			ROOT::Geom
+			ROOT::FFTW
+			ROOT::Tree
+			ROOT::Core
 	  SERVICE_LIBRARIES icaruscode_Geometry
 	  		icaruscode_Geometry
 			${MF_UTILITIES}
 	                ${FHICLCPP}
 			${CETLIB}
 			${CETLIB_EXCEPT}
-                        ${ROOT_GEOM}
-                        ${ROOT_XMLIO}
-                        ${ROOT_GDML}
-			${ROOT_BASIC_LIB_LIST}
+			ROOT::ROOTDataFrame
+			ROOT::Physics
+			ROOT::Geom
+			ROOT::XMLIO
+			ROOT::Gdml
+			ROOT::FFTW
+			ROOT::Tree
+			ROOT::Core
         )							   
 
 
diff --git a/icaruscode/PMT/Algorithms/CMakeLists.txt b/icaruscode/PMT/Algorithms/CMakeLists.txt
index ed3e933..f07cd92 100644
--- a/icaruscode/PMT/Algorithms/CMakeLists.txt
+++ b/icaruscode/PMT/Algorithms/CMakeLists.txt
@@ -16,9 +16,15 @@ art_make(
      ${FHICLCPP}
      ${CETLIB}
      cetlib_except
-     ${ROOT_FFTW}
-     ${ROOT_GENVECTOR}
-     ${ROOT_BASIC_LIB_LIST}
+	ROOT::ROOTDataFrame
+	ROOT::Physics
+	ROOT::Geom
+	ROOT::GenVector
+	ROOT::FFTW
+	ROOT::Tree
+	ROOT::Core
+        CLHEP::Random
+	CLHEP::RandomObjects
      ${CLHEP}
 #    ${Boost_SYSTEM_LIBRARY}
   )
diff --git a/icaruscode/PMT/CMakeLists.txt b/icaruscode/PMT/CMakeLists.txt
index 8fd7f09..a0f00f5 100644
--- a/icaruscode/PMT/CMakeLists.txt
+++ b/icaruscode/PMT/CMakeLists.txt
@@ -25,8 +25,13 @@ simple_plugin(SimPMTIcarus "module"
   ${FHICLCPP}
   ${CETLIB}
   cetlib_except
-  ${ROOT_BASIC_LIB_LIST}
-  ${Boost_SYSTEM_LIBRARY}
+  ROOT::ROOTDataFrame
+  ROOT::Physics
+  ROOT::Geom
+  ROOT::FFTW
+  ROOT::Tree
+  ROOT::Core
+  Boost::system
   )
 
 
@@ -38,8 +43,13 @@ art_make(
             lardata_Utilities
             ${FHICLCPP}
 			${CETLIB}
-			${ROOT_BASIC_LIB_LIST}
-			${Boost_SYSTEM_LIBRARY}
+			ROOT::ROOTDataFrame
+			ROOT::Physics
+			ROOT::Geom
+			ROOT::FFTW
+			ROOT::Tree
+			ROOT::Core
+			Boost::system
 	        MODULE_LIBRARIES larcorealg_Geometry
 	  		larcore_Geometry_Geometry_service
 	        lardata_Utilities
@@ -65,12 +75,16 @@ art_make(
 	        ${FHICLCPP}
 			${CETLIB}
 			cetlib_except
-            ${ROOT_GEOM}
-            ${ROOT_XMLIO}
-            ${ROOT_GDML}
-			${ROOT_FFTW}
-			${ROOT_BASIC_LIB_LIST}
-			${Boost_SYSTEM_LIBRARY}
+			ROOT::ROOTDataFrame
+			ROOT::Physics
+			ROOT::Geom
+			ROOT::XMLIO
+			ROOT::FFTW
+			ROOT::Tree
+			ROOT::Core 
+			Boost::system
+			CLHEP::Random
+			CLHEP::RandomObjects
         )
 
 install_headers()
diff --git a/icaruscode/PMT/LibraryMappingTools/CMakeLists.txt b/icaruscode/PMT/LibraryMappingTools/CMakeLists.txt
index 5d88d11..8760ea2 100644
--- a/icaruscode/PMT/LibraryMappingTools/CMakeLists.txt
+++ b/icaruscode/PMT/LibraryMappingTools/CMakeLists.txt
@@ -10,9 +10,14 @@ art_make_library(
     ${CETLIB_EXCEPT}
     ${MF_MESSAGELOGGER}
     ${FHICLCPP}
-    ${ROOT_GENVECTOR}
-    ${ROOT_BASIC_LIB_LIST}
-    ${Boost_SYSTEM_LIBRARY}
+    ROOT::ROOTDataFrame
+    ROOT::Physics
+    ROOT::Geom
+    ROOT::GenVector
+    ROOT::FFTW
+    ROOT::Tree
+    ROOT::Core
+    Boost::system
   )
 
 simple_plugin(ICARUSPhotonMappingTransformations "tool"
diff --git a/icaruscode/PMT/OpReco/CMakeLists.txt b/icaruscode/PMT/OpReco/CMakeLists.txt
index 53a8c0c..9caf540 100644
--- a/icaruscode/PMT/OpReco/CMakeLists.txt
+++ b/icaruscode/PMT/OpReco/CMakeLists.txt
@@ -12,7 +12,7 @@ art_make(
                         ${ART_FRAMEWORK_CORE}
 			${ART_FRAMEWORK_PRINCIPAL}
 			${ART_FRAMEWORK_SERVICES_REGISTRY}
-			${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+			${ART_ROOT_IO_TFILE_SUPPORT} ROOT::Core
 			${ART_ROOT_IO_TFILESERVICE_SERVICE}
 			art_Persistency_Common
 			art_Persistency_Provenance
@@ -23,11 +23,15 @@ art_make(
 	                ${FHICLCPP}
 			${CETLIB}
 			${CLHEP}
-                        ${ROOT_GEOM}
-                        ${ROOT_XMLIO}
-                        ${ROOT_GDML}
-			${ROOT_FFTW}
-			${ROOT_BASIC_LIB_LIST}
+		     CLHEP::Random
+                        ROOT::Geom
+                        ROOT::XMLIO
+                        ROOT::Gdml
+			ROOT::FFTW
+			ROOT::ROOTDataFrame
+			ROOT::Physics
+			ROOT::Tree
+			ROOT::Core
         )
 
 install_headers()
diff --git a/icaruscode/PMT/OpReco/FlashFinder/CMakeLists.txt b/icaruscode/PMT/OpReco/FlashFinder/CMakeLists.txt
index efd0c73..20516d2 100644
--- a/icaruscode/PMT/OpReco/FlashFinder/CMakeLists.txt
+++ b/icaruscode/PMT/OpReco/FlashFinder/CMakeLists.txt
@@ -15,7 +15,7 @@ art_make(
 		   ${ART_FRAMEWORK_UTILITIES}
                    ${ART_FRAMEWORK_BASIC}
 		   ${ART_FRAMEWORK_SERVICES_REGISTRY}
-		   ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+		   ${ART_ROOT_IO_TFILE_SUPPORT} ROOT::Core
 		   ${ART_ROOT_IO_TFILESERVICE_SERVICE}
 		   art_Persistency_Common
 		   art_Persistency_Provenance
@@ -23,10 +23,14 @@ art_make(
 		   ${MF_MESSAGELOGGER}
 		   ${MF_UTILITIES}
 		   ${FHICLCPP}
-		   ${ROOT_GEOM}
-		   ${ROOT_XMLIO}
-		   ${ROOT_GDML}
-		   ${ROOT_BASIC_LIB_LIST}
+			ROOT::ROOTDataFrame
+			ROOT::Physics
+			ROOT::Geom
+			ROOT::XMLIO
+			ROOT::Gdml
+			ROOT::FFTW
+			ROOT::Tree
+			ROOT::Core
           MODULE_LIBRARIES
 		   icaruscode_PMT_OpReco_FlashFinder
 		   icaruscode_Geometry
@@ -41,7 +45,7 @@ art_make(
 		   ${ART_FRAMEWORK_PRINCIPAL}
                    ${ART_FRAMEWORK_BASIC}
 		   ${ART_FRAMEWORK_SERVICES_REGISTRY}
-		   ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+		   ${ART_ROOT_IO_TFILE_SUPPORT} ROOT::Core
 		   ${ART_ROOT_IO_TFILESERVICE_SERVICE}
 		   art_Persistency_Common
 		   art_Persistency_Provenance
@@ -49,10 +53,14 @@ art_make(
 		   ${MF_MESSAGELOGGER}
 		   ${MF_UTILITIES}
 		   ${FHICLCPP}
-		   ${ROOT_GEOM}
-		   ${ROOT_XMLIO}
-		   ${ROOT_GDML}
-		   ${ROOT_BASIC_LIB_LIST}
+		   ROOT::ROOTDataFrame
+		   ROOT::Physics
+		   ROOT::Geom
+		   ROOT::XMLIO
+		   ROOT::Gdml
+		   ROOT::FFTW
+		   ROOT::Tree
+		   ROOT::Core
 
         )							   
 
diff --git a/icaruscode/PMT/OpticalTools/CMakeLists.txt b/icaruscode/PMT/OpticalTools/CMakeLists.txt
index ff9ffff..a586d7b 100644
--- a/icaruscode/PMT/OpticalTools/CMakeLists.txt
+++ b/icaruscode/PMT/OpticalTools/CMakeLists.txt
@@ -16,7 +16,7 @@ art_make(
 			              art_Persistency_Common
 			              art_Persistency_Provenance
 			              ${ART_FRAMEWORK_SERVICES_BASIC}
-			              ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+			              ${ART_ROOT_IO_TFILE_SUPPORT} ROOT::Core
                           ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
 			              ${ART_ROOT_IO_TFILESERVICE_SERVICE}
 			              ${ART_FRAMEWORK_SERVICES_SYSTEM_FILECATALOGMETADATA_SERVICE}
@@ -30,7 +30,13 @@ art_make(
 			              ${Boost_FILESYSTEM_LIBRARY}
 			              ${Boost_SYSTEM_LIBRARY}
                           ${CLHEP}
-			              ${ROOT_BASIC_LIB_LIST}
+				      ROOT::ROOTDataFrame
+				      ROOT::Physics
+				      ROOT::Geom
+				      ROOT::FFTW
+				      ROOT::Tree
+				      ROOT::Core
+
     )
 
 install_headers()
diff --git a/icaruscode/PMT/Trigger/Algorithms/CMakeLists.txt b/icaruscode/PMT/Trigger/Algorithms/CMakeLists.txt
index c0da0ad..ecfd8d8 100644
--- a/icaruscode/PMT/Trigger/Algorithms/CMakeLists.txt
+++ b/icaruscode/PMT/Trigger/Algorithms/CMakeLists.txt
@@ -6,6 +6,7 @@ cet_make(
     lardataobj_RawData
     MF_MessageLogger
     fhiclcpp
+    range-v3
   )
 
 install_headers()
diff --git a/icaruscode/PMT/Trigger/CMakeLists.txt b/icaruscode/PMT/Trigger/CMakeLists.txt
index eea20a1..3b46f58 100644
--- a/icaruscode/PMT/Trigger/CMakeLists.txt
+++ b/icaruscode/PMT/Trigger/CMakeLists.txt
@@ -12,8 +12,12 @@ art_make(
     ${ART_FRAMEWORK_SERVICES_REGISTRY}
     ${MF_MESSAGELOGGER}
     ${FHICLCPP}
-    ${ROOT_CORE}
-    ${ROOT_PHYSICS}
+    ROOT::ROOTDataFrame
+    ROOT::Physics
+    ROOT::Geom
+    ROOT::FFTW
+    ROOT::Tree
+    ROOT::Core
   TOOL_LIBRARIES
     icaruscode_PMT_Trigger_Algorithms
     icaruscode_PMT_Trigger_Data
@@ -41,10 +45,13 @@ simple_plugin(TriggerEfficiencyPlots module
   ${MF_MESSAGELOGGER}
   ${FHICLCPP}
   ${CETLIB_EXCEPT}
-  ${ROOT_GENVECTOR}
-  ${ROOT_HIST}
-  ${ROOT_TREE}
-  ${ROOT_CORE}
+  ROOT::ROOTDataFrame
+  ROOT::Hist
+  ROOT::GenVector
+  ROOT::FFTW
+  ROOT::Tree
+  ROOT::Core
+  range-v3
   )
 
 install_headers()
diff --git a/icaruscode/RecoUtils/CMakeLists.txt b/icaruscode/RecoUtils/CMakeLists.txt
index aa70ef5..73d7770 100644
--- a/icaruscode/RecoUtils/CMakeLists.txt
+++ b/icaruscode/RecoUtils/CMakeLists.txt
@@ -15,7 +15,7 @@ art_make(
                            ${ART_FRAMEWORK_CORE}
                            ${ART_FRAMEWORK_PRINCIPAL}
                            ${ART_FRAMEWORK_SERVICES_REGISTRY}
-                           ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+                           ${ART_ROOT_IO_TFILE_SUPPORT}
                            ${ART_ROOT_IO_TFILESERVICE_SERVICE}
                            art_Persistency_Common canvas
                            art_Persistency_Provenance canvas
@@ -23,11 +23,11 @@ art_make(
                            ${MF_MESSAGELOGGER}
                            ${MF_UTILITIES}
                            ${FHICLCPP}
-                           ${ROOT_GEOM}
-                           ${ROOT_XMLIO}
-                           ${ROOT_GDML}
-                           ${ROOT_BASIC_LIB_LIST}
-                           ${Boost_SYSTEM_LIBRARY}
+                           ROOT::Geom
+                           ROOT::XMLIO
+                           ROOT::Gdml
+                           ROOT::Core
+                           Boost::system
         )
 
 install_headers()
diff --git a/icaruscode/TPC/SignalProcessing/HitFinder/CMakeLists.txt b/icaruscode/TPC/SignalProcessing/HitFinder/CMakeLists.txt
index c74f002..3ac7e1b 100644
--- a/icaruscode/TPC/SignalProcessing/HitFinder/CMakeLists.txt
+++ b/icaruscode/TPC/SignalProcessing/HitFinder/CMakeLists.txt
@@ -7,7 +7,12 @@ art_make(
 	                   lardata_Utilities
 	                   ${FHICLCPP}
 			           ${CETLIB}
-			           ${ROOT_BASIC_LIB_LIST}
+				   ROOT::ROOTDataFrame
+				   ROOT::Physics
+				   ROOT::Geom
+				   ROOT::FFTW
+				   ROOT::Tree
+				   ROOT::Core
 	  MODULE_LIBRARIES larcorealg_Geometry
 	  		           larcore_Geometry_Geometry_service
 	                   lardata_Utilities
@@ -22,7 +27,7 @@ art_make(
                        ${ART_FRAMEWORK_CORE}
 			           ${ART_FRAMEWORK_PRINCIPAL}
 			           ${ART_FRAMEWORK_SERVICES_REGISTRY}
-			           ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+			           ${ART_ROOT_IO_TFILE_SUPPORT}
 			           ${ART_ROOT_IO_TFILESERVICE_SERVICE}
 			           art_Persistency_Common
 			           art_Persistency_Provenance
@@ -32,11 +37,15 @@ art_make(
 			           ${MF_UTILITIES}
 	                   ${FHICLCPP}
 			           ${CETLIB}
-                       ${ROOT_GEOM}
-                       ${ROOT_XMLIO}
-                       ${ROOT_GDML}
-			           ${ROOT_FFTW}
-			           ${ROOT_BASIC_LIB_LIST}
+		       ROOT::ROOTDataFrame
+		       ROOT::Gdml
+		       ROOT::XMLIO
+		       ROOT::Physics
+		       ROOT::Geom
+		       ROOT::FFTW
+		       ROOT::Tree
+		       ROOT::Core
+
         )
 
 install_headers()
diff --git a/icaruscode/TPC/SignalProcessing/HitFinder/HitFinderTools/CMakeLists.txt b/icaruscode/TPC/SignalProcessing/HitFinder/HitFinderTools/CMakeLists.txt
index 1601ea8..6fe8452 100644
--- a/icaruscode/TPC/SignalProcessing/HitFinder/HitFinderTools/CMakeLists.txt
+++ b/icaruscode/TPC/SignalProcessing/HitFinder/HitFinderTools/CMakeLists.txt
@@ -12,7 +12,7 @@ set( hitfinder_tool_lib_list
 			art_Persistency_Common
 			art_Persistency_Provenance
 			${ART_FRAMEWORK_SERVICES_BASIC}
-			${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+			${ART_ROOT_IO_TFILE_SUPPORT} 
             ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
 			${ART_ROOT_IO_TFILESERVICE_SERVICE}
 			${ART_FRAMEWORK_SERVICES_SYSTEM_FILECATALOGMETADATA_SERVICE}
@@ -22,10 +22,16 @@ set( hitfinder_tool_lib_list
 			${MF_UTILITIES}
 			${FHICLCPP}
 			${CETLIB}
-			${Boost_FILESYSTEM_LIBRARY}
-			${Boost_SYSTEM_LIBRARY}
+			Boost::filesystem
+			Boost::system
             ${CLHEP}
-			${ROOT_BASIC_LIB_LIST}
+			ROOT::ROOTDataFrame
+			ROOT::Physics
+			ROOT::Geom
+			ROOT::FFTW
+			ROOT::Tree
+			ROOT::Core
+
     )
 
 art_make(TOOL_LIBRARIES ${hitfinder_tool_lib_list}
diff --git a/icaruscode/TPC/SignalProcessing/RecoWire/SimTestPulse/CMakeLists.txt b/icaruscode/TPC/SignalProcessing/RecoWire/SimTestPulse/CMakeLists.txt
index 50fee5d..582bf47 100644
--- a/icaruscode/TPC/SignalProcessing/RecoWire/SimTestPulse/CMakeLists.txt
+++ b/icaruscode/TPC/SignalProcessing/RecoWire/SimTestPulse/CMakeLists.txt
@@ -6,7 +6,12 @@ art_make(   LIB_LIBRARIES
                 lardata_Utilities
                 ${FHICLCPP}
                 ${CETLIB}
-                ${ROOT_BASIC_LIB_LIST}
+		ROOT::ROOTDataFrame
+		ROOT::Physics
+		ROOT::Geom
+		ROOT::FFTW
+		ROOT::Tree
+		ROOT::Core
             MODULE_LIBRARIES
                 icaruscode_TPC_SignalProcessing_RecoWire_SimTestPulse
                 icaruscode_Utilities_SignalShapingServiceICARUS_service
@@ -33,7 +38,7 @@ art_make(   LIB_LIBRARIES
                 ${ART_FRAMEWORK_CORE}
                 ${ART_FRAMEWORK_PRINCIPAL}
                 ${ART_FRAMEWORK_SERVICES_REGISTRY}
-                ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+                ${ART_ROOT_IO_TFILE_SUPPORT} 
                 ${ART_ROOT_IO_TFILESERVICE_SERVICE}
                 art_Persistency_Common
                 art_Persistency_Provenance
@@ -43,8 +48,12 @@ art_make(   LIB_LIBRARIES
                 ${MF_UTILITIES}
                 ${FHICLCPP}
                 ${CETLIB}
-                ${ROOT_FFTW}
-                ${ROOT_BASIC_LIB_LIST}
+		ROOT::ROOTDataFrame
+		ROOT::Physics
+		ROOT::Geom
+		ROOT::FFTW
+		ROOT::Tree
+		ROOT::Core
 )
 
 install_fhicl()
diff --git a/icaruscode/TPC/Simulation/DetSim/CMakeLists.txt b/icaruscode/TPC/Simulation/DetSim/CMakeLists.txt
index 3a1b625..64a8722 100644
--- a/icaruscode/TPC/Simulation/DetSim/CMakeLists.txt
+++ b/icaruscode/TPC/Simulation/DetSim/CMakeLists.txt
@@ -16,7 +16,7 @@ art_make(
                            ${ART_FRAMEWORK_CORE}
                            ${ART_FRAMEWORK_PRINCIPAL}
                            ${ART_FRAMEWORK_SERVICES_REGISTRY}
-                           ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+                           ${ART_ROOT_IO_TFILE_SUPPORT}
                            ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
                            ${ART_ROOT_IO_TFILESERVICE_SERVICE}
                            art_Persistency_Common
@@ -28,12 +28,18 @@ art_make(
                            ${FHICLCPP}
                            ${CETLIB}
                            ${CLHEP}
-                           ${ROOT_GEOM}
-                           ${ROOT_XMLIO}
-                           ${ROOT_GDML}
-                           ${ROOT_BASIC_LIB_LIST}
+                           CLHEP::Random
+                           ROOT::ROOTDataFrame
+                           ROOT::Physics
+                           ROOT::Gdml
+                           ROOT::Geom
+                           ROOT::FFTW
+                           ROOT::Tree
+                           ROOT::Core
+
         )
 
+
 simple_plugin(SimWireICARUS "module"
                            larcorealg_Geometry
                            larcore_Geometry_Geometry_service
@@ -61,10 +67,15 @@ simple_plugin(SimWireICARUS "module"
                            ${FHICLCPP}
                            ${CETLIB}
                            ${CLHEP}
-                           ${ROOT_GEOM}
-                           ${ROOT_XMLIO}
-                           ${ROOT_GDML}
-                           ${ROOT_BASIC_LIB_LIST}
+                           CLHEP::Random
+                           ROOT::ROOTDataFrame
+                           ROOT::Physics
+                           ROOT::Geom
+                           ROOT::FFTW
+                           ROOT::Tree
+                           ROOT::XMLIO
+                           ROOT::Gdml
+                           ROOT::Core
         )
 
 #install_headers()
diff --git a/icaruscode/TPC/Simulation/DetSim/tools/CMakeLists.txt b/icaruscode/TPC/Simulation/DetSim/tools/CMakeLists.txt
index 149a6d9..a8d8a3e 100644
--- a/icaruscode/TPC/Simulation/DetSim/tools/CMakeLists.txt
+++ b/icaruscode/TPC/Simulation/DetSim/tools/CMakeLists.txt
@@ -26,10 +26,18 @@ art_make(
 			              ${MF_UTILITIES}
 			              ${FHICLCPP}
 			              ${CETLIB}
-			              ${Boost_FILESYSTEM_LIBRARY}
-			              ${Boost_SYSTEM_LIBRARY}
-                          ${CLHEP}
-			              ${ROOT_BASIC_LIB_LIST}
+				   ${CLHEP}
+				   CLHEP::Random
+				   Boost::filesystem
+				   Boost::system
+				   ROOT::ROOTDataFrame
+				   ROOT::Physics
+				   ROOT::Geom
+				   ROOT::FFTW
+				   ROOT::Tree
+				   ROOT::Core
+                                   Eigen3::Eigen
+
     )
 
 install_headers()
diff --git a/icaruscode/TPC/Tracking/MCS/CMakeLists.txt b/icaruscode/TPC/Tracking/MCS/CMakeLists.txt
index 3a93ab4..ff0cfb9 100644
--- a/icaruscode/TPC/Tracking/MCS/CMakeLists.txt
+++ b/icaruscode/TPC/Tracking/MCS/CMakeLists.txt
@@ -19,10 +19,14 @@ art_make(
                               larreco_RecoAlg_CMTool_CMTAlgPriority
                               larreco_RecoAlg_CMTool_CMTAlgMatch
                               larreco_RecoAlg_CMTool_CMToolApp
-                              ${ROOT_MATHCORE}
-                              ${ROOT_GENVECTOR}
-                              ${ROOT_GEOM}
-                              ${ROOT_BASIC_LIB_LIST}
+			      ROOT::ROOTDataFrame
+			      ROOT::MathCore
+			      ROOT::GenVector
+			      ROOT::Physics
+			      ROOT::Geom
+			      ROOT::FFTW
+			      ROOT::Tree
+			      ROOT::Core
                               larreco_RecoAlg_TCAlg
                               larreco_RecoAlg_ImagePatternAlgs_DataProvider
                               larreco_RecoAlg_PMAlg
@@ -42,9 +46,8 @@ art_make(
                               ${FHICLCPP}
                               cetlib cetlib_except
                               ${CLHEP}
-                              ${ROOT_BASIC_LIB_LIST}
-                              ${ROOT_MINUIT}
-                              ${ROOT_MINUIT2}
+                              ROOT::Minuit
+                              ROOT::Minuit2
              MODULE_LIBRARIES icaruscode_TPC_Tracking_MCS
                               larsim_MCCheater_BackTrackerService_service
                               larsim_MCCheater_ParticleInventoryService_service
@@ -65,10 +68,14 @@ art_make(
                               larreco_RecoAlg_CMTool_CMTAlgPriority
                               larreco_RecoAlg_CMTool_CMTAlgMatch
                               larreco_RecoAlg_CMTool_CMToolApp
-                              ${ROOT_MATHCORE}
-                              ${ROOT_GENVECTOR}
-                              ${ROOT_GEOM}
-                              ${ROOT_BASIC_LIB_LIST}
+			      ROOT::ROOTDataFrame
+			      ROOT::MathCore
+			      ROOT::GenVector
+			      ROOT::Physics
+			      ROOT::Geom
+			      ROOT::FFTW
+			      ROOT::Tree
+			      ROOT::Core
                               larreco_RecoAlg_TCAlg
                               larreco_RecoAlg_ImagePatternAlgs_DataProvider
                               larreco_RecoAlg_PMAlg
@@ -88,9 +95,8 @@ art_make(
                               ${FHICLCPP}
                               cetlib cetlib_except
                               ${CLHEP}
-                              ${ROOT_BASIC_LIB_LIST}
-                              ${ROOT_MINUIT}
-                              ${ROOT_MINUIT2}
+                              ROOT::Minuit
+                              ROOT::Minuit2
         )
 
 
diff --git a/icaruscode/gallery/CMakeLists.txt b/icaruscode/gallery/CMakeLists.txt
index 59192d1..e1be379 100644
--- a/icaruscode/gallery/CMakeLists.txt
+++ b/icaruscode/gallery/CMakeLists.txt
@@ -4,14 +4,6 @@ add_subdirectory(MCTruthBase)
 
 foreach(ExampleDir IN ITEMS galleryAnalysis )
   
-  # take *all* the files from the listed directory
-  file(GLOB_RECURSE GalleryExamples RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} ${ExampleDir}/* )
-  
-  # cet_install_files() will reproduce the intermediate subdirectories from the
-  # source list into the destination directory
-  cet_install_files(
-    LIST ${GalleryExamples}
-    DIRNAME examples/gallery/${ExampleDir}
-    )
+ install(DIRECTORY ${ExampleDir}  DESTINATION examples/gallery/${ExampleDir})
   
 endforeach()
diff --git a/icaruscode/gallery/MCTruthBase/CMakeLists.txt b/icaruscode/gallery/MCTruthBase/CMakeLists.txt
index 3b0d008..1ce6a9c 100644
--- a/icaruscode/gallery/MCTruthBase/CMakeLists.txt
+++ b/icaruscode/gallery/MCTruthBase/CMakeLists.txt
@@ -9,10 +9,14 @@ art_make( LIB_LIBRARIES lardataobj_RecoBase
                         ${FHICLCPP}
                         ${CETLIB}
                         cetlib_except
-                        ${Boost_FILESYSTEM_LIBRARY}
-                        ${Boost_SYSTEM_LIBRARY}
                         ${CLHEP}
-                        ${ROOT_BASIC_LIB_LIST}
+                        Boost::filesystem
+                        Boost::system
+                        ROOT::Physics
+                        ROOT::Geom
+                        ROOT::FFTW
+                        ROOT::Tree
+                        ROOT::Core
         )
 
 install_headers()
diff --git a/icaruscode/gallery/galleryAnalysis/CMakeLists.txt b/icaruscode/gallery/galleryAnalysis/CMakeLists.txt
index bcad54a..e5f05ff 100644
--- a/icaruscode/gallery/galleryAnalysis/CMakeLists.txt
+++ b/icaruscode/gallery/galleryAnalysis/CMakeLists.txt
@@ -55,7 +55,8 @@ set(CMAKE_INSTALL_PREFIX ${CMAKE_CURRENT_SOURCE_DIR})
 #
 # include directories
 #
-include_directories($ENV{BOOST_INC})
+get_target_property(BoostIncludes Boost::filesystem INTERFACE_INCLUDE_DIRECTORIES)
+include_directories(${BoostIncludes})
 include_directories($ENV{ROOT_INC} $ENV{CLHEP_INC})
 include_directories($ENV{CETLIB_INC} $ENV{CETLIB_EXCEPT_INC} $ENV{CANVAS_INC} $ENV{CANVAS_ROOT_IO_INC})
 include_directories($ENV{FHICLCPP_INC} $ENV{MESSAGEFACILITY_INC})
diff a/icaruscode/TPC/SignalProcessing/RawDigitFilter/Algorithms/CMakeLists.txt	 b/icaruscode/TPC/SignalProcessing/RawDigitFilter/Algorithms/CMakeLists.txt
--- a/icaruscode/TPC/SignalProcessing/RawDigitFilter/Algorithms/CMakeLists.txt	2020-04-20 15:58:08.000000000 -0500
+++ b/icaruscode/TPC/SignalProcessing/RawDigitFilter/Algorithms/CMakeLists.txt	2020-05-02 17:03:23.314662000 -0500
@@ -8,7 +8,7 @@
                           lardata_Utilities
                           ${FHICLCPP}
                           cetlib cetlib_except
-                          ${ROOT_BASIC_LIB_LIST}
+                          ${ROOT_LIBRARIES}
                           larcorealg_Geometry
                           larcore_Geometry_Geometry_service
                           lardata_Utilities
@@ -61,9 +61,13 @@
                           ${MF_UTILITIES}
                           ${FHICLCPP}
                           cetlib cetlib_except
-                          ${ROOT_GEOM}
-                          ${ROOT_FFTW}
-                          ${ROOT_BASIC_LIB_LIST}
+                          ROOT::Physics
+                          ROOT::ROOTDataFrame
+                          ROOT::Geom
+                          ROOT::FFTW
+                          ROOT::Tree
+                          ROOT::Core
+                          Eigen3::Eigen
        )
 
 install_headers()
diff a/icaruscode/TPC/SignalProcessing/RawDigitFilter/CMakeLists.txt	a/icaruscode/TPC/SignalProcessing/RawDigitFilter/CMakeLists.txt	
--- a/icaruscode/TPC/SignalProcessing/RawDigitFilter/CMakeLists.txt	2020-04-20 15:58:08.000000000 -0500
+++ b/icaruscode/TPC/SignalProcessing/RawDigitFilter/CMakeLists.txt	2020-05-02 17:04:23.831502000 -0500
@@ -9,7 +9,13 @@
                         lardata_Utilities
                         ${FHICLCPP}
                         cetlib cetlib_except
-                        ${ROOT_BASIC_LIB_LIST}
+			ROOT::ROOTDataFrame
+			ROOT::Physics
+			ROOT::Geom
+			ROOT::FFTW
+			ROOT::Tree
+			ROOT::Core
+
           MODULE_LIBRARIES
                         larcorealg_Geometry
                         icaruscode_TPC_SignalProcessing_RawDigitFilter_Algorithms
@@ -34,11 +40,11 @@
                         ${MF_UTILITIES}
                         ${FHICLCPP}
                         cetlib cetlib_except
-                        ${ROOT_GEOM}
-                        ${ROOT_XMLIO}
-                        ${ROOT_GDML}
-			${ROOT_FFTW}
-			${ROOT_BASIC_LIB_LIST}
+                        ROOT::Geom
+                        ROOT::XMLIO
+                        ROOT::Gdml
+			ROOT::Fftw
+			${ROOT_LIBRARIES}
 			${TBB}
         )
 
diff a/icaruscode/TPC/SignalProcessing/RecoWire/DeconTools/CMakeLists.txt b/icaruscode/TPC/SignalProcessing/RecoWire/DeconTools/CMakeLists.txt
--- a/icaruscode/TPC/SignalProcessing/RecoWire/DeconTools/CMakeLists.txt	2020-05-02 15:28:27.223103000 -0500
+++ b/icaruscode/TPC/SignalProcessing/RecoWire/DeconTools/CMakeLists.txt	2020-05-02 17:05:08.207502000 -0500
@@ -27,10 +27,10 @@
                           ${MF_UTILITIES}
                           ${FHICLCPP}
                           ${CETLIB}
-                          ${Boost_FILESYSTEM_LIBRARY}
-                          ${Boost_SYSTEM_LIBRARY}
-                          ${CLHEP}
-                          ${ROOT_BASIC_LIB_LIST}
+                          Boost::filesystem
+                          Boost::system
+                          ${CLHEP_LIBRARIES}
+                          ${ROOT_LIBRARIES}
     )
 
 install_headers()
--- a/icaruscode/TPC/SignalProcessing/RecoWire/CMakeLists.txt	2020-04-20 15:58:08.000000000 -0500
+++ b/icaruscode/TPC/SignalProcessing/RecoWire/CMakeLists.txt	2020-05-02 17:05:45.564670000 -0500
@@ -10,8 +10,8 @@
             lardataobj_RecoBase
             lardata_Utilities
             ${FHICLCPP}
-            ${CETLIB}
-            ${ROOT_BASIC_LIB_LIST}
+            ${CETLIB_LIBRARIES}
+            ${ROOT_LIBRARIES}
           MODULE_LIBRARIES 
             larcorealg_Geometry
             larcore_Geometry_Geometry_service
@@ -26,7 +26,7 @@
             ${ART_FRAMEWORK_CORE}
             ${ART_FRAMEWORK_PRINCIPAL}
             ${ART_FRAMEWORK_SERVICES_REGISTRY}
-            ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+            ${ART_ROOT_IO_TFILE_SUPPORT} 
             ${ART_ROOT_IO_TFILESERVICE_SERVICE}
             art_Persistency_Common
             art_Persistency_Provenance
@@ -36,8 +36,8 @@
             ${MF_UTILITIES}
             ${FHICLCPP}
             ${CETLIB}
-            ${ROOT_FFTW}
-            ${ROOT_BASIC_LIB_LIST}
+            ROOT::Fftw
+            ${ROOT_LIBRARIES}
             ${TBB}
         )
 
diff a/icaruscode/CRT/CRTProducts/CMakeLists.txt b/icaruscode/CRT/CRTProducts/CMakeLists.txt
--- a/icaruscode/CRT/CRTProducts/CMakeLists.txt	2020-05-02 15:28:27.217103000 -0500
+++ b/icaruscode/CRT/CRTProducts/CMakeLists.txt	2020-05-02 17:07:24.790186000 -0500
@@ -1,6 +1,6 @@
 # for classes_def.xml!!
 include_directories( ${CMAKE_CURRENT_SOURCE_DIR} )
-include_directories( $ENV{NURANDOM_INC} )
+include_directories( ${NURANDOM_INC} )
 
 art_make( 
           LIBRARY_NAME icaruscode_CRTProducts
diff --git a/icaruscode/Geometry/CMakeLists.txt b/icaruscode/Geometry/CMakeLists.txt
index fc1511d..989993d 100644
--- a/icaruscode/Geometry/CMakeLists.txt
+++ b/icaruscode/Geometry/CMakeLists.txt
@@ -11,9 +11,9 @@ art_make(
     ${ART_FRAMEWORK_MODULES}
     ${ART_FRAMEWORK_PRINCIPAL}
     ${ART_FRAMEWORK_SERVICES_REGISTRY}
-    ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+    ${ART_ROOT_IO_TFILE_SUPPORT} ROOT::Core
     ${ART_ROOT_IO_TFILESERVICE_SERVICE}
-    ${BOOST_LIB}
+    Boost::Boost
     ${FHICLCPP}
   LIB_LIBRARIES
     larcorealg_Geometry
@@ -22,9 +22,9 @@ art_make(
     ${CETLIB}
     ${CETLIB_EXCEPT}
     ${FHICLCPP}
-    ${ROOT_GEOM}
-    ${ROOT_GENVECTOR}
-    ${ROOT_BASIC_LIB_LIST}
+    ROOT::Geom
+    ROOT::GenVector
+    ${ROOT_LIBRARIES}
   SERVICE_LIBRARIES
     icaruscode_Geometry
     ${ART_UTILITIES}
@@ -32,11 +32,10 @@ art_make(
     ${FHICLCPP}
     ${CETLIB}
     ${CETLIB_EXCEPT}
-    ${ROOT_GEOM}
-    ${ROOT_XMLIO}
-    ${ROOT_GDML}
-    ${ROOT_BASIC_LIB_LIST}
-    
+    ROOT::Geom
+    ROOT::XMLIO
+    ROOT::GDML
+    ${ROOT_LIBRARIES}
   )
 
 
diff --git a/icaruscode/Utilities/CMakeLists.txt b/icaruscode/Utilities/CMakeLists.txt
index 0828085..e3a9af7 100644
--- a/icaruscode/Utilities/CMakeLists.txt
+++ b/icaruscode/Utilities/CMakeLists.txt
@@ -3,12 +3,13 @@ art_make(
   EXCLUDE
     "SaveConfigurationIntoTFile_module.cc"
   LIB_LIBRARIES
-    ${CLHEP}
+    ${CLHEP_LIBRARIES}
+    CLHEP::Random
   )
 
 simple_plugin(SaveConfigurationIntoTFile "module"
               ${ART_ROOT_IO_TFILESERVICE_SERVICE}
-              ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
+              ${ART_ROOT_IO_TFILE_SUPPORT} ROOT::Core
               ${ART_FRAMEWORK_SERVICES_REGISTRY}
               ${ART_FRAMEWORK_PRINCIPAL}
               ${ART_FRAMEWORK_CORE}
@@ -17,7 +18,8 @@ simple_plugin(SaveConfigurationIntoTFile "module"
               hep_concurrency
               ${FHICLCPP}
               ${CETLIB}
-              ${ROOT_CORE}
+	      ROOT::Core
+              ${SQLITE3}
 )
 
 
diff --git a/test/Utilities/CMakeLists.txt b/test/Utilities/CMakeLists.txt
index 53e7f3b..7d02921 100644
--- a/test/Utilities/CMakeLists.txt
+++ b/test/Utilities/CMakeLists.txt
@@ -1,9 +1,9 @@
 cet_test(FastAndPoorGauss_test
   LIBRARIES
-    ${ROOT_MATHCORE}
-    ${ROOT_HIST}
-    ${ROOT_RIO}
-    ${ROOT_CORE}
+    ROOT::MatchCore
+    ROOT::Hist
+    ROOT::RIO
+    ROOT::Core
   USE_BOOST_UNIT
   )
 cet_test(SampledFunction_test USE_BOOST_UNIT)
@@ -11,6 +11,6 @@ cet_test(SampledFunction_test USE_BOOST_UNIT)
 cet_test(NonRandomCounter_test
   LIBRARIES
     icaruscode_Utilities
-    ${CLHEP}
+    ${CLHEP_LIBRARIES}
   USE_BOOST_UNIT
   )
