diff -Naur log4cpp/tests/Makefile.am log4cpp/tests/Makefile.am
--- log4cpp/tests/Makefile.am	2017-04-18 00:50:54.000000000 -0500
+++ log4cpp/tests/Makefile.am	2017-05-31 21:34:14.568398758 -0500
@@ -1,6 +1,6 @@
 TESTS = testCategory testFixedContextCategory testNDC testPattern \
 	testErrorCollision testPriority testFilter testProperties \
-	testConfig testPropertyConfig testRollingFileAppender testDailyRollingFileAppender
+	testConfig testPropertyConfig testRollingFileAppender
 
 check_PROGRAMS = $(TESTS)
 check_DATA = log4cpp.init log4cpp.properties testProperties.properties \
diff -Naur log4cpp/tests/Makefile.in log4cpp/tests/Makefile.in
--- log4cpp/tests/Makefile.in	2017-04-18 00:53:01.000000000 -0500
+++ log4cpp/tests/Makefile.in	2017-05-31 21:37:40.567816882 -0500
@@ -39,8 +39,7 @@
 	testErrorCollision$(EXEEXT) testPriority$(EXEEXT) \
 	testFilter$(EXEEXT) testProperties$(EXEEXT) \
 	testConfig$(EXEEXT) testPropertyConfig$(EXEEXT) \
-	testRollingFileAppender$(EXEEXT) \
-	testDailyRollingFileAppender$(EXEEXT)
+	testRollingFileAppender$(EXEEXT)
 check_PROGRAMS = $(am__EXEEXT_1)
 noinst_PROGRAMS = testmain$(EXEEXT) testbench$(EXEEXT)
 subdir = tests
@@ -70,8 +69,7 @@
 	testErrorCollision$(EXEEXT) testPriority$(EXEEXT) \
 	testFilter$(EXEEXT) testProperties$(EXEEXT) \
 	testConfig$(EXEEXT) testPropertyConfig$(EXEEXT) \
-	testRollingFileAppender$(EXEEXT) \
-	testDailyRollingFileAppender$(EXEEXT)
+	testRollingFileAppender$(EXEEXT)
 PROGRAMS = $(noinst_PROGRAMS)
 am_testCategory_OBJECTS = testCategory.$(OBJEXT)
 testCategory_OBJECTS = $(am_testCategory_OBJECTS)
