diff --git a/CMakeLists.txt b/CMakeLists.txt
index 729b2a3..51aa577 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,16 +16,13 @@ cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
 
 project(larwirecell VERSION 09.04.03 LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
-
-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules $ENV{ART_DIR}/Modules)
+find_package( cetmodules REQUIRED )
+find_package( canvas_root_io )
+find_package( art )
 
 include(CetCMakeEnv)
 cet_cmake_env()
 
-set_install_root()
-
 cet_set_compiler_flags(DIAGS CAUTIOUS
   WERROR
   NO_UNDEFINED
@@ -35,31 +32,30 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( larcore )
-find_ups_product( lardata )
-find_ups_product( larevt )
-find_ups_product( lardataobj )
-find_ups_product( lardataalg )
-find_ups_product( larcoreobj )
-find_ups_product( larcorealg )
-find_ups_product( art )
-find_ups_product(art_root_io)
-
-find_ups_product( jsoncpp )
-find_ups_product( gojsonnet )
-find_ups_product( eigen )
-find_ups_product( spdlog )
-find_ups_product( wirecell )
-
-
-cet_find_library( WIRECELL_SIGPROC_LIB NAMES WireCellSigProc PATHS ENV WIRECELL_LIB NO_DEFAULT_PATH )
-cet_find_library( WIRECELL_GEN_LIB NAMES WireCellGen PATHS ENV WIRECELL_LIB NO_DEFAULT_PATH )
-cet_find_library( WIRECELL_IFACE_LIB NAMES WireCellIface PATHS ENV WIRECELL_LIB NO_DEFAULT_PATH )
-cet_find_library( WIRECELL_UTIL_LIB NAMES WireCellUtil PATHS ENV WIRECELL_LIB NO_DEFAULT_PATH )
-cet_find_library( WIRECELL_APPS_LIB NAMES WireCellApps PATHS ENV WIRECELL_LIB NO_DEFAULT_PATH )
-set(WIRECELL_LIBS WIRECELL_APPS_LIB WIRECELL_SIGPROC_LIB WIRECELL_IFACE_LIB WIRECELL_UTIL_LIB WIRECELL_GEN_LIB)
-
-cet_find_library( JSONCPP NAMES jsoncpp PATHS ENV JSONCPP_LIB NO_DEFAULT_PATH )
+find_package( larevt )
+find_package( larcore )
+find_package( lardata )
+find_package( lardataalg )
+find_package( lardataobj )
+find_package( larcoreobj )
+find_package( larcorealg )
+find_package( nusimdata )
+find_package( art_root_io )
+find_package( jsoncpp )
+find_package( jsonnet )
+find_package( eigen )
+find_package( spdlog )
+find_package( wirecell )
+
+
+find_library( WIRECELL_SIGPROC_LIB NAMES WireCellSigProc PATHS $ENV{WIRECELL_LIB}  )
+find_library( WIRECELL_GEN_LIB NAMES WireCellGen PATHS $ENV{WIRECELL_LIB}  )
+find_library( WIRECELL_IFACE_LIB NAMES WireCellIface PATHS $ENV{WIRECELL_LIB}  )
+find_library( WIRECELL_UTIL_LIB NAMES WireCellUtil PATHS $ENV{WIRECELL_LIB}  )
+find_library( WIRECELL_APPS_LIB NAMES WireCellApps PATHS $ENV{WIRECELL_LIB}  )
+set(WIRECELL_LIBS ${WIRECELL_APPS_LIB} ${WIRECELL_SIGPROC_LIB} ${WIRECELL_IFACE_LIB} ${WIRECELL_UTIL_LIB} ${WIRECELL_GEN_LIB})
+
+find_library( JSONCPP NAMES jsoncpp PATHS $ENV{JSONCPP_LIB}  )
 
 # macros for dictionary and simple_plugin
 include(ArtDictionary)
@@ -68,10 +64,9 @@ include(BuildPlugins)
 
 # sources
 add_subdirectory(larwirecell)
-
 # ups - table and config files
-add_subdirectory(ups)
+#add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
-
+#include(UseCPack)
+cet_cmake_config()
diff --git a/larwirecell/Components/CMakeLists.txt b/larwirecell/Components/CMakeLists.txt
index 4e7393c..abb4916 100644
--- a/larwirecell/Components/CMakeLists.txt
+++ b/larwirecell/Components/CMakeLists.txt
@@ -1,28 +1,41 @@
 # Build the libWireCellLarsoft WCT plugin library.
 # See README for some details.
 
-file(GLOB wcls_components_sources "*.cxx")
+set(wcls_components_sources 
+      ChannelNoiseDB.cxx
+      ChannelSelectorDB.cxx
+      CookedFrameSink.cxx
+      CookedFrameSource.cxx
+      FrameSaver.cxx
+      LazyFrameSource.cxx
+      MultiChannelNoiseDB.cxx
+      RawFrameSource.cxx
+      SimChannelSink.cxx
+      SimDepoSource.cxx
+)
 
 # This is what's needed for building he WireCellLarsoft WCT component
 # plugin library.
-art_make_library(
+cet_make_library(
   LIBRARY_NAME WireCellLarsoft
   SOURCE ${wcls_components_sources}
+  HEADERS_TARGET
   LIBRARIES
     art::Framework_Core
     art::Framework_Principal
     art::Framework_Services_Registry
-    ART_PERSISTENCY_PROVENANCE
+    art::Persistency_Provenance
     art::Utilities
     JSONCPP
+    art::Utilities
     ROOT::Core
     ${WIRECELL_LIBS}
     canvas::canvas
     cetlib_except::cetlib_except
-    larcorealg_Geometry
-    lardataalg_DetectorInfo
-    lardataobj_RawData
-    lardataobj_RecoBase
-    lardataobj_Simulation
-    larevt_CalibrationDBI_IOVData
+    larcorealg::Geometry
+    lardataalg::DetectorInfo
+    lardataobj::RawData
+    lardataobj::RecoBase
+    lardataobj::Simulation
+    larevt::CalibrationDBI_IOVData
 )
diff --git a/larwirecell/Interfaces/CMakeLists.txt b/larwirecell/Interfaces/CMakeLists.txt
index d8ae4ed..2a0805e 100644
--- a/larwirecell/Interfaces/CMakeLists.txt
+++ b/larwirecell/Interfaces/CMakeLists.txt
@@ -3,4 +3,3 @@
 # has no library.
 
 install_headers()
-install_source()
diff --git a/larwirecell/LArInterface/CMakeLists.txt b/larwirecell/LArInterface/CMakeLists.txt
index d7f1767..bc4659e 100644
--- a/larwirecell/LArInterface/CMakeLists.txt
+++ b/larwirecell/LArInterface/CMakeLists.txt
@@ -5,11 +5,10 @@
 #
 # CMake instructions for larsoft service interface classes.
 
-art_make(
-  MODULE_LIBRARIES
-    lardataobj_RawData
-    larevt_CalibrationDBI_IOVData
-    larcorealg_Geometry
+cet_build_plugin(WireCellNoiseFilter art::module LIBRARIES
+    lardataobj::RawData
+    larevt::CalibrationDBI_IOVData
+    larcorealg::Geometry
     art::Framework_Services_Registry
     art_root_io::TFileService_service
     ${WIRECELL_LIBS}
@@ -18,4 +17,3 @@ art_make(
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/larwirecell/Modules/CMakeLists.txt b/larwirecell/Modules/CMakeLists.txt
index 04ac76a..346bd20 100644
--- a/larwirecell/Modules/CMakeLists.txt
+++ b/larwirecell/Modules/CMakeLists.txt
@@ -1,13 +1,20 @@
 # This is what's needed for building he WireCellLarsoft WCT component
 # plugin library.
-art_make(
-  MODULE_LIBRARIES
-    lardataobj_RawData
-    lardataobj_RecoBase
+cet_build_plugin(BlipMaker art::module LIBRARIES lardataobj::RawData
     ${WIRECELL_LIBS}
 )
 
+cet_build_plugin(EventButcher art::module LIBRARIES lardataobj::RawData
+    lardataobj::RecoBase
+    ${WIRECELL_LIBS}
+)
+
+cet_build_plugin(WireCellToolkit art::module LIBRARIES lardataobj::RawData
+    lardataobj::RecoBase
+    ${WIRECELL_LIBS}
+)
+
+
 add_subdirectory(Geometry)
 
 install_headers()
-install_source()
diff --git a/larwirecell/Modules/Geometry/CMakeLists.txt b/larwirecell/Modules/Geometry/CMakeLists.txt
index 828459d..ad2b5a3 100644
--- a/larwirecell/Modules/Geometry/CMakeLists.txt
+++ b/larwirecell/Modules/Geometry/CMakeLists.txt
@@ -1,31 +1,34 @@
 #add_subdirectory(tools)
-
-art_make( MODULE_LIBRARIES larcorealg_Geometry
-                           larcore_Geometry_Geometry_service
-                           lardataobj_Simulation
-                           lardata_Utilities
-                           larevt_Filters
-                           lardataobj_RawData
-                           larevt_CalibrationDBI_Providers
+cet_build_plugin(CTreeGeometry art::module LIBRARIES
+                           larcorealg::Geometry
+                           larcore::Geometry_Geometry_service
+                           lardataobj::Simulation
+                           lardata::Utilities
+                           larevt::Filters
+                           lardataobj::RawData
+                           larevt::CalibrationDBI_Providers
                            art::Framework_Core
                            art::Framework_Principal
                            art::Framework_Services_Registry
-                           ${ART_FRAMEWORK_SERVICES_OPTIONAL}
-                           ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
-                           ${ART_FRAMEWORK_SERVICES_OPTIONAL_TFILESERVICE_SERVICE}
+                           art::Framework_Services_System_FileCatalogMetadata_service art::Framework_Services_System_FloatingPointControl_service art::Framework_Services_System_TriggerNamesService_service
+                           art::Framework_Services_System_FloatingPointControl_service
+                           art_root_io::TFileService_service
+                           art_root_io::tfile_support 
                            art::Persistency_Common
                            art::Persistency_Provenance
                            art::Utilities
                            canvas::canvas
                            messagefacility::MF_MessageLogger
-                           ${MF_UTILITIES}
-                           fhiclcpp::fhiclcpp
+                           messagefacility::headers
+                           fhiclcpp::fhiclcpp fhiclcpp::types
                            cetlib::cetlib
                            cetlib_except::cetlib_except
+                           ROOT::Core
                            ROOT::Geom
                            ROOT::XMLIO
                            ROOT::Gdml
-                           ${ROOT_BASIC_LIB_LIST}
+                           ROOT::MathCore
+                           ROOT::Core
         )
 
 #install_headers()
diff --git a/larwirecell/Tools/CMakeLists.txt b/larwirecell/Tools/CMakeLists.txt
index fec5d24..57e45f6 100644
--- a/larwirecell/Tools/CMakeLists.txt
+++ b/larwirecell/Tools/CMakeLists.txt
@@ -1 +1 @@
-simple_plugin(WCLS "tool" ${WIRECELL_LIBS})
+cet_build_plugin(WCLS art::tool LIBRARIES ${WIRECELL_LIBS})
