commit cd05f00d76e38195535c82830954004c5a7f644d
Author: Patrick Gartung <gartung@fnal.gov>
Date:   Fri Apr 3 13:56:56 2020 -0500

    Cetmodules

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e9ded4b..0b4e380 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,12 +13,12 @@
 #  make package (builds distribution tarfile)
 # ======================================================================
 
-cmake_minimum_required(VERSION 3.11)  # use cmake 3.11 or later
+cmake_minimum_required(VERSION 3.12)  # use cmake 3.12 or later
 
-project(nurandom)
+project(nurandom VERSION 1.04.01 LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
+set(fcl_dir job)
+find_package(cetmodules REQUIRED)
 
 list(APPEND CMAKE_MODULE_PATH
   $ENV{CANVAS_ROOT_IO_DIR}/Modules
@@ -34,9 +34,8 @@ cet_set_compiler_flags(DIAGS CAUTIOUS WERROR
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product(art_root_io)
-find_ups_root()
-find_ups_product( cetbuildtools )
+find_package(art_root_io REQUIRED)
+find_package(ROOT REQUIRED)
 
 # macros
 include(ArtMake)
@@ -47,8 +46,8 @@ add_subdirectory (nurandom)
 add_subdirectory (test)
 
 # ups - table and config files
-add_subdirectory(ups)
-
+#add_subdirectory(ups)
+add_subdirectory(cmakeConfig)
 # packaging utility
-include(UseCPack)
+#include(UseCPack)
 
diff --git a/cmakeConfig/CMakeLists.txt b/cmakeConfig/CMakeLists.txt
new file mode 100644
index 0000000..e3d411c
--- /dev/null
+++ b/cmakeConfig/CMakeLists.txt
@@ -0,0 +1,3 @@
+
+# create package configuration and version files
+cet_cmake_config(CONFIG_FRAGMENTS nurandom-fragment.cmake.in)
diff --git a/cmakeConfig/nurandom-fragment.cmake.in b/cmakeConfig/nurandom-fragment.cmake.in
new file mode 100644
index 0000000..d022213
--- /dev/null
+++ b/cmakeConfig/nurandom-fragment.cmake.in
@@ -0,0 +1,8 @@
+find_package( ROOT REQUIRED COMPONENTS MathCore Geom GeomPainter GX11 EG EGPythia6 Gui)
+find_package( art REQUIRED)
+find_package( art_root_io REQUIRED)
+find_package( cetlib REQUIRED)
+find_package( cetlib_except REQUIRED)
+find_package( nusimdata REQUIRED )
+find_package( Geant4 REQUIRED)
+
diff --git a/nurandom/RandomUtils/CMakeLists.txt b/nurandom/RandomUtils/CMakeLists.txt
index 230c930..d6be40d 100644
--- a/nurandom/RandomUtils/CMakeLists.txt
+++ b/nurandom/RandomUtils/CMakeLists.txt
@@ -13,7 +13,7 @@ art_make(
     fhiclcpp
     cetlib
     cetlib_except
-    ${ROOT_CORE}
+    ROOT::Core
   PLUGIN_LIBRARIES
     art_Framework_Core
     art_Persistency_Provenance
@@ -22,10 +22,9 @@ art_make(
     fhiclcpp
     cetlib
     cetlib_except
-    ${Boost_FILESYSTEM_LIBRARY}
-    ${Boost_SYSTEM_LIBRARY}
+    Boost::filesystem
+    Boost::system
 )
 
-install_source()
 install_headers()
 install_fhicl()
diff --git a/nurandom/RandomUtils/Providers/CMakeLists.txt b/nurandom/RandomUtils/Providers/CMakeLists.txt
index 4c2cfe8..ce3bf72 100644
--- a/nurandom/RandomUtils/Providers/CMakeLists.txt
+++ b/nurandom/RandomUtils/Providers/CMakeLists.txt
@@ -1,5 +1,4 @@
 # currently only headers here
 
-install_source()
 install_headers()
 install_fhicl()
diff --git a/test/RandomUtils/CMakeLists.txt b/test/RandomUtils/CMakeLists.txt
index 14be885..6934473 100644
--- a/test/RandomUtils/CMakeLists.txt
+++ b/test/RandomUtils/CMakeLists.txt
@@ -12,7 +12,7 @@ art_make(
     ${FHICLCPP}
     cetlib
     cetlib_except
-    ${CLHEP}
+    ${CLHEP_LIBRARIES}
     NO_INSTALL
   MODULE_LIBRARIES
     nurandom_test_RandomUtils
@@ -26,7 +26,7 @@ art_make(
     ${FHICLCPP}
     cetlib
     cetlib_except
-    ${CLHEP}
+    ${CLHEP_LIBRARIES}
     NO_INSTALL
   SERVICE_LIBRARIES
     nurandom_RandomUtils_NuRandomService_service
@@ -38,8 +38,8 @@ art_make(
     ${FHICLCPP}
     cetlib
     cetlib_except
-    ${ROOT_CORE}
-    ${ROOT_MATHCORE}
+    ROOT::Core
+    ROOT::MathCore
     NO_INSTALL
   USE_PRODUCT_NAME
 )
