diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1d9862c..042e62f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,6 +15,8 @@
 # It sets up all of the larsoft ups products (lardata, larsim, larreco, etc.)
 # ======================================================================
 
+# if the CETBUILDTOOLS_VERSION environmental variable is defined, use cetbuildtools
+if(DEFINED ENV{CETBUILDTOOLS_VERSION})
 
 # use cmake 2.8 or later
 cmake_minimum_required (VERSION 2.8)
@@ -58,3 +60,31 @@ add_subdirectory(ups)
 
 # packaging utility
 include(UseCPack)
+
+else() # We are using cetmodules.
+
+cmake_minimum_required (VERSION 3.12)
+
+project( larsoft VERSION 08.43.00 LANGUAGES CXX )
+
+find_package(cetmodules REQUIRED)
+
+set(fcl_dir job)
+set(gdml_dir gdml)
+
+include(CetCMakeEnv)
+cet_cmake_env()
+
+# useful scripts
+add_subdirectory(bin)
+
+# documentation
+add_subdirectory(doc)
+
+# release database
+add_subdirectory(releaseDB)
+
+# generate packageConfig
+add_subdirectory(cmakeConfig)
+
+endif()
diff --git a/cmakeConfig/CMakeLists.txt b/cmakeConfig/CMakeLists.txt
new file mode 100644
index 0000000..c64db59
--- /dev/null
+++ b/cmakeConfig/CMakeLists.txt
@@ -0,0 +1 @@
+cet_cmake_config( CONFIG_FRAGMENTS larsoft-fragment.cmake.in )
diff --git a/cmakeConfig/larsoft-fragment.cmake.in b/cmakeConfig/larsoft-fragment.cmake.in
new file mode 100644
index 0000000..4632468
--- /dev/null
+++ b/cmakeConfig/larsoft-fragment.cmake.in
@@ -0,0 +1,10 @@
+find_package( lareventdisplay REQUIRED )
+find_package( larexamples REQUIRED )
+find_package( larana REQUIRED )
+find_package( larreco REQUIRED )
+find_package( larg4 REQUIRED )
+find_package( larpandora REQUIRED )
+find_package( larwirecell REQUIRED )
+find_package( larsoftobj REQUIRED )
+find_package( ifdh_art REQUIRED )
+find_package( larsoft_data )
diff --git a/doc/CMakeLists.txt b/doc/CMakeLists.txt
index 521a7fd..381d854 100644
--- a/doc/CMakeLists.txt
+++ b/doc/CMakeLists.txt
@@ -1,2 +1,2 @@
 # install the documentation
-install_source(EXTRAS LArSoftLogo-tiny.png)
+#install_source(EXTRAS LArSoftLogo-tiny.png)
diff --git a/releaseDB/CMakeLists.txt b/releaseDB/CMakeLists.txt
index 2748032..5c1eaad 100644
--- a/releaseDB/CMakeLists.txt
+++ b/releaseDB/CMakeLists.txt
@@ -1,3 +1,6 @@
+
+# if the CETBUILDTOOLS_VERSION environmental variable is defined, use cetbuildtools
+if(DEFINED ENV{CETBUILDTOOLS_VERSION})
 # variables to be used in cmake templates
 macro( create_product_variables productname productversion )
    string(TOUPPER ${productname} PRODUCTNAME_UC )
@@ -92,3 +95,4 @@ CONFIGURE_FILE ( ${CMAKE_CURRENT_SOURCE_DIR}/larsoft.html.in
 INSTALL( FILES base_dependency_database
                README
          DESTINATION "${product}/${version}/releaseDB" )
+endif()
