diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1d9862c..9d4da09 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,6 +15,8 @@
 # It sets up all of the larsoft ups products (lardata, larsim, larreco, etc.)
 # ======================================================================
 
+# if the CETBUILDTOOLS_VERSION environmental variable is defined, use cetbuildtools
+if(DEFINED ENV{CETBUILDTOOLS_VERSION})
 
 # use cmake 2.8 or later
 cmake_minimum_required (VERSION 2.8)
@@ -58,3 +60,31 @@ add_subdirectory(ups)
 
 # packaging utility
 include(UseCPack)
+
+else() # We are using cetmodules.
+
+cmake_minimum_required (VERSION 3.11)
+
+project( larsoft VERSION 08.05.00 LANGUAGES NONE )
+
+find_package(cetmodules REQUIRED)
+
+set(fcl_dir job)
+set(gdml_dir gdml)
+
+include(CetCMakeEnv)
+cet_cmake_env()
+
+# useful scripts
+add_subdirectory(bin)
+
+# documentation
+add_subdirectory(doc)
+
+# release database
+add_subdirectory(releaseDB)
+
+# generate packageConfig
+cet_cmake_config()
+
+endif()
