diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7f18d08..c99642d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,21 +13,24 @@ cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
 
 project(larsoft VERSION 09.35.00 LANGUAGES CXX C)
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
+set(fcl_dir job)
+set(gdml_dir gdml)
+set(larsoft_bin_dir "${CMAKE_INSTALL_PREFIX}/bin")
 
-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules $ENV{ART_DIR}/Modules)
+find_package(cetmodules REQUIRED)
+find_package(canvas_root_io)
+find_package(art)
 
 include(CetCMakeEnv)
 cet_cmake_env()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( lareventdisplay )
-find_ups_product( larexamples     )
-find_ups_product( larana          )
-find_ups_product( larreco         )
-find_ups_product( larrecodnn      )
-find_ups_product( larpandora      )
+find_package( lareventdisplay )
+find_package( larexamples     )
+find_package( larana          )
+find_package( larreco         )
+find_package( larrecodnn      )
+find_package( larpandora      )
 
 # useful scripts
 add_subdirectory(bin)
@@ -39,7 +42,7 @@ add_subdirectory(doc)
 add_subdirectory(releaseDB)
 
 # ups - table and config files
+if(WANT_UPS)
 add_subdirectory(ups)
-
-# packaging utility
-include(UseCPack)
+endif()
+cet_cmake_config()
diff --git a/doc/CMakeLists.txt b/doc/CMakeLists.txt
index 521a7fd..381d854 100644
--- a/doc/CMakeLists.txt
+++ b/doc/CMakeLists.txt
@@ -1,2 +1,2 @@
 # install the documentation
-install_source(EXTRAS LArSoftLogo-tiny.png)
+#install_source(EXTRAS LArSoftLogo-tiny.png)
diff --git a/releaseDB/CMakeLists.txt b/releaseDB/CMakeLists.txt
index 365b4fc..1386c76 100644
--- a/releaseDB/CMakeLists.txt
+++ b/releaseDB/CMakeLists.txt
@@ -1,3 +1,6 @@
+
+# if the CETBUILDTOOLS_VERSION environmental variable is defined, use cetbuildtools
+if(DEFINED ENV{CETBUILDTOOLS_VERSION})
 # variables to be used in cmake templates
 macro( create_product_variables productname productversion )
    string(TOUPPER ${productname} PRODUCTNAME_UC )
@@ -156,4 +159,5 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/stan-cfg.in
 # install releaseDB
 INSTALL( FILES base_dependency_database
                README
-         DESTINATION "${product}/${version}/releaseDB" )
+         DESTINATION "${CMAKE_INSTALL_PREFIX}/releaseDB" )
+endif()
