diff --git a/CMakeLists.txt b/CMakeLists.txt
index 22b8f43..6c85e3c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,3 @@
-#
 #  fhiclcpp/CMakeLists.txt
 #  -------------------------------------------------------------------
 #
@@ -15,10 +14,10 @@
 #  make package (builds distribution tarfile)
 
 cmake_minimum_required(VERSION 3.12...3.14 FATAL_ERROR)
-project(fhiclcpp LANGUAGES CXX C)
+project(fhiclcpp VERSION 4.11.01 LANGUAGES CXX C)
 
 # cetbuildtools contains our cmake modules
-find_package(cetbuildtools 7.13.01 REQUIRED)
+find_package(cetmodules REQUIRED)
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -62,20 +61,15 @@
 
 
 # these are minimum required versions, not the actual product versions
-find_ups_product(cetlib_except v1_00_00)
-find_ups_product(cetlib v2_01_00)
-find_ups_product(python)
-find_ups_product(sqlite v3_07_15_00)
-find_ups_boost(BOOST_TARGETS v1_50_0)
-
-# SQLite
-cet_find_library(SQLITE3 NAMES sqlite3_ups PATHS ENV SQLITE_LIB NO_DEFAULT_PATH)
+find_package(hep_concurrency REQUIRED)
+find_package(cetlib_except REQUIRED)
+find_package(cetlib REQUIRED)
+find_package(Boost 1.50.0 REQUIRED COMPONENTS program_options)
+find_library(SQLITE3 NAMES sqlite3 REQUIRED)
 
 # source code
 add_subdirectory(fhiclcpp)
 add_subdirectory(python)
 add_subdirectory(tools)
-add_subdirectory(ups)
 
-# packaging utility
-include(UseCPack)
+cet_cmake_config()
diff --git a/cmakeConfig/CMakeLists.txt b/cmakeConfig/CMakeLists.txt
new file mode 100644
index 0000000..937d12d
--- /dev/null
+++ b/cmakeConfig/CMakeLists.txt
@@ -0,0 +1,2 @@
+# Create package configuration and version files.
+cet_cmake_config()
diff --git a/cmakeConfig/fhiclcpp-fragment.cmake.in b/cmakeConfig/fhiclcpp-fragment.cmake.in
new file mode 100644
index 0000000..74caf8c
--- /dev/null
+++ b/cmakeConfig/fhiclcpp-fragment.cmake.in
@@ -0,0 +1,5 @@
+find_package(Boost 1.50.0 REQUIRED COMPONENTS)
+find_library(SQLITE3 NAMES sqlite3 REQUIRED)
+find_package(cetlib_except REQUIRED)
+find_package(cetlib REQUIRED)
+find_package(hep_concurrency REQUIRED)
diff --git a/xcompile/CMakeLists.txt b/xcompile/CMakeLists.txt
index 16af271..75d6b3d 100644
--- a/xcompile/CMakeLists.txt
+++ b/xcompile/CMakeLists.txt
@@ -1,4 +1,4 @@
 install (FILES
     LinuxPPC.cmake
-    DESTINATION ${product}/${version}/src/xcompile
+    DESTINATION src/xcompile
     )
