diff --git a/fhiclcpp/types/CMakeLists.txt b/fhiclcpp/types/CMakeLists.txt
index 0fdcfe4..ac7a77a 100644
--- a/fhiclcpp/types/CMakeLists.txt
+++ b/fhiclcpp/types/CMakeLists.txt
@@ -19,6 +19,8 @@ cet_make_library(LIBRARY_NAME fhiclcpp_types
       TBB::tbb
 )
 
+add_library(fhiclcpp::types ALIAS fhiclcpp_types)
+
 cet_make_library(LIBRARY_NAME plugin_config_macros INTERFACE
   EXPORT_SET PluginSupport
   SOURCE AllowedConfigurationMacro.h
