diff --git a/CMakeLists.txt b/CMakeLists.txt
index aeac6a1..82abba4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,13 +16,11 @@
 # use cmake 3.11 or later
 cmake_minimum_required (VERSION 3.11)
 
-project(art)
+project(art VERSION 3.04.00 LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools 7.13.01 REQUIRED)
-
-list(APPEND CMAKE_MODULE_PATH
-  ${CMAKE_CURRENT_SOURCE_DIR}/Modules)
+find_package(cetmodules REQUIRED)
+ 
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/Modules)
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -31,7 +31,7 @@
   EXTRA_FLAGS -pedantic
   EXTRA_CXX_FLAGS -Wnon-virtual-dtor -Wdelete-non-virtual-dtor)
 
-cet_have_qual(sse2 SSE2)
+#cet_have_qual(sse2 SSE2)
 if (SSE2)
   cet_add_compiler_flags(CXX -msse2 -ftree-vectorizer-verbose=2)
 endif()
@@ -69,20 +67,24 @@
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product(canvas v1_06_00)
-find_ups_product(messagefacility)
-find_ups_product(fhiclcpp)
-find_ups_product(cetlib v2_01_00)
-find_ups_product(cetlib_except v1_01_00)
-find_ups_product(clhep)
-find_ups_product(cppunit)
-find_ups_product(hep_concurrency)
-find_ups_product(sqlite)
-find_ups_product(range)
-find_ups_boost(BOOST_TARGETS)
+
+find_package(CLHEP REQUIRED COMPONENTS Matrix Vector Random)
+find_package(Boost REQUIRED
+  COMPONENTS date_time filesystem graph program_options
+  regex system unit_test_framework)
+find_package(TBB REQUIRED)
+
+find_library(SQLITE3 NAMES sqlite3 REQUIRED)
 
 # SQLite
 cet_find_library(SQLITE3 NAMES sqlite3_ups PATHS ENV SQLITE_LIB NO_DEFAULT_PATH)
+#These have to happend after to avoid calling find_package(TBB) twice
+find_package(hep_concurrency REQUIRED)
+find_package(cetlib_except REQUIRED)
+find_package(cetlib REQUIRED)
+find_package(canvas REQUIRED)
+find_package(fhiclcpp REQUIRED)
+find_package(messagefacility REQUIRED)
 
 # macros for simple_plugin
 include(BuildPlugins)
@@ -99,7 +99,7 @@
 add_subdirectory(art)
 
 # ups - table and config files
-add_subdirectory(ups)
+#Eadd_subdirectory(ups)
 
 # CMake modules
 add_subdirectory(Modules)
@@ -105,4 +105,5 @@
 add_subdirectory(Modules)
 
 # packaging utility
-include(UseCPack)
+cet_cmake_config()
+
--- a/perllib/CetSkelPlugins/CMakeLists.txt	2020-02-17 10:48:46.508360927 -0600
+++ b/perllib/CetSkelPlugins/CMakeLists.txt	2020-02-17 10:49:09.183770304 -0600
@@ -1 +1,3 @@
+ find_package(cetlib REQUIRED)
+ set(cetlib_PLUGINVERSIONINFO_PM_IN "${cetlib_DIR}/../../../perllib/PluginVersionInfo.pm.in")
 install_perllib()
--- a/Modules/BuildPlugins.cmake	2020-02-19 10:10:47.000000000 -0600
+++ b/Modules/BuildPlugins.cmake	2020-02-19 10:10:16.000000000 -0600
@@ -73,22 +73,22 @@
       art_Framework_IO_Sources
       Boost::filesystem)
   endif()
-  check_ups_version(cetbuildtools ${cetbuildtools_UPS_VERSION} v4_05_00 PRODUCT_MATCHES_VAR BP_HAS_SOURCE)
-  if(SP_SOURCE)
-    if (BP_HAS_SOURCE)
-      list(INSERT SP_SOURCE 0 SOURCE)
-    else()
-      message(FATAL_ERROR "SOURCE option specified, but not supported by cetbuildtools ${CETBUILDTOOLS_VERSION}")
-    endif()
-  endif()
-  check_ups_version(cetbuildtools ${cetbuildtools_UPS_VERSION} v4_06_00 PRODUCT_MATCHES_VAR BP_HAS_NOP)
-  if (BP_HAS_NOP AND NOT SP_NOP)
-    # Set it anyway so we have a good separator.
-    set(SP_NOP TRUE)
-  elseif(SP_NOP AND NOT BP_HAS_NOP)
-    # Not a problem, it's already done its job.
-    unset(SP_NOP)
-  endif()
+  #check_ups_version(cetbuildtools ${cetbuildtools_UPS_VERSION} v4_05_00 PRODUCT_MATCHES_VAR BP_HAS_SOURCE)
+  #if(SP_SOURCE)
+  #  if (BP_HAS_SOURCE)
+  #    list(INSERT SP_SOURCE 0 SOURCE)
+  #  else()
+  #    message(FATAL_ERROR "SOURCE option specified, but not supported by cetbuildtools ${CETBUILDTOOLS_VERSION}")
+  #  endif()
+  #endif()
+  #check_ups_version(cetbuildtools ${cetbuildtools_UPS_VERSION} v4_06_00 PRODUCT_MATCHES_VAR BP_HAS_NOP)
+  #if (BP_HAS_NOP AND NOT SP_NOP)
+  #  # Set it anyway so we have a good separator.
+  #  set(SP_NOP TRUE)
+  #elseif(SP_NOP AND NOT BP_HAS_NOP)
+  #  # Not a problem, it's already done its job.
+  #  unset(SP_NOP)
+  #endif()
   if (SP_NOP)
     set (NOP_ARG NOP)
   endif()
diff --git a/art/test/Utilities/CMakeLists.txt b/art/test/Utilities/CMakeLists.txt
index d874569..c56f5c3 100644
--- a/art/test/Utilities/CMakeLists.txt
+++ b/art/test/Utilities/CMakeLists.txt
@@ -13,7 +13,7 @@ add_subdirectory(tools)
 cet_test(make_tool_t USE_BOOST_UNIT LIBRARIES ${default_art_test_libraries})
 
 cet_test(ParameterSet_get_CLHEP_t
-  LIBRARIES ${default_art_test_libraries} ${CLHEP}
+  LIBRARIES ${default_art_test_libraries} CLHEP::Matrix CLHEP::Vector
   )
 
 cet_test(pointersEqual_t USE_BOOST_UNIT
diff --git a/art/Framework/Core/CMakeLists.txt b/art/Framework/Core/CMakeLists.txt
index 4949e9d..11e18b0 100644
--- a/art/Framework/Core/CMakeLists.txt
+++ b/art/Framework/Core/CMakeLists.txt
@@ -15,8 +15,8 @@
   canvas
   cetlib
   hep_concurrency
-  ${CLHEP}
-  ${TBB})
+  CLHEP::Random
+  TBB)
 
 install_headers(SUBDIRS detail)
 install_source(SUBDIRS detail)
--- c/art/Framework/Services/Optional/CMakeLists.txt	2020-02-19 13:01:21.000000000 -0600
+++ d/art/Framework/Services/Optional/CMakeLists.txt	2020-02-19 13:01:51.000000000 -0600
@@ -7,7 +7,7 @@
 simple_plugin(RandomNumberGenerator "service"
   art_Framework_Principal
   art_Persistency_Common
-  ${CLHEP})
+  CLHEP::Random)
 
 basic_plugin(MemoryTracker "service"
   art_Framework_Principal
@@ -18,7 +18,7 @@
 
 simple_plugin(TimeTracker "service"
   art_Framework_Principal
-  ${TBB}
+  TBB::tbb
   ${SQLITE3})
 
 simple_plugin(Tracer "service" art_Framework_Principal MF_MessageLogger)
