--- spack-src/art/Utilities/ScheduleID.h.orig	2021-03-23 18:57:06.182392000 +0000
+++ spack-src/art/Utilities/ScheduleID.h	2021-03-23 18:57:42.639080000 +0000
@@ -83,6 +83,7 @@
     // Comparison operators.
     bool operator==(ScheduleID const& other) const noexcept;
     bool operator<(ScheduleID const& other) const noexcept;
+    operator long int() const { return (long int)id_;}
 
   private: // Implementation Details
     constexpr ScheduleID(PremadeTypeFlag flag)
