diff --git a/CMakeLists.txt b/CMakeLists.txt
index 46e3ff3..8d3666f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -49,9 +49,23 @@ cet_report_compiler_flags(REPORT_THRESHOLD VERBOSE)
 
 cet_register_export_set(SET_NAME art NAMESPACE art)
 
-add_subdirectory(third_party/RapidJSON)
+find_package(RapidJSON)
+if(RapidJSON_FOUND)
+
+  message("using found RapidJson")
+  add_library(rapidjson INTERFACE)
+  target_include_directories(rapidjson PUBLIC ${RapidJSON_INCLUDE})
+endif()
+
+
+if(NOT RapidJSON_FOUND)
+
+  message("using third_party/RapidJson")
+  add_subdirectory(third_party/RapidJSON)
+  list(PREPEND CMAKE_PREFIX_PATH ${PROJECT_BINARY_DIR}/third_party/RapidJSON/rapidjson)
+
+endif()
 
-list(PREPEND CMAKE_PREFIX_PATH ${PROJECT_BINARY_DIR}/third_party/RapidJSON/rapidjson)
 
 # Plugin skeleton generators for cetskelgen.
 add_subdirectory(perllib)
