diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1d0d021..f5bf9a0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,10 +10,9 @@ cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
 
 project(larsimrad VERSION 09.03.04 LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
-
-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules $ENV{ART_DIR}/Modules)
+find_package( cetmodules REQUIRED )
+find_package( canvas_root_io )
+find_package( art )
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -29,17 +28,20 @@ cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
 unset(GSL_FOUND CACHE) # Workaround infelicity in CMake's FindPkgConfig.cmake.
-find_ups_product(gsl)
-find_ups_product(art_root_io)
-find_ups_product( larcore )
-find_ups_product( larcoreobj )
-find_ups_product( lardataobj )
-find_ups_product( lardataalg )
-find_ups_product( lardata )
-find_ups_product( larsim )
-find_ups_product( bxdecay0 )
-find_ups_product( nugen )
-find_ups_product( nurandom )
+find_package( gsl )
+find_package( art_root_io )
+find_package( larcore )
+find_package( larcorealg )
+find_package( larcoreobj )
+find_package( lardataobj )
+find_package( lardataalg )
+find_package( lardata )
+find_package( larsim )
+find_package( bxdecay0 )
+find_package( nugen )
+find_package( nurandom  )
+find_package( nusimdata  )
+
 
 # macros for dictionary and simple_plugin
 include(ArtDictionary)
@@ -53,7 +55,9 @@ add_subdirectory(larsimrad)
 #add_subdirectory(test)
 
 # ups - table and config files
+if(WANT_UPS)
 add_subdirectory(ups)
+endif()
 
 # packaging utility
 include(UseCPack)
diff --git a/larsimrad/BxDecay0/CMakeLists.txt b/larsimrad/BxDecay0/CMakeLists.txt
index 4d790d9..d697e59 100644
--- a/larsimrad/BxDecay0/CMakeLists.txt
+++ b/larsimrad/BxDecay0/CMakeLists.txt
@@ -1,16 +1,16 @@
 include_directories ( ENV BXDECAY0_INC )
-cet_find_library( BXDECAY0 NAMES BxDecay0 PATHS ENV BXDECAY0_LIB NO_DEFAULT_PATH )
+cet_find_library( BXDECAY0 NAMES BxDecay0 PATHS ENV BXDECAY0_LIB )
 #cet_find_library( GSL NAMES gsl PATHS ENV GSL_LIB NO_DEFAULT_PATH )
 
 art_make(MODULE_LIBRARIES
-         larsim_Simulation
-         lardataalg_DetectorInfo
-         lardataobj_Simulation
-         nusimdata_SimulationBase
-         larcoreobj_SummaryData
-         larcorealg_Geometry
-         larcore_Geometry_Geometry_service
-         nurandom_RandomUtils_NuRandomService_service
+         larsim::Simulation
+         lardataalg::DetectorInfo
+         lardataobj::Simulation
+         nusimdata::SimulationBase
+         larcoreobj::SummaryData
+         larcorealg::Geometry
+         larcore::Geometry_Geometry_service
+         nurandom::RandomUtils_NuRandomService_service
          BXDECAY0
          art::Framework_Services_Registry
          art_root_io::tfile_support
