diff --git a/CMakeLists.txt b/CMakeLists.txt
index 481cc1c..d52844e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,44 +12,29 @@ cmake_minimum_required (VERSION 3.12...3.14 FATAL_ERROR)
 
 project(lardata LANGUAGES CXX)
 
-# cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
-IF (NOT CETBUILDTOOLS_VERSION)
-    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
-ENDIF()
-
-set(CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules
-		      $ENV{CETBUILDTOOLS_DIR}/Modules
-		      ${CMAKE_MODULE_PATH})
-
+set(fcl_dir job)
+set(gdml_dir gdml)
+find_package( cetmodules )
 include(CetCMakeEnv)
 cet_cmake_env()
 
 cet_set_compiler_flags(DIAGS CAUTIOUS
   WERROR
   NO_UNDEFINED
-  EXTRA_FLAGS -pedantic -I $ENV{CLANG_FQ_DIR}/include/c++/v1
+  EXTRA_FLAGS -pedantic
 )
 
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( nusimdata )
-find_ups_product( larcoreobj )
-find_ups_product( larcorealg )
-find_ups_product( lardataobj )
-find_ups_product( lardataalg )
-find_ups_product( larcore  )
-find_ups_product( art  )
-find_ups_product(art_root_io)
-find_ups_product( postgresql  )
-find_ups_product( range )
-find_ups_product( cetbuildtools )
-find_ups_product( fftw )
-
-#  Find all the libraries needed by our dependent CMakeList.txt files
-# cet_find_library( NUSIMDATA_SIMULATIONBASE NAMES nusimdata_SimulationBase PATHS ENV NUSIMDATA_LIB NO_DEFAULT_PATH )
-cet_find_library( PQ                  NAMES pq                  PATHS ENV POSTGRESQL_LIBRARIES NO_DEFAULT_PATH )
+find_package( PostgreSQL REQUIRED )
+find_package( Range-v3 )
+find_package( ROOT REQUIRED COMPONENTS Core GenVector FFTW Cling RIO Net Hist Geom XMLIO Gdml Graf Graf3d Gpad Tree EG Thread MathCore Physics Postscript Rint)
+find_package( Boost REQUIRED COMPONENTS date_time filesystem graph program_options serialization regex system unit_test_framework)
+find_package( lardataobj REQUIRED)
+find_package( lardataalg REQUIRED)
+find_package( larcore  REQUIRED)
+find_package( nutools  REQUIRED)
 
 # macros for artdaq_dictionary and simple_plugin
 include(ArtDictionary)
@@ -61,9 +46,9 @@ add_subdirectory(lardata)
 
 # tests
 add_subdirectory(test)
-
+add_subdirectory(cmakeConfig)
 # ups - table and config files
-add_subdirectory(ups)
+#add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
+#include(UseCPack)
diff --git a/cmakeConfig/CMakeLists.txt b/cmakeConfig/CMakeLists.txt
new file mode 100644
index 0000000..9b911ea
--- /dev/null
+++ b/cmakeConfig/CMakeLists.txt
@@ -0,0 +1,4 @@
+
+# create package configuration and version files
+
+cet_cmake_config(CONFIG_FRAGMENTS lardata-fragment.cmake.in)
diff --git a/cmakeConfig/lardata-fragment.cmake.in b/cmakeConfig/lardata-fragment.cmake.in
new file mode 100644
index 0000000..f89a140
--- /dev/null
+++ b/cmakeConfig/lardata-fragment.cmake.in
@@ -0,0 +1,9 @@
+find_package( PostgreSQL REQUIRED )
+find_package( Range-v3 )
+find_package( ROOT REQUIRED COMPONENTS Core GenVector FFTW Cling RIO Net Hist Geom XMLIO Gdml Graf Graf3d Gpad Tree EG Thread MathCore Physics Postscript Rint)
+find_package(Boost REQUIRED COMPONENTS date_time filesystem graph program_options
+serialization regex system unit_test_framework)
+find_package( lardataobj REQUIRED)
+find_package( lardataalg REQUIRED)
+find_package( larcore  REQUIRED)
+find_package( nutools  REQUIRED)
diff --git a/lardata/RawData/utils/CMakeLists.txt b/lardata/RawData/utils/CMakeLists.txt
index 7a43433..980e3fa 100644
--- a/lardata/RawData/utils/CMakeLists.txt
+++ b/lardata/RawData/utils/CMakeLists.txt
@@ -10,9 +10,9 @@ art_make(NO_PLUGINS
                        art_Persistency_Provenance
                        canvas
                        cetlib_except
-                       ${PQ}
-                       ${Boost_SERIALIZATION_LIBRARY}
-                       ${Boost_DATE_TIME_LIBRARY}
+                       ${PostgreSQL_LIBRARY}
+                       ${Boost_serialization_LIBRARY}
+                       ${Boost_data_time_LIBRARY}
                        ROOT::Core
                        ROOT::RIO
                        ROOT::Tree
diff --git a/lardata/RecoObjects/CMakeLists.txt b/lardata/RecoObjects/CMakeLists.txt
index af41ecd..d382ea1 100644
--- a/lardata/RecoObjects/CMakeLists.txt
+++ b/lardata/RecoObjects/CMakeLists.txt
@@ -1,3 +1,4 @@
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-misleading-indentation")
 art_make(LIB_LIBRARIES lardataobj_AnalysisBase
                        lardataobj_RecoBase
                        larcore_Geometry_Geometry_service
diff --git a/lardata/Utilities/CMakeLists.txt b/lardata/Utilities/CMakeLists.txt
index c3b4977..835c061 100644
--- a/lardata/Utilities/CMakeLists.txt
+++ b/lardata/Utilities/CMakeLists.txt
@@ -13,7 +13,7 @@ art_make(NO_PLUGINS
 
 simple_plugin(DatabaseUtil "service"
               ${MF_MESSAGELOGGER}
-              ${PQ}
+              ${PostgreSQL_LIBRARY}
               ${ART_FRAMEWORK_CORE}
               ${ART_FRAMEWORK_PRINCIPAL}
               ${ART_ROOT_IO_ROOTDB}
diff --git a/test/ArtDataHelper/CMakeLists.txt b/test/ArtDataHelper/CMakeLists.txt
index 27e88c0..aff9a67 100644
--- a/test/ArtDataHelper/CMakeLists.txt
+++ b/test/ArtDataHelper/CMakeLists.txt
@@ -3,7 +3,7 @@ art_make(
     lardata_ArtDataHelper
     lardataobj_RecoBase
     ${MF_MESSAGELOGGER}
-    
+    ${ROOT_Core_Library}
     ${FHICLCPP}
     cetlib_except
 )
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index b49bd59..0ff3962 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -28,5 +28,3 @@ cet_test(donothing HANDBUILT
   TEST_EXEC lar
   TEST_ARGS --rethrow-all --config ./test_donothing.fcl
 )
-
-
diff --git a/test/RecoBaseProxy/CMakeLists.txt b/test/RecoBaseProxy/CMakeLists.txt
index bd312da..dad91f3 100644
--- a/test/RecoBaseProxy/CMakeLists.txt
+++ b/test/RecoBaseProxy/CMakeLists.txt
@@ -2,6 +2,7 @@
 ###  ProxyBase tests
 ###
 add_subdirectory(ProxyBase)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-sign-compare -Wno-pedantic")
 
 simple_plugin(ProxyBaseTest "module"
   lardata_RecoBaseProxy
diff --git a/test/RecoBaseProxy/ProxyBase/CMakeLists.txt b/test/RecoBaseProxy/ProxyBase/CMakeLists.txt
index f8e1fad..b5480c1 100644
--- a/test/RecoBaseProxy/ProxyBase/CMakeLists.txt
+++ b/test/RecoBaseProxy/ProxyBase/CMakeLists.txt
@@ -1,6 +1,7 @@
 ###############################################################################
 ###  Support utility tests
 ###
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-sign-compare -Wno-pedantic")
 cet_test(AssnsTraits_test
   USE_BOOST_UNIT
   )
diff --git a/test/RecoObjects/CMakeLists.txt b/test/RecoObjects/CMakeLists.txt
index e16fc02..ab1db22 100644
--- a/test/RecoObjects/CMakeLists.txt
+++ b/test/RecoObjects/CMakeLists.txt
@@ -1,3 +1,5 @@
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-sign-compare -Wno-pedantic")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-misleading-indentation")
 cet_test( SurfYZTest USE_BOOST_UNIT LIBRARIES lardata_RecoObjects )
 cet_test( SurfXYZTest USE_BOOST_UNIT LIBRARIES lardata_RecoObjects )
 cet_test( SurfYZLineTest USE_BOOST_UNIT LIBRARIES lardata_RecoObjects )
diff --git a/test/Utilities/CMakeLists.txt b/test/Utilities/CMakeLists.txt
index e1b351f..0bbd78a 100644
--- a/test/Utilities/CMakeLists.txt
+++ b/test/Utilities/CMakeLists.txt
@@ -1,8 +1,9 @@
 add_subdirectory( testPtrMaker )
 add_subdirectory( testForEachAssociatedGroup )
 add_subdirectory( testAssnsChainUtils )
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-sign-compare -Wno-pedantic")
 
-# BulkAllocator_test, NestedIterator_test, CountersMap_test 
+# BulkAllocator_test, NestedIterator_test, CountersMap_test
 # and test pure header libraries (they are templates)
 
 # test removed per issue #19494
@@ -30,7 +31,6 @@ cet_test(timingreference_test HANDBUILT
   OPTIONAL_GROUPS ONLY_SLF
 )
 
-
 install_headers()
 install_fhicl()
 install_source()
diff --git a/test/Utilities/testForEachAssociatedGroup/CMakeLists.txt b/test/Utilities/testForEachAssociatedGroup/CMakeLists.txt
index e5845f4..fca14bc 100644
--- a/test/Utilities/testForEachAssociatedGroup/CMakeLists.txt
+++ b/test/Utilities/testForEachAssociatedGroup/CMakeLists.txt
@@ -1,7 +1,7 @@
 simple_plugin( AssnProducer1          "module"  NO_INSTALL cetlib_except art_Persistency_Provenance)
 simple_plugin( AssnProducer2          "module"  NO_INSTALL cetlib_except art_Persistency_Provenance)
 simple_plugin( AssnAnalyzer           "module"  NO_INSTALL cetlib_except art_Persistency_Provenance)
-
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-sign-compare -Wno-pedantic")
 cet_test(associated_groups_test USE_BOOST_UNIT
   LIBRARIES
     canvas
diff --git a/test/Utilities/testPtrMaker/CMakeLists.txt b/test/Utilities/testPtrMaker/CMakeLists.txt
index 8ded552..395b59c 100644
--- a/test/Utilities/testPtrMaker/CMakeLists.txt
+++ b/test/Utilities/testPtrMaker/CMakeLists.txt
@@ -4,26 +4,26 @@ simple_plugin(PtrMakerProducer2          "module"  NO_INSTALL cetlib_except art_
 simple_plugin(PtrMakerAnalyzer           "module"  NO_INSTALL cetlib_except art_Persistency_Provenance)
 
 cet_test(test_ptrmaker HANDBUILT
-  TEST_EXEC lar 
+  TEST_EXEC lar
   TEST_ARGS --rethrow-all --config test_ptrmaker.fcl
   DATAFILES test_ptrmaker.fcl
 )
 
 cet_test(test_ptrmaker2 HANDBUILT
-  TEST_EXEC lar 
+  TEST_EXEC lar
   TEST_ARGS --rethrow-all --config test_ptrmaker2.fcl
   DATAFILES test_ptrmaker2.fcl
 )
 
 cet_test(test_ptrmaker3 HANDBUILT
-  TEST_EXEC lar 
+  TEST_EXEC lar
   TEST_ARGS --rethrow-all --config test_ptrmaker3.fcl
   DATAFILES test_ptrmaker3.fcl
   TEST_PROPERTIES DEPENDS test_ptrmaker
 )
 
 cet_test(test_ptrmaker4 HANDBUILT
-  TEST_EXEC lar 
+  TEST_EXEC lar
   TEST_ARGS --rethrow-all --config test_ptrmaker4.fcl
   DATAFILES test_ptrmaker4.fcl
   TEST_PROPERTIES DEPENDS test_ptrmaker2
