diff --git a/CMakeLists.txt b/CMakeLists.txt
index 72de522..9c0ea27 100644
@@ -10,10 +10,9 @@ cmake_minimum_required(VERSION 3.19 FATAL_ERROR)

 project(lardata VERSION 09.04.02 LANGUAGES CXX)

-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
-
-list(APPEND CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules $ENV{ART_DIR}/Modules)
+find_package(cetmodules REQUIRED)
+find_package(canvas_root_io)
+find_package(art)

 include(CetCMakeEnv)
 cet_cmake_env()
@@ -27,41 +26,43 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()

 # these are minimum required versions, not the actual product versions
-find_ups_product( nusimdata )
-find_ups_product( larcoreobj )
-find_ups_product( larcorealg )
-find_ups_product( lardataobj )
-find_ups_product( lardataalg )
-find_ups_product( larcore  )
-find_ups_product( art  )
-find_ups_product(art_root_io)
-find_ups_product( postgresql  )
-find_ups_product( range )
-find_ups_product( larvecutils )
-find_ups_product( fftw )
-find_ups_product( canvas  )
-find_ups_product(messagefacility)
-find_ups_product(cetlib)
-find_ups_product(cetlib_except)
-find_ups_product( clhep  )
+find_package( nusimdata )
+find_package( larcoreobj )
+find_package( larcorealg )
+find_package( lardataobj )
+find_package( lardataalg )
+find_package( larcore )
+find_package( larvecutils )
+find_package( art )
+find_package( art_root_io )
+find_package( FFTW3 )
+find_package( canvas )
+find_package( messagefacility )
+find_package( cetlib)
+find_package( cetlib_except)
+find_package( CLHEP  )
+find_package( PostgreSQL )
+find_package( Range-v3 )
+find_package( Boost COMPONENTS serialization)

 #  Find all the libraries needed by our dependent CMakeList.txt files
-# cet_find_library( NUSIMDATA_SIMULATIONBASE NAMES nusimdata_SimulationBase PATHS ENV NUSIMDATA_LIB NO_DEFAULT_PATH )
-cet_find_library( PQ                  NAMES pq                  PATHS ENV POSTGRESQL_LIBRARIES NO_DEFAULT_PATH )
+# cet_find_library( NUSIMDATA_SIMULATIONBASE NAMES nusimdata_SimulationBase PATHS ENV NUSIMDATA_LIB )
+find_library( PQ  NAMES pq PATHS ${POSTGRESQL_LIBRARIES} )

 # macros for artdaq_dictionary and simple_plugin
 include(ArtDictionary)
 include(ArtMake)
 include(BuildPlugins)
+find_package(messagefacility)

 # source
 add_subdirectory(lardata)

 # tests
-add_subdirectory(test)
-
+#add_subdirectory(test)
 # ups - table and config files
-add_subdirectory(ups)
+#add_subdirectory(ups)

 # packaging utility
-include(UseCPack)
+#include(UseCPack)
+cet_cmake_config()
diff --git a/lardata/ArtDataHelper/CMakeLists.txt b/lardata/ArtDataHelper/CMakeLists.txt
index ae00594..5987c3b 100644
--- a/lardata/ArtDataHelper/CMakeLists.txt
+++ b/lardata/ArtDataHelper/CMakeLists.txt
@@ -1,8 +1,8 @@
-art_make(LIB_LIBRARIES lardataobj_RawData
-                       lardataobj_RecoBase
-                       lardataobj_AnalysisBase
-                       larcorealg_Geometry
-                       larcore_Geometry_Geometry_service
+art_make(LIB_LIBRARIES lardataobj::RawData
+                       lardataobj::RecoBase
+                       lardataobj::AnalysisBase
+                       larcorealg::Geometry
+                       larcore::Geometry_Geometry_service
                        art::Framework_Core
                        art::Framework_Principal
                        art::Framework_Services_Registry
diff --git a/lardata/ArtDataHelper/Dumpers/CMakeLists.txt b/lardata/ArtDataHelper/Dumpers/CMakeLists.txt
index b30834f..c1f7d50 100644
--- a/lardata/ArtDataHelper/Dumpers/CMakeLists.txt
+++ b/lardata/ArtDataHelper/Dumpers/CMakeLists.txt
@@ -24,29 +24,37 @@ set(SimulationDumpers
 set(MCDumper)
 
 
-art_make(NO_PLUGINS LIB_LIBRARIES lardataobj_RecoBase)
+cet_make_library(SOURCE
+                       PCAxisDumpers.cc
+                       SpacePointDumpers.cc
+                HEADERS_TARGET
+                LIBRARIES 
+                       lardataobj_RecoBase)
 
 foreach(Dumper IN LISTS RawDataDumpers)
-  simple_plugin(${Dumper} "module"
-      lardataobj_RawData
+  cet_build_plugin(${Dumper} art::module
+      LIBRARIES
+      lardataobj::RawData
       art::Framework_Services_Registry
       messagefacility::MF_MessageLogger)
 endforeach()
 
 
 foreach(Dumper IN LISTS RecoBaseDumpers)
-  simple_plugin(${Dumper} "module"
-      lardataobj_RecoBase
-      lardata_ArtDataHelper_Dumpers
-      lardata_RecoBaseProxy
+  cet_build_plugin(${Dumper} art::module
+      LIBRARIES
+      lardataobj::RecoBase
+      lardata::ArtDataHelper_Dumpers
+      lardata::RecoBaseProxy
       messagefacility::MF_MessageLogger)
 endforeach()
 
 
 foreach(Dumper IN LISTS SimulationDumpers)
-  simple_plugin(${Dumper} "module"
-      lardataalg_MCDumpers
-      lardataobj_Simulation
+  cet_build_plugin(${Dumper} art::module
+      LIBRARIES
+      lardataalg::MCDumpers
+      lardataobj::Simulation
       messagefacility::MF_MessageLogger
       ROOT::GenVector
       ROOT::Core)
@@ -54,7 +62,7 @@ endforeach(Dumper IN LISTS SimulationDumpers)
 
 
 foreach(Dumper IN LISTS MCDumpers)
-  simple_plugin(${Dumper} "module" lardataobj_MCBase)
+  cet_build_plugin(${Dumper} art::module LIBRARIES lardataobj::MCBase)
 endforeach()
 
 
diff --git a/lardata/DetectorInfoServices/CMakeLists.txt b/lardata/DetectorInfoServices/CMakeLists.txt
index e74459b..824bf32 100644
--- a/lardata/DetectorInfoServices/CMakeLists.txt
+++ b/lardata/DetectorInfoServices/CMakeLists.txt
@@ -1,28 +1,30 @@
-simple_plugin(DetectorClocksServiceStandard "service"
-              art_root_io_detail
-              lardataalg_DetectorInfo
-              lardataobj_RawData
+cet_build_plugin(DetectorClocksServiceStandard art::service
+              LIBRARIES
+              art_root_io::detail
+              lardataalg::DetectorInfo
+              lardataobj::RawData
               art::Framework_Core
               art::Framework_Principal
-              ${ART_ROOT_IO_ROOTDB}
               art::Persistency_Provenance
+              art_root_io::RootDB
               ${SQLITE3}
               ROOT::Core
               ROOT::RIO
               ROOT::Tree)
-
-simple_plugin(DetectorPropertiesServiceStandard "service"
-              lardataalg_DetectorInfo
-              larcore_Geometry_Geometry_service
-              larcorealg_Geometry
-              ${ART_ROOT_IO_ROOTDB}
+cet_build_plugin(DetectorPropertiesServiceStandard art::service
+              LIBRARIES
+              lardataalg::DetectorInfo
+              larcore::Geometry_Geometry_service
+              larcorealg::Geometry
+              art_root_io::RootDB
               ${SQLITE3}
               messagefacility::MF_MessageLogger
               ROOT::Core
               ROOT::RIO)
 
-simple_plugin(LArPropertiesServiceStandard "service"
-              lardataalg_DetectorInfo
+cet_build_plugin(LArPropertiesServiceStandard art::service
+              LIBRARIES
+              lardataalg::DetectorInfo
               art::Framework_Principal)
 
 install_headers()
diff --git a/lardata/RawData/utils/CMakeLists.txt b/lardata/RawData/utils/CMakeLists.txt
index 1d62aed..8114598 100644
--- a/lardata/RawData/utils/CMakeLists.txt
+++ b/lardata/RawData/utils/CMakeLists.txt
@@ -1,8 +1,8 @@
 art_make(NO_PLUGINS
          EXCLUDE LArRawInputDriverUBooNE.cxx
-         LIB_LIBRARIES lardataobj_RawData
-                       larcorealg_Geometry
-                       larcoreobj_SummaryData
+         LIB_LIBRARIES lardataobj::RawData
+                       larcorealg::Geometry
+                       larcoreobj::SummaryData
                        art::Framework_Core
                        art::Framework_IO_Sources
                        art::Framework_Principal
@@ -10,17 +10,18 @@ art_make(NO_PLUGINS
                        art::Persistency_Provenance
                        canvas::canvas
                        cetlib_except::cetlib_except
-                       PQ
-                       ${Boost_SERIALIZATION_LIBRARY}
-                       ${Boost_DATE_TIME_LIBRARY}
+                       ${PQ}
+                       Boost::serialization
+                       Boost::date_time
                        ROOT::Core
                        ROOT::RIO
                        ROOT::Tree
         )
 
-simple_plugin(LArRawInputSource "source"
-  lardataobj_RawData
-  lardata_RawData_utils
+cet_build_plugin(LArRawInputSource art::source
+  LIBRARIES
+  lardataobj::RawData
+  lardata::RawData_utils
   art::Framework_IO_Sources
 )
 
diff --git a/lardata/RecoBaseProxy/CMakeLists.txt b/lardata/RecoBaseProxy/CMakeLists.txt
index 6ae53fc..6ed933f 100644
--- a/lardata/RecoBaseProxy/CMakeLists.txt
+++ b/lardata/RecoBaseProxy/CMakeLists.txt
@@ -1,6 +1,6 @@
 add_subdirectory(ProxyBase)
 
-art_make(LIB_LIBRARIES lardataobj_RecoBase ROOT::Core)
+art_make(LIB_LIBRARIES lardataobj::RecoBase ROOT::Core)
 
 install_headers()
 install_source()
diff --git a/lardata/RecoObjects/CMakeLists.txt b/lardata/RecoObjects/CMakeLists.txt
index fc3df4d..4bc01a5 100644
--- a/lardata/RecoObjects/CMakeLists.txt
+++ b/lardata/RecoObjects/CMakeLists.txt
@@ -1,12 +1,12 @@
 art_make(LIB_LIBRARIES
-           lardataobj_AnalysisBase
-           lardataobj_RecoBase
-           lardataalg_DetectorInfo
-           larcore_Geometry_Geometry_service
-           larcorealg_Geometry
+           lardataobj::AnalysisBase
+           lardataobj::RecoBase
+           lardataalg::DetectorInfo
+           larcore::Geometry_Geometry_service
+           larcorealg::Geometry
            art::Framework_Core
            art::Framework_Services_Registry
-           ${ART_UTILITIES}
+           art::Utilities
            canvas::canvas
            fhiclcpp::fhiclcpp
            cetlib_except::cetlib_except
diff --git a/lardata/Utilities/CMakeLists.txt b/lardata/Utilities/CMakeLists.txt
index 9c4e691..119f17f 100644
--- a/lardata/Utilities/CMakeLists.txt
+++ b/lardata/Utilities/CMakeLists.txt
@@ -1,37 +1,44 @@
 include_directories(${FFTW_INCLUDE_DIR})
-cet_find_library(FFTW_LIBRARY NAMES fftw3 fftw3-3 PATHS ENV FFTW_LIBRARY )
+find_library(FFTW_LIBRARY NAMES fftw3 PATHS ${FFTW_LIBRARIES} )
+
+include_directories("${CMAKE_CURRENT_SOURCE_DIR}/../..")
 
 art_make(NO_PLUGINS
          LIB_LIBRARIES
-            lardata_Utilities_LArFFT_service
-            lardataobj_RecoBase
-            larcorealg_Geometry
-              art::Framework_Core
-              art::Framework_Principal
-              art::Persistency_Provenance
+            lardata::Utilities_LArFFT_service
+            lardataobj::RecoBase
+            larcorealg::Geometry
+            art::Framework_Core
+            art::Framework_Principal
+            art::Persistency_Provenance
             canvas::canvas
             messagefacility::MF_MessageLogger
             cetlib_except::cetlib_except
-            FFTW_LIBRARY)
+            ${FFTW_LIBRARY})
 
-simple_plugin(DatabaseUtil "service"
+cet_build_plugin(DatabaseUtil art::service
+         LIBRARIES
               messagefacility::MF_MessageLogger
-              PQ
+              ${PQ}
               art::Framework_Core
               art::Framework_Principal
-              ${ART_ROOT_IO_ROOTDB}
+              art_root_io::RootDB
               art::Persistency_Provenance)
 
-simple_plugin(LArFFT "service"
-              lardataalg_DetectorInfo
+cet_build_plugin(LArFFT art::service
+         LIBRARIES
+              lardataalg::DetectorInfo
               messagefacility::MF_MessageLogger
               ROOT::Core
               ROOT::FFTW
               ROOT::Hist
               ROOT::MathCore
-              ROOT::Physics)
+              ROOT::Physics
+              ${FFTW_LIBRARY})
 
-simple_plugin(ComputePi "module"
+cet_build_plugin(ComputePi art::module
+         LIBRARIES
+              messagefacility::headers
               messagefacility::MF_MessageLogger)
 
 include(FindOpenMP)
diff --git a/lardata/Utilities/ComputePi_module.cc b/lardata/Utilities/ComputePi_module.cc
index e1086e4..e6a0f07 100644
--- a/lardata/Utilities/ComputePi_module.cc
+++ b/lardata/Utilities/ComputePi_module.cc
@@ -15,6 +15,7 @@
 #include "fhiclcpp/ParameterSet.h"
 #include "art/Framework/Core/EDAnalyzer.h"
 #include "art/Framework/Core/ModuleMacros.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
 
 
 namespace lar {
diff --git a/test/ArtDataHelper/CMakeLists.txt b/test/ArtDataHelper/CMakeLists.txt
index f0cca4d..116825c 100644
--- a/test/ArtDataHelper/CMakeLists.txt
+++ b/test/ArtDataHelper/CMakeLists.txt
@@ -1,9 +1,8 @@
 art_make(
   MODULE_LIBRARIES
-    lardata_ArtDataHelper
-    lardataobj_RecoBase
+    lardata::ArtDataHelper
+    lardataobj::RecoBase
     messagefacility::MF_MessageLogger
-    
     fhiclcpp::fhiclcpp
     cetlib_except::cetlib_except
 )
diff --git a/test/DetectorInfoServices/CMakeLists.txt b/test/DetectorInfoServices/CMakeLists.txt
index 168fc2e..dcda35c 100644
--- a/test/DetectorInfoServices/CMakeLists.txt
+++ b/test/DetectorInfoServices/CMakeLists.txt
@@ -1,9 +1,10 @@
 art_make(
   MODULE_LIBRARIES
-    larcorealg_Geometry
-    lardataalg_DetectorInfo
+    larcorealg::Geometry
+    lardataalg::DetectorInfo
     art::Framework_Services_Registry
     messagefacility::MF_MessageLogger
+    messagefacility::headers
     ROOT::Core
 )
 
diff --git a/test/RecoBaseProxy/CMakeLists.txt b/test/RecoBaseProxy/CMakeLists.txt
index 3c1ca8d..c0b045f 100644
--- a/test/RecoBaseProxy/CMakeLists.txt
+++ b/test/RecoBaseProxy/CMakeLists.txt
@@ -5,9 +5,10 @@ add_subdirectory(ProxyBase)
 
 simple_plugin(ProxyBaseTest "module"
   lardata_RecoBaseProxy
-  lardataobj_RecoBase
+  lardataobj::RecoBase
   art::Persistency_Provenance
   messagefacility::MF_MessageLogger
+  messagefacility::headers
 
   ROOT::GenVector
   USE_BOOST_UNIT
@@ -26,8 +27,8 @@ cet_test(ProxyBase_test
 ###  TrackProxy tests
 ###
 simple_plugin(TrackProxyTest "module"
-  lardata_RecoBaseProxy
-  lardataobj_RecoBase
+  lardata::RecoBaseProxy
+  lardataobj::RecoBase
   art::Persistency_Provenance
   messagefacility::MF_MessageLogger
 
@@ -43,7 +44,7 @@ simple_plugin(TrackProxyHitMaker "module"
   )
 
 simple_plugin(TrackProxyTrackMaker "module"
-  lardataobj_RecoBase
+  lardataobj::RecoBase
   art::Persistency_Provenance
   messagefacility::MF_MessageLogger
   ROOT::Core
@@ -62,7 +63,7 @@ cet_test(TrackProxy_test
 ###
 simple_plugin(ChargedSpacePointProxyTest "module"
   lardata_RecoBaseProxy
-  lardataobj_RecoBase
+  lardataobj::RecoBase
   art::Persistency_Provenance
   messagefacility::MF_MessageLogger
 
@@ -71,10 +72,9 @@ simple_plugin(ChargedSpacePointProxyTest "module"
   )
 
 simple_plugin(ChargedSpacePointProxyInputMaker "module"
-  lardataobj_RecoBase
+  lardataobj::RecoBase
   lardata_ArtDataHelper
   messagefacility::MF_MessageLogger
-
   USE_BOOST_UNIT
   )
 
diff --git a/test/RecoBaseProxy/ProxyBase/CMakeLists.txt b/test/RecoBaseProxy/ProxyBase/CMakeLists.txt
index f8e1fad..b5480c1 100644
--- a/test/RecoBaseProxy/ProxyBase/CMakeLists.txt
+++ b/test/RecoBaseProxy/ProxyBase/CMakeLists.txt
@@ -1,6 +1,7 @@
 ###############################################################################
 ###  Support utility tests
 ###
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-sign-compare -Wno-pedantic")
 cet_test(AssnsTraits_test
   USE_BOOST_UNIT
   )
diff --git a/test/RecoObjects/CMakeLists.txt b/test/RecoObjects/CMakeLists.txt
index e16fc02..f1541e3 100644
--- a/test/RecoObjects/CMakeLists.txt
+++ b/test/RecoObjects/CMakeLists.txt
@@ -1,9 +1,10 @@
-cet_test( SurfYZTest USE_BOOST_UNIT LIBRARIES lardata_RecoObjects )
-cet_test( SurfXYZTest USE_BOOST_UNIT LIBRARIES lardata_RecoObjects )
-cet_test( SurfYZLineTest USE_BOOST_UNIT LIBRARIES lardata_RecoObjects )
-cet_test( TrackTest LIBRARIES lardata_RecoObjects )
-cet_test( LATest LIBRARIES lardata_RecoObjects )
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-sign-compare -Wno-pedantic")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-misleading-indentation")
+cet_test( SurfYZTest USE_BOOST_UNIT LIBRARIES lardata::RecoObjects )
+cet_test( SurfXYZTest USE_BOOST_UNIT LIBRARIES lardata::RecoObjects )
+cet_test( SurfYZLineTest USE_BOOST_UNIT LIBRARIES lardata::RecoObjects )
+cet_test( TrackTest LIBRARIES lardata::RecoObjects )
+cet_test( LATest LIBRARIES lardata::RecoObjects )
 
 install_headers()
 install_fhicl()
-install_source()
diff --git a/test/Utilities/CMakeLists.txt b/test/Utilities/CMakeLists.txt
index 950c8d8..5f160ce 100644
--- a/test/Utilities/CMakeLists.txt
+++ b/test/Utilities/CMakeLists.txt
@@ -1,8 +1,9 @@
 add_subdirectory( testPtrMaker )
 add_subdirectory( testForEachAssociatedGroup )
 add_subdirectory( testAssnsChainUtils )
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-sign-compare -Wno-pedantic")
 
-# BulkAllocator_test, NestedIterator_test, CountersMap_test 
+# BulkAllocator_test, NestedIterator_test, CountersMap_test
 # and test pure header libraries (they are templates)
 
 # test removed per issue #19494
@@ -30,7 +31,6 @@ cet_test(timingreference_test HANDBUILT
   OPTIONAL_GROUPS ONLY_SLF
 )
 
-
 install_headers()
 install_fhicl()
 install_source()
diff --git a/test/Utilities/testAssnsChainUtils/CMakeLists.txt b/test/Utilities/testAssnsChainUtils/CMakeLists.txt
index 8f64cd4..bb989d6 100644
--- a/test/Utilities/testAssnsChainUtils/CMakeLists.txt
+++ b/test/Utilities/testAssnsChainUtils/CMakeLists.txt
@@ -1,7 +1,7 @@
 simple_plugin(
   AssnsChainHitMaker "module"
-    lardataobj_RecoBase
-    art_Framework_Core
+    lardataobj::RecoBase
+    art::Framework_Core
     art::Framework_Principal
     art::Persistency_Provenance
     messagefacility::MF_MessageLogger
@@ -11,8 +11,8 @@ simple_plugin(
   )
 simple_plugin(
   AssnsChainClusterMaker "module"
-    lardataobj_RecoBase
-    art_Framework_Core
+    lardataobj::RecoBase
+    art::Framework_Core
     art::Framework_Principal
     art::Persistency_Provenance
     messagefacility::MF_MessageLogger
@@ -22,8 +22,8 @@ simple_plugin(
   )
 simple_plugin(
   AssnsChainPFParticleMaker "module"
-    lardataobj_RecoBase
-    art_Framework_Core
+    lardataobj::RecoBase
+    art::Framework_Core
     art::Framework_Principal
     art::Persistency_Provenance
     messagefacility::MF_MessageLogger
@@ -33,19 +33,20 @@ simple_plugin(
   )
 simple_plugin(
   AssnsChainShowerMaker "module"
-    lardataobj_RecoBase
-    art_Framework_Core
+    lardataobj::RecoBase
+    art::Framework_Core
     art::Framework_Principal
     art::Persistency_Provenance
     messagefacility::MF_MessageLogger
     fhiclcpp::fhiclcpp
     cetlib_except::cetlib_except
+    ROOT::Core
   NO_INSTALL
   )
 simple_plugin(
   AssnsChainTest "module"
-    lardataobj_RecoBase
-    art_Framework_Core
+    lardataobj::RecoBase
+    art::Framework_Core
     art::Framework_Principal
     art::Persistency_Provenance
     messagefacility::MF_MessageLogger
diff --git a/test/Utilities/testForEachAssociatedGroup/CMakeLists.txt b/test/Utilities/testForEachAssociatedGroup/CMakeLists.txt
index 4abf69a..59f538b 100644
--- a/test/Utilities/testForEachAssociatedGroup/CMakeLists.txt
+++ b/test/Utilities/testForEachAssociatedGroup/CMakeLists.txt
@@ -1,6 +1,6 @@
-simple_plugin( AssnProducer1          "module"  NO_INSTALL cetlib_except art_Persistency_Provenance)
-simple_plugin( AssnProducer2          "module"  NO_INSTALL cetlib_except art_Persistency_Provenance)
-simple_plugin( AssnAnalyzer           "module"  NO_INSTALL cetlib_except art_Persistency_Provenance)
+simple_plugin( AssnProducer1          "module"  NO_INSTALL cetlib_except::cetlib_except art::Persistency_Provenance)
+simple_plugin( AssnProducer2          "module"  NO_INSTALL cetlib_except::cetlib_except art::Persistency_Provenance)
+simple_plugin( AssnAnalyzer           "module"  NO_INSTALL cetlib_except::cetlib_except art::Persistency_Provenance)
 
 cet_test(associated_groups_test USE_BOOST_UNIT
   LIBRARIES
@@ -55,4 +55,4 @@ cet_test(test_associatedgroupswithleft HANDBUILT
   TEST_PROPERTIES DEPENDS test_feag2
 )
 
-art_dictionary(DICTIONARY_LIBRARIES cetlib::cetlib cetlib_except NO_INSTALL )
+art_dictionary(DICTIONARY_LIBRARIES cetlib::cetlib cetlib_except::cetlib_except NO_INSTALL )
diff --git a/test/Utilities/testPtrMaker/CMakeLists.txt b/test/Utilities/testPtrMaker/CMakeLists.txt
index 8ded552..19691bb 100644
--- a/test/Utilities/testPtrMaker/CMakeLists.txt
+++ b/test/Utilities/testPtrMaker/CMakeLists.txt
@@ -1,7 +1,7 @@
-simple_plugin(IntVectorProducer          "module"  NO_INSTALL cetlib_except art_Persistency_Provenance)
-simple_plugin(PtrMakerProducer           "module"  NO_INSTALL cetlib_except art_Persistency_Provenance)
-simple_plugin(PtrMakerProducer2          "module"  NO_INSTALL cetlib_except art_Persistency_Provenance)
-simple_plugin(PtrMakerAnalyzer           "module"  NO_INSTALL cetlib_except art_Persistency_Provenance)
+simple_plugin(IntVectorProducer          "module"  NO_INSTALL cetlib_except::cetlib_except art::Persistency_Provenance)
+simple_plugin(PtrMakerProducer           "module"  NO_INSTALL cetlib_except::cetlib_except art::Persistency_Provenance)
+simple_plugin(PtrMakerProducer2          "module"  NO_INSTALL cetlib_except::cetlib_except art::Persistency_Provenance)
+simple_plugin(PtrMakerAnalyzer           "module"  NO_INSTALL cetlib_except::cetlib_except art::Persistency_Provenance)
 
 cet_test(test_ptrmaker HANDBUILT
   TEST_EXEC lar 
