diff --git a/CMakeLists.txt b/CMakeLists.txt
index 745168c..085fc9f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,18 +11,16 @@
 #  make install
 #  make package (builds distribution tarfile)
 # ======================================================================
-
-
 cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
 
 
-project(sbnanaobj VERSION 09.17.06.06 LANGUAGES CXX)
 
+find_package(cetmodules 2.29.07 REQUIRED)
+project(sbnanaobj VERSION 09.17.06.06 LANGUAGES CXX)
 
 message(STATUS "\n\n  ==========================   ${PROJECT_NAME}   ==========================")
 
-# cetbuildtools contains our cmake modules
-find_package(cetbuildtools REQUIRED)
+# cetbuildtools contains our cmake modules ### MIGRATE-ACTION-REQUIRED: remove
 
 include(CetCMakeEnv)
 cet_cmake_env()
@@ -39,15 +37,14 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
   EXTRA_FLAGS -pedantic -Wno-unused-local-typedefs -Wno-undefined-var-template -I $ENV{CLANG_FQ_DIR}/include/c++/v1
 )
 
-cet_report_compiler_flags()
+cet_report_compiler_flags(REPORT_THRESHOLD VERBOSE)
 
 # these are minimum required versions, not the actual product versions
-find_ups_root(v6_00_00)
+find_package(ROOT REQUIRED)
 
 include(BuildDictionary)
 
 add_subdirectory(sbnanaobj)
-add_subdirectory(ups)
 
 # packaging utility
-include(UseCPack)
+cet_cmake_config()
diff --git a/sbnanaobj/StandardRecord/CMakeLists.txt b/sbnanaobj/StandardRecord/CMakeLists.txt
index 121c262..34463d7 100644
--- a/sbnanaobj/StandardRecord/CMakeLists.txt
+++ b/sbnanaobj/StandardRecord/CMakeLists.txt
@@ -1,5 +1,5 @@
 # for classes_def.xml!!
-include_directories( ${CMAKE_CURRENT_SOURCE_DIR} )
+include_directories( ${CMAKE_CURRENT_SOURCE_DIR} ) ### MIGRATE-ACTION-RECOMMENDED: use target_link_directories() with target semantics
 
 set( PACKAGE sbnanaobj_StandardRecord )
 
@@ -10,7 +10,7 @@ FILE( GLOB src_files *.cxx )
 
 cet_make_library( LIBRARY_NAME sbnanaobj_StandardRecord
                   SOURCE       ${src_files}
-                  LIBRARIES    ${ROOT_BASIC_LIB_LIST}
+                  LIBRARIES    ROOT::Core ROOT::Physics
                 )
 
 build_dictionary( sbnanaobj_StandardRecord
diff --git a/sbnanaobj/StandardRecord/Flat/CMakeLists.txt b/sbnanaobj/StandardRecord/Flat/CMakeLists.txt
index 35cfc7c..588aa23 100644
--- a/sbnanaobj/StandardRecord/Flat/CMakeLists.txt
+++ b/sbnanaobj/StandardRecord/Flat/CMakeLists.txt
@@ -13,4 +13,4 @@ cet_make_library(LIBRARY_NAME sbnanaobj_StandardRecordFlat
                  LIBRARIES    ${ROOT_BASIC_LIB_LIST} ROOT::TreePlayer
                  )
 
-install_headers(EXTRAS $ENV{MRB_BUILDDIR}/sbnanaobj/sbnanaobj/StandardRecord/Flat/FlatRecord.h $ENV{MRB_BUILDDIR}/sbnanaobj/sbnanaobj/StandardRecord/Flat/FwdDeclare.h)
+install_headers(EXTRAS ${CMAKE_BINARY_DIR}/sbnanaobj/StandardRecord/Flat/FlatRecord.h ${CMAKE_BINARY_DIR}/sbnanaobj/StandardRecord/Flat/FwdDeclare.h)
diff --git a/sbnanaobj/StandardRecord/Proxy/CMakeLists.txt b/sbnanaobj/StandardRecord/Proxy/CMakeLists.txt
index 310cdae..beb8b37 100644
--- a/sbnanaobj/StandardRecord/Proxy/CMakeLists.txt
+++ b/sbnanaobj/StandardRecord/Proxy/CMakeLists.txt
@@ -15,4 +15,4 @@ cet_make_library(LIBRARY_NAME sbnanaobj_StandardRecordProxy
                  SOURCE       SRProxy.cxx Instantiations.cxx
                  LIBRARIES    ${ROOT_BASIC_LIB_LIST} ROOT::TreePlayer)
 
-install_headers(EXTRAS $ENV{MRB_BUILDDIR}/sbnanaobj/sbnanaobj/StandardRecord/Proxy/SRProxy.h $ENV{MRB_BUILDDIR}/sbnanaobj/sbnanaobj/StandardRecord/Proxy/FwdDeclare.h)
+install_headers(EXTRAS ${CMAKE_BINARY_DIR}/sbnanaobj/StandardRecord/Proxy/SRProxy.h ${CMAKE_BINARY_DIR}/sbnanaobj/StandardRecord/Proxy/FwdDeclare.h)
