diff --git a/CMakeLists.txt b/CMakeLists.txt
index b94c69a..1e0a933 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,18 +16,9 @@
 # use cmake 3.3 or later
 cmake_minimum_required (VERSION 3.3)
 
-project(larcoreobj)
-
-# cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
-IF (NOT CETBUILDTOOLS_VERSION)
-    MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
-ENDIF()
-
-set(CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules
-		      $ENV{CETBUILDTOOLS_DIR}/Modules
-		      ${CMAKE_MODULE_PATH})
+project(larcoreobj VERSION 08.02.02 LANGUAGES CXX)
 
+find_package(cetmodules)
 include(CetCMakeEnv)
 cet_cmake_env()
 
@@ -40,9 +31,8 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( canvas )
-find_ups_product( cetbuildtools )
-find_ups_root( )
+find_package(canvas_root_io REQUIRED)
+
 
 # macros for artdaq_dictionary and simple_plugin
 include(ArtDictionary)
@@ -55,8 +45,4 @@ add_subdirectory(larcoreobj)
 # tests 
 add_subdirectory(test)
 
-# ups - table and config files
-add_subdirectory(ups)
-
-# packaging utility
-include(UseCPack)
+cet_cmake_config()
