diff --git a/CMakeLists.txt b/CMakeLists.txt
index 765bdb0..b52b785 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,12 +17,15 @@
 
 
 # use cmake 2.8 or later
-cmake_minimum_required (VERSION 2.8)
+cmake_minimum_required(VERSION 3.11)
 
-project(dunetpc)
+project(dunetpc VERSION 08.59.00 LANGUAGES CXX C)
+#some need this for install_fhicl(), install_gdml()
+set(fcl_dir job)
+set(gdml_dir gdml)
 
 # cetbuildtools contains our cmake modules
-SET ( CETBUILDTOOLS_VERSION $ENV{CETBUILDTOOLS_VERSION} )
+SET ( CETBUILDTOOLS_VERSION $CMAKE_CACHE_MAJOR_VERSION )
 IF (NOT CETBUILDTOOLS_VERSION)
     MESSAGE (FATAL_ERROR "ERROR: setup cetbuildtools to get the cmake modules")
 ENDIF()
@@ -31,6 +34,7 @@ set(CMAKE_MODULE_PATH $ENV{CANVAS_ROOT_IO_DIR}/Modules
                       $ENV{CETBUILDTOOLS_DIR}/Modules
                       ${CMAKE_MODULE_PATH})
 
+find_package(cetmodules)
 include(CetCMakeEnv)
 cet_cmake_env()
 
@@ -46,55 +50,52 @@ cet_set_compiler_flags(DIAGS CAUTIOUS
 cet_report_compiler_flags()
 
 # these are minimum required versions, not the actual product versions
-find_ups_product( larcoreobj )
-find_ups_product( larcore v1_00_00 )
-find_ups_product( larevt v1_00_00 )
-find_ups_product( lardataobj )
-find_ups_product( lardataalg )
-find_ups_product( lardata v1_00_00 )
-find_ups_product( larpandora v1_00_00 )
-find_ups_product( pandora )
-find_ups_product( larreco v1_00_00 )
-find_ups_product( larana v1_00_00 )
-find_ups_product( nutools v1_00_00 )
-find_ups_product( nuevdb )
-find_ups_product( nug4 )
-find_ups_product( nugen )
-find_ups_product( genie )
-find_ups_product( nurandom )
-find_ups_product( art v1_09_00 )
-find_ups_product(art_root_io)
-find_ups_product( clhep v2_1_3_0 )
-find_ups_product( messagefacility v1_10_00 )
-find_ups_product( cetbuildtools v3_10_00 )
-find_ups_product( postgresql v9_1_5 )
-find_ups_product( dune_raw_data )
-find_ups_product( dunepdsprce v0_0_3 )
-find_ups_product( artdaq_core v01_03_6 )
-find_ups_product( ifdh_art )
-find_ups_product( ifdhc )
-find_ups_geant4( )
-find_ups_product( systematicstools v00_01_01 )
-find_ups_product( nusystematics v00_01_01 )
+find_package( larcoreobj )
+find_package( larcore )
+find_package( larevt )
+find_package( lardataobj )
+find_package( lardataalg )
+find_package( lardata )
+find_package( larpandora )
+find_package( pandora )
+find_package( larreco )
+find_package( larana )
+find_package( nutools )
+find_package( nuevdb )
+find_package( nug4 )
+find_package( nugen )
+find_package( genie )
+find_package( nurandom )
+find_package( art )
+find_package( art_root_io )
+find_package( CLHEP )
+find_package( messagefacility )
+find_package( cetmodules )
+find_package( postgresql )
+find_package( dune_raw_data )
+find_package( dunepdsprce )
+find_package( artdaq_core )
+find_package( ifdh_art )
+cet_find_simple_package( ifdh INCPATH_SUFFIXES inc INCPATH_VAR IFDHC_INC )
+find_package( Geant4 )
+find_package( systematicstools )
+find_package( nusystematics )
 if(DEFINED ENV{CAFFE_LIB} )
-  find_ups_product(caffe)
+find_package( caffe )
 endif()
-if(DEFINED ENV{LAPACK_LIB} )
-  find_ups_product( lapack )
-endif()
-if(( DEFINED ENV{TENSORFLOW_DIR} ) AND ( DEFINED ENV{TRTIS_CLIENTS_DIR} ))
+find_package( lapack )
   # find tensorflow library
-  find_ups_product( protobuf )
-  find_ups_product( tensorflow )
-  find_ups_product( trtis_clients )
-  find_ups_product( larrecodnn v1_00_00 )
-  cet_find_library(TENSORFLOW NAMES tensorflow-core PATHS ENV TENSORFLOW_LIB NO_DEFAULT_PATH )
-  cet_find_library(PROTOBUF NAMES protobuf PATHS ENV PROTOBUF_LIB NO_DEFAULT_PATH )
-endif()
+find_package( protobuf )
+find_package( tensorflow )
+find_package( trtis_clients )
+find_package( larrecodnn )
+cet_find_library(TENSORFLOW NAMES tensorflow-core)
+cet_find_library(PROTOBUF NAMES protobuf)
 # artdaq_core Utility library
-cet_find_library( ARTDAQ-CORE_UTILITIES NAMES artdaq-core_Utilities PATHS ENV ARTDAQ_CORE_LIB )
+find_package( artdaq-core )
 
 # macros for dictionary and simple_plugin
+find_package(canvas_root_io)
 include(ArtDictionary)
 include(ArtMake)
 include(BuildPlugins)
@@ -106,9 +107,8 @@ add_subdirectory(releaseDB)
 # subdirectory for test code
 add_subdirectory(test)
 # ups - table and config files
-add_subdirectory(ups)
 # subdirectory for fcl files
 add_subdirectory(fcl)
 
 # packaging utility
-include(UseCPack)
+cet_cmake_config()
diff --git a/dune/3x1x1dp/DataImport/ImportFullFile/CMakeLists.txt b/dune/3x1x1dp/DataImport/ImportFullFile/CMakeLists.txt
index 18e54f7..f76bb60 100644
--- a/dune/3x1x1dp/DataImport/ImportFullFile/CMakeLists.txt
+++ b/dune/3x1x1dp/DataImport/ImportFullFile/CMakeLists.txt
@@ -12,7 +12,7 @@ simple_plugin(ImportFull311File "source"
 			art_Utilities canvas
 			${FHICLCPP}
 			${MF_MESSAGELOGGER}
-			
+
 			${CETLIB}
 			${ROOT_BASIC_LIB_LIST}
 			BASENAME_ONLY
diff --git a/dune/3x1x1dp/DataImport/ImportSingleEvent/CMakeLists.txt b/dune/3x1x1dp/DataImport/ImportSingleEvent/CMakeLists.txt
index 45b8dca..26bb89f 100644
--- a/dune/3x1x1dp/DataImport/ImportSingleEvent/CMakeLists.txt
+++ b/dune/3x1x1dp/DataImport/ImportSingleEvent/CMakeLists.txt
@@ -10,7 +10,7 @@ simple_plugin(ImportSingle311Event "module"
 			larcore_Geometry_Geometry_service
 			lardata_Utilities
 			larsim_Simulation nug4_ParticleNavigation lardataobj_Simulation
-			
+
 			nusimdata_SimulationBase
 			${ART_FRAMEWORK_CORE}
 			${ART_FRAMEWORK_PRINCIPAL}
@@ -21,7 +21,7 @@ simple_plugin(ImportSingle311Event "module"
 			art_Persistency_Provenance canvas
 			art_Utilities canvas
 			${MF_MESSAGELOGGER}
-			
+
 			${CETLIB}
 			${ROOT_BASIC_LIB_LIST}
 			BASENAME_ONLY
diff --git a/dune/3x1x1dp/DataImport/Services/CMakeLists.txt b/dune/3x1x1dp/DataImport/Services/CMakeLists.txt
index 969ee50..344dfc3 100644
--- a/dune/3x1x1dp/DataImport/Services/CMakeLists.txt
+++ b/dune/3x1x1dp/DataImport/Services/CMakeLists.txt
@@ -13,7 +13,7 @@ simple_plugin(RawData311InputDriver "service"
 			art_Persistency_Provenance canvas
 			art_Utilities canvas
 			${MF_MESSAGELOGGER}
-			
+
 			${CETLIB}
 			${ROOT_BASIC_LIB_LIST}
 			BASENAME_ONLY
@@ -29,7 +29,7 @@ simple_plugin(dlardaq "service"
 			larcore_Geometry_Geometry_service
 			lardata_Utilities
 			larsim_Simulation nug4_ParticleNavigation lardataobj_Simulation
-			
+
 			nusimdata_SimulationBase
 			${ART_FRAMEWORK_CORE}
 			${ART_FRAMEWORK_PRINCIPAL}
@@ -40,7 +40,7 @@ simple_plugin(dlardaq "service"
 			art_Persistency_Provenance canvas
 			art_Utilities canvas
 			${MF_MESSAGELOGGER}
-			
+
 			${CETLIB}
 			${ROOT_BASIC_LIB_LIST}
 			BASENAME_ONLY
@@ -50,7 +50,7 @@ simple_plugin(dlardaq "service"
 simple_plugin(EventDecoder "service"
 			dlardaq_service
 			HuffDataCompressor_service
-			
+
 			lardataobj_RawData
 			lardataobj_RecoBase
 			larreco_RecoAlg
@@ -60,7 +60,7 @@ simple_plugin(EventDecoder "service"
 			larcore_Geometry_Geometry_service
 			lardata_Utilities
 			larsim_Simulation nug4_ParticleNavigation lardataobj_Simulation
-			
+
 			nusimdata_SimulationBase
 			${ART_FRAMEWORK_CORE}
 			${ART_FRAMEWORK_PRINCIPAL}
@@ -71,7 +71,7 @@ simple_plugin(EventDecoder "service"
 			art_Persistency_Provenance canvas
 			art_Utilities canvas
 			${MF_MESSAGELOGGER}
-			
+
 			${CETLIB}
 			${ROOT_BASIC_LIB_LIST}
 			BASENAME_ONLY
@@ -90,7 +90,7 @@ simple_plugin(HuffDataCompressor "service"
 			larcore_Geometry_Geometry_service
 			lardata_Utilities
 			larsim_Simulation nug4_ParticleNavigation lardataobj_Simulation
-			
+
 			nusimdata_SimulationBase
 			${ART_FRAMEWORK_CORE}
 			${ART_FRAMEWORK_PRINCIPAL}
@@ -101,7 +101,7 @@ simple_plugin(HuffDataCompressor "service"
 			art_Persistency_Provenance canvas
 			art_Utilities canvas
 			${MF_MESSAGELOGGER}
-			
+
 			${CETLIB}
 			${ROOT_BASIC_LIB_LIST}
 			BASENAME_ONLY
diff --git a/dune/AnaTree/CMakeLists.txt b/dune/AnaTree/CMakeLists.txt
index 0bd52f1..eb5ad7d 100644
--- a/dune/AnaTree/CMakeLists.txt
+++ b/dune/AnaTree/CMakeLists.txt
@@ -26,7 +26,7 @@ art_make( MODULE_LIBRARIES larcorealg_Geometry
 	  art_Persistency_Provenance canvas
 	  art_Utilities canvas
 	  ${MF_MESSAGELOGGER}
-	  
+
 	  ${FHICLCPP}
 	  cetlib cetlib_except
           ${ROOT_GEOM}
diff --git a/dune/AnaUtils/CMakeLists.txt b/dune/AnaUtils/CMakeLists.txt
index 2b6c955..19edd48 100644
--- a/dune/AnaUtils/CMakeLists.txt
+++ b/dune/AnaUtils/CMakeLists.txt
@@ -1,4 +1,4 @@
-art_make( 
+art_make(
     LIB_LIBRARIES larcorealg_Geometry
     larcore_Geometry_Geometry_service
     larsim_Simulation lardataobj_Simulation
diff --git a/dune/BeamData/CMakeLists.txt b/dune/BeamData/CMakeLists.txt
index 1de9d13..59086c0 100644
--- a/dune/BeamData/CMakeLists.txt
+++ b/dune/BeamData/CMakeLists.txt
@@ -4,13 +4,13 @@ include_directories( $ENV{IFDH_ART_INC} )
 include_directories( $ENV{IFDHC_FQ_DIR}/inc )
 include_directories( $ENV{IFBEAM_FQ_DIR}/include )
 
-cet_find_library( WDALIB NAMES wda PATHS ENV LIBWDA_LIB NO_DEFAULT_PATH )
+cet_find_library( WDALIB NAMES wda)
 
 FILE (GLOB src_files *.cxx)
 
 art_make_library(  LIBRARY_NAME  BeamData
 		             SOURCE ${src_files}
-			     LIBRARIES			
+			     LIBRARIES
 	                wda
 			${IFDHC}
 			${IFHC_ART}
@@ -25,7 +25,7 @@ art_make_library(  LIBRARY_NAME  BeamData
 			art_Persistency_Provenance canvas
 			art_Utilities canvas
                         ${MF_MESSAGELOGGER}
-                        
+
                         cetlib cetlib_except
 			${FHICLCPP}
                         ${ROOT_BASIC_LIB_LIST}
diff --git a/dune/CFilter/CMakeLists.txt b/dune/CFilter/CMakeLists.txt
index b03b635..5b53237 100644
--- a/dune/CFilter/CMakeLists.txt
+++ b/dune/CFilter/CMakeLists.txt
@@ -16,7 +16,7 @@ simple_plugin(CFilter "module"
 			art_Persistency_Provenance canvas
 			art_Utilities canvas
 			${MF_MESSAGELOGGER}
-			
+
 			cetlib cetlib_except
 			${ROOT_BASIC_LIB_LIST}
               BASENAME_ONLY
diff --git a/dune/CVN/adcutils/CMakeLists.txt b/dune/CVN/adcutils/CMakeLists.txt
index e5f8c31..d948fc4 100644
--- a/dune/CVN/adcutils/CMakeLists.txt
+++ b/dune/CVN/adcutils/CMakeLists.txt
@@ -1,7 +1,7 @@
 
 if( DEFINED ENV{TENSORFLOW_DIR} )
 
-art_make_library( 
+art_make_library(
   LIBRARY_NAME     GlobImage
   SOURCE EventImageData.h EventImageData.cxx
   LIBRARIES    dune_CVN_func
@@ -27,7 +27,7 @@ simple_plugin(SPMultiTpcDump "module"
 			art_Utilities
 			canvas
 			${MF_MESSAGELOGGER}
-			
+
 			cetlib cetlib_except
 			${ROOT_BASIC_LIB_LIST}
       ${GLOBIMAGE}
diff --git a/dune/CVN/art/CMakeLists.txt b/dune/CVN/art/CMakeLists.txt
index 3a30485..4c316c3 100644
--- a/dune/CVN/art/CMakeLists.txt
+++ b/dune/CVN/art/CMakeLists.txt
@@ -8,7 +8,7 @@ include_directories( $ENV{GSL_INC}/gsl )
 #include_directories( $ENV{GLOG_INC} )
 
 #cet_find_library( CAFFELIB NAMES caffe-d caffe PATHS ENV CAFFE_LIB NO_DEFAULT_PATH )
-cet_find_library( GSLLIB NAMES gsl PATHS ENV GSL_LIB NO_DEFAULT_PATH )
+cet_find_library( GSLLIB NAMES gsl)
 #cet_find_library( PROTOBUFLIB NAMES protobuf PATHS ENV PROTOBUF_LIB NO_DEFAULT_PATH )
 #cet_find_library( LEVELDBLIB NAMES leveldb PATHS ENV LEVELDB_LIB NO_DEFAULT_PATH )
 #cet_find_library( GFLAGSLIB NAMES gflags PATHS ENV GFLAGS_LIB NO_DEFAULT_PATH )
@@ -30,8 +30,8 @@ endif (DEFINED ENV{TENSORFLOW_DIR})
 if (DEFINED ENV{HEP_HPC_DIR} AND DEFINED ENV{HDF5_DIR})
 include_directories( $ENV{HEP_HPC_INC} )
 include_directories( $ENV{HDF5_INC} )
-cet_find_library( HEPHPCLIB NAMES hep_hpc_hdf5 PATHS ENV HEP_HPC_LIB NO_DEFAULT_PATH )
-cet_find_library( HDF5LIB NAMES hdf5 PATHS ENV HDF5_LIB NO_DEFAULT_PATH )
+cet_find_library( HEPHPCLIB NAMES hep_hpc_hdf5)
+cet_find_library( HDF5LIB NAMES hdf5)
 else (DEFINED ENV{HEP_HPC_DIR} AND DEFINED ENV{HDF5_DIR})
 set (EXCLUDE_HEPHPC GCNH5_module.cc)
 endif (DEFINED ENV{HEP_HPC_DIR} AND DEFINED ENV{HDF5_DIR})
diff --git a/dune/CVN/func/CMakeLists.txt b/dune/CVN/func/CMakeLists.txt
index d0141f4..f0b12e5 100644
--- a/dune/CVN/func/CMakeLists.txt
+++ b/dune/CVN/func/CMakeLists.txt
@@ -8,7 +8,7 @@ art_make( BASENAME_ONLY
   larcorealg_Geometry
   larcorealg_GeoAlgo
   lardataobj_RecoBase
-  larsim_MCCheater_BackTrackerService_service 
+  larsim_MCCheater_BackTrackerService_service
   larsim_MCCheater_ParticleInventoryService_service
   ${ART_FRAMEWORK_CORE}
   ${ART_FRAMEWORK_PRINCIPAL}
@@ -21,7 +21,7 @@ art_make( BASENAME_ONLY
   art_Utilities canvas
   cetlib cetlib_except
   canvas
-  ${Boost_FILESYSTEM_LIBRARY}            
+  ${Boost_FILESYSTEM_LIBRARY}
   ${Boost_SYSTEM_LIBRARY}
   ${ROOT_BASIC_LIB_LIST}
   DICT_LIBRARIES   lardataobj_RecoBase
diff --git a/dune/CVN/levelDB/CMakeLists.txt b/dune/CVN/levelDB/CMakeLists.txt
index 460681f..33bdce5 100644
--- a/dune/CVN/levelDB/CMakeLists.txt
+++ b/dune/CVN/levelDB/CMakeLists.txt
@@ -15,14 +15,14 @@ include_directories( $ENV{BOOST_INC} )
 #cet_find_library( PROTOBUFLIB NAMES protobuf PATHS ENV PROTOBUF_LIB NO_DEFAULT_PATH )
 #cet_find_library( LEVELDBLIB NAMES leveldb PATHS ENV LEVELDB_LIB NO_DEFAULT_PATH )
 #cet_find_library( LMDBLIB NAMES lmdb PATHS ENV LMDB_LIB NO_DEFAULT_PATH )
-cet_find_library( BOOSTLIB NAMES boost_program_options PATHS ENV BOOST_LIB NO_DEFAULT_PATH )
+cet_find_library( BOOSTLIB NAMES boost_program_options)
 #cet_find_library( GFLAGSLIB NAMES gflags PATHS ENV GFLAGS_LIB NO_DEFAULT_PATH )
 #cet_find_library( GLOGLIB NAMES glog PATHS ENV GLOG_LIB NO_DEFAULT_PATH )
 
 #message("HDF5: ${HDF5_CPPLIB}")
 message("Boost manual lib      : ${BOOSTLIB}")
 
-art_make_exec( cvnCreateZlibImages       
+art_make_exec( cvnCreateZlibImages
                SOURCE    cvnCreateZlibImages.cc
                LIBRARIES ${ART_FRAMEWORK_CORE}
                          ${ART_FRAMEWORK_PRINCIPAL}
diff --git a/dune/CalData/CMakeLists.txt b/dune/CalData/CMakeLists.txt
index f9f9097..0637ce1 100644
--- a/dune/CalData/CMakeLists.txt
+++ b/dune/CalData/CMakeLists.txt
@@ -19,7 +19,7 @@ art_make( BASENAME_ONLY
 			art_Persistency_Provenance canvas
 			art_Utilities canvas
 			${MF_MESSAGELOGGER}
-			
+
 	                ${FHICLCPP}
 			cetlib cetlib_except
                         ${ROOT_GEOM}
diff --git a/dune/Calib/CMakeLists.txt b/dune/Calib/CMakeLists.txt
index 3462082..4ba15bb 100644
--- a/dune/Calib/CMakeLists.txt
+++ b/dune/Calib/CMakeLists.txt
@@ -2,7 +2,7 @@ include_directories ( ${CMAKE_CURRENT_SOURCE_DIR} )
 include_directories ( $ENV{NUTOOLS_DIR}/include )
 include_directories( $ENV{LIBWDA_FQ_DIR}/include )
 #SET(ENV{LIBWDA_LIB} "$ENV{LIBWDA_DIR}/lib")
-cet_find_library( WDALIB NAMES wda PATHS ENV LIBWDA_LIB NO_DEFAULT_PATH )
+cet_find_library( WDALIB NAMES wda)
 
 FILE (GLOB src_files *.cxx)
 
@@ -17,15 +17,15 @@ art_make_library(  LIBRARY_NAME  dune_Calib
                         ${ROOT_BASIC_LIB_LIST}
 			${Boost_SYSTEM_LIBRARY}
         )
- 
+
 art_make_exec( NAME getXYZCalibProtoDUNE
-               SOURCE getXYZCalibProtoDUNE.cc                 
+               SOURCE getXYZCalibProtoDUNE.cc
                LIBRARIES
 			dune_Calib
                )
 
 art_make_exec( NAME getLinConstsProtoDUNE
-               SOURCE getLinConstsProtoDUNE.cc                 
+               SOURCE getLinConstsProtoDUNE.cc
                LIBRARIES
 			dune_Calib
                )
diff --git a/dune/CalibServices/CMakeLists.txt b/dune/CalibServices/CMakeLists.txt
index 20775ba..6387999 100644
--- a/dune/CalibServices/CMakeLists.txt
+++ b/dune/CalibServices/CMakeLists.txt
@@ -11,10 +11,10 @@ simple_plugin(XYZCalibServiceProtoDUNE "service"
 	      ${ART_FRAMEWORK_SERVICES_SYSTEM_FILECATALOGMETADATA_SERVICE}
 	      art_Utilities canvas
 	      ${MF_MESSAGELOGGER}
-	      
+
   	      cetlib cetlib_except
    	      ${ROOT_BASIC_LIB_LIST}
-              
+
 )
 
 simple_plugin(LifetimeCalibServiceProtoDUNE "service"
@@ -34,7 +34,7 @@ simple_plugin(LifetimeCalibServiceProtoDUNE "service"
       cetlib cetlib_except
       ${ROOT_BASIC_LIB_LIST}
 
-) 
+)
 
 
 install_headers()
diff --git a/dune/DAQSimAna/CMakeLists.txt b/dune/DAQSimAna/CMakeLists.txt
index 19d3ddd..e31337c 100644
--- a/dune/DAQSimAna/CMakeLists.txt
+++ b/dune/DAQSimAna/CMakeLists.txt
@@ -13,7 +13,7 @@ art_make( BASENAME_ONLY MODULE_LIBRARIES
   larsim_MCCheater_PhotonBackTrackerService_service
   larsim_Simulation nug4_ParticleNavigation lardataobj_Simulation
   lardata_ArtDataHelper
-  nusimdata_SimulationBase	
+  nusimdata_SimulationBase
   ${ART_FRAMEWORK_CORE}
   ${ART_FRAMEWORK_PRINCIPAL}
   ${ART_FRAMEWORK_SERVICES_BASIC}
@@ -24,7 +24,7 @@ art_make( BASENAME_ONLY MODULE_LIBRARIES
   art_Persistency_Provenance canvas
   art_Utilities canvas
   ${MF_MESSAGELOGGER}
-  
+
   cetlib cetlib_except
   ${ROOT_BASIC_LIB_LIST}
   EXCLUDE read_packed.cxx TriggerPrimitiveFinderPass1_tool.cc TriggerPrimitiveFinderPass2_tool.cc TriggerPrimitiveFinderTemplate_tool.cc
diff --git a/dune/DAQSimAna/SNAnaClustering/CMakeLists.txt b/dune/DAQSimAna/SNAnaClustering/CMakeLists.txt
index 087302a..6c8bdac 100644
--- a/dune/DAQSimAna/SNAnaClustering/CMakeLists.txt
+++ b/dune/DAQSimAna/SNAnaClustering/CMakeLists.txt
@@ -8,7 +8,7 @@ art_make( BASENAME_ONLY MODULE_LIBRARIES
   larsim_Simulation
   nug4_ParticleNavigation
   lardataobj_Simulation
-  nusimdata_SimulationBase	
+  nusimdata_SimulationBase
   ${ART_FRAMEWORK_CORE}
   ${ART_FRAMEWORK_PRINCIPAL}
   ${ART_FRAMEWORK_SERVICES_BASIC}
diff --git a/dune/DAQTriggerSim/AnalyzeTriggers/CMakeLists.txt b/dune/DAQTriggerSim/AnalyzeTriggers/CMakeLists.txt
index ecae3bc..f64ffd6 100644
--- a/dune/DAQTriggerSim/AnalyzeTriggers/CMakeLists.txt
+++ b/dune/DAQTriggerSim/AnalyzeTriggers/CMakeLists.txt
@@ -5,7 +5,7 @@ art_make( BASENAME_ONLY
                            larcore_Geometry_Geometry_service
                            larcorealg_Geometry
                            #lardataobj_RawData
-                           nusimdata_SimulationBase	
+                           nusimdata_SimulationBase
                            ${ART_FRAMEWORK_CORE}
                            ${ART_FRAMEWORK_PRINCIPAL}
                            ${ART_FRAMEWORK_SERVICES_REGISTRY}
@@ -15,7 +15,7 @@ art_make( BASENAME_ONLY
                            art_Persistency_Provenance canvas
                            art_Utilities canvas
                            ${MF_MESSAGELOGGER}
-                           
+
                            cetlib cetlib_except
                            ${ROOT_BASIC_LIB_LIST}
 	 )
diff --git a/dune/DAQTriggerSim/HardwareMapperTest/CMakeLists.txt b/dune/DAQTriggerSim/HardwareMapperTest/CMakeLists.txt
index 67f9798..e78edc6 100644
--- a/dune/DAQTriggerSim/HardwareMapperTest/CMakeLists.txt
+++ b/dune/DAQTriggerSim/HardwareMapperTest/CMakeLists.txt
@@ -4,7 +4,7 @@ art_make( BASENAME_ONLY
           larcore_Geometry_Geometry_service
           larcorealg_Geometry
           lardataobj_RawData
-          nusimdata_SimulationBase	
+          nusimdata_SimulationBase
           ${ART_FRAMEWORK_CORE}
           ${ART_FRAMEWORK_PRINCIPAL}
           ${ART_FRAMEWORK_SERVICES_REGISTRY}
@@ -14,7 +14,7 @@ art_make( BASENAME_ONLY
           art_Persistency_Provenance canvas
           art_Utilities canvas
           ${MF_MESSAGELOGGER}
-          
+
           cetlib cetlib_except
           ${ROOT_BASIC_LIB_LIST}
         )
diff --git a/dune/DAQTriggerSim/TriggerAlgorithms/CMakeLists.txt b/dune/DAQTriggerSim/TriggerAlgorithms/CMakeLists.txt
index 4673a96..bba8339 100644
--- a/dune/DAQTriggerSim/TriggerAlgorithms/CMakeLists.txt
+++ b/dune/DAQTriggerSim/TriggerAlgorithms/CMakeLists.txt
@@ -18,7 +18,7 @@ art_make( BASENAME_ONLY
   art_Persistency_Provenance canvas
   art_Utilities canvas
   cetlib cetlib_except
-  ${Boost_FILESYSTEM_LIBRARY}            
+  ${Boost_FILESYSTEM_LIBRARY}
   ${Boost_SYSTEM_LIBRARY}
   ${ROOT_BASIC_LIB_LIST}
   ${MF_MESSAGELOGGER}
diff --git a/dune/DAQTriggerSim/TriggerDataProducts/CMakeLists.txt b/dune/DAQTriggerSim/TriggerDataProducts/CMakeLists.txt
index 83db4e3..70b3f34 100644
--- a/dune/DAQTriggerSim/TriggerDataProducts/CMakeLists.txt
+++ b/dune/DAQTriggerSim/TriggerDataProducts/CMakeLists.txt
@@ -1,12 +1,12 @@
 # for classes_def.xml:
 include_directories( ${CMAKE_CURRENT_SOURCE_DIR} )
 
-art_make( 
+art_make(
           LIB_LIBRARIES lardataobj_RawData
 	  art_Persistency_Common canvas
 	  art_Persistency_Provenance canvas
 	  ${MF_MESSAGELOGGER}
-	  
+
 	  cetlib cetlib_except
 	  ${ROOT_BASIC_LIB_LIST}
 	)
diff --git a/dune/DAQTriggerSim/TriggerModules/CMakeLists.txt b/dune/DAQTriggerSim/TriggerModules/CMakeLists.txt
index 7216f70..17b0be7 100644
--- a/dune/DAQTriggerSim/TriggerModules/CMakeLists.txt
+++ b/dune/DAQTriggerSim/TriggerModules/CMakeLists.txt
@@ -4,7 +4,7 @@ art_make( BASENAME_ONLY
                            larcore_Geometry_Geometry_service
                            larcorealg_Geometry
                            #lardataobj_RawData
-                           nusimdata_SimulationBase	
+                           nusimdata_SimulationBase
                            ${ART_FRAMEWORK_CORE}
                            ${ART_FRAMEWORK_PRINCIPAL}
                            ${ART_FRAMEWORK_SERVICES_REGISTRY}
@@ -14,7 +14,7 @@ art_make( BASENAME_ONLY
                            art_Persistency_Provenance canvas
                            art_Utilities canvas
                            ${MF_MESSAGELOGGER}
-                           
+
                            cetlib cetlib_except
                            ${ROOT_BASIC_LIB_LIST}
 	 )
diff --git a/dune/DUNEPandora/CMakeLists.txt b/dune/DUNEPandora/CMakeLists.txt
index 3060b09..206c754 100644
--- a/dune/DUNEPandora/CMakeLists.txt
+++ b/dune/DUNEPandora/CMakeLists.txt
@@ -26,7 +26,7 @@ art_make( BASENAME_ONLY
                         art_Persistency_Provenance canvas
                         art_Utilities canvas
                         ${MF_MESSAGELOGGER}
-                        
+
                         ${FHICLCPP}
                         cetlib cetlib_except
                         ${ROOT_BASIC_LIB_LIST}
diff --git a/dune/DataPrep/Module/CMakeLists.txt b/dune/DataPrep/Module/CMakeLists.txt
index f758fe9..5ecf61f 100644
--- a/dune/DataPrep/Module/CMakeLists.txt
+++ b/dune/DataPrep/Module/CMakeLists.txt
@@ -21,7 +21,7 @@ art_make( BASENAME_ONLY
 			art_Persistency_Provenance canvas
 			art_Utilities canvas
 			${MF_MESSAGELOGGER}
-			
+
 	                ${FHICLCPP}
 			cetlib cetlib_except
                         ${ROOT_GEOM}
diff --git a/dune/DetSim/CMakeLists.txt b/dune/DetSim/CMakeLists.txt
index aa80608..6a47495 100644
--- a/dune/DetSim/CMakeLists.txt
+++ b/dune/DetSim/CMakeLists.txt
@@ -28,7 +28,7 @@ art_make( BASENAME_ONLY
 			    art_Persistency_Provenance canvas
 			    art_Utilities canvas
 			    ${MF_MESSAGELOGGER}
-			    
+
 	        ${FHICLCPP}
 			    cetlib cetlib_except
 			    ${CLHEP}
diff --git a/dune/DetSim/Module/CMakeLists.txt b/dune/DetSim/Module/CMakeLists.txt
index 1d81d30..c9498d6 100644
--- a/dune/DetSim/Module/CMakeLists.txt
+++ b/dune/DetSim/Module/CMakeLists.txt
@@ -28,7 +28,7 @@ art_make(
     art_Persistency_Provenance canvas
     art_Utilities canvas
     ${MF_MESSAGELOGGER}
-    
+
     ${FHICLCPP}
     cetlib cetlib_except
     ${CLHEP}
diff --git a/dune/EnergyStudies/CMakeLists.txt b/dune/EnergyStudies/CMakeLists.txt
index 9a9407e..c7f16ba 100644
--- a/dune/EnergyStudies/CMakeLists.txt
+++ b/dune/EnergyStudies/CMakeLists.txt
@@ -13,7 +13,7 @@ simple_plugin(dunezsanalysis "module"
   art_Persistency_Provenance canvas
   art_Utilities canvas
   ${MF_MESSAGELOGGER}
-  
+
   cetlib cetlib_except
   ${ROOT_BASIC_LIB_LIST}
   BASENAME_ONLY
@@ -36,7 +36,7 @@ simple_plugin(dEdx "module"
   art_Persistency_Provenance canvas
   art_Utilities canvas
   ${MF_MESSAGELOGGER}
-  
+
   cetlib cetlib_except
   ${ROOT_BASIC_LIB_LIST}
   BASENAME_ONLY
@@ -61,7 +61,7 @@ simple_plugin(EMEnergyCalib "module"
   art_Persistency_Provenance canvas
   art_Utilities canvas
   ${MF_MESSAGELOGGER}
-  
+
   cetlib cetlib_except
   ${ROOT_BASIC_LIB_LIST}
   BASENAME_ONLY
@@ -86,7 +86,7 @@ simple_plugin(EMPi0Energy "module"
   art_Persistency_Provenance canvas
   art_Utilities canvas
   ${MF_MESSAGELOGGER}
-  
+
   cetlib cetlib_except
   ${ROOT_BASIC_LIB_LIST}
   BASENAME_ONLY
diff --git a/dune/EventFilters/CMakeLists.txt b/dune/EventFilters/CMakeLists.txt
index 82029ad..0c719c3 100644
--- a/dune/EventFilters/CMakeLists.txt
+++ b/dune/EventFilters/CMakeLists.txt
@@ -5,7 +5,7 @@ simple_plugin(DuneEventFilter "module"
                         lardataobj_RecoBase
 	                larcorealg_Geometry
 	                larcore_Geometry_Geometry_service
-                        lardata_Utilities    
+                        lardata_Utilities
                         ${ART_FRAMEWORK_CORE}
 			${ART_FRAMEWORK_PRINCIPAL}
 			${ART_FRAMEWORK_SERVICES_REGISTRY}
diff --git a/dune/EventGenerator/CMakeLists.txt b/dune/EventGenerator/CMakeLists.txt
index bbee6db..efbb98a 100644
--- a/dune/EventGenerator/CMakeLists.txt
+++ b/dune/EventGenerator/CMakeLists.txt
@@ -1,7 +1,7 @@
 include_directories( $ENV{IFDHC_FQ_DIR}/inc )
 include_directories( ${CMAKE_CURRENT_SOURCE_DIR} )
 art_make( BASENAME_ONLY
-          MODULE_LIBRARIES 
+          MODULE_LIBRARIES
                         larcoreobj_SummaryData
                         larcorealg_Geometry
                         larcore_Geometry_Geometry_service
diff --git a/dune/EventGenerator/ProtoDUNEbeamDataProducts/CMakeLists.txt b/dune/EventGenerator/ProtoDUNEbeamDataProducts/CMakeLists.txt
index 94cb7b9..19c775c 100644
--- a/dune/EventGenerator/ProtoDUNEbeamDataProducts/CMakeLists.txt
+++ b/dune/EventGenerator/ProtoDUNEbeamDataProducts/CMakeLists.txt
@@ -10,9 +10,9 @@ art_make(
           ${ART_ROOT_IO_TFILE_SUPPORT} ${ROOT_CORE}
           ${ART_ROOT_IO_TFILESERVICE_SERVICE}
           art_Persistency_Common
-          art_Utilities 
+          art_Utilities
           art_Persistency_Common
-          art_Persistency_Provenance 
+          art_Persistency_Provenance
 	  canvas
           ${MF_MESSAGELOGGER}
           cetlib cetlib_except
@@ -21,7 +21,7 @@ art_make(
 
 install_headers()
 install_source()
-install_fhicl()   
+install_fhicl()
 install_scripts()
 
 
diff --git a/dune/FDSensOpt/CMakeLists.txt b/dune/FDSensOpt/CMakeLists.txt
index 554e882..ad39f3b 100644
--- a/dune/FDSensOpt/CMakeLists.txt
+++ b/dune/FDSensOpt/CMakeLists.txt
@@ -6,57 +6,59 @@ add_subdirectory(NeutrinoEnergyRecoAlg)
 include_directories ( $ENV{GENIE_INC}/GENIE )
 
 # genie
-  cet_find_library( GFWMSG       NAMES GFwMsg       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFWREG       NAMES GFwReg       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFWALG       NAMES GFwAlg       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFWINT       NAMES GFwInt       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFWGHEP      NAMES GFwGHEP      PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFWNUM       NAMES GFwNum       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFWUTL       NAMES GFwUtl       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFWPARDAT    NAMES GFwParDat    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFWEG        NAMES GFwEG        PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GFWNTP       NAMES GFwNtp       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHXSIG      NAMES GPhXSIg      PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHPDF       NAMES GPhPDF       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHNUCLST    NAMES GPhNuclSt    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHCMN       NAMES GPhCmn       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHDCY       NAMES GPhDcy       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHHADTRANSP NAMES GPhHadTransp PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHHADNZ     NAMES GPhHadnz     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHDEEX      NAMES GPhDeEx      PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHAMNGXS    NAMES GPhAMNGXS    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHAMNGEG    NAMES GPhAMNGEG    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHCHMXS     NAMES GPhChmXS     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHCOHXS     NAMES GPhCohXS     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHCOHEG     NAMES GPhCohEG     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHDISXS     NAMES GPhDISXS     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHDISEG     NAMES GPhDISEG     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHDFRCXS    NAMES GPhDfrcXS    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHDFRCEG    NAMES GPhDfrcEG    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHGLWRESXS  NAMES GPhGlwResXS  PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHGLWRESEG  NAMES GPhGlwResEG  PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHIBDXS     NAMES GPhIBDXS     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHIBDEG     NAMES GPhIBDEG     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHMNUCXS    NAMES GPhMNucXS    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHMNUCEG    NAMES GPhMNucEG    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHMEL       NAMES GPhMEL       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHNUELXS    NAMES GPhNuElXS    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHNUELEG    NAMES GPhNuElEG    PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHQELXS     NAMES GPhQELXS     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHQELEG     NAMES GPhQELEG     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHRESXS     NAMES GPhResXS     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHRESEG     NAMES GPhResEG     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHSTRXS     NAMES GPhStrXS     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHSTREG     NAMES GPhStrEG     PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GPHNDCY      NAMES GPhNDcy      PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GTLGEO       NAMES GTlGeo       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GTLFLX       NAMES GTlFlx       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GRWFWK       NAMES GRwFwk       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GRWIO        NAMES GRwIO        PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
-  cet_find_library( GRWCLC       NAMES GRwClc       PATHS ENV GENIE_LIB NO_DEFAULT_PATH )
+find_package(cetmodules)
+include(CetCMakeEnv)
+cet_find_library( GFWMSG       NAMES GFwMsg      )
+cet_find_library( GFWREG       NAMES GFwReg      )
+cet_find_library( GFWALG       NAMES GFwAlg      )
+cet_find_library( GFWINT       NAMES GFwInt      )
+cet_find_library( GFWGHEP      NAMES GFwGHEP     )
+cet_find_library( GFWNUM       NAMES GFwNum      )
+cet_find_library( GFWUTL       NAMES GFwUtl      )
+cet_find_library( GFWPARDAT    NAMES GFwParDat   )
+cet_find_library( GFWEG        NAMES GFwEG       )
+cet_find_library( GFWNTP       NAMES GFwNtp      )
+cet_find_library( GPHXSIG      NAMES GPhXSIg     )
+cet_find_library( GPHPDF       NAMES GPhPDF      )
+cet_find_library( GPHNUCLST    NAMES GPhNuclSt   )
+cet_find_library( GPHCMN       NAMES GPhCmn      )
+cet_find_library( GPHDCY       NAMES GPhDcy      )
+cet_find_library( GPHHADTRANSP NAMES GPhHadTransp)
+cet_find_library( GPHHADNZ     NAMES GPhHadnz    )
+cet_find_library( GPHDEEX      NAMES GPhDeEx     )
+cet_find_library( GPHAMNGXS    NAMES GPhAMNGXS   )
+cet_find_library( GPHAMNGEG    NAMES GPhAMNGEG   )
+cet_find_library( GPHCHMXS     NAMES GPhChmXS    )
+cet_find_library( GPHCOHXS     NAMES GPhCohXS    )
+cet_find_library( GPHCOHEG     NAMES GPhCohEG    )
+cet_find_library( GPHDISXS     NAMES GPhDISXS    )
+cet_find_library( GPHDISEG     NAMES GPhDISEG    )
+cet_find_library( GPHDFRCXS    NAMES GPhDfrcXS   )
+cet_find_library( GPHDFRCEG    NAMES GPhDfrcEG   )
+cet_find_library( GPHGLWRESXS  NAMES GPhGlwResXS )
+cet_find_library( GPHGLWRESEG  NAMES GPhGlwResEG )
+cet_find_library( GPHIBDXS     NAMES GPhIBDXS    )
+cet_find_library( GPHIBDEG     NAMES GPhIBDEG    )
+cet_find_library( GPHMNUCXS    NAMES GPhMNucXS   )
+cet_find_library( GPHMNUCEG    NAMES GPhMNucEG   )
+cet_find_library( GPHMEL       NAMES GPhMEL      )
+cet_find_library( GPHNUELXS    NAMES GPhNuElXS   )
+cet_find_library( GPHNUELEG    NAMES GPhNuElEG   )
+cet_find_library( GPHQELXS     NAMES GPhQELXS    )
+cet_find_library( GPHQELEG     NAMES GPhQELEG    )
+cet_find_library( GPHRESXS     NAMES GPhResXS    )
+cet_find_library( GPHRESEG     NAMES GPhResEG    )
+cet_find_library( GPHSTRXS     NAMES GPhStrXS    )
+cet_find_library( GPHSTREG     NAMES GPhStrEG    )
+cet_find_library( GPHNDCY      NAMES GPhNDcy     )
+cet_find_library( GTLGEO       NAMES GTlGeo      )
+cet_find_library( GTLFLX       NAMES GTlFlx      )
+cet_find_library( GRWFWK       NAMES GRwFwk      )
+cet_find_library( GRWIO        NAMES GRwIO       )
+cet_find_library( GRWCLC       NAMES GRwClc      )
 
   # GENIE pre-R-3 used GSL via ROOT's Math libs, now makes direct call not available via that route
-  cet_find_library( GSL NAMES gsl PATHS ENV GSL_LIB NO_DEFAULT_PATH )
+cet_find_library( GSL NAMES gsl)
 
   set (GENIE_LIB_LIST ${GFWMSG}
                       ${GFWREG}
@@ -133,7 +135,7 @@ simple_plugin(NueAna   "module"
 			art_Persistency_Provenance canvas
 			art_Utilities canvas
 			${MF_MESSAGELOGGER}
-			
+
 			cetlib cetlib_except
 			${ROOT_BASIC_LIB_LIST}
               BASENAME_ONLY
@@ -156,7 +158,7 @@ simple_plugin(MVASelect   "module"
                         art_Persistency_Provenance canvas
                         art_Utilities canvas
                         ${MF_MESSAGELOGGER}
-                        
+
                         cetlib cetlib_except
                         ${ROOT_BASIC_LIB_LIST}
 			${GENIE_LIB_LIST}
@@ -185,7 +187,7 @@ simple_plugin(IniSegReco "module"
 		      art_Persistency_Provenance canvas
 		      art_Utilities canvas
 		      ${MF_MESSAGELOGGER}
-		      
+
 		      cetlib cetlib_except
 		      ${ROOT_BASIC_LIB_LIST}
               BASENAME_ONLY
@@ -202,8 +204,8 @@ simple_plugin(ShSeg   "module"
 			larcore_Geometry_Geometry_service
 			lardata_Utilities
 			larsim_Simulation nug4_ParticleNavigation lardataobj_Simulation
-			
-			
+
+
 			larsim_MCCheater_BackTrackerService_service
            larsim_MCCheater_ParticleInventoryService_service
 			nusimdata_SimulationBase
@@ -216,7 +218,7 @@ simple_plugin(ShSeg   "module"
 			art_Persistency_Provenance canvas
 			art_Utilities canvas
 			${MF_MESSAGELOGGER}
-			
+
 			cetlib cetlib_except
 			${ROOT_BASIC_LIB_LIST}
 			BASENAME_ONLY
@@ -245,7 +247,7 @@ simple_plugin(EnergyReco   "module"
                         art_Persistency_Provenance canvas
                         art_Utilities canvas
                         ${MF_MESSAGELOGGER}
-                        
+
                         ${CETLIB}
                         ${ROOT_BASIC_LIB_LIST}
                          ${GENIE_LIB_LIST}
@@ -268,7 +270,7 @@ simple_plugin(CheckRecoEnergy   "module"
                         art_Persistency_Provenance canvas
                         art_Utilities canvas
                         ${MF_MESSAGELOGGER}
-                        
+
                         cetlib cetlib_except
                         ${ROOT_BASIC_LIB_LIST}
                         ${GENIE_LIB_LIST}
diff --git a/dune/FDSensOpt/IniSegAlg/CMakeLists.txt b/dune/FDSensOpt/IniSegAlg/CMakeLists.txt
index ff296b1..790270e 100644
--- a/dune/FDSensOpt/IniSegAlg/CMakeLists.txt
+++ b/dune/FDSensOpt/IniSegAlg/CMakeLists.txt
@@ -23,7 +23,7 @@ art_make( BASENAME_ONLY
                         art_Utilities canvas
                         ${FHICLCPP}
 			${MF_MESSAGELOGGER}
-			
+
                         cetlib cetlib_except
                         ${ROOT_BASIC_LIB_LIST}
 			${Boost_SYSTEM_LIBRARY}
diff --git a/dune/FDSensOpt/MVAAlg/CMakeLists.txt b/dune/FDSensOpt/MVAAlg/CMakeLists.txt
index 0046c79..2861dcf 100644
--- a/dune/FDSensOpt/MVAAlg/CMakeLists.txt
+++ b/dune/FDSensOpt/MVAAlg/CMakeLists.txt
@@ -28,7 +28,7 @@ art_make( BASENAME_ONLY
   art_Utilities canvas
   ${FHICLCPP}
   ${MF_MESSAGELOGGER}
-  
+
   cetlib cetlib_except
   ${ROOT_BASIC_LIB_LIST}
   ${Boost_SYSTEM_LIBRARY}
diff --git a/dune/FDSensOpt/NeutrinoEnergyRecoAlg/CMakeLists.txt b/dune/FDSensOpt/NeutrinoEnergyRecoAlg/CMakeLists.txt
index 2d010d4..b5bc3b2 100644
--- a/dune/FDSensOpt/NeutrinoEnergyRecoAlg/CMakeLists.txt
+++ b/dune/FDSensOpt/NeutrinoEnergyRecoAlg/CMakeLists.txt
@@ -2,7 +2,7 @@ include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 
 art_make( BASENAME_ONLY
   LIBRARY_NAME  NeutrinoEnergyRecoAlg
-  LIB_LIBRARIES 
+  LIB_LIBRARIES
   larcorealg_Geometry
   dune_AnaUtils
   larcore_Geometry_Geometry_service
@@ -30,7 +30,7 @@ art_make( BASENAME_ONLY
   art_Utilities canvas
   ${FHICLCPP}
   ${MF_MESSAGELOGGER}
-  
+
   ${ROOT_EG}
   cetlib cetlib_except
   ${ROOT_BASIC_LIB_LIST}
diff --git a/dune/Gaps/CMakeLists.txt b/dune/Gaps/CMakeLists.txt
index 1d55988..10da57c 100644
--- a/dune/Gaps/CMakeLists.txt
+++ b/dune/Gaps/CMakeLists.txt
@@ -13,7 +13,7 @@ simple_plugin(GapFilter "module"
                         art_Persistency_Provenance canvas
                         art_Utilities canvas
                         ${MF_MESSAGELOGGER}
-                        
+
                         ${FHICLCPP}
                         cetlib cetlib_except
                         ${ROOT_GEOM}
@@ -41,7 +41,7 @@ simple_plugin(GapWidth "module"
                        art_Persistency_Provenance canvas
                        art_Utilities canvas
                        ${MF_MESSAGELOGGER}
-                       
+
                        cetlib cetlib_except
                        ${ROOT_BASIC_LIB_LIST}
                        ${ROOT_MINUIT}
diff --git a/dune/Geometry/test/testIssue19191/CMakeLists.txt b/dune/Geometry/test/testIssue19191/CMakeLists.txt
index 5f9e2a6..8eea9eb 100644
--- a/dune/Geometry/test/testIssue19191/CMakeLists.txt
+++ b/dune/Geometry/test/testIssue19191/CMakeLists.txt
@@ -1,7 +1,7 @@
 simple_plugin(issue19191           "module"  NO_INSTALL cetlib_except larcorealg_Geometry larcore_Geometry_Geometry_service dune_ArtSupport)
 
 cet_test(test_issue19191 HANDBUILT
-  TEST_EXEC art 
+  TEST_EXEC art
   TEST_ARGS --rethrow-all --config test_issue19191.fcl
   DATAFILES test_issue19191.fcl
 )
diff --git a/dune/HitAnalysis/CMakeLists.txt b/dune/HitAnalysis/CMakeLists.txt
index 6821556..37bc8a7 100644
--- a/dune/HitAnalysis/CMakeLists.txt
+++ b/dune/HitAnalysis/CMakeLists.txt
@@ -17,7 +17,7 @@ art_make(BASENAME_ONLY
  	 art_Persistency_Common canvas
 	 art_Persistency_Provenance canvas
 	 art_Utilities canvas
-	 ${MF_MESSAGELOGGER}	 
+	 ${MF_MESSAGELOGGER}
 	 cetlib cetlib_except
 	 ${ROOT_BASIC_LIB_LIST}
          ${Boost_SYSTEM_LIBRARY}
diff --git a/dune/HitFinderDUNE/CMakeLists.txt b/dune/HitFinderDUNE/CMakeLists.txt
index 0eab2a9..13b4e1e 100644
--- a/dune/HitFinderDUNE/CMakeLists.txt
+++ b/dune/HitFinderDUNE/CMakeLists.txt
@@ -23,7 +23,7 @@ art_make( BASENAME_ONLY
 		       	   art_Persistency_Provenance canvas
 			   art_Utilities canvas
 			   ${MF_MESSAGELOGGER}
-			   
+
 	                   ${FHICLCPP}
 			   cetlib cetlib_except
                            ${CLHEP}
@@ -49,7 +49,7 @@ art_make( BASENAME_ONLY
 			   art_Persistency_Common canvas
 			   art_Utilities canvas
 			   ${MF_MESSAGELOGGER}
-			   
+
 	                   ${FHICLCPP}
 			   cetlib cetlib_except
                            ${ROOT_BASIC_LIB_LIST}
diff --git a/dune/Iceberg/RawDecoding/CMakeLists.txt b/dune/Iceberg/RawDecoding/CMakeLists.txt
index f82d772..8238d51 100644
--- a/dune/Iceberg/RawDecoding/CMakeLists.txt
+++ b/dune/Iceberg/RawDecoding/CMakeLists.txt
@@ -1,6 +1,6 @@
 
-include_directories( "$ENV{DUNEPDSPRCE_INC}" ) 
-cet_find_library( RCEDAMLIB NAMES protodune-dam PATHS ENV DUNEPDSPRCE_LIB NO_DEFAULT_PATH )
+include_directories( "$ENV{DUNEPDSPRCE_INC}" )
+cet_find_library( RCEDAMLIB NAMES protodune-dam)
 
 simple_plugin(IcebergTPCRawDecoder "module"
                         lardataobj_RawData
diff --git a/dune/Iceberg/Signal2Noise/CMakeLists.txt b/dune/Iceberg/Signal2Noise/CMakeLists.txt
index 9baba30..bf7541c 100644
--- a/dune/Iceberg/Signal2Noise/CMakeLists.txt
+++ b/dune/Iceberg/Signal2Noise/CMakeLists.txt
@@ -1,5 +1,5 @@
-art_make( 
-  LIB_LIBRARIES 
+art_make(
+  LIB_LIBRARIES
   lardataobj_RecoBase
   larana_OpticalDetector_OpHitFinder
   larsim_MCSTReco
diff --git a/dune/LArG4/CMakeLists.txt b/dune/LArG4/CMakeLists.txt
index f3f7b03..9fa1755 100644
--- a/dune/LArG4/CMakeLists.txt
+++ b/dune/LArG4/CMakeLists.txt
@@ -13,7 +13,7 @@ art_make( BASENAME_ONLY
 			art_Persistency_Provenance canvas
 			art_Utilities canvas
 			${MF_MESSAGELOGGER}
-			
+
 	                ${FHICLCPP}
 			cetlib cetlib_except
                         ${ROOT_GEOM}
diff --git a/dune/OpticalDetector/BackTrackingTests/CMakeLists.txt b/dune/OpticalDetector/BackTrackingTests/CMakeLists.txt
index ef2eed9..9d86ce9 100644
--- a/dune/OpticalDetector/BackTrackingTests/CMakeLists.txt
+++ b/dune/OpticalDetector/BackTrackingTests/CMakeLists.txt
@@ -9,7 +9,7 @@ art_make( BASENAME_ONLY MODULE_LIBRARIES
   larsim_MCCheater_ParticleInventoryService_service
   larsim_MCCheater_PhotonBackTrackerService_service
   larsim_Simulation nug4_ParticleNavigation lardataobj_Simulation
-  nusimdata_SimulationBase	
+  nusimdata_SimulationBase
   ${ART_FRAMEWORK_CORE}
  ${ART_FRAMEWORK_PRINCIPAL}
  ${ART_FRAMEWORK_SERVICES_BASIC}
diff --git a/dune/OpticalDetector/CMakeLists.txt b/dune/OpticalDetector/CMakeLists.txt
index b437ffa..ec5a76e 100644
--- a/dune/OpticalDetector/CMakeLists.txt
+++ b/dune/OpticalDetector/CMakeLists.txt
@@ -2,7 +2,7 @@
 
 art_make( BASENAME_ONLY
                 LIBRARY_NAME dune_OpticalDetector
-                LIB_LIBRARIES 
+                LIB_LIBRARIES
                                 lardataobj_RecoBase
                                 larcorealg_Geometry
                                 larcore_Geometry_Geometry_service
@@ -19,11 +19,11 @@ art_make( BASENAME_ONLY
                                 ${ART_FRAMEWORK_SERVICES_OPTIONAL_RANDOMNUMBERGENERATOR_SERVICE}
                                 art_Persistency_Common canvas
                                 art_Utilities canvas
-                                cetlib cetlib_except 
+                                cetlib cetlib_except
                                 ${ROOT_BASIC_LIB_LIST}
                                 ${Boost_SYSTEM_LIBRARY}
                                 ${Boost_FILESYSTEM_LIBRARY}
-  
+
                 MODULE_LIBRARIES
                                 dune_DuneObj
                                 lardataobj_RecoBase
@@ -33,7 +33,7 @@ art_make( BASENAME_ONLY
                                 larcore_Geometry_Geometry_service
                                 dune_OpticalDetector
                                 larsim_MCCheater_ParticleInventoryService_service
-                                larsim_MCCheater_PhotonBackTrackerService_service 
+                                larsim_MCCheater_PhotonBackTrackerService_service
                                 nusimdata_SimulationBase
                                 ${FHICLCPP}
                                 ${MF_MESSAGELOGGER}
diff --git a/dune/ProductFilters/CMakeLists.txt b/dune/ProductFilters/CMakeLists.txt
index 498b5eb..57350ef 100644
--- a/dune/ProductFilters/CMakeLists.txt
+++ b/dune/ProductFilters/CMakeLists.txt
@@ -19,7 +19,7 @@ art_make( BASENAME_ONLY
 			   art_Persistency_Common canvas
 			   art_Utilities canvas
 			   ${MF_MESSAGELOGGER}
-			   
+
 	       ${FHICLCPP}
 			   cetlib cetlib_except
          ${ROOT_BASIC_LIB_LIST}
diff --git a/dune/Protodune/dualphase/CMakeLists.txt b/dune/Protodune/dualphase/CMakeLists.txt
index d0e8c77..9d97bc4 100644
--- a/dune/Protodune/dualphase/CMakeLists.txt
+++ b/dune/Protodune/dualphase/CMakeLists.txt
@@ -43,7 +43,7 @@ simple_plugin(
                         art_Persistency_Provenance canvas
                         art_Utilities canvas
                         ${MF_MESSAGELOGGER}
-                        
+
                         cetlib cetlib_except
                         ${ROOT_BASIC_LIB_LIST}
                         BASENAME_ONLY
@@ -69,7 +69,7 @@ simple_plugin(
                         art_Persistency_Provenance canvas
                         art_Utilities canvas
                         ${MF_MESSAGELOGGER}
-                        
+
                         cetlib cetlib_except
                         ${ROOT_BASIC_LIB_LIST}
                         BASENAME_ONLY
@@ -103,7 +103,7 @@ simple_plugin(
 	  art_Persistency_Provenance canvas
 	  art_Utilities canvas
 	  ${MF_MESSAGELOGGER}
-	  
+
 	  ${FHICLCPP}
 	  cetlib cetlib_except
           ${ROOT_GEOM}
diff --git a/dune/Protodune/dualphase/Light/CMakeLists.txt b/dune/Protodune/dualphase/Light/CMakeLists.txt
index 087302a..6c8bdac 100644
--- a/dune/Protodune/dualphase/Light/CMakeLists.txt
+++ b/dune/Protodune/dualphase/Light/CMakeLists.txt
@@ -8,7 +8,7 @@ art_make( BASENAME_ONLY MODULE_LIBRARIES
   larsim_Simulation
   nug4_ParticleNavigation
   lardataobj_Simulation
-  nusimdata_SimulationBase	
+  nusimdata_SimulationBase
   ${ART_FRAMEWORK_CORE}
   ${ART_FRAMEWORK_PRINCIPAL}
   ${ART_FRAMEWORK_SERVICES_BASIC}
diff --git a/dune/Protodune/dualphase/RawDecoding/CMakeLists.txt b/dune/Protodune/dualphase/RawDecoding/CMakeLists.txt
index 718597f..f297adb 100644
--- a/dune/Protodune/dualphase/RawDecoding/CMakeLists.txt
+++ b/dune/Protodune/dualphase/RawDecoding/CMakeLists.txt
@@ -61,7 +61,7 @@ simple_plugin(PDDPTPCRawDecoderTest "module"
 			art_Persistency_Provenance canvas
 			art_Utilities canvas
 			${MF_MESSAGELOGGER}
-			
+
 			cetlib cetlib_except
 			${ROOT_BASIC_LIB_LIST}
               		BASENAME_ONLY
diff --git a/dune/Protodune/dualphase/test/CMakeLists.txt b/dune/Protodune/dualphase/test/CMakeLists.txt
index f90fde4..14a1eee 100644
--- a/dune/Protodune/dualphase/test/CMakeLists.txt
+++ b/dune/Protodune/dualphase/test/CMakeLists.txt
@@ -11,7 +11,7 @@ simple_plugin(		recobWireCheck "module"
                         art_Persistency_Provenance canvas
                         art_Utilities canvas
                         ${MF_MESSAGELOGGER}
-                        
+
                         cetlib cetlib_except
                         ${ROOT_BASIC_LIB_LIST}
                         BASENAME_ONLY
diff --git a/dune/Protodune/singlephase/BeamReco/CMakeLists.txt b/dune/Protodune/singlephase/BeamReco/CMakeLists.txt
index 0a38017..3b6f39e 100644
--- a/dune/Protodune/singlephase/BeamReco/CMakeLists.txt
+++ b/dune/Protodune/singlephase/BeamReco/CMakeLists.txt
@@ -3,9 +3,9 @@ include_directories( $ENV{IFDH_ART_INC} )
 include_directories( $ENV{IFDHC_FQ_DIR}/inc )
 include_directories( $ENV{IFBEAM_FQ_DIR}/include )
 
-cet_find_library( WDALIB NAMES wda PATHS ENV LIBWDA_LIB NO_DEFAULT_PATH )
-cet_find_library( IFBEAMSERVICE NAMES IFBeam_service PATHS ENV IFDH_ART_LIB NO_DEFAULT_PATH )
-cet_find_library( IFBEAM NAMES ifbeam PATHS ENV IFBEAM_LIB   NO_DEFAULT_PATH )
+cet_find_library( WDALIB NAMES wda)
+cet_find_library( IFBEAMSERVICE NAMES IFBeam_service)
+cet_find_library( IFBEAM NAMES ifbeam)
 
 
 art_make(  MODULE_LIBRARIES
@@ -21,14 +21,14 @@ art_make(  MODULE_LIBRARIES
                         larcore_Geometry_Geometry_service
                         lardata_ArtDataHelper
                         lardata_Utilities
-                        larsim_Simulation 
+                        larsim_Simulation
                         lardataobj_Simulation
                         dune_DuneObj
                         dune-raw-data_Services_ChannelMap_PdspChannelMapService_service
                         larsim_MCCheater_PhotonBackTrackerService_service
                         larsim_MCCheater_BackTrackerService_service
                         larsim_MCCheater_ParticleInventoryService_service
-                        nug4_ParticleNavigation 
+                        nug4_ParticleNavigation
                         nusimdata_SimulationBase
                         ${ART_FRAMEWORK_CORE}
                         ${ART_FRAMEWORK_PRINCIPAL}
diff --git a/dune/Protodune/singlephase/CMakeLists.txt b/dune/Protodune/singlephase/CMakeLists.txt
index b963875..f13fa2f 100644
--- a/dune/Protodune/singlephase/CMakeLists.txt
+++ b/dune/Protodune/singlephase/CMakeLists.txt
@@ -3,9 +3,9 @@ include_directories( $ENV{IFDH_ART_INC} )
 include_directories( $ENV{IFDHC_FQ_DIR}/inc )
 include_directories( $ENV{IFBEAM_FQ_DIR}/include )
 
-cet_find_library( WDALIB NAMES wda PATHS ENV LIBWDA_LIB NO_DEFAULT_PATH )
-cet_find_library( IFBEAMSERVICE NAMES IFBeam_service PATHS ENV IFDH_ART_LIB NO_DEFAULT_PATH )
-cet_find_library( IFBEAM NAMES ifbeam PATHS ENV IFBEAM_LIB   NO_DEFAULT_PATH )
+cet_find_library( WDALIB NAMES wda)
+cet_find_library( IFBEAMSERVICE NAMES IFBeam_service)
+cet_find_library( IFBEAM NAMES ifbeam)
 
 
 install_headers()
diff --git a/dune/Protodune/singlephase/DataUtils/CMakeLists.txt b/dune/Protodune/singlephase/DataUtils/CMakeLists.txt
index 4089f6a..3143332 100644
--- a/dune/Protodune/singlephase/DataUtils/CMakeLists.txt
+++ b/dune/Protodune/singlephase/DataUtils/CMakeLists.txt
@@ -3,9 +3,9 @@ include_directories( $ENV{IFDH_ART_INC} )
 include_directories( $ENV{IFDHC_FQ_DIR}/inc )
 include_directories( $ENV{IFBEAM_FQ_DIR}/include )
 
-cet_find_library( WDALIB NAMES wda PATHS ENV LIBWDA_LIB NO_DEFAULT_PATH )
-cet_find_library( IFBEAMSERVICE NAMES IFBeam_service PATHS ENV IFDH_ART_LIB NO_DEFAULT_PATH )
-cet_find_library( IFBEAM NAMES ifbeam PATHS ENV IFBEAM_LIB   NO_DEFAULT_PATH )
+cet_find_library( WDALIB NAMES wda)
+cet_find_library( IFBEAMSERVICE NAMES IFBeam_service)
+cet_find_library( IFBEAM NAMES ifbeam)
 
 
 art_make( LIBRARY_NAME ProtoDUNEDataUtils
@@ -22,14 +22,14 @@ art_make( LIBRARY_NAME ProtoDUNEDataUtils
                         larcore_Geometry_Geometry_service
                         lardata_ArtDataHelper
                         lardata_Utilities
-                        larsim_Simulation 
+                        larsim_Simulation
                         lardataobj_Simulation
                         dune_DuneObj
                         dune-raw-data_Services_ChannelMap_PdspChannelMapService_service
                         larsim_MCCheater_PhotonBackTrackerService_service
                         larsim_MCCheater_BackTrackerService_service
                         larsim_MCCheater_ParticleInventoryService_service
-                        nug4_ParticleNavigation 
+                        nug4_ParticleNavigation
                         nusimdata_SimulationBase
                         ${ART_FRAMEWORK_CORE}
                         ${ART_FRAMEWORK_PRINCIPAL}
diff --git a/dune/Protodune/singlephase/DetectorServices/Providers/CMakeLists.txt b/dune/Protodune/singlephase/DetectorServices/Providers/CMakeLists.txt
index 59a7c77..4a4d2d7 100644
--- a/dune/Protodune/singlephase/DetectorServices/Providers/CMakeLists.txt
+++ b/dune/Protodune/singlephase/DetectorServices/Providers/CMakeLists.txt
@@ -2,7 +2,7 @@ cet_enable_asserts()
 art_make( NO_PLUGINS
           LIBRARY_NAME  ProtoDUNEspDataProviders
           LIB_LIBRARIES
-          
+
   	                wda
 	                nuevdb_IFDatabase
                         larcorealg_Geometry
diff --git a/dune/Protodune/singlephase/DetectorServices/Services/CMakeLists.txt b/dune/Protodune/singlephase/DetectorServices/Services/CMakeLists.txt
index ff88451..1ded67f 100644
--- a/dune/Protodune/singlephase/DetectorServices/Services/CMakeLists.txt
+++ b/dune/Protodune/singlephase/DetectorServices/Services/CMakeLists.txt
@@ -1,4 +1,4 @@
-simple_plugin(DetectorPropertiesServiceProtoDUNEsp "service" 
+simple_plugin(DetectorPropertiesServiceProtoDUNEsp "service"
 	          ProtoDUNEspDataProviders
               lardataalg_DetectorInfo
               larcore_Geometry_Geometry_service
@@ -12,12 +12,12 @@ simple_plugin(DetectorPropertiesServiceProtoDUNEsp "service"
 	      canvas
 	      ${SQLITE3}
 	      ${MF_MESSAGELOGGER}
-	      
+
 	      cetlib cetlib_except
 	      ${ROOT_BASIC_LIB_LIST}
-              
+
 )
 
 install_headers()
 install_fhicl()
-install_source()
\ No newline at end of file
+install_source()
diff --git a/dune/Protodune/singlephase/NearlineMonitor/CMakeLists.txt b/dune/Protodune/singlephase/NearlineMonitor/CMakeLists.txt
index 864b0e2..7048d19 100644
--- a/dune/Protodune/singlephase/NearlineMonitor/CMakeLists.txt
+++ b/dune/Protodune/singlephase/NearlineMonitor/CMakeLists.txt
@@ -84,7 +84,7 @@ simple_plugin(PlotEventDetails "module"
       ${ART_ROOT_IO_TFILESERVICE_SERVICE}
       ${ART_PERSISTENCY_PROVENANCE}
       art_Persistency_Common
-      art_Utilities 
+      art_Utilities
       canvas
       ${MF_MESSAGELOGGER}
       cetlib cetlib_except
diff --git a/dune/Protodune/singlephase/RawDecoding/CMakeLists.txt b/dune/Protodune/singlephase/RawDecoding/CMakeLists.txt
index 796efd5..5785013 100644
--- a/dune/Protodune/singlephase/RawDecoding/CMakeLists.txt
+++ b/dune/Protodune/singlephase/RawDecoding/CMakeLists.txt
@@ -1,5 +1,5 @@
 include_directories( "$ENV{DUNEPDSPRCE_INC}" )
-cet_find_library( RCEDAMLIB NAMES protodune-dam PATHS ENV DUNEPDSPRCE_LIB NO_DEFAULT_PATH )
+cet_find_library( RCEDAMLIB NAMES protodune-dam)
 
 simple_plugin(PDSPTPCDataInterface   "tool"
 				     art_utilities
diff --git a/dune/Protodune/singlephase/dEdxcalibration/CMakeLists.txt b/dune/Protodune/singlephase/dEdxcalibration/CMakeLists.txt
index 9baba30..bf7541c 100644
--- a/dune/Protodune/singlephase/dEdxcalibration/CMakeLists.txt
+++ b/dune/Protodune/singlephase/dEdxcalibration/CMakeLists.txt
@@ -1,5 +1,5 @@
-art_make( 
-  LIB_LIBRARIES 
+art_make(
+  LIB_LIBRARIES
   lardataobj_RecoBase
   larana_OpticalDetector_OpHitFinder
   larsim_MCSTReco
diff --git a/dune/RecoAlgDUNE/CMakeLists.txt b/dune/RecoAlgDUNE/CMakeLists.txt
index 4c5d271..d2cd484 100644
--- a/dune/RecoAlgDUNE/CMakeLists.txt
+++ b/dune/RecoAlgDUNE/CMakeLists.txt
@@ -32,13 +32,13 @@ art_make(
 			art_Persistency_Provenance canvas
 			art_Utilities canvas
 			${MF_MESSAGELOGGER}
-			
+
 	                ${FHICLCPP}
 			cetlib cetlib_except
                         ${CLHEP}
 			${ROOT_BASIC_LIB_LIST}
                         ${ROOT_MINUIT}
-			${ROOT_MINUIT2}	
+			${ROOT_MINUIT2}
                         ${Boost_FILESYSTEM_LIBRARY}
                         ${Boost_SYSTEM_LIBRARY}
         )
diff --git a/dune/RecoAlgDUNE/Cluster3DAlgs/CMakeLists.txt b/dune/RecoAlgDUNE/Cluster3DAlgs/CMakeLists.txt
index 214c4e0..f9fa29e 100644
--- a/dune/RecoAlgDUNE/Cluster3DAlgs/CMakeLists.txt
+++ b/dune/RecoAlgDUNE/Cluster3DAlgs/CMakeLists.txt
@@ -1,5 +1,5 @@
 
-art_make( 
+art_make(
 	  LIBRARY_NAME  dune_RecoAlgDUNE_Cluster3DAlgs
           LIB_LIBRARIES larreco_VertexFinder_HarrisVertexFinder_module
                         lardataobj_RecoBase
@@ -21,13 +21,13 @@ art_make(
 			art_Persistency_Provenance canvas
 			art_Utilities canvas
 			${MF_MESSAGELOGGER}
-			
+
 	                ${FHICLCPP}
 			cetlib cetlib_except
                         ${CLHEP}
 			${ROOT_BASIC_LIB_LIST}
                         ${ROOT_MINUIT}
-			${ROOT_MINUIT2}	
+			${ROOT_MINUIT2}
                         ${Boost_FILESYSTEM_LIBRARY}
                         ${Boost_SYSTEM_LIBRARY}
         )
diff --git a/dune/RegCNN/art/CMakeLists.txt b/dune/RegCNN/art/CMakeLists.txt
index 93b1268..16615da 100644
--- a/dune/RegCNN/art/CMakeLists.txt
+++ b/dune/RegCNN/art/CMakeLists.txt
@@ -22,7 +22,7 @@ art_make(BASENAME_ONLY
   lardata_Utilities
   larpandora_LArPandoraInterface
   nusimdata_SimulationBase
-  larsim_MCCheater_BackTrackerService_service           
+  larsim_MCCheater_BackTrackerService_service
   larsim_MCCheater_ParticleInventoryService_service
   MVAAlg
   larreco_RecoAlg
diff --git a/dune/RegCNN/macro/CMakeLists.txt b/dune/RegCNN/macro/CMakeLists.txt
index cbc0647..b783436 100644
--- a/dune/RegCNN/macro/CMakeLists.txt
+++ b/dune/RegCNN/macro/CMakeLists.txt
@@ -18,7 +18,7 @@ art_make(MODULE_LIBRARIES
   lardataobj_RecoBase
   lardata_Utilities
   nusimdata_SimulationBase
-  larsim_MCCheater_BackTrackerService_service           
+  larsim_MCCheater_BackTrackerService_service
   larsim_MCCheater_ParticleInventoryService_service
   larreco_RecoAlg
   larcorealg_Geometry
diff --git a/dune/SNSlicer/CMakeLists.txt b/dune/SNSlicer/CMakeLists.txt
index aa8d887..68c280d 100644
--- a/dune/SNSlicer/CMakeLists.txt
+++ b/dune/SNSlicer/CMakeLists.txt
@@ -22,7 +22,7 @@ simple_plugin(SNSlicer "module"
                         art_Persistency_Provenance canvas_Persistency_Provenance
                         art_Utilities canvas_Utilities
                         ${MF_MESSAGELOGGER}
-                        
+
                         ${CETLIB}
                         ${ROOT_BASIC_LIB_LIST}
 )
@@ -32,7 +32,7 @@ cet_make( LIBRARIES lardataobj_RawData
                     canvas_Persistency_Provenance
                     canvas_Utilities
                     MF_MessageLogger
-                    
+
                     cetlib
                     ${ROOT_BASIC_LIB_LIST}
                     NO_DICTIONARY )
diff --git a/dune/ShowerAna/CMakeLists.txt b/dune/ShowerAna/CMakeLists.txt
index 9324e52..3929030 100644
--- a/dune/ShowerAna/CMakeLists.txt
+++ b/dune/ShowerAna/CMakeLists.txt
@@ -18,7 +18,7 @@ simple_plugin(ShowerAnalysis "module"
 			     art_Persistency_Provenance canvas
 			     art_Utilities canvas
 			     ${MF_MESSAGELOGGER}
-			     
+
 			     cetlib cetlib_except
 			     ${ROOT_BASIC_LIB_LIST}
 )
diff --git a/dune/SimFilter/CMakeLists.txt b/dune/SimFilter/CMakeLists.txt
index 58f6c29..af8a6cb 100644
--- a/dune/SimFilter/CMakeLists.txt
+++ b/dune/SimFilter/CMakeLists.txt
@@ -15,7 +15,7 @@ simple_plugin(GenExtCounterFilter "module"
 			art_Persistency_Provenance canvas
 			art_Utilities canvas
 			${MF_MESSAGELOGGER}
-			
+
 			cetlib cetlib_except
 			${ROOT_BASIC_LIB_LIST}
               BASENAME_ONLY
@@ -38,7 +38,7 @@ simple_plugin(LArG4ExtCounterFilter "module"
 			art_Persistency_Provenance canvas
 			art_Utilities canvas
 			${MF_MESSAGELOGGER}
-			
+
 			cetlib cetlib_except
 			${ROOT_BASIC_LIB_LIST}
               BASENAME_ONLY
diff --git a/dune/SpaceCharge/CMakeLists.txt b/dune/SpaceCharge/CMakeLists.txt
index 6f7cde0..86d6d9e 100644
--- a/dune/SpaceCharge/CMakeLists.txt
+++ b/dune/SpaceCharge/CMakeLists.txt
@@ -1,5 +1,5 @@
 art_make( NO_PLUGINS
-          
+
           LIB_LIBRARIES
           				lardataobj_RecoBase
                         lardata_Utilities
@@ -7,7 +7,7 @@ art_make( NO_PLUGINS
               larevt_SpaceCharge
 			larcorealg_Geometry
                         ${MF_MESSAGELOGGER}
-                        
+
                         cetlib cetlib_except
             ${ART_FRAMEWORK_CORE}
 			    ${ART_FRAMEWORK_PRINCIPAL}
diff --git a/dune/SpaceChargeServices/CMakeLists.txt b/dune/SpaceChargeServices/CMakeLists.txt
index 647672b..4200a7d 100644
--- a/dune/SpaceChargeServices/CMakeLists.txt
+++ b/dune/SpaceChargeServices/CMakeLists.txt
@@ -12,10 +12,10 @@ simple_plugin(SpaceChargeServiceProtoDUNE "service"
 	      ${ART_FRAMEWORK_SERVICES_SYSTEM_FILECATALOGMETADATA_SERVICE}
 	      art_Utilities canvas
 	      ${MF_MESSAGELOGGER}
-	      
+
   	      cetlib cetlib_except
    	      ${ROOT_BASIC_LIB_LIST}
-)              
+)
 simple_plugin(SpaceChargeServiceProtoDUNEdp "service"
               dune_SpaceCharge
               lardataalg_DetectorInfo
@@ -50,10 +50,10 @@ simple_plugin(SpaceChargeServiceDUNE35t "service"
 	      ${ART_FRAMEWORK_SERVICES_SYSTEM_FILECATALOGMETADATA_SERVICE}
 	      art_Utilities canvas
 	      ${MF_MESSAGELOGGER}
-	      
+
   	      cetlib cetlib_except
    	      ${ROOT_BASIC_LIB_LIST}
-              
+
 )
 
 simple_plugin(SpaceChargeService3x1x1dphase "service"
@@ -69,10 +69,10 @@ simple_plugin(SpaceChargeService3x1x1dphase "service"
 	      ${ART_FRAMEWORK_SERVICES_SYSTEM_FILECATALOGMETADATA_SERVICE}
 	      art_Utilities canvas
 	      ${MF_MESSAGELOGGER}
-	      
+
   	      cetlib cetlib_except
    	      ${ROOT_BASIC_LIB_LIST}
-              
+
 )
 
 install_headers()
diff --git a/dune/SupernovaAna/CMakeLists.txt b/dune/SupernovaAna/CMakeLists.txt
index 31eecc5..4c5472f 100644
--- a/dune/SupernovaAna/CMakeLists.txt
+++ b/dune/SupernovaAna/CMakeLists.txt
@@ -17,7 +17,7 @@ art_make(BASENAME_ONLY
          ${ART_PERSISTENCY_PROVENANCE}
          ${ART_UTILITIES}
          ${MF_MESSAGELOGGER}
-         
+
          cetlib cetlib_except
          ${ROOT_BASIC_LIB_LIST}
          )
diff --git a/dune/TrackPID/algorithms/CMakeLists.txt b/dune/TrackPID/algorithms/CMakeLists.txt
index 65f4cad..415f59e 100644
--- a/dune/TrackPID/algorithms/CMakeLists.txt
+++ b/dune/TrackPID/algorithms/CMakeLists.txt
@@ -1,6 +1,6 @@
 include_directories( ${CMAKE_CURRENT_SOURCE_DIR} )
 
-art_make( 
+art_make(
     LIB_LIBRARIES larcorealg_Geometry
     larcore_Geometry_Geometry_service
     larsim_Simulation lardataobj_Simulation
@@ -9,7 +9,7 @@ art_make(
     lardataobj_AnalysisBase
     lardata_Utilities
     nusimdata_SimulationBase
-    larsim_MCCheater_BackTrackerService_service           
+    larsim_MCCheater_BackTrackerService_service
     larsim_MCCheater_ParticleInventoryService_service
     ${ART_FRAMEWORK_CORE}
     ${ART_FRAMEWORK_PRINCIPAL}
diff --git a/dune/TrackPID/modules/CMakeLists.txt b/dune/TrackPID/modules/CMakeLists.txt
index a791985..d6b82da 100644
--- a/dune/TrackPID/modules/CMakeLists.txt
+++ b/dune/TrackPID/modules/CMakeLists.txt
@@ -1,6 +1,6 @@
 include_directories( ${CMAKE_CURRENT_SOURCE_DIR} )
 
-art_make( 
+art_make(
     MODULE_LIBRARIES larcorealg_Geometry
     larcore_Geometry_Geometry_service
     larsim_Simulation lardataobj_Simulation
@@ -9,7 +9,7 @@ art_make(
     lardataobj_AnalysisBase
     lardata_Utilities
     nusimdata_SimulationBase
-    larsim_MCCheater_BackTrackerService_service           
+    larsim_MCCheater_BackTrackerService_service
     larsim_MCCheater_ParticleInventoryService_service
     ${ART_FRAMEWORK_CORE}
     ${ART_FRAMEWORK_PRINCIPAL}
diff --git a/dune/TrackPID/products/CMakeLists.txt b/dune/TrackPID/products/CMakeLists.txt
index e1b7ebc..2f16a4d 100644
--- a/dune/TrackPID/products/CMakeLists.txt
+++ b/dune/TrackPID/products/CMakeLists.txt
@@ -1,7 +1,7 @@
 include_directories( ${CMAKE_CURRENT_SOURCE_DIR} )
 
-art_make( 
-    LIB_LIBRARIES 
+art_make(
+    LIB_LIBRARIES
 #    larcorealg_Geometry
 #    larcore_Geometry_Geometry_service
 #    larsim_Simulation lardataobj_Simulation
diff --git a/dune/TrackingAna/CMakeLists.txt b/dune/TrackingAna/CMakeLists.txt
index ac1ec9f..6cd7531 100644
--- a/dune/TrackingAna/CMakeLists.txt
+++ b/dune/TrackingAna/CMakeLists.txt
@@ -17,7 +17,7 @@ art_make(BASENAME_ONLY
 	 art_Persistency_Provenance canvas
 	 art_Utilities canvas
 	 ${MF_MESSAGELOGGER}
-	 
+
 	 cetlib cetlib_except
 	 ${ROOT_BASIC_LIB_LIST}
 	 ${ROOT_GEOM}
diff --git a/dune/Utilities/CMakeLists.txt b/dune/Utilities/CMakeLists.txt
index 62fc43d..e09cdf5 100644
--- a/dune/Utilities/CMakeLists.txt
+++ b/dune/Utilities/CMakeLists.txt
@@ -15,7 +15,7 @@ set( dune_util_lib_list lardata_Utilities_LArFFT_service
 			${ART_FRAMEWORK_SERVICES_SYSTEM_FILECATALOGMETADATA_SERVICE}
 			art_Utilities canvas
 			${MF_MESSAGELOGGER}
-			
+
 			cetlib cetlib_except
 			${ROOT_BASIC_LIB_LIST}
                         ${ROOT_GEOM}
@@ -36,12 +36,12 @@ simple_plugin( SignalShapingServiceDUNEDPhase  "service"
                ${dune_util_lib_list}
 	       BASENAME_ONLY
         )
-	             
+
 simple_plugin( SignalShapingServiceDUNE34kt  "service"
                ${dune_util_lib_list}
 	       BASENAME_ONLY
         )
-              
+
 
 simple_plugin( SignalShapingServiceDUNE35t  "service"
                ${dune_util_lib_list}
diff --git a/dune/Utilities/test/CMakeLists.txt b/dune/Utilities/test/CMakeLists.txt
index 06812e6..393a26e 100644
--- a/dune/Utilities/test/CMakeLists.txt
+++ b/dune/Utilities/test/CMakeLists.txt
@@ -13,7 +13,7 @@ art_make( NO_PLUGINS
 			art_Persistency_Provenance canvas
 			art_Utilities canvas
 			${MF_MESSAGELOGGER}
-			
+
 	                ${FHICLCPP}
 			cetlib cetlib_except
                         ${ROOT_GEOM}
diff --git a/fcl/protodunedp/mcc/CMakeLists.txt b/fcl/protodunedp/mcc/CMakeLists.txt
index a182aec..d63aa58 100644
--- a/fcl/protodunedp/mcc/CMakeLists.txt
+++ b/fcl/protodunedp/mcc/CMakeLists.txt
@@ -1,3 +1,3 @@
 add_subdirectory(mcc10)
 
-install_fhicl()
\ No newline at end of file
+install_fhicl()
diff --git a/fcl/protodunedp/mcc/mcc10/CMakeLists.txt b/fcl/protodunedp/mcc/mcc10/CMakeLists.txt
index eca2a52..1335578 100644
--- a/fcl/protodunedp/mcc/mcc10/CMakeLists.txt
+++ b/fcl/protodunedp/mcc/mcc10/CMakeLists.txt
@@ -1 +1 @@
-install_fhicl()
\ No newline at end of file
+install_fhicl()
diff --git a/test/Geometry/CMakeLists.txt b/test/Geometry/CMakeLists.txt
index 5356839..ffdcd43 100644
--- a/test/Geometry/CMakeLists.txt
+++ b/test/Geometry/CMakeLists.txt
@@ -54,7 +54,7 @@ cet_test(geometry_crmchannelmapping_test
             GeometryTestLib
             dune_Geometry
             ${MF_MESSAGELOGGER}
-            
+
             ${FHICLCPP}
             cetlib cetlib_except
   USE_BOOST_UNIT
diff --git a/test/OpDet/CMakeLists.txt b/test/OpDet/CMakeLists.txt
index 86339b2..9400f27 100644
--- a/test/OpDet/CMakeLists.txt
+++ b/test/OpDet/CMakeLists.txt
@@ -23,27 +23,29 @@
 # crash with a SEGFAULT on Mac OS,
 # they will be executed only on SLF for now
 if (NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+find_package(cetmodules)
+include(CetCMakeEnv)
    cet_test(optical_sim_dune35t HANDBUILT
      DATAFILES test_optical_sim_dune35t.fcl
      TEST_EXEC lar
-     TEST_ARGS --rethrow-all --config ./test_optical_sim_dune35t.fcl 
+     TEST_ARGS --rethrow-all --config ./test_optical_sim_dune35t.fcl
    )
 
    cet_test(optical_digi_dune35t HANDBUILT
      DATAFILES test_optical_digi_dune35t.fcl
      TEST_EXEC lar
-     TEST_ARGS --rethrow-all --config ./test_optical_digi_dune35t.fcl 
+     TEST_ARGS --rethrow-all --config ./test_optical_digi_dune35t.fcl
    )
 
    cet_test(optical_reco_dune35t HANDBUILT
      DATAFILES test_optical_reco_dune35t.fcl
      TEST_EXEC lar
-     TEST_ARGS --rethrow-all --config ./test_optical_reco_dune35t.fcl 
+     TEST_ARGS --rethrow-all --config ./test_optical_reco_dune35t.fcl
    )
 
    cet_test(build_oplib_dune35t HANDBUILT
      DATAFILES test_build_oplib_dune35t.fcl
      TEST_EXEC lar
-     TEST_ARGS --rethrow-all --config ./test_build_oplib_dune35t.fcl 
+     TEST_ARGS --rethrow-all --config ./test_build_oplib_dune35t.fcl
    )
 endif()
