diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5dc8aa4..d0587da 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,4 @@
-# use cmake 3.12 or later
-cmake_minimum_required (VERSION 3.12)
+cmake_minimum_required (VERSION 3.19)
 
 # if the CETBUILDTOOLS_VERSION environmental variable is defined, use cetbuildtools
 if(DEFINED ENV{CETBUILDTOOLS_VERSION})
@@ -16,11 +15,11 @@ if(DEFINED ENV{CETBUILDTOOLS_VERSION})
     #  make install
     #  make package (builds distribution tarfile)
     # ======================================================================
-    project(larpandoracontent VERSION 03.26.01.01)
+    project(larpandoracontent VERSION 03.26.01.01 LANGUAGES CXX )
 
     # cetbuildtools contains our cmake modules
     # Note that find package will add cetbuildtools Modules to CMAKE_MODULE_PATH
-    find_package(cetbuildtools REQUIRED)
+    find_package(cetmodules REQUIRED)
     ##message(STATUS "larpandoracontent: CMAKE_MODULE_PATH is ${CMAKE_MODULE_PATH}")
 
     include(CetCMakeEnv)
@@ -59,10 +58,14 @@ if(DEFINED ENV{CETBUILDTOOLS_VERSION})
     #add_subdirectory(test)
 
     # ups - table and config files
+    if(WANT_UPS)
     add_subdirectory(ups)
+    endif()
 
     # packaging utility
     include(UseCPack)
+    # create package configuration and version files
+    cet_cmake_config()
 else()
     # cmake file for building in Pandora standalone cmake setup, distinct from cetbuildtools setup
     #-------------------------------------------------------------------------------------------------------------------------------------------
@@ -76,7 +79,7 @@ else()
     if(NOT LAR_CONTENT_LIBRARY_NAME STREQUAL "LArPandoraContent")
         set(LAR_CONTENT_LIBRARY_NAME "LArContent")
     endif()
-    project(${LAR_CONTENT_LIBRARY_NAME})
+    project(${LAR_CONTENT_LIBRARY_NAME} VERSION 03.26.01 LANGUAGES CXX )
 
     # DL Content
     option(PANDORA_LIBTORCH "Flag for building against LibTorch" OFF)
diff --git a/larpandoracontent/CMakeLists.txt b/larpandoracontent/CMakeLists.txt
index f3684d7..92b5297 100644
--- a/larpandoracontent/CMakeLists.txt
+++ b/larpandoracontent/CMakeLists.txt
@@ -1,4 +1,5 @@
-include_directories( $ENV{EIGEN_INC} )
+include_directories( ${EIGEN3_INCLUDE_DIRS} )
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-shadow")
 
 set( subdir_list LArCheating
                  LArControlFlow
